---
title: "Function main figure"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(janitor)
library(rstatix)
library(vegan)
library(vegan)
library(ape)
library(mixOmics)
library(tidyverse)
library(gplots)
library(viridis)
library(cowplot)
library(patchwork)
library(pander)
library(ggrepel)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

```{r load_data}
## load the species and genus tables generated with HUMAnN3
load("../../05-results.backup/CMC_humann2_tables.RData")


humann2_path <- fread("../../05-results.backup/pathabundance_joined_cpm.tsv")
humann2_path <- as_tibble(humann2_path)
humann2_KOs_full <- fread("../../05-results.backup/genefamilies_joined_cpm_ur90rxn.tsv.gz")
humann2_KOs_full <- as_tibble(humann2_KOs_full)

# clean the file names
humann2_KOs_full <- rename(humann2_KOs_full, Ortholog = `# Gene Family`)
colnames(humann2_KOs_full) <- gsub(".unmapped_Abundance-RPKs","", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub(".SG1","", colnames(humann2_KOs_full))

humann2_path <- rename(humann2_path, Pathway = `# Pathway`)
colnames(humann2_path) <- gsub(".unmapped_Abundance","", colnames(humann2_path))
colnames(humann2_path) <- gsub(".SG1","", colnames(humann2_path))

# remove unmapped and ungrouped reads
humann2_KOs <- humann2_KOs_full %>% filter(Ortholog != "UNMAPPED") %>% filter(!str_detect(Ortholog, "UNGROUPED"))


```


```{r ancomBC_top}
# read in ancom-bc table for CMC vs. Industrial plaque
ancom_path_ip_cmc <- fread("../../05-results.backup/ancomBC_h3_noblanks_path_IP_CMC.tsv") %>%
  rename(Pathways = Proteins) %>%
  filter(diff.abn == TRUE) %>%
  filter(`log fold change (Industrial plaque - CMC)` >= 1.0 | `log fold change (Industrial plaque - CMC)` <= -1.0) %>%
  arrange(`log fold change (Industrial plaque - CMC)`) %>%
  mutate(Direction = ifelse(`log fold change (Industrial plaque - CMC)` > 0, "IP","CMC")) 
# %>%
#   select(Pathways, Direction, `log fold change (Industrial plaque - CMC)`)

# read in ancom-bc table for CMC vs. Industrial calculus
ancom_path_ic_cmc <- fread("../../05-results.backup/ancomBC_h3_noblanks_path_IC_CMC.tsv") %>%
  filter(diff.abn == TRUE) %>%
  filter(`log fold change (Industrial calculus - CMC)` >= 1.0 | `log fold change (Industrial calculus - CMC)` <= -1.0) %>%
  arrange(`log fold change (Industrial calculus - CMC)`) %>%
  mutate(Direction = ifelse(`log fold change (Industrial calculus - CMC)` > 0, "IC","CMC")) 
  
# read in ancom-bc table for CMC anterior vs. posterior
ancom_path_cmc_ts <- fread("../../05-results.backup/ancomBC_h3_paths_cmc_toothsite.tsv") %>%
  filter(diff.abn == TRUE) %>%
  filter(`log fold change (Anterior - Posterior)` >= 1.0 | `log fold change (Anterior - Posterior)` <= -1.0) %>%
  arrange(`log fold change (Anterior - Posterior)`) %>%
  mutate(Direction = ifelse(`log fold change (Anterior - Posterior)` > 0, "Anterior","Posterior"))
  

```


```{r load_metadata}

load("../../05-results.backup/CMC_metadata.RData")

```

```{r}
# set colors

species_colors <- c("Other" = "#bdbdbd",
                    "unclassified" = "#737373",
                    "Aggregatibacter" = "#FF3200",
                    "Burkholderia" = "#FF8466",
                    "Campylobacter" = "#BF6746",
                    "Corynebacterium" = "#F5B39A",
                    "Enterobacter" = "#FEB24C",
                    "Haemophilus" = "#FEE0B7",
                    "Lautropia" = "#A2B085",
                    "Neisseria" = "#DFEACA",
                    "Ottowia" = "#2C968A",
                    "Propionibacterium" = "#87D6CD",
                    "Pseudopropionibacterium" = "#C70E7B",
                    "Selenomonas" = "#DD6EAF",
                    "Streptococcus" = "#065FA3",
                    "Tetragenococcus" = "#699FC7")

```


# MetaCyc Pathways
## PCAs
```{r pca_fxn}
# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, color_group, shape_group, size_group) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))
    metadata_group_size <- get(paste(size_group, "_size", sep = ""))

    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_integration, Market_integration_split, Village, Tooth_site, Type), by = "SampleID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]
    size_group = df_X[[size_group]]
                      
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group, size = size_group)) +
                        geom_point() +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        scale_size_manual(values = metadata_group_size) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 14) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

```

```{r mr_paths}
# factor analysis MRs
humann2_path_noblanks.fa.df <- fread("../../05-results.backup/minres_oblimin_noblanks.fa_path.tsv")
humann2_path_cmc.fa.df <- fread("../../05-results.backup/minres_oblimin_cmc.fa_path.tsv")

```

```{r mr1_paths_noblanks}
fa_paths_noblanks <- humann2_path_noblanks.fa.df %>% 
  filter(Residual %in% c("MR1")) %>%
  pull(Ortholog)

# sub-sample the gene family table for only those gene families
humann2_path_noblanks_mr1 <- humann2_path.decontam %>%
  select(-matches("ERR|SRR164|SRS|EXB|LIB|10M")) %>%
  filter(Pathway %in% fa_paths_noblanks) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.humann2_path_noblanks_mr1_pca <- tune.pca(humann2_path_noblanks_mr1, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path_noblanks_mr1.pca <- pca(humann2_path_noblanks_mr1, ncomp = 3, logratio = 'CLR')

# plot the PCA
cmc_i_pca <- plot_pca(humann2_path_noblanks_mr1.pca, "PC1", "PC2", "Market_integration_split", "Market_integration_split", "Market_integration_split")
cmc_i_pca

```

```{r mr1_paths_noblanks}
fa_paths_cmc <- humann2_path_cmc.fa.df %>% 
  filter(Residual %in% c("MR1")) %>%
  pull(Ortholog)

# sub-sample the gene family table for only those gene families
humann2_path_cmc_mr1 <- humann2_path.decontam %>%
  select(matches("Pathway|CMC")) %>%
  filter(Pathway %in% fa_paths_cmc) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.humann2_path_cmc_mr1_pca <- tune.pca(humann2_path_cmc_mr1, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path_cmc_mr1.pca <- pca(humann2_path_cmc_mr1, ncomp = 3, logratio = 'CLR')

# plot the PCA colored by market integration
cmc_pca_mi <- plot_pca(humann2_path_cmc_mr1.pca, "PC1", "PC2", "Market_integration", "Market_integration", "Market_integration")
cmc_pca_mi

# plot the PCA colored by anterior/posterior
cmc_pca_ts <- plot_pca(humann2_path_cmc_mr1.pca, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group")
cmc_pca_ts

```


## CMC and Industrial samples ANCOM-BC differentially abundant pathways

## CMC vs IP
```{r path_cmc_ip}
ancom_path_ip_cmc_paths <- ancom_path_ip_cmc %>%
  separate(., Pathways, into = "Pathway", sep = ":", extra = "drop") %>%
  pull(Pathway) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those differentially abundant pathways from the list, and split the column with names into 3 (Pathway, Genus, Species)
humann3_path_ancombc_cmc_ip <- humann2_path %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  filter(str_detect(Path, ancom_path_ip_cmc_paths)) %>%
  select(-Path) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  arrange(Pathway)

# calculate the % for each ortholog contributed by each genus         
humann3_path_ancombc_cmc_ip_stats <- humann3_path_ancombc_cmc_ip %>%
  group_by(Pathway, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.) %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop")

# create the list of 10 orthologs again, but don't collapse the list as above
ancom_path_ip_cmc_paths <- ancom_path_ip_cmc %>%
  separate(., Pathways, into = "Pathway", sep = ":", extra = "drop") %>%
  pull(Pathway)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_cmc_ip_stats_extra <- lapply(ancom_path_ip_cmc_paths, function(eclass) {
 high_percent <- humann3_path_ancombc_cmc_ip_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 5) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_cmc_ip_bar_df <- humann3_path_ancombc_cmc_ip_stats %>%
  select(-c("Sum","Pathway")) %>%
  bind_rows(., humann2_path_cmc_ip_stats_extra %>%
              rename(Percent = Remaining) %>%
              select(Genus, Percent, Path)) %>%
  arrange(Path) %>%
  # unclassified for PWY-6470 got added to "Other" so we can remove "Other" for that path 
  filter(Genus != "Other" | Path != "PWY-6470")

```


```{r}
humann2_path_cmc_ip_bar_df_l <- humann2_path_cmc_ip_bar_df %>%
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Streptococcus"),
         Path = fct_relevel(Path, ancom_path_ip_cmc %>%
                              separate(., Pathways, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  full_join(., ancom_path_ip_cmc %>%
              mutate(newcol = Pathways) %>%
              separate(., newcol, into = "Path", sep = ":", extra = "drop")) %>%
  mutate(star_up = ifelse(q.val<0.001 & Direction == "IP", "***", ""),
         star_down = ifelse(q.val<0.001 & Direction == "CMC", "***", ""))

humann2_path_cmc_ip_bar_df_l

```

```{r}
# set colors for the plot

cmcip_colors <- c("unclassified" = "#737373",
                  "Streptococcus" = "#065FA3")

cmcipsp <- humann2_path_cmc_ip_bar_df_l %>%
  mutate(Prop = (Percent / 100) * `log fold change (Industrial plaque - CMC)`) %>%
  ggplot(., aes(x = Path, y = Prop, ymin = ci.lo.adj, ymax = ci.up.adj, fill = Genus)) + 
    # geom_col(position = "stack", color = "black", size = 0.25) +
    geom_bar(position = "stack", stat = "identity", color = "black", size = 0.25) +
    geom_errorbar(width=0.2, size=0.25) +
    geom_abline(slope = 0, intercept = 0, linetype="dashed", alpha=0.5) +
    theme_minimal(base_size = 14) +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, size = 14),
          axis.text.y = element_text(size = 12)) +
    scale_y_continuous(breaks=scales::pretty_breaks(n=6)) +
    scale_fill_manual(values = cmcip_colors) +
    # ylim(-4,4) +
    ylab("Log fold change") +
    xlab("") +
    coord_flip() +
    theme(legend.position="bottom")

cmcipsp

```

```{r}
# now calculate the mean proportion of these pathways in each group
cmcippm <- humann2_path %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  filter(str_detect(Path, ancom_path_ip_cmc_paths %>%
                      str_c(collapse = "|"))) %>%
  select(-Path) %>%
  # mutate(Pathway = str_replace_all(Pathway, "\\|unclassified","")) %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  pivot_longer(!Pathway, names_to = "Library_ID", values_to = "CPM") %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  left_join(., metadata %>%
              select(SampleID, Env, Tooth_site) %>%
              rename(Library_ID = SampleID), by = "Library_ID") %>%
  group_by(Env, Path) %>%
  summarize(Mean_cpm = mean(CPM),
            sd_cpm = sd(CPM)) %>%
  ungroup() %>%
  filter(str_detect(Env, "CMC|plaque")) %>%
  mutate(Path = fct_relevel(Path, ancom_path_ip_cmc %>%
                              separate(., Pathways, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  ggplot(., aes(x = Mean_cpm, y = Path, shape = Env, bg = Env)) + 
    # geom_bar(position = "dodge", stat = "identity", color = "black", size = 0.25) +
    geom_point(position = position_dodge(0.9), size = 3) +
    geom_errorbar(aes(xmax = (Mean_cpm + sd_cpm), xmin = (Mean_cpm - sd_cpm)), width = 0.2, position=position_dodge(0.9), color = "grey20") +
    theme_minimal(base_size = 14) +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, size = 14),
          axis.text.y = element_text(size = 14)) +
    # scale_y_continuous(breaks=scales::pretty_breaks(n=10)) +
    # ylim(-4,4) +
    scale_shape_manual(values = c(21, 21)) +
    ylab("") +
    xlab("Mean CPM") +
    scale_fill_manual(values = c("#FF3200","#969696")) +
    theme(legend.position="bottom",
          axis.text.y = element_blank())
cmcippm

```

## CMC vs. IC
```{r path_cmc_ic}
ancom_path_ic_cmc_paths <- ancom_path_ic_cmc %>%
  separate(., Pathways, into = "Pathway", sep = ":", extra = "drop") %>%
  pull(Pathway) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those differentially abundant pathways from the list, and split the column with names into 3 (Pathway, Genus, Species)
humann3_path_ancombc_cmc_ic <- humann2_path %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  filter(str_detect(Path, ancom_path_ic_cmc_paths)) %>%
  select(-Path) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  arrange(Pathway)

# calculate the % for each ortholog contributed by each genus         
humann3_path_ancombc_cmc_ic_stats <- humann3_path_ancombc_cmc_ic %>%
  group_by(Pathway, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.) %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop")

# create the list of 10 orthologs again, but don't collapse the list as above
ancom_path_ic_cmc_paths <- ancom_path_ic_cmc %>%
  separate(., Pathways, into = "Pathway", sep = ":", extra = "drop") %>%
  pull(Pathway)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_cmc_ic_stats_extra <- lapply(ancom_path_ic_cmc_paths, function(eclass) {
 high_percent <- humann3_path_ancombc_cmc_ic_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 10) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_cmc_ic_bar_df <- humann3_path_ancombc_cmc_ic_stats %>%
  select(-c("Sum","Pathway")) %>%
  filter(Percent >= 10) %>%
  bind_rows(., humann2_path_cmc_ic_stats_extra %>%
              rename(Percent = Remaining) %>%
              select(Genus, Percent, Path)) %>%
  arrange(Path) 
# %>%
#   # unclassified for PWY-6470 got added to "Other" so we can remove "Other" for that path 
#   filter(Genus != "Other" | Path != "PWY-6470")

```

```{r}
humann2_path_cmc_ic_bar_df_l <- humann2_path_cmc_ic_bar_df %>%
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Burkholderia","Corynebacterium","Enterobacter","Haemophilus","Lautropia","Neisseria","Propionibacterium","Pseudomonas","Pseudopropionibacterium","Streptococcus"),
         Path = fct_relevel(Path, ancom_path_ic_cmc %>%
                              separate(., Pathways, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  full_join(., ancom_path_ic_cmc %>%
              mutate(newcol = Pathways) %>%
              separate(., newcol, into = "Path", sep = ":", extra = "drop")) %>%
  mutate(star_up = ifelse(q.val<0.001 & Direction == "IC", "***", ""),
         star_down = ifelse(q.val<0.001 & Direction == "CMC", "***", ""))

humann2_path_cmc_ic_bar_df_l

```

```{r}
cmcic_colors <- c("Other" = "#bdbdbd",
                    "unclassified" = "#737373",
                    "Burkholderia" = "#FF8466",
                    "Corynebacterium" = "#F5B39A",
                    "Enterobacter" = "#FEB24C",
                    "Haemophilus" = "#FEE0B7",
                    "Lautropia" = "#A2B085",
                    "Neisseria" = "#DFEACA",
                    "Propionibacterium" = "#87D6CD",
                    "Pseudopropionibacterium" = "#C70E7B",
                    "Streptococcus" = "#065FA3")

cmcicsp <- humann2_path_cmc_ic_bar_df_l %>%
  mutate(Prop = (Percent / 100) * `log fold change (Industrial calculus - CMC)`,
         Path = fct_relevel(Path, ancom_path_ic_cmc %>%
                              separate(., Pathways, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  ggplot(., aes(x = Path, y = Prop, ymin = ci.lo.adj, ymax = ci.up.adj, fill = Genus)) + 
    # geom_col(position = "stack", color = "black", size = 0.25) +
    geom_bar(position = "stack", stat = "identity", color = "black", size = 0.25) +
    geom_errorbar(width=0.2, size=0.25) +
    geom_abline(slope = 0, intercept = 0, linetype="dashed", alpha=0.5) +
    theme_minimal(base_size = 14) +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, size = 14),
          axis.text.y = element_text(size = 12)) +
    scale_y_continuous(breaks=scales::pretty_breaks(n=6)) +
    scale_fill_manual(values = cmcic_colors) +
    ylab("Log fold change") +
    xlab("") +
    coord_flip() +
    theme(legend.position="bottom")

cmcicsp

```

```{r}
# now calculate the mean proportion of these pathways in each group
cmcicpm <- humann2_path %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  filter(str_detect(Path, ancom_path_ic_cmc_paths %>%
                      str_c(collapse = "|"))) %>%
  select(-Path) %>%
  # mutate(Pathway = str_replace_all(Pathway, "\\|unclassified","")) %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  pivot_longer(!Pathway, names_to = "Library_ID", values_to = "CPM") %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  left_join(., metadata %>%
              select(SampleID, Env, Tooth_site) %>%
              rename(Library_ID = SampleID), by = "Library_ID") %>%
  group_by(Env, Path) %>%
  summarize(Mean_cpm = mean(CPM),
            sd_cpm = sd(CPM)) %>%
  ungroup() %>%
  filter(str_detect(Env, "CMC|calculus")) %>%
  mutate(Path = fct_relevel(Path, ancom_path_ic_cmc %>%
                              separate(., Pathways, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  ggplot(., aes(x = Mean_cpm, y = Path, shape = Env, bg = Env)) + 
    # geom_bar(position = "dodge", stat = "identity", color = "black", size = 0.25) +
    geom_point(position = position_dodge(0.9), size = 3) +
    geom_errorbar(aes(xmax = (Mean_cpm + sd_cpm), xmin = (Mean_cpm - sd_cpm)), width = 0.2, position=position_dodge(0.9), color = "grey20") +
    theme_minimal(base_size = 14) +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, size = 14),
          axis.text.y = element_text(size = 14)) +
    # scale_y_continuous(breaks=scales::pretty_breaks(n=10)) +
    # ylim(-4,4) +
    scale_shape_manual(values = c(21, 21)) +
    ylab("") +
    xlab("Mean CPM") +
    scale_fill_manual(values = c("#FF3200","#feb24c")) +
    theme(legend.position="bottom",
          axis.text.y = element_blank())

cmcicpm

```


## CMC anterior vs. posterior
```{r path_cmc_ts}
ancom_path_ts_cmc_paths <- ancom_path_cmc_ts %>%
  separate(., Pathway, into = "Path", sep = ":", extra = "drop") %>%
  pull(Path) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those differentially abundant pathways from the list, and split the column with names into 3 (Pathway, Genus, Species)
humann3_path_ancombc_cmc_ic <- humann2_path %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  filter(str_detect(Path, ancom_path_ts_cmc_paths)) %>%
  select(-Path) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  arrange(Pathway)

# calculate the % for each ortholog contributed by each genus         
humann3_path_ancombc_cmc_ts_stats <- humann3_path_ancombc_cmc_ic %>%
  group_by(Pathway, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.) %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop")

# create the list of 10 orthologs again, but don't collapse the list as above
ancom_path_ts_cmc_paths <- ancom_path_cmc_ts %>%
  separate(., Pathway, into = "Path", sep = ":", extra = "drop") %>%
  pull(Path)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_cmc_ts_stats_extra <- lapply(ancom_path_ts_cmc_paths, function(eclass) {
 high_percent <- humann3_path_ancombc_cmc_ts_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 10) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_cmc_ts_bar_df <- humann3_path_ancombc_cmc_ts_stats %>%
  select(-c("Sum","Pathway")) %>%
  filter(Percent >= 10) %>%
  bind_rows(., humann2_path_cmc_ts_stats_extra %>%
              rename(Percent = Remaining) %>%
              select(Genus, Percent, Path)) %>%
  arrange(Path) 

```


```{r}
humann2_path_cmc_ts_bar_df_l <- humann2_path_cmc_ts_bar_df %>%
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Aggregatibacter","Campylobacter","Enterobacter","Haemophilus","Neisseria","Ottowia","Selenomonas","Streptococcus","Tetragenococcus"),
         Path = fct_relevel(Path, ancom_path_cmc_ts %>%
                              separate(., Pathway, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  full_join(., ancom_path_cmc_ts %>%
              mutate(newcol = Pathway) %>%
              separate(., newcol, into = "Path", sep = ":", extra = "drop")) %>%
  mutate(star_up = ifelse(q.val<0.001 & Direction == "IC", "***", ""),
         star_down = ifelse(q.val<0.001 & Direction == "CMC", "***", ""))

humann2_path_cmc_ts_bar_df_l

```

```{r}

# set minimal color palette for just this plot
cmcts_colors <- c("Other" = "#bdbdbd",
                    "unclassified" = "#737373",
                    "Aggregatibacter" = "#FF3200",
                    "Campylobacter" = "#BF6746",
                    "Enterobacter" = "#FEB24C",
                    "Haemophilus" = "#FEE0B7",
                    "Neisseria" = "#DFEACA",
                    "Ottowia" = "#2C968A",
                    "Selenomonas" = "#DD6EAF",
                    "Streptococcus" = "#065FA3",
                    "Tetragenococcus" = "#699FC7")

tssp <- humann2_path_cmc_ts_bar_df_l %>%
  mutate(Prop = (Percent / 100) * `log fold change (Anterior - Posterior)`,
         Path = fct_relevel(Path, ancom_path_cmc_ts %>%
                              separate(., Pathway, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  ggplot(., aes(x = Path, y = Prop, ymin = ci.lo.adj, ymax = ci.up.adj, fill = Genus)) + 
    # geom_col(position = "stack", color = "black", size = 0.25) +
    geom_bar(position = "stack", stat = "identity", color = "black", size = 0.25) +
    geom_errorbar(width=0.2, size=0.25) +
    geom_abline(slope = 0, intercept = 0, linetype="dashed", alpha=0.5) +
    theme_minimal(base_size = 14) +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, size = 14),
          axis.text.y = element_text(size = 12)) +
    scale_y_continuous(breaks=scales::pretty_breaks(n=6)) +
    scale_fill_manual(values = cmcts_colors) +
    ylab("Log fold change") +
    xlab("") +
    coord_flip() +
    theme(legend.position="bottom")

tssp

```

```{r}
# now calculate the mean proportion of these pathways in each group
tscpm <- humann2_path %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  filter(str_detect(Path, ancom_path_ts_cmc_paths %>%
                      str_c(collapse = "|"))) %>%
  select(-Path) %>%
  # mutate(Pathway = str_replace_all(Pathway, "\\|unclassified","")) %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  pivot_longer(!Pathway, names_to = "Library_ID", values_to = "CPM") %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  left_join(., metadata %>%
              select(SampleID, Env, Tooth_site) %>%
              rename(Library_ID = SampleID), by = "Library_ID") %>%
  filter(str_detect(Tooth_site, "Anterior|Posterior"),
         str_detect(Env, "CMC")) %>%
  group_by(Tooth_site, Path) %>%
  summarize(Mean_cpm = mean(CPM),
            sd_cpm = sd(CPM)) %>%
  ungroup() %>%
  mutate(Path = fct_relevel(Path, ancom_path_cmc_ts %>%
                              separate(., Pathway, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  ggplot(., aes(x = Mean_cpm, y = Path, shape = Tooth_site, bg = Tooth_site)) + 
    # geom_bar(position = "dodge", stat = "identity", color = "black", size = 0.25) +
    geom_point(position = position_dodge(0.9), size = 3) +
    geom_errorbar(aes(xmax = (Mean_cpm + sd_cpm), xmin = (Mean_cpm - sd_cpm)), width = 0.2, position=position_dodge(0.9), color = "grey20") +
    theme_minimal(base_size = 14) +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, size = 14),
          axis.text.y = element_text(size = 14)) +
    scale_shape_manual(values = c(21,21)) +
    # scale_y_continuous(breaks=scales::pretty_breaks(n=10)) +
    # ylim(-4,4) +
    ylab("") +
    xlab("Mean CPM") +
    scale_fill_manual(values = c("#FF3200","#065FA3")) +
    theme(legend.position="bottom",
          axis.text.y = element_blank())
tscpm

```


```{r}
pca_plots <- cmc_i_pca + cmc_pca_mi + cmc_pca_ts + 
  plot_annotation(tag_levels = 'A')

pca_plots

ggsave("../../06-publication/main_figures/Figure_8/panel_parts/h3_path_pca.pdf", plot = pca_plots,
       device = "pdf", scale = 1, width = 14, height = 4, units = c("in"), dpi = 300)


h3_species_ancombc <- cmcipsp + cmcippm + cmcicsp + cmcicpm + tssp + tscpm +
  plot_layout(nrow = 3) + 
  plot_layout(widths = c(2, 1), heights = c(1, 2, 3)) + 
  plot_annotation(tag_levels = 'A')

h3_species_ancombc

# ggsave("../../06-publication/main_figures/Figure_8/panel_parts/h3_species_ancombc.pdf", plot = h3_species_ancombc,
#        device = "pdf", scale = 1, width = 11, height = 10, units = c("in"), dpi = 300)



```































