---
title: "Cameroon hunter-gatherer calculus/plaque HUMAnN2 species contributions"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(decontam)
library(data.table)
library(vegan)
library(ape)
library(mixOmics)
library(janitor)
library(rstatix)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

```{r load_data}
## load the species and genus tables generated with HUMAnN2
humann2_path <- fread("./05-results.backup/humann2.pathabundance.all.tss.tsv")
humann2_path <- as_tibble(humann2_path)
humann2_KOs_full <- fread("./05-results.backup/humann2.genefamilies.all.i.tss.renorm.ko.tsv")
humann2_KOs_full <- as_tibble(humann2_KOs_full)

# clean the file names
humann2_KOs_full <- rename(humann2_KOs_full, Ortholog = `# Gene Family`)
colnames(humann2_KOs_full) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M_Abundance-RPKs","_10M", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","_10M", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub(".fastq.truncated_Abundance-RPKs","", colnames(humann2_KOs_full))

humann2_path <- rename(humann2_path, Pathway = `# Pathway`)
colnames(humann2_path) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M_Abundance","_10M", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","_10M", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","", colnames(humann2_path))
colnames(humann2_path) <- gsub(".fastq.truncated_Abundance","", colnames(humann2_path))

# remove unmapped and ungrouped reads
humann2_KOs <- humann2_KOs_full %>% filter(Ortholog != "UNMAPPED") %>% filter(!str_detect(Ortholog, "UNGROUPED"))


humann2_pathway_biplot_list <- fread("./05-results.backup/humann2_pathway_biplot_list.tsv") 
humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>%
  rename(Pathway = pathway) %>%
  mutate(Path = sapply(Pathway, function(f) {
                                  unlist(str_split(f, ":"))[1]
                                  })) %>%
  select(Pathway, Path, everything())

# read in the list of orthologs that are the top 10 loading in PC1 and PC2, based on the MR1 orthologs
humann2_KO_biplot_list <- fread("./05-results.backup/humann2_KO_biplot_list.tsv") # file "./05-results.backup/humann2_KO_biplot_list_abund.tsv" has the KOs from the abundance cut-off table
humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
  rename(Ortholog = Function)

```

```{r load_metadata}

load("./05-results.backup/CMC_metadata.RData")

```

# MetaCyc Pathways - maybe don't filter these?
## CMC and industrial samples
```{r path_pc1pws_cmc_indust}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_path_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC1+", Samples == "Industrial+CMC") %>%
  arrange(Path) %>%
  pull(Path) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_path_pc1pws <- humann2_path %>%
  filter(str_detect(Pathway, humann2_path_biplot_pc)) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  inner_join(., humann2_pathway_biplot_list %>%
              select(Pathway, Path) %>%
               distinct(.), by = "Pathway") %>%
  select(-Pathway) %>%
  select(Path, everything()) %>%
  arrange(Path)

# calculate the % for each ortholog contributed by each genus         
humann2_path_pc1pws_stats <- humann2_path_pc1pws %>%
  group_by(Path, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_path_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC1+", Samples == "Industrial+CMC") %>%
  arrange(Path) %>%
  pull(Path)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_pc1pws_stats_extra <- lapply(humann2_path_biplot_pc, function(eclass) {
 high_percent <- humann2_path_pc1pws_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 5) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_pcbi_bar_df <- humann2_path_pc1pws_stats_extra %>%
  rename(Percent = Remaining) %>%
  bind_rows(., humann2_path_pc1pws_stats %>%
              select(-Sum)) %>% 
  select(Path, Genus, Percent) %>%
  mutate(Direction = "PC1+",
         Samples = "CMC+Industrial")
  
# plot the values in a bar chart
# **Note** PWY-5941 has no species-level designations, and therefore is a blank space instead of a bar
humann2_path_pcbi_bar_df %>%
  filter(Direction == "PC1+", Genus != "Other") %>% # removing Other plots all species/unassigned - no need to filter the pathways
  # filter(Percent >= 5 | (Percent <= 5 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Campylobacter","Centipeda","Corynebacterium","Prevotella","Propionibacterium","Selenomonas","Veillonella")) %>%
  ggplot(., aes(x=Path, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to KEGG Orthologs, PC1 positive") +
    theme(title = element_text(size=10))


```

```{r path_pc1neg_cmc_indust}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_path_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC1-", Samples == "Industrial+CMC") %>%
  arrange(Path) %>%
  pull(Path) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_path_pc1neg <- humann2_path %>%
  filter(str_detect(Pathway, humann2_path_biplot_pc)) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  inner_join(., humann2_pathway_biplot_list %>%
              select(Pathway, Path) %>%
               distinct(.), by = "Pathway") %>%
  select(-Pathway) %>%
  select(Path, everything()) %>%
  arrange(Path)

# calculate the % for each ortholog contributed by each genus         
humann2_path_pc1neg_stats <- humann2_path_pc1neg %>%
  group_by(Path, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_path_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC1-", Samples == "Industrial+CMC") %>%
  arrange(Path) %>%
  pull(Path)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_pc1neg_stats_extra <- lapply(humann2_path_biplot_pc, function(eclass) {
 high_percent <- humann2_path_pc1neg_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 5) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_pcbi_bar_df <- humann2_path_pcbi_bar_df %>%
  bind_rows(humann2_path_pc1neg_stats_extra %>%
            rename(Percent = Remaining) %>%
            bind_rows(., humann2_path_pc1neg_stats %>%
                      select(-Sum)) %>% 
            select(Path, Genus, Percent) %>%
            mutate(Direction = "PC1-",
                   Samples = "CMC+Industrial"))

# plot the values in a bar chart
humann2_path_pcbi_bar_df %>%
  filter(Direction == "PC1-", Genus != "Other") %>% # removing Other plots all species/unassigned - no need to filter the pathways
  # filter(Percent >= 5 | (Percent <= 5 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Campylobacter","Enterobacter","Kingella","Neisseria","Olsenella","Streptococcus")) %>%
  ggplot(., aes(x=Path, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    # facet_wrap(~pathrtholog, nrow=2) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to Metacyc pathways - CMC + Industrial, PC1 negative") +
    theme(title = element_text(size=10))


```

```{r path_pc2pws_cmc_indust}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_path_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC2+", Samples == "Industrial+CMC") %>%
  arrange(Path) %>%
  pull(Path) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_path_pc2pws <- humann2_path %>%
  filter(str_detect(Pathway, humann2_path_biplot_pc)) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  inner_join(., humann2_pathway_biplot_list %>%
              select(Pathway, Path) %>%
               distinct(.), by = "Pathway") %>%
  select(-Pathway) %>%
  select(Path, everything()) %>%
  arrange(Path)

# calculate the % for each ortholog contributed by each genus         
humann2_path_pc2pws_stats <- humann2_path_pc2pws %>%
  group_by(Path, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_path_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC2+", Samples == "Industrial+CMC") %>%
  arrange(Path) %>%
  pull(Path)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_pc2pws_stats_extra <- lapply(humann2_path_biplot_pc, function(eclass) {
 high_percent <- humann2_path_pc2pws_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 5) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_pcbi_bar_df <- humann2_path_pcbi_bar_df %>%
  bind_rows(humann2_path_pc2pws_stats_extra %>%
            rename(Percent = Remaining) %>%
            bind_rows(., humann2_path_pc2pws_stats %>%
                      select(-Sum)) %>% 
            select(Path, Genus, Percent) %>%
            mutate(Direction = "PC2+",
                   Samples = "CMC+Industrial"))
 
# plot the values in a bar chart
# Most of these come from 'unclassified'
humann2_path_pcbi_bar_df %>%
  filter(Direction == "PC2+", Genus != "Other") %>%  # removing Other plots all species/unassigned - no need to filter the pathways
  # filter(Percent >= 5 | (Percent <= 5 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Actinomyces")) %>%
  ggplot(., aes(x=Path, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    # facet_wrap(~pathrtholog, nrow=2) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to Metacyc pathways - CMC + Industrial, PC2 positive") +
    theme(title = element_text(size=10))


```

```{r path_pc2neg_cmc_indust}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_path_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC2-", Samples == "Industrial+CMC") %>%
  arrange(Path) %>%
  pull(Path) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_path_pc2neg <- humann2_path %>%
  filter(str_detect(Pathway, humann2_path_biplot_pc)) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  inner_join(., humann2_pathway_biplot_list %>%
              select(Pathway, Path) %>%
               distinct(.), by = "Pathway") %>%
  select(-Pathway) %>%
  select(Path, everything()) %>%
  arrange(Path)

# calculate the % for each ortholog contributed by each genus         
humann2_path_pc2neg_stats <- humann2_path_pc2neg %>%
  group_by(Path, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_path_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC2-", Samples == "Industrial+CMC") %>%
  arrange(Path) %>%
  pull(Path)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_pc2neg_stats_extra <- lapply(humann2_path_biplot_pc, function(eclass) {
 high_percent <- humann2_path_pc2neg_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 5) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_pcbi_bar_df <- humann2_path_pcbi_bar_df %>%
  bind_rows(humann2_path_pc2neg_stats_extra %>%
            rename(Percent = Remaining) %>%
            bind_rows(., humann2_path_pc2neg_stats %>%
                      select(-Sum)) %>% 
            select(Path, Genus, Percent) %>%
            mutate(Direction = "PC2-",
                   Samples = "CMC+Industrial"))

  
# plot the values in a bar chart
humann2_path_pcbi_bar_df %>%
  filter(Direction == "PC2-", Genus != "Other") %>% # removing Other plots all species/unassigned - no need to filter the pathways
  # filter(Percent >= 5 | (Percent <= 5 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Campylobacter","Haemophilus","Prevotella")) %>%
  ggplot(., aes(x=Path, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    # facet_wrap(~pathrtholog, nrow=2) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to Metacyc pathways - CMC + Industrial, PC2 negative") +
    theme(title = element_text(size=10))


```

## CMC samples only
```{r path_pc1pws_cmc}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_path_cmc_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC1+", Samples == "CMC") %>%
  arrange(Path) %>%
  pull(Path) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_path_cmc_pc1pws <- humann2_path %>%
  filter(str_detect(Pathway, humann2_path_cmc_biplot_pc)) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  inner_join(., humann2_pathway_biplot_list %>%
              select(Pathway, Path) %>%
               distinct(.), by = "Pathway") %>%
  select(-Pathway) %>%
  select(Path, everything()) %>%
  arrange(Path)

# calculate the % for each ortholog contributed by each genus         
humann2_path_cmc_pc1pws_stats <- humann2_path_cmc_pc1pws %>%
  group_by(Path, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_path_cmc_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC1+", Samples == "CMC") %>%
  arrange(Path) %>%
  pull(Path)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_cmc_pc1pws_stats_extra <- lapply(humann2_path_cmc_biplot_pc, function(eclass) {
 high_percent <- humann2_path_cmc_pc1pws_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 5) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_cmc_pcbi_bar_df <- humann2_path_cmc_pc1pws_stats_extra %>%
  rename(Percent = Remaining) %>%
  bind_rows(., humann2_path_cmc_pc1pws_stats %>%
              select(-Sum)) %>% 
  select(Path, Genus, Percent) %>%
  mutate(Direction = "PC1+",
         Samples = "CMC")
  
# plot the values in a bar chart
# **Note** PWY-5941 and PWY-7209 have no species-level designations, and therefore are blank spaces instead of bars
humann2_path_cmc_pcbi_bar_df %>%
  filter(Direction == "PC1+", Genus != "Other") %>%
  # filter(Percent >= 5 | (Percent <= 5 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Campylobacter","Centipeda","Corynebacterium","Prevotella","Propionibacterium","Selenomonas","Veillonella")) %>%
  ggplot(., aes(x=Path, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to Metacyc pathways - CMC only, PC1 positive") +
    theme(title = element_text(size=10))


```

```{r path_pc1neg_cmc}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_path_cmc_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC1-", Samples == "CMC") %>%
  arrange(Path) %>%
  pull(Path) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_path_cmc_pc1neg <- humann2_path %>%
  filter(str_detect(Pathway, humann2_path_cmc_biplot_pc)) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  inner_join(., humann2_pathway_biplot_list %>%
              select(Pathway, Path) %>%
               distinct(.), by = "Pathway") %>%
  select(-Pathway) %>%
  select(Path, everything()) %>%
  arrange(Path)

# calculate the % for each ortholog contributed by each genus         
humann2_path_cmc_pc1neg_stats <- humann2_path_cmc_pc1neg %>%
  group_by(Path, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_path_cmc_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC1-", Samples == "CMC") %>%
  arrange(Path) %>%
  pull(Path)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_cmc_pc1neg_stats_extra <- lapply(humann2_path_cmc_biplot_pc, function(eclass) {
 high_percent <- humann2_path_cmc_pc1neg_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 5) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_cmc_pcbi_bar_df <- humann2_path_cmc_pcbi_bar_df %>%
  bind_rows(humann2_path_cmc_pc1neg_stats_extra %>%
            rename(Percent = Remaining) %>%
            bind_rows(., humann2_path_cmc_pc1neg_stats %>%
                      select(-Sum)) %>% 
            select(Path, Genus, Percent) %>%
            mutate(Direction = "PC1-",
                   Samples = "CMC"))

# plot the values in a bar chart
humann2_path_cmc_pcbi_bar_df %>%
  filter(Direction == "PC1-", Genus != "Other") %>%
  # filter(Percent >= 5 | (Percent <= 5 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Campylobacter","Enterobacter","Kingella","Neisseria","Olsenella","Streptococcus")) %>%
  ggplot(., aes(x=Path, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    # facet_wrap(~pathrtholog, nrow=2) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to Metacyc pathways - CMC only, PC1 negative") +
    theme(title = element_text(size=10))


```

```{r path_pc2pws_cmc}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_path_cmc_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC2+", Samples == "CMC") %>%
  arrange(Path) %>%
  pull(Path) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_path_cmc_pc2pws <- humann2_path %>%
  filter(str_detect(Pathway, humann2_path_cmc_biplot_pc)) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  inner_join(., humann2_pathway_biplot_list %>%
              select(Pathway, Path) %>%
               distinct(.), by = "Pathway") %>%
  select(-Pathway) %>%
  select(Path, everything()) %>%
  arrange(Path)

# calculate the % for each ortholog contributed by each genus         
humann2_path_cmc_pc2pws_stats <- humann2_path_cmc_pc2pws %>%
  group_by(Path, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_path_cmc_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC2+", Samples == "CMC") %>%
  arrange(Path) %>%
  pull(Path)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_cmc_pc2pws_stats_extra <- lapply(humann2_path_cmc_biplot_pc, function(eclass) {
 high_percent <- humann2_path_cmc_pc2pws_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 5) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_cmc_pcbi_bar_df <- humann2_path_cmc_pcbi_bar_df %>%
  bind_rows(humann2_path_cmc_pc2pws_stats_extra %>%
            rename(Percent = Remaining) %>%
            bind_rows(., humann2_path_cmc_pc2pws_stats %>%
                      select(-Sum)) %>% 
            select(Path, Genus, Percent) %>%
            mutate(Direction = "PC2+",
                   Samples = "CMC"))
 
# plot the values in a bar chart
# **Note** PWY-5941 and PWY-7209 have no species-level designations, and therefore are blank spaces instead of bars
humann2_path_cmc_pcbi_bar_df %>%
  filter(Direction == "PC2+", Genus != "Other") %>%
  # filter(Percent >= 5 | (Percent <= 5 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Actinomyces")) %>%
  ggplot(., aes(x=Path, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    # facet_wrap(~pathrtholog, nrow=2) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to Metacyc pathways - CMC only, PC2 positive") +
    theme(title = element_text(size=10))


```

```{r path_pc2neg_cmc}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_path_cmc_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC2-", Samples == "CMC") %>%
  arrange(Path) %>%
  pull(Path) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_path_cmc_pc2neg <- humann2_path %>%
  filter(str_detect(Pathway, humann2_path_cmc_biplot_pc)) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  inner_join(., humann2_pathway_biplot_list %>%
              select(Pathway, Path) %>%
               distinct(.), by = "Pathway") %>%
  select(-Pathway) %>%
  select(Path, everything()) %>%
  arrange(Path)

# calculate the % for each ortholog contributed by each genus         
humann2_path_cmc_pc2neg_stats <- humann2_path_cmc_pc2neg %>%
  group_by(Path, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_path_cmc_biplot_pc <- humann2_pathway_biplot_list %>%
  filter(Direction == "PC2-", Samples == "CMC") %>%
  arrange(Path) %>%
  pull(Path)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_cmc_pc2neg_stats_extra <- lapply(humann2_path_cmc_biplot_pc, function(eclass) {
 high_percent <- humann2_path_cmc_pc2neg_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 5) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_cmc_pcbi_bar_df <- humann2_path_cmc_pcbi_bar_df %>%
  bind_rows(humann2_path_cmc_pc2neg_stats_extra %>%
            rename(Percent = Remaining) %>%
            bind_rows(., humann2_path_cmc_pc2neg_stats %>%
                      select(-Sum)) %>% 
            select(Path, Genus, Percent) %>%
            mutate(Direction = "PC2-",
                   Samples = "CMC"))

  
# plot the values in a bar chart
humann2_path_cmc_pcbi_bar_df %>%
  filter(Direction == "PC2-", Genus != "Other") %>%
  # filter(Percent >= 5 | (Percent <= 5 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Campylobacter","Haemophilus","Prevotella")) %>%
  ggplot(., aes(x=Path, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    # facet_wrap(~pathrtholog, nrow=2) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to Metacyc pathways - CMC only, PC2 negative") +
    theme(title = element_text(size=10))


```


# KEGG Orthologs
## CMC and industrial samples
```{r KO_pc1pws_cmc_indust}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC1+", Samples == "Industrial+CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_KOs_pc1pws <- humann2_KOs_full %>%
  filter(str_detect(Ortholog, humann2_KO_biplot_pc)) %>%
  filter(str_detect(Ortholog, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Ortholog = str_replace_all(Ortholog, "\\.", "|")) %>%
  separate(., Ortholog, into = c("Ortholog", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", ""))

# calculate the % for each ortholog contributed by each genus         
humann2_KOs_pc1pws_stats <- humann2_KOs_pc1pws %>%
  group_by(Ortholog, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC1+", Samples == "Industrial+CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_KOs_pc1pws_stats_extra <- lapply(humann2_KO_biplot_pc, function(eclass) {
 high_percent <- humann2_KOs_pc1pws_stats %>%
   # group_by(Host_Genus) %>%
   filter(Ortholog == eclass) %>%
   filter(Percent < 2) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Ortholog = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_KOs_pcbi_bar_df <- humann2_KOs_pc1pws_stats_extra %>%
  rename(Percent = Remaining) %>%
  bind_rows(., humann2_KOs_pc1pws_stats %>%
              select(-Sum)) %>% 
  select(Ortholog, Genus, Percent) %>%
  mutate(Direction = "PC1+",
         Samples = "CMC+Industrial")
  
# plot the values in a bar chart
humann2_KOs_pcbi_bar_df %>%
  filter(Percent >= 2 | (Percent <= 2 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Aggregatibacter","Capnocytophaga","Fusobacterium","Haemophilus","Leptotrichia","Neisseria","Prevotella","Rothia","Streptococcus","Veillonella")) %>%
  ggplot(., aes(x=Ortholog, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    # facet_wrap(~KOrtholog, nrow=2) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to KEGG Orthologs, PC1 positive") +
    theme(title = element_text(size=10))


```

```{r pc1neg_cmc_indust}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC1-", Samples == "Industrial+CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_KOs_pc1neg <- humann2_KOs_full %>%
  filter(str_detect(Ortholog, humann2_KO_biplot_pc)) %>%
  filter(str_detect(Ortholog, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Ortholog = str_replace_all(Ortholog, "\\.", "|")) %>%
  separate(., Ortholog, into = c("Ortholog", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", ""))

# calculate the % for each ortholog contributed by each genus         
humann2_KOs_pc1neg_stats <- humann2_KOs_pc1neg %>%
  group_by(Ortholog, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC1-", Samples == "Industrial+CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_KOs_pc1neg_stats_extra <- lapply(humann2_KO_biplot_pc, function(eclass) {
 high_percent <- humann2_KOs_pc1neg_stats %>%
   # group_by(Host_Genus) %>%
   filter(Ortholog == eclass) %>%
   filter(Percent < 2) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Ortholog = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_KOs_pcbi_bar_df <- humann2_KOs_pcbi_bar_df %>%
  bind_rows(., humann2_KOs_pc1neg_stats_extra %>%
            rename(Percent = Remaining) %>%
            bind_rows(., humann2_KOs_pc1neg_stats %>%
              select(-Sum) %>% 
              filter(Percent >= 2)) %>%
            select(Ortholog, Genus, Percent) %>%
            mutate(Direction = "PC1-",
                   Samples = "CMC+Industrial")) %>%
  arrange(Direction, Ortholog)

# plot the values in a bar chart
humann2_KOs_pcbi_bar_df %>%
  filter(Direction == "PC1-") %>%
  filter(Percent >= 2 | (Percent <= 2 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Aggregatibacter","Capnocytophaga","Eikenella","Haemophilus","Neisseria","Porphyromonas","Prevotella","Propionibacterium","Rothia","Streptococcus")) %>%
  ggplot(., aes(x=Ortholog, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    # facet_wrap(~KOrtholog, nrow=2) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to KEGG Orthologs, PC1 negative") +
    theme(title = element_text(size=10))
  
```

```{r pc2pws_cmc_indust}
# list the 10 orthologs with strongest loading in pc2 + values
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC2+", Samples == "Industrial+CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_KOs_pc2pws <- humann2_KOs_full %>%
  filter(str_detect(Ortholog, humann2_KO_biplot_pc)) %>%
  filter(str_detect(Ortholog, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Ortholog = str_replace_all(Ortholog, "\\.", "|")) %>%
  separate(., Ortholog, into = c("Ortholog", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", ""))

# calculate the % for each ortholog contributed by each genus         
humann2_KOs_pc2pws_stats <- humann2_KOs_pc2pws %>%
  group_by(Ortholog, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC2+", Samples == "Industrial+CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_KOs_pc2pws_stats_extra <- lapply(humann2_KO_biplot_pc, function(eclass) {
 high_percent <- humann2_KOs_pc2pws_stats %>%
   # group_by(Host_Genus) %>%
   filter(Ortholog == eclass) %>%
   filter(Percent < 2) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Ortholog = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_KOs_pcbi_bar_df <- humann2_KOs_pcbi_bar_df %>%
  bind_rows(., humann2_KOs_pc2pws_stats_extra %>%
            rename(Percent = Remaining) %>%
            bind_rows(., humann2_KOs_pc2pws_stats %>%
              select(-Sum) %>% 
              filter(Percent >= 2)) %>%
            select(Ortholog, Genus, Percent) %>%
            mutate(Direction = "PC2+",
                   Samples = "CMC+Industrial")) %>%
  arrange(Direction, Ortholog)

# plot the values in a bar chart
humann2_KOs_pcbi_bar_df %>%
  filter(Direction == "PC2+") %>%
  filter(Percent >= 2 | (Percent <= 2 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Capnocytophaga","Fusobacterium","Leptotrichia","Porphyromonas","Prevotella","Propionibacterium","Rothia","Streptococcus","Veillonella")) %>%
  ggplot(., aes(x=Ortholog, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    # facet_wrap(~KOrtholog, nrow=2) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to KEGG Orthologs, PC2 positive") +
    theme(title = element_text(size=10))
  
```

```{r pc2neg_cmc_indust}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC2-", Samples == "Industrial+CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_KOs_pc2neg <- humann2_KOs_full %>%
  filter(str_detect(Ortholog, humann2_KO_biplot_pc)) %>%
  filter(str_detect(Ortholog, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Ortholog = str_replace_all(Ortholog, "\\.", "|")) %>%
  separate(., Ortholog, into = c("Ortholog", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", ""))

# calculate the % for each ortholog contributed by each genus         
humann2_KOs_pc2neg_stats <- humann2_KOs_pc2neg %>%
  group_by(Ortholog, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC2-", Samples == "Industrial+CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_KOs_pc2neg_stats_extra <- lapply(humann2_KO_biplot_pc, function(eclass) {
 high_percent <- humann2_KOs_pc2neg_stats %>%
   # group_by(Host_Genus) %>%
   filter(Ortholog == eclass) %>%
   filter(Percent < 2) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Ortholog = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_KOs_pcbi_bar_df <- humann2_KOs_pcbi_bar_df %>%
  bind_rows(., humann2_KOs_pc2neg_stats_extra %>%
            rename(Percent = Remaining) %>%
            bind_rows(., humann2_KOs_pc2neg_stats %>%
              select(-Sum) %>% 
              filter(Percent >= 2)) %>%
            select(Ortholog, Genus, Percent) %>%
            mutate(Direction = "PC2-",
                   Samples = "CMC+Industrial")) %>%
  arrange(Direction, Ortholog)
 
# plot the values in a bar chart
humann2_KOs_pcbi_bar_df %>%
  filter(Direction == "PC2-") %>%
  filter(Percent >= 2 | (Percent <= 2 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Aggregatibacter","Capnocytophaga","Fusobacterium","Haemophilus","Leptotrichia","Neisseria","Prevotella","Propionibacterium","Rothia","Streptococcus","Veillonella")) %>%
  ggplot(., aes(x=Ortholog, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    # facet_wrap(~KOrtholog, nrow=2) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to KEGG Orthologs, PC2 negative") +
    theme(title = element_text(size=10))
  
```

```{r cmc_industrial_ortholog_genera, eval = F}
# this won't work b/c there are different orthologs in each direction, and facet wrap puts all together on the x-axis
humann2_KOs_pcbi_bar_df %>%
  filter(Percent >= 5 | (Percent <= 5 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  # mutate(Genus = fct_relevel(Genus, "Other","unclassified","Aggregatibacter","Capnocytophaga","Fusobacterium","Haemophilus","Leptotrichia","Neisseria","Prevotella","Propionibacterium","Rothia","Streptococcus","Veillonella")) %>%
  ggplot(., aes(x=Ortholog, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    facet_wrap(~Direction, nrow=2) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to KEGG Orthologs, PC2 negative") +
    theme(title = element_text(size=10))


```

# KEGG Orthologs
## CMC samples only
```{r pc1pws_cmc}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC1+", Samples == "CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_KOs_pc1pws <- humann2_KOs_full %>%
  filter(str_detect(Ortholog, humann2_KO_biplot_pc)) %>%
  filter(str_detect(Ortholog, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Ortholog = str_replace_all(Ortholog, "\\.", "|")) %>%
  separate(., Ortholog, into = c("Ortholog", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", ""))

# calculate the % for each ortholog contributed by each genus         
humann2_KOs_pc1pws_stats <- humann2_KOs_pc1pws %>%
  group_by(Ortholog, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC1+", Samples == "CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_KOs_pc1pws_stats_extra <- lapply(humann2_KO_biplot_pc, function(eclass) {
 high_percent <- humann2_KOs_pc1pws_stats %>%
   # group_by(Host_Genus) %>%
   filter(Ortholog == eclass) %>%
   filter(Percent < 2) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Ortholog = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_KOs_pcbi_bar_df <- humann2_KOs_pcbi_bar_df %>%
  bind_rows(., humann2_KOs_pc1pws_stats_extra %>%
            rename(Percent = Remaining) %>%
            bind_rows(., humann2_KOs_pc1pws_stats %>%
              select(-Sum) %>% 
              filter(Percent >= 2)) %>%
            select(Ortholog, Genus, Percent) %>%
            mutate(Direction = "PC1+",
                   Samples = "CMC")) %>%
  arrange(Direction, Ortholog)

  
# plot the values in a bar chart
humann2_KOs_pcbi_bar_df %>%
  filter(Samples == "CMC", Direction == "PC1+") %>%
  filter(Percent >= 2 | (Percent <= 2 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Aggregatibacter","Candidatus_Saccharibacteria_noname","Fusobacterium","Haemophilus","Leptotrichia","Neisseria","Rothia","Selenomonas","Streptococcus","Veillonella")) %>%
  ggplot(., aes(x=Ortholog, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to KEGG Orthologs - CMC only, PC1 positive") +
    theme(title = element_text(size=10))


```

```{r pc1neg_cmc}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC1-", Samples == "CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_KOs_pc1neg <- humann2_KOs_full %>%
  filter(str_detect(Ortholog, humann2_KO_biplot_pc)) %>%
  filter(str_detect(Ortholog, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Ortholog = str_replace_all(Ortholog, "\\.", "|")) %>%
  separate(., Ortholog, into = c("Ortholog", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", ""))

# calculate the % for each ortholog contributed by each genus         
humann2_KOs_pc1neg_stats <- humann2_KOs_pc1neg %>%
  group_by(Ortholog, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC1-", Samples == "CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_KOs_pc1neg_stats_extra <- lapply(humann2_KO_biplot_pc, function(eclass) {
 high_percent <- humann2_KOs_pc1neg_stats %>%
   filter(Ortholog == eclass) %>%
   filter(Percent < 2) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Ortholog = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_KOs_pcbi_bar_df <- humann2_KOs_pcbi_bar_df %>%
  bind_rows(., humann2_KOs_pc1neg_stats_extra %>%
            rename(Percent = Remaining) %>%
            bind_rows(., humann2_KOs_pc1neg_stats %>%
              select(-Sum) %>% 
              filter(Percent >= 2)) %>%
            select(Ortholog, Genus, Percent) %>%
            mutate(Direction = "PC1-",
                   Samples = "CMC")) %>%
  arrange(Direction, Ortholog)

# plot the values in a bar chart
humann2_KOs_pcbi_bar_df %>%
  filter(Samples == "CMC", Direction == "PC1-") %>%
  filter(Percent >= 2 | (Percent <= 2 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Aggregatibacter","Capnocytophaga","Eikenella","Haemophilus","Neisseria","Porphyromonas","Prevotella","Propionibacterium","Rothia","Streptococcus","Veillonella")) %>%
  ggplot(., aes(x=Ortholog, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to KEGG Orthologs - CMC only, PC1 negative") +
    theme(title = element_text(size=10))
  
```

```{r pc2pws_cmc}
# list the 10 orthologs with strongest loading in pc2 + values
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC2+", Samples == "CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_KOs_pc2pws <- humann2_KOs_full %>%
  filter(str_detect(Ortholog, humann2_KO_biplot_pc)) %>%
  filter(str_detect(Ortholog, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Ortholog = str_replace_all(Ortholog, "\\.", "|")) %>%
  separate(., Ortholog, into = c("Ortholog", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", ""))

# calculate the % for each ortholog contributed by each genus         
humann2_KOs_pc2pws_stats <- humann2_KOs_pc2pws %>%
  group_by(Ortholog, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC2+", Samples == "CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_KOs_pc2pws_stats_extra <- lapply(humann2_KO_biplot_pc, function(eclass) {
 high_percent <- humann2_KOs_pc2pws_stats %>%
   filter(Ortholog == eclass) %>%
   filter(Percent < 2) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Ortholog = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_KOs_pcbi_bar_df <- humann2_KOs_pcbi_bar_df %>%
  bind_rows(., humann2_KOs_pc2pws_stats_extra %>%
            rename(Percent = Remaining) %>%
            bind_rows(., humann2_KOs_pc2pws_stats %>%
              select(-Sum) %>% 
              filter(Percent >= 2)) %>%
            select(Ortholog, Genus, Percent) %>%
            mutate(Direction = "PC2+",
                   Samples = "CMC")) %>%
  arrange(Direction, Ortholog)

# plot the values in a bar chart
humann2_KOs_pcbi_bar_df %>%
  filter(Samples == "CMC", Direction == "PC2+") %>%
  filter(Percent >= 2 | (Percent <= 2 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Aggretatibacter","Candidatus_Saccharibacteria_noname","Capnocytophaga","Fusobacterium","Haemophilus","Leptotrichia","Neisseria","Porphyromonas","Prevotella","Propionibacterium","Rhodopseudomonas","Selenomonas","Streptococcus","Veillonella")) %>%
  ggplot(., aes(x=Ortholog, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to KEGG Orthologs - CMC only, PC2 positive") +
    theme(title = element_text(size=10))
  
```

```{r pc2neg_cmc}
# list the 10 orthologs with strongest loading in PC1 + values
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC2-", Samples == "CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those 10 orthologs from the list, and split the column with names into 3 (Ortholog, Genus, Species)
humann2_KOs_pc2neg <- humann2_KOs_full %>%
  filter(str_detect(Ortholog, humann2_KO_biplot_pc)) %>%
  filter(str_detect(Ortholog, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Ortholog = str_replace_all(Ortholog, "\\.", "|")) %>%
  separate(., Ortholog, into = c("Ortholog", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", ""))

# calculate the % for each ortholog contributed by each genus         
humann2_KOs_pc2neg_stats <- humann2_KOs_pc2neg %>%
  group_by(Ortholog, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.)

# create the list of 10 orthologs again, but don't collapse the list as above
humann2_KO_biplot_pc <- humann2_KO_biplot_list %>%
  filter(Direction == "PC2-", Samples == "CMC") %>%
  arrange(Ortholog) %>%
  pull(Ortholog)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_KOs_pc2neg_stats_extra <- lapply(humann2_KO_biplot_pc, function(eclass) {
 high_percent <- humann2_KOs_pc2neg_stats %>%
   filter(Ortholog == eclass) %>%
   filter(Percent < 2) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Ortholog = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_KOs_pcbi_bar_df <- humann2_KOs_pcbi_bar_df %>%
  bind_rows(., humann2_KOs_pc2neg_stats_extra %>%
            rename(Percent = Remaining) %>%
            bind_rows(., humann2_KOs_pc2neg_stats %>%
              select(-Sum) %>% 
              filter(Percent >= 2)) %>%
            select(Ortholog, Genus, Percent) %>%
            mutate(Direction = "PC2-",
                   Samples = "CMC")) %>%
  arrange(Direction, Ortholog)
 
# plot the values in a bar chart
humann2_KOs_pcbi_bar_df %>%
  filter(Samples == "CMC", Direction == "PC2-") %>%
  filter(Percent >= 2 | (Percent <= 2 & Genus == "Other")) %>% # filter out the genera with % < 5, but keep Other < 5
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Aggregatibacter","Capnocytophaga","Fusobacterium","Haemophilus","Leptotrichia","Neisseria","Prevotella","Propionibacterium","Rothia","Streptococcus","Veillonella")) %>%
  ggplot(., aes(x=Ortholog, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF","#74D055FF",
    #                              "#07071DFF","#6B1D81FF","#D6456CFF","#FEB37BFF",
    #                              "#453781FF","#287D8EFF","#3CBC75FF","#DCE318FF",
    #                              "#FA815FFF","#FCFDBFFF","#bdbdbd")) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    ylab("Percent") +
    ggtitle("Genus-level assignents to KEGG Orthologs - CMC only, PC2 negative") +
    theme(title = element_text(size=10))
  
```
































