---
title: "Cameroon hunter-gatherer calculus/plaque MALT beta diversity"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(psych)
library(ape)
library(mixOmics)
library(compositions)
library(vegan)
library(janitor)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

# MALT read stats
The number of reads assigned in each sample appears very low (~50%). We want to check this so we know how many were assinged in each sample. We used the cleaned and merged reads that didn't align to the human genome (hg19), with the following database: /projects1/malt/databases/indexed/index040/full-nt_2017-10. 
```{r load_data}
## load the species and genus tables generated in MEGAN
# malt_species <- fread("05-results.backup/CMC_MALT_nt_species_bacteria_archaea.txt")
# malt_genus <- fread("05-results.backup/CMC_MALT_nt_genus_bacteria_archaea.txt")

# malt_species <- fread("05-results.backup/CMC_MALT_RSC_species_bacteria_archaea.txt")
# malt_genus <- fread("05-results.backup/CMC_MALT_RSC_genus_bacteria_archaea.txt")

# malt_species <- fread("05-results.backup/CMC_someHMP_MALT_RSC_species_bacteria_archaea.txt")
# malt_genus <- fread("05-results.backup/CMC_someHMP_MALT_RSC_genus_bacteria_archaea.txt")

# malt_species <- fread("./05-results.backup/CMC_HMP_MALT_RSC_species_bacteria_archaea.txt")
# malt_genus <- fread("./05-results.backup/CMC_HMP_MALT_RSC_genus_bacteria_archaea.txt")

# malt_species <- fread("./05-results.backup/CMC_HMP_JAE_MALT_RSC_species_bacteria_archaea.txt")
# malt_genus <- fread("./05-results.backup/CMC_HMP_JAE_MALT_RSC_genus_bacteria_archaea.txt")

malt_species <- fread("./05-results.backup/CMC_HMP_JAE_Yano_MALT_RSC_species_bacteria_archaea.txt")
malt_genus <- fread("./05-results.backup/CMC_HMP_JAE_Yano_MALT_RSC_genus_bacteria_archaea.txt")

# clean the file names
malt_species <- rename(malt_species, Species = `#Datasets`)
colnames(malt_species) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M","_10M", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam","_10M", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub(".fastq.truncated","", colnames(malt_species))


# remove human and unassigned reads
malt_species <- malt_species %>% filter(!str_detect(Species, "Homo sapiens"))
malt_species <- malt_species %>% filter(!str_detect(Species, "Not assigned"))
# get rid of the extra column
malt_species <- malt_species %>% select(-Comparison_CMC_HMP_JAE_MALT_RSC)

malt_genus <- rename(malt_genus, Species = `#Datasets`)
colnames(malt_genus) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam","_10M", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M","_10M", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub(".fastq.truncated","", colnames(malt_genus))

# remove human and unassigned reads
malt_genus <- malt_genus %>% filter(!str_detect(Species, "Homo"))
malt_genus <- malt_genus %>% filter(!str_detect(Species, "Not assigned"))
# get rid of the extra column at the begining
malt_genus <- malt_genus %>% select(-Comparison_CMC_HMP_JAE_MALT_RSC)

# load the metadata file
metadata <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata <- metadata %>%
  rename(SampleID = `#SampleID`)  %>%
  as_tibble() %>%
  mutate(Market_economy = str_replace(Market_economy, "HMP","Industrial")) %>%
  mutate(Description = fct_relevel(Description, "HunterGatherer", "Yanomami", "HMP_supPlaque", "HMP_subPlaque",
                                   "HMP10M_supPlaque", "HMP10M_subPlaque","JAE","JAE10M",
                                   "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime","Yanomami","HMP","HMP10M","JAE","JAE10M",
                                     "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Env = fct_relevel(Env, "HunterGatherer","HMP","HMP10M","JAE","JAE10M",
                                     "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Market_economy = fct_relevel(Market_economy, "Logging road","Forrest camp","Farming","Yanomami",
                                      "Industrial","ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Age_group = fct_relevel(Age_group, "10-18","18-29","30-39","40-49","50-59","60+","HMP","HMP10M"),
         Sex = fct_relevel(Sex, "Male","Female"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior","Buccal_mucosa","HMP","HMP10M"),
         Type = fct_relevel(Type, "Calculus","Plaque","Blank"),
         Village = fct_relevel(Village, "202","205","204","203","206","Yanomami","HMP","HMP10M","JAE","JAE10M",
                               "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Industry = fct_relevel(Industry, "Non-industrial","Industrial"))

fullSRR <- metadata %>%
  select(SampleID) %>%
  filter(str_detect(SampleID, "SRR")) %>%
  filter(!str_detect(SampleID, "10M")) %>%
  pull
```

```{r colors_shapes_passionfruit, eval = F}
Village_colors <- c(`202` = "#C70E7B", `204` = "#60CB57", `205` = "#577EA1", `203` = "#E68561", `206` = "#172869",
                     ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
village_shapes <- c(`202` = 15, `205` = 18, `204` = 17, `203` = 16, `206` = 25,
                    ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Ethnic_Group_colors <- c(Baka = "#C70E7B", Nzime = "#172869",
                         ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Ethnic_Group_shapes <- c(Baka = 19, Nzime = 17, 
                         ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Env_colors <- c(HunterGatherer = "#C70E7B", ExtractionGauze = "#881C00",
                ExtractionBlank = "#A34D1A", LibraryBlank = "#CC986A")
Env_shapes <- c(HunterGatherer = 19, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

market_economy_colors <- c(`Logging road` = "#C70E7B", `Forrest camp` = "#172869", Farming = "#60CB57",
                           ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
market_economy_shapes <- c(`Logging road` = 19, `Forrest camp` = 17, Farming = 18,
                           ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

age_group_colors <- c(`18-29` = "#C70E7B", `30-39` = "#E68561", `40-49` = "#60CB57", `50-59` = "#577EA1", `60+` = "#172869",
                      ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
age_group_shapes <- c(`18-29` = 15, `30-39` = 16, `40-49` = 17, `50-59` = 18, `60+` = 25,
                      ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

sex_colors <- c(Female = "#C70E7B", Male = "#172869", ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
sex_shapes <- c(Female = 19, Male = 17, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

tooth_site_colors <- c(Anterior = "#C70E7B", Posterior = "#172869", 
                       ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
tooth_site_shapes <- c(Anterior = 19, Posterior = 17, 
                       ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

```

```{r colors_shapes_peachpear}
# lacroix_palette("PeachPear", n = 8, type = "continuous")
# "#FF3200" "#EF8158" "#E9BD8E" "#CBDDA7" "#38BCAD" "#0B91B5" "#065FA3" "#172869"

Village_colors <- c(`202` = "#FF3200", `205` = "#EF8158", `204` = "#E9BD8E", `203` = "#CBDDA7", `206` = "#38BCAD",
                    Yanomami = "#C70E7B", HMP = "#172869", HMP10M = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                     ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Village_shapes <- c(`202` = 15, `203` = 16, `205` = 17,`204` = 18, `206` = 25, Yanomami = 14, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9,
                    ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Village_size <- c(`202` = 4.5, `203` = 4, `205` = 4,`204` = 6, `206` = 4, Yanomami = 4, HMP = 4, HMP10M = 4, JAE = 4, JAE10M = 4,
                    ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Ethnic_Group_colors <- c(Baka = "#FF3200", Nzime = "#172869", Yanomami = "#C70E7B", HMP = "#969696", HMP10M = "#d9d9d9", 
                         JAE10M = "#ffeda0", JAE = "#feb24c",ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Ethnic_Group_shapes <- c(Baka = 19, Nzime = 17,  Yanomami = 14, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9,
                         ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Ethnic_Group_size <- c(Baka = 4, Nzime = 4,  Yanomami = 4, HMP = 4, HMP10M = 4, JAE = 4, JAE10M = 4,
                         ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Env_colors <- c(HunterGatherer = "#FF3200", HMP = "#172869", HMP10M = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Env_shapes <- c(HunterGatherer = 19, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Env_size <- c(HunterGatherer = 4, HMP = 4, HMP10M = 4, JAE = 4, JAE10M = 4, ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Description_colors <- c(HunterGatherer = "#FF3200", Yanomami = "#C70E7B", HMP_supPlaque = "#969696", HMP_subPlaque = "#969696", # HMP_supPlaque = "#172869", HMP_subPlaque = "#172869", 
                        HMP10M_supPlaque = "#065FA3", HMP10M_subPlaque = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                        ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Description_shapes <- c(HunterGatherer = 19, Yanomami = 14, HMP_supPlaque = 10, HMP_subPlaque = 13, HMP10M_supPlaque = 10, 
                        HMP10M_subPlaque = 13, JAE = 7, JAE10M = 9, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Description_size <- c(HunterGatherer = 4, Yanomami = 4, HMP_supPlaque = 4, HMP_subPlaque = 4, HMP10M_supPlaque = 4, 
                        HMP10M_subPlaque = 4, JAE = 4, JAE10M = 4, ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Market_economy_colors <- c(`Logging road` = "#FF3200", `Forrest camp` = "#172869", Farming = "#82CDAA", Yanomami = "#C70E7B", Industrial = "#969696",
                           ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Market_economy_shapes <- c(`Logging road` = 19, `Forrest camp` = 17, Farming = 18, Yanomami = 14, Industrial = 15,
                           ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Market_economy_size <- c(`Logging road` = 4, `Forrest camp` = 4, Farming = 6, Yanomami = 4, Industrial = 4,
                           ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Age_group_colors <- c(`18-29` = "#FF3200", `30-39` = "#E9B186", `40-49` = "#82CDAA", `50-59` = "#0686B8", `60+` = "#172869",
                      `10-18` = "#C70E7B", HMP = "#969696", HMP10M = "#d9d9d9",
                      ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Age_group_shapes <- c(`18-29` = 15, `30-39` = 16, `40-49` = 17, `50-59` = 18, `60+` = 25, `10-18` = 14, HMP = 7, HMP10M = 9,
                      ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Age_group_size <- c(`18-29` = 4.5, `30-39` = 4, `40-49` = 4, `50-59` = 6, `60+` = 4, `10-18` = 4, HMP = 4, HMP10M = 4,
                      ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Sex_colors <- c(Female = "#FF3200", Male = "#172869",
                ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Sex_shapes <- c(Female = 19, Male = 17, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Sex_size <- c(Female = 4, Male = 4, ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Tooth_site_colors <- c(Anterior = "#FF3200", Posterior = "#172869", Buccal_mucosa = "#C70E7B", HMP = "#969696", HMP10M = "#d9d9d9",
                       ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Tooth_site_shapes <- c(Anterior = 19, Posterior = 17,  Buccal_mucosa = 14, HMP = 7, HMP10M = 9,
                       ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Tooth_site_size <- c(Anterior = 4, Posterior = 4,  Buccal_mucosa = 4, HMP = 4, HMP10M = 4,
                       ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Industry_colors <- c(`Non-industrial` = "#FF3200", Industrial = "#969696",
                           ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Industry_shapes <- c(`Non-industrial` = 19, Industrial = 17,
                           ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Industry_size <- c(`Non-industrial` = 4, Industrial = 4,
                           ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

```

```{r pca_plotting_functions}
# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group, shape = metadata_group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$SampleID), size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}


# plot PCA with colored dots and the title including the # of species or genera
plot_pca_titles <- function(df, pc1, pc2, color_group, shape_group, size_group, ncomps) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))
    metadata_group_size <- get(paste(size_group, "_size", sep = ""))

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$explained_variance * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]
    size_group = df_X[[size_group]]
   
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    title_text <- df %>%
                     rownames_to_column("SampleID") %>%
                     gather("Species","Counts",2:ncol(.)) %>%
                     spread(SampleID, Counts) %>%
                     count
    
    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group, size = size_group)) +
                        geom_point() +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        scale_size_manual(values = metadata_group_size) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        ggtitle(paste(title_text))

    return(pca_plot)
}

```

```{r pca_biplot_functions}
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$explained_variance * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }
    
    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(10, !!arrow_pc)

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(-10, !!arrow_pc)


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 10)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

# pca_test_biplot <- plot_pca_bi(malt_species.pca, "PC3", "PC2", "Village", PC2)
# pca_test_biplot


```


Before running decontam, let's plot the samples in a PCA to see if there are any that are clearly not good-quality
```{r raw_plot}

malt_species_raw <- malt_species %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species_raw) <- malt_species_raw$SampleID

malt_species_raw <- malt_species_raw %>% select(-SampleID)

malt_species_raw = malt_species_raw+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species_raw, logratio = 'CLR')

# perform a PCA to see how the data cluster
malt_species.pca <- mixOmics::pca(malt_species_raw, ncomp = 3, logratio = 'CLR')
plot(malt_species.pca)


plot_pca(malt_species.pca, "PC1", "PC2", "Description")
plot_pca_names(malt_species.pca,  "PC1", "PC2", "Description")

```


List the outliers from the plot in section `raw_plot` in `Cameroon_h-g_taxonomy.Rmd` (above) and remove them from the taxonomy table. 
```{r outliers}
outliers <- c("CMC002.B0201","CMC032.B0101","CMC037.B0201","CMC047.A0201",
              "EXB059.A1901","EXB059.A1301","EXB059.A1101","EXB059.A0901",
              "SRR1646041","SRR1646032","SRR1646042","SRR1646045","SRR1646030","SRR1646047")

outliersF <- str_c(outliers, collapse = "|")

```

Read in the contaminant list from `Cameroon_taxonomy_decontam.Rmd` and remove them from the taxonomy table. 
```{r contaminants}

contaminants_species <- fread("05-results.backup/malt_contaminants_species.tsv", header = T, sep = "\t")
contaminants_genus <- fread("05-results.backup/malt_contaminants_genus.tsv")

malt_species.decontam <- malt_species %>%
  anti_join(., contaminants_species)

malt_genus.decontam <- malt_genus %>%
  anti_join(., contaminants_genus)


```

#Sample Clustering with PCA

Step-by-step removal of taxa based on decontam and a filter threshold, followed
by PCA to see how contaminants/low abundance species affect the relationships of
samples to eachother between groups.
```{r pca_all_species}
# unfiltered and not decontaminated
malt_species_unfiltered <- malt_species %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("10M")) %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species_unfiltered) <- malt_species_unfiltered$SampleID

malt_species_unfiltered <- malt_species_unfiltered %>% select(-SampleID)

malt_species_unfiltered = malt_species_unfiltered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species_unfiltered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_species_unfiltered, "PC1", "PC2", "Description", "Env", "Env", 3)
plot_pca_titles(malt_species_unfiltered, "PC3", "PC2", "Description", "Env", "Env", 3)


# ggsave("05-results.backup/pca_species_unfiltered_pc1pc2.pdf", plot = pca_species_unfiltered_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)


# filtered only
malt_species_filtered <- malt_species %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("10M")) %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.005) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species_filtered) <- malt_species_filtered$SampleID

malt_species_filtered <- malt_species_filtered %>% select(-SampleID)

malt_species_filtered = malt_species_filtered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species_filtered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_species_filtered, "PC1", "PC2", "Description", "Env", "Env", 3)
plot_pca_titles(malt_species_filtered, "PC3", "PC2", "Description", "Env", "Env", 3)


# decontaminated only
malt_species.decontam_unfilt <- malt_species.decontam %>%
                                         adorn_totals(where = "col", name = "Sum_species.decontam") %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("10M")) %>%
                                         mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
                                         select(-one_of(c("Sum_species.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_unfilt) <- malt_species.decontam_unfilt$SampleID

malt_species.decontam_unfilt <- malt_species.decontam_unfilt %>% select(-SampleID)

malt_species.decontam_unfilt = malt_species.decontam_unfilt+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_unfilt, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_species.decontam_unfilt, "PC1", "PC2", "Description", "Env", "Env", 3)
plot_pca_titles(malt_species.decontam_unfilt, "PC3", "PC2", "Description", "Env", "Env", 3)

# decontaminated and filtered
malt_species.decontam_filtered <- malt_species.decontam %>%
                                         adorn_totals(where = "col", name = "Sum_species.decontam_filtered") %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("10M")) %>%
                                         mutate(Percent = Sum_species.decontam_filtered/sum(Sum_species.decontam_filtered)*100) %>%
                                         filter(Percent > 0.005) %>%
                                         select(-one_of(c("Sum_species.decontam_filtered","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered) <- malt_species.decontam_filtered$SampleID

malt_species.decontam_filtered <- malt_species.decontam_filtered %>% select(-SampleID)

malt_species.decontam_filtered = malt_species.decontam_filtered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_species.decontam_filtered, "PC1", "PC2", "Description", "Env", "Env", 3)
plot_pca_titles(malt_species.decontam_filtered, "PC3", "PC2", "Description", "Env", "Env", 3)

```
The 10M read subsampled HMP and JAE samples behave similarly to their equivalent
full samples, so for downstream species analysis we will use the full samples.
For the JAE samples, there is still a small but clear "movement" of 10M read-subsampled
samples toward the CMC samples, which were sequened to a depth of 3M-8M reads,
so there may be some diversity lost in calculus samples that are not deeply
sequenced. We would have to sequence the CMC samples deeper to see if this is
the case for these mature plaque samples as well, but I suspect it would be.

```{r pca_all_genus}
# unfiltered and not decontaminated
malt_genus_unfiltered <- malt_genus %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("10M")) %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus_unfiltered) <- malt_genus_unfiltered$SampleID

malt_genus_unfiltered <- malt_genus_unfiltered %>% select(-SampleID)

malt_genus_unfiltered = malt_genus_unfiltered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_genus_unfiltered, logratio = 'CLR')

# perform a PCA to see how the data cluster
nt_genus.pca <- mixOmics::pca(malt_genus_unfiltered, ncomp = 3, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_genus_unfiltered, "PC1", "PC2", "Description", "Env", "Env", 3)
plot_pca_titles(malt_genus_unfiltered, "PC3", "PC2", "Description", "Env", "Env", 3)


# filtered only
malt_genus_filtered <- malt_genus %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("10M")) %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.01) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus_filtered) <- malt_genus_filtered$SampleID

malt_genus_filtered <- malt_genus_filtered %>% select(-SampleID)

malt_genus_filtered = malt_genus_filtered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_genus_filtered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_genus_filtered, "PC1", "PC2", "Description", "Env", "Env", 3)
plot_pca_titles(malt_genus_filtered, "PC3", "PC2", "Description", "Env", "Env", 3)


# ggsave("05-results.backup/pca_nt_genus_pc1pc2.pdf", plot = pca_nt_genus_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)


# decontaminated only
malt_genus.decontam_unfiltered <- malt_genus.decontam %>%
                                         adorn_totals(where = "col", name = "Sum_genus.decontam") %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("10M")) %>%
                                         mutate(Percent = Sum_genus.decontam/sum(Sum_genus.decontam)*100) %>%
                                         select(-one_of(c("Sum_genus.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_unfiltered) <- malt_genus.decontam_unfiltered$SampleID

malt_genus.decontam_unfiltered <- malt_genus.decontam_unfiltered %>% select(-SampleID)

malt_genus.decontam_unfiltered = malt_genus.decontam_unfiltered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_genus.decontam_unfiltered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_genus.decontam_unfiltered, "PC1", "PC2", "Description", "Env", "Env", 3)
plot_pca_titles(malt_genus.decontam_unfiltered, "PC3", "PC2", "Description", "Env", "Env", 3)


# decontaminated and filtered
malt_genus.decontam_filtered <- malt_genus.decontam %>%
                                         adorn_totals(where = "col", name = "Sum_genus.decontam") %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("10M")) %>%
                                         mutate(Percent = Sum_genus.decontam/sum(Sum_genus.decontam)*100) %>%
                                         filter(Percent > 0.01) %>%
                                         select(-one_of(c("Sum_genus.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered) <- malt_genus.decontam_filtered$SampleID

malt_genus.decontam_filtered <- malt_genus.decontam_filtered %>% select(-SampleID)

malt_genus.decontam_filtered = malt_genus.decontam_filtered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_genus.decontam_filtered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Description", "Env", "Env", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Description", "Env", "Env", 3)

```
The 10M read subsampled HMP and JAE samples behave similarly to their equivalent
full samples once the data have been decontaminated and filtered, so for
downstream genus analysis we will use the full samples.

## Species-level analyses

What is the relationship of the groups in a PCA without the blanks?
```{r pca_noblanks_species}
malt_species.decontam_filtered_noblanks <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-contains("10M")) %>%
                                         # select(-one_of(fullSRR)) %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_noblanks) <- malt_species.decontam_filtered_noblanks$SampleID

malt_species.decontam_filtered_noblanks <- malt_species.decontam_filtered_noblanks %>% select(-SampleID)

malt_species.decontam_filtered_noblanks = malt_species.decontam_filtered_noblanks+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered_noblanks, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Description (do HMP supragingival and HMP subgingival plaque samples separate?)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Description", "Description", "Description", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_species_decontam_filtered_noblanks.pca <- mixOmics::pca(malt_species.decontam_filtered_noblanks, ncomp = 3, logratio = 'CLR')
plot(malt_species_decontam_filtered_noblanks.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_species_decontam_filtered_noblanks.pca.variates <- as.data.frame(malt_species_decontam_filtered_noblanks.pca$variates$X)
malt_species_decontam_filtered_noblanks.pca.variates <- malt_species_decontam_filtered_noblanks.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_species_decontam_filtered_noblanks.pca.varmet <- inner_join(malt_species_decontam_filtered_noblanks.pca.variates, metadata, by = "SampleID")

pca_malt_species_pc1pc2 <- ggplot(malt_species_decontam_filtered_noblanks.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[2], 2)), "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_species_pc1pc2

# ggsave("05-results.backup/pca_nt_species_pc1pc2.pdf", plot = pca_nt_species_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)

```
The CMC samples span a similar diversity across PC1 as HMP and JAE, although
there are clearly some samples with much less overlap (the HMP and JAE samples
to the far left, and the CMC samples to the far right). Plaque and calculus
samples separate along PC2, which is similar to the difference reported in
Velsko, *et al.* 2019 *Microbiome*. The CMC samples fall between the plaque and
calculus, suggesting they are mature plaque samples on the way to calculus.  
Although the Yanomami samples appear to fall within the variation of the HMP
samples in PC1/PC2 figures, they are clearly separated from all plaque/calculus
in PC3. This may be because the samples are buccal swabs, which have a different
species profile, or because they were MDA treated before sequencing. Downstream
analyses will be split into 2 sets, one including the Yanomami samples and one
excluding them, to see how the relationships of the plaque/calculus samples to
eachother changes when including a very different, but still oral, sample type.


Now we will look at the species that have strongest loadings in PC1, PC2, and
PC3, to see what appears to drive differences between the sample groups. We will
visualize the loadings of these species with biplots. In the first set we
include the Yanomami samples, and in the second set we exclude them.
```{r biplot_noblanks_species}

malt_species_decontam_filtered_biplot_121 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_species_decontam_filtered_biplot_121

pandoc.table(malt_species_decontam_filtered_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_species_decontam_filtered_biplot_122 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_biplot_122

pandoc.table(malt_species_decontam_filtered_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_species_decontam_filtered_biplot_322 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_biplot_322

malt_species_decontam_filtered_biplot_323 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_species_decontam_filtered_biplot_323

pandoc.table(malt_species_decontam_filtered_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")


```
There are some clear biological differences in the taxa with highest loadings in
each component, positive and negative. In PC1, the positive loading taxa are all
anaerobic, while the negative loading taxa are all *Neisseria* plus 1 *Rothia*, wich
are aerobic. The HMP samples that fall farther to the left are mostly
supragingival plaque, which is consistent with the aerobic species driving
separation this direction. Supragingival plaque samples have higher levels of
aerobes than subgingival plaque samples. All Yanomami buccal swab samples have
negative PC2 values, and it makes sense that *Neisseria* and other aerobic taxa
characterize this direction, as they are more abundant on buccal epithelia. In
PC2, the positive loading taxa are anaerobic, later colonizers, which
corresponds with the calculus samples that plot in positice PC2 values. The PC2
negative loading taxa are a mix of aerobes, facultative anaerobes, and
anaerobes, characteristic of a younger biofilm community such as found in plaque
compared to calculus. In PC3, the positive loading taxa are mostly *Neisseria*,
with one *Strep* & one *Rothia*, while the negative loading taxa are mostly
*Rothia*, with 2 *Actinomyces.* Both of these groups are mostly aerobic, but
suggests that *Rothia* are more predominant in plaque than *Neisseria*, while
*Neisseria* are more predominant in buccal swabs.


```{r biplot_noblanks_noyanomami_species}
# remove the Yanomami samples from the input table
malt_species.decontam_filtered_noblanks_noyanomami <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-contains("10M")) %>%
                                         select(-contains("SRR16460")) %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_noblanks_noyanomami) <- malt_species.decontam_filtered_noblanks_noyanomami$SampleID

malt_species.decontam_filtered_noblanks_noyanomami <- malt_species.decontam_filtered_noblanks_noyanomami %>% select(-SampleID)

malt_species.decontam_filtered_noblanks_noyanomami = malt_species.decontam_filtered_noblanks_noyanomami+1

# perform PCA
malt_species.decontam_filtered_noblanks_noyanomami.pca <- mixOmics::pca(malt_species.decontam_filtered_noblanks_noyanomami, ncomp = 3, logratio = 'CLR')

# First plot by Description (do HMP supragingival and HMP subgingival plaque samples separate?) to see what it looks like before adding the biplot over the points
plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC3", "PC2", "Description", "Description", "Description", 3)

# Then plot by Tooth_site (do anterior teeth and posterior teeth samples separate?) to see what it looks like before adding the biplot over the points
plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# Now plot biplots, and make tables of the top 10 species in each loading
malt_species_decontam_filtered_noyanomami_biplot_121 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_species_decontam_filtered_noyanomami_biplot_121

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_species_decontam_filtered_noyanomami_biplot_122 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_noyanomami_biplot_122

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_species_decontam_filtered_noyanomami_biplot_322 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_noyanomami_biplot_322

malt_species_decontam_filtered_noyanomami_biplot_323 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_species_decontam_filtered_noyanomami_biplot_323

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
When we look only at samples from tooth surfaces, we still see some biological
differences between the taxa in the different PCs with positive and negative
loadings. In PC1, the positive loading taxa are 9 *Niesseria* and 1 *Rothia*,
suggesting these samples are less mature biofilms, or are exposed to higher
oxygen tension. The negative loading taxa are a mix of anaerobic species,
suggesting they are more mature or exposed to lower oxygen tension. The CMC
samples separate along PC1 by tooth site, anterior to the right (positive
loadings) and posterior to the left (negative loadings), which corresponds with
the species loading in these directions. The HMP samples also separate this way,
with the supragingival plaque samples plotting farther to the right (more
positive) and subgingival plaque samples plotting more to the left (less
positive), which also corresponds with the species that load in positive and
negative values.

In PC2 positive loadings there is a mix of aerobic and anaerobic taxa, with more
*Rothia/Veillonella* than the others, while the negative loading taxa are mostly
late colonizer, anaerobic taxa. This distinction corresponds to the separation
of plaque in positive values and calculus in negative values. The CMC samples
also sort of separate by tooth site along PC2, when viewed in PC2/PC3 figures.
The anterior samples plot in more positive values and the posterior samples plot
in more negative values, corresponding with the species that drive plotting. In
PC3, the positive loading taxa are a mix of aerobic and anaerobic taxa, while
the negative loading taxa are also a mix, but *Neisseria* make up 6 of the 10.

```{r metadata_betadisper_noblanks}

metadata_noblanks <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata_noblanks <- metadata_noblanks %>%
  rename(SampleID = `#SampleID`)  %>%
  filter(!SampleID %in% outliers) %>%
  filter(!str_detect(SampleID, "SRR16460|EXB|LIB|10M")) %>%
  as_tibble() %>%
  mutate(Market_economy = str_replace(Market_economy, "HMP","Industrial"))%>%
  mutate(Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime","HMP","JAE"),
         Env = fct_relevel(Env, "HunterGatherer","HMP","JAE"),
         Industry = fct_relevel(Industry, "Non-industrial","Industrial"),
         Market_economy = fct_relevel(Market_economy, "Logging road","Forrest camp","Farming","Industrial"),
         Age_group = fct_relevel(Age_group, "18-29","30-39","40-49","50-59","60+","HMP"),
         Sex = fct_relevel(Sex, "Male","Female"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior","HMP"),
         Village = fct_relevel(Village, "202","205","204","203","206","HMP","JAE")) %>%
  select(SampleID, Age_group, Ethnic_Group, Env, Industry, Market_economy, Sex, Tooth_site, Village)

```

Run a permanova with function adonis to test if the clusters are significantly different.
```{r adonis_noblanks_noyanomami_species}
# use the CLR-transformed matrix from the mixOmics PCA in the last section

# Euclidean distance
malt_species.decontam_filtered_noblanks_noyanomami.euc <- vegdist(malt_species.decontam_filtered_noblanks_noyanomami.pca$X, method = "euclidean", na.rm = T)
malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoa <- ape::pcoa(malt_species.decontam_filtered_noblanks_noyanomami.euc)

malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoavectors <- as.data.frame(malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoa$vectors)
malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoavectors <- malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_species.decontam_filtered_noblanks_noyanomami.euc.vecmet <- inner_join(malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoavectors, metadata, by = "SampleID")

malt_species.decontam_filtered_noblanks_noyanomami.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Industry, shape = Industry)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Industry_colors) +
  scale_shape_manual(values = Industry_shapes) +
  stat_ellipse(aes(colour = Industry, group = Industry)) +
  theme_minimal(base_size = 7) +
  ggtitle("Industry")

# test for difference between different metadata categories 
# Sex can't be run b/c for JAE012 sex is unknown, and is NA
adonis(malt_species.decontam_filtered_noblanks_noyanomami.euc ~ Industry + Tooth_site + Ethnic_Group + Age_group + Market_economy + Village, metadata_noblanks, perm=999)

```


Then run betadisper to see if the groups with significant differences in
clustering have different dispersion ("tight" vs. "loose" clustering of groups)
- is one group more variable than the other(s)?
```{r betadisper_noblanks_noyanomami_species}
# Test the 3 metadata categories that are significant (p < 0.05) based on adonis for significant differences in dispersion
# Ok, sometimes Village is not significantly different, it fluxuates around 0.05. Maybe better to leave it out as not significant

# Tooth site
groups_tooth_site <- metadata_noblanks %>%
  select(Tooth_site) %>% 
  pull()

## Calculate multivariate dispersions
toothsite_betadisper_noblanks_noyanomami <- betadisper(malt_species.decontam_filtered_noblanks_noyanomami.euc, groups_tooth_site, bias.adjust =TRUE)
toothsite_betadisper_noblanks_noyanomami

## Perform test
anova(toothsite_betadisper_noblanks_noyanomami)

## Permutation test for F
permutest(toothsite_betadisper_noblanks_noyanomami, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
toothsite_betadisper_noblanks_noyanomami.HSD <- TukeyHSD(toothsite_betadisper_noblanks_noyanomami)
plot(toothsite_betadisper_noblanks_noyanomami.HSD)

## Plot the groups and distances to centroids on the
## first two PCoA axes
## with data ellipses instead of hulls
plot(toothsite_betadisper_noblanks_noyanomami, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## can also specify which axes to plot, ordering respected
# plot(toothsite_betadisper_noblanks_noyanomami, axes = c(3,1), seg.col = "forestgreen", seg.lty = "dashed")

## Draw a boxplot of the distances to centroid for each group
boxplot(toothsite_betadisper_noblanks_noyanomami)


# Industry
groups_industry <- metadata_noblanks %>%
  select(Industry) %>% 
  pull()

## Calculate multivariate dispersions
industry_betadisper_noblanks_noyanomami <- betadisper(malt_species.decontam_filtered_noblanks_noyanomami.euc, groups_industry, bias.adjust =TRUE)
industry_betadisper_noblanks_noyanomami

## Perform test
anova(industry_betadisper_noblanks_noyanomami)

## Permutation test for F
permutest(industry_betadisper_noblanks_noyanomami, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
industry_betadisper_noblanks_noyanomami.HSD <- TukeyHSD(industry_betadisper_noblanks_noyanomami)
plot(industry_betadisper_noblanks_noyanomami.HSD)

## Plot with data ellipses instead of hulls
plot(industry_betadisper_noblanks_noyanomami, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(industry_betadisper_noblanks_noyanomami)


# Village
groups_village <- metadata_noblanks %>%
  select(Village) %>% 
  pull()

## Calculate multivariate dispersions
village_betadisper_noblanks_noyanomami <- betadisper(malt_species.decontam_filtered_noblanks_noyanomami.euc, groups_village, bias.adjust =TRUE)
village_betadisper_noblanks_noyanomami

## Perform test
anova(village_betadisper_noblanks_noyanomami)

## Permutation test for F
permutest(village_betadisper_noblanks_noyanomami, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
village_betadisper_noblanks_noyanomami.HSD <- TukeyHSD(village_betadisper_noblanks_noyanomami)
plot(village_betadisper_noblanks_noyanomami.HSD)

## Plot with data ellipses instead of hulls
plot(village_betadisper_noblanks_noyanomami, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(village_betadisper_noblanks_noyanomami)


```



What is the relationship of Cameroon plaque samples to eachother without any
other sample types? Do they cluster by any of the metadata categories?
```{r pca_cmc_species}
malt_species.decontam_filtered_cmc <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-starts_with("SRR")) %>%
                                         select(-starts_with("JAE")) %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_cmc) <- malt_species.decontam_filtered_cmc$SampleID

malt_species.decontam_filtered_cmc <- malt_species.decontam_filtered_cmc %>% select(-SampleID)

malt_species.decontam_filtered_cmc = malt_species.decontam_filtered_cmc+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered_cmc, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_species.decontam_filtered_cmc.pca <- mixOmics::pca(malt_species.decontam_filtered_cmc, ncomp = 3, logratio = 'CLR')
plot(malt_species.decontam_filtered_cmc.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_species.decontam_filtered_cmc.pca.variates <- as.data.frame(malt_species.decontam_filtered_cmc.pca$variates$X)
malt_species.decontam_filtered_cmc.pca.variates <- malt_species.decontam_filtered_cmc.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_species.decontam_filtered_cmc.pca.varmet <- inner_join(malt_species.decontam_filtered_cmc.pca.variates, metadata, by = "SampleID")

pca_malt_species_pc1pc2 <- ggplot(malt_species.decontam_filtered_cmc.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_species.decontam_filtered_cmc.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_species.decontam_filtered_cmc.pca$explained_variance[2], 2))< "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_species_pc1pc2

# ggsave("05-results.backup/pca_nt_species_pc1pc2.pdf", plot = pca_nt_species_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)
```

```{r metadata_betadisper_cmc}
# Need special metadata that has only the categories of CMC samples, b/c of issues w/ extra factors from removed groups
metadata_cmc <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata_cmc <- metadata_cmc %>%
  rename(SampleID = `#SampleID`) %>%
  filter(!SampleID %in% outliers) %>%
  filter(Description == "HunterGatherer") %>%
  select(SampleID, Age_group, Ethnic_Group, Market_economy, Sex, Tooth_site, Village) %>%
  as_tibble() %>%
  mutate(Age_group = fct_relevel(Age_group, "18-29","30-39","40-49","50-59","60+"),
         Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime"),
         Market_economy = fct_relevel(Market_economy, "Logging road","Forrest camp","Farming"),
         Sex = fct_relevel(Sex, "Male","Female"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior"),
         Village = fct_relevel(Village, "202","205","204","203","206"))
```


Run a permanova with function adonis to test if the clusters are significantly different.
```{r adonis_cmc_species}
# use the CLR-transformed matrix from the mixOmics PCA in the last section

# Euclidean distance
malt_species.decontam_filtered_cmc.euc <- vegdist(malt_species.decontam_filtered_cmc.pca$X, method = "euclidean", na.rm = T)
malt_species.decontam_filtered_cmc.euc.pcoa <- ape::pcoa(malt_species.decontam_filtered_cmc.euc)

malt_species.decontam_filtered_cmc.euc.pcoavectors <- as.data.frame(malt_species.decontam_filtered_cmc.euc.pcoa$vectors)
malt_species.decontam_filtered_cmc.euc.pcoavectors <- malt_species.decontam_filtered_cmc.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_species.decontam_filtered_cmc.euc.vecmet <- inner_join(malt_species.decontam_filtered_cmc.euc.pcoavectors, metadata, by = "SampleID")

malt_species.decontam_filtered_cmc.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis(malt_species.decontam_filtered_cmc.euc ~ Tooth_site + Ethnic_Group + Market_economy + Sex + Village + Age_group, metadata_cmc, perm=999)

```

Then run betadisper to see if the groups with significant differences in
clustering have different dispersion ("tight" vs. "loose" clustering of groups)
- is one group more variable than the other(s)?
```{r betadisper_cmc_species}
# Test the 4 metadata categories that are significant (p < 0.05) based on adonis for significant differences in dispersion
# Ok, sometimes Sex is not significantly different, it fluxuates around 0.05. Maybe better to leave it out as not significant

# Tooth site
groups_tooth_site <- metadata_cmc %>%
  select(Tooth_site) %>% 
  pull()

## Calculate multivariate dispersions
toothsite_betadisper_cmc <- betadisper(malt_species.decontam_filtered_cmc.euc, groups_tooth_site, bias.adjust =TRUE)
toothsite_betadisper_cmc

## Perform test
anova(toothsite_betadisper_cmc)

## Permutation test for F
permutest(toothsite_betadisper_cmc, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
toothsite_betadisper_cmc.HSD <- TukeyHSD(toothsite_betadisper_cmc)
plot(toothsite_betadisper_cmc.HSD)

## Plot the groups and distances to centroids on the
## first two PCoA axes
## with data ellipses instead of hulls
plot(toothsite_betadisper_cmc, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## can also specify which axes to plot, ordering respected
# plot(toothsite_betadisper_cmc, axes = c(3,1), seg.col = "forestgreen", seg.lty = "dashed")

## Draw a boxplot of the distances to centroid for each group
boxplot(toothsite_betadisper_cmc)


# Market economy
groups_market_economy <- metadata_cmc %>%
  select(Market_economy) %>% 
  pull()

## Calculate multivariate dispersions
market_economy_betadisper_cmc <- betadisper(malt_species.decontam_filtered_cmc.euc, groups_market_economy, bias.adjust =TRUE)
market_economy_betadisper_cmc

## Perform test
anova(market_economy_betadisper_cmc)

## Permutation test for F
permutest(market_economy_betadisper_cmc, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
market_economy_betadisper_cmc.HSD <- TukeyHSD(market_economy_betadisper_cmc)
plot(market_economy_betadisper_cmc.HSD)

## Plot with data ellipses instead of hulls
plot(market_economy_betadisper_cmc, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(market_economy_betadisper_cmc)


# Village
groups_village <- metadata_cmc %>%
  select(Village) %>% 
  pull()

## Calculate multivariate dispersions
village_betadisper_cmc <- betadisper(malt_species.decontam_filtered_cmc.euc, groups_village, bias.adjust =TRUE)
village_betadisper_cmc

## Perform test
anova(village_betadisper_cmc)

## Permutation test for F
permutest(village_betadisper_cmc, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
village_betadisper_cmc.HSD <- TukeyHSD(village_betadisper_cmc)
plot(village_betadisper_cmc.HSD)

## Plot with data ellipses instead of hulls
plot(village_betadisper_cmc, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(village_betadisper_cmc)



```


```{r biplots_cmc_species}

malt_species.decontam_filtered_nocontrols <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-starts_with("JAE")) %>%
                                         select(-starts_with("SRR")) %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_nocontrols) <- malt_species.decontam_filtered_nocontrols$SampleID

malt_species.decontam_filtered_nocontrols <- malt_species.decontam_filtered_nocontrols %>% select(-SampleID)

malt_species.decontam_filtered_nocontrols = malt_species.decontam_filtered_nocontrols+1

malt_species.decontam_filtered_nocontrols.pca <- mixOmics::pca(malt_species.decontam_filtered_nocontrols, ncomp = 3, logratio = 'CLR')
plot(malt_species.decontam_filtered_nocontrols.pca)

malt_species.decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Tooth_site", PC1)
malt_species.decontam_filtered_nocontrols_biplot_121

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_species.decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Tooth_site", PC2)
malt_species.decontam_filtered_nocontrols_biplot_122

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_species.decontam_filtered_nocontrols_biplot_322 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Tooth_site", PC2)
malt_species.decontam_filtered_nocontrols_biplot_322

malt_species.decontam_filtered_nocontrols_biplot_323 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Tooth_site", PC3)
malt_species.decontam_filtered_nocontrols_biplot_323

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
In PC1, the positive loading taxa are mostly anaerobic taxa, while the negative
loading taxa are a mix of aerobic and facultative taxa. This corresonds to the
separation of samples by posterior (positive values) and anterior (negative
values) tooth sites. In PC2, the positive loading taxa are largely aerobic, 8
*Rothia*, while the negative loading taxa are mostly anaerobic. In PC3, the
positive loading taxa are 8 aerobic *Rothia* in positive PC2 values and
anaerobic(?) *[Eubacterium]* and *Thermoplasmatales* (archaeon), while the
negative loading taxa are a mix of facultative anaerobes and anaerobes that can
be found in early biofilms (*Streptococcus*, *Porphyromonas*). The split in PC3
positive taxa corresonds with the taxa loading in PC2 (aerobic in positive,
anaerobic in negative).


## Genus-level analyses

What is the relationship of the groups in a PCA without the blanks? Are
relationships similar to those observed using species data? Is there better
resolution with genus data?
```{r pca_noblanks_genus}
malt_genus.decontam_filtered <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-contains("10M")) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.01) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered) <- malt_genus.decontam_filtered$SampleID

malt_genus.decontam_filtered <- malt_genus.decontam_filtered %>% select(-SampleID)

malt_genus.decontam_filtered = malt_genus.decontam_filtered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_genus.decontam_filtered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Description (do HMP supragingial and HMP subgingival plaque separate?)
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Description", "Description", "Description", 3)


# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_genus.decontam_filtered.pca <- mixOmics::pca(malt_genus.decontam_filtered, ncomp = 3, logratio = 'CLR')
plot(malt_genus.decontam_filtered.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_genus.decontam_filtered.pca.variates <- as.data.frame(malt_genus.decontam_filtered.pca$variates$X)
malt_genus.decontam_filtered.pca.variates <- malt_genus.decontam_filtered.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_genus.decontam_filtered.pca.varmet <- inner_join(malt_genus.decontam_filtered.pca.variates, metadata, by = "SampleID")

pca_malt_genus_pc1pc2 <- ggplot(malt_genus.decontam_filtered.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_genus.decontam_filtered.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_genus.decontam_filtered.pca$explained_variance[2], 2)), "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_genus_pc1pc2


# ggsave("05-results.backup/pca_malt_genus_pc1pc2.pdf", plot = pca_malt_genus_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)


```
Just like we saw in the species PCA plots, the CMC samples span a similar
diversity across PC1 as HMP and JAE, although there are clearly some samples
with much less overlap (the HMP and JAE samples to the far left, and the CMC
samples to the far right). Plaque and calculus samples still separate along PC2,
with the Yanomami samples falling directly between them on PC2. The CMC samples
again fall between the plaque and calculus, suggesting they are mature plaque
samples on the way to calculus. At the genus level, the Yanomami samples appear
more similar to the HMP plaque samples than they did at the species level. In
PC3, the Yanomami samples are clearly separated from JAE calculus and CMC plaque
samples, yet overlap with HMP samples. They still separate from HMP along PC2.
Perhaps buccal swab profiles are more simlar to plaque samples at the genus
level than the species level. Those HMP samples that overlap with Yanomami
buccal swabs along PC3 are actually subgingival plaque, not supragingival
plaque. I would expect supragingival plaque, because buccal swab samples have
higher proportions of aerobes, which we expect in supragingival plaque samples
compared to subgingival plaque.


Now we will look at the genera that have strongest loadings in PC1, PC2, and
PC3, to see what appears to drive differences between the sample groups. We will
visualize the loadings of these species with biplots. In the first set we
include the Yanomami samples, and in the second set we exclude them. Do the
genera with strongest loadings include the species with strongest loadings in
sections `biplots_samples_species` and `biplots_samples_species_no_yanomami`
above?
```{r biplot_noblanks_genus}

malt_genus.decontam_filtered_biplot_121 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_genus.decontam_filtered_biplot_121

pandoc.table(malt_genus.decontam_filtered_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")



malt_genus.decontam_filtered_biplot_122 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_genus.decontam_filtered_biplot_122

pandoc.table(malt_genus.decontam_filtered_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_genus.decontam_filtered_biplot_322 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_genus.decontam_filtered_biplot_322

malt_genus.decontam_filtered_biplot_323 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_genus.decontam_filtered_biplot_323

pandoc.table(malt_genus.decontam_filtered_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
Loadings of genera show a similar pattern to what we saw with species. In PC1,
the positive loading taxa are mostly anaerobic, while the negative loading taxa
are a mix of aerobic and anaerobic. There is some overlap in the genera between
species and genus loadings for positive values, but not all species are also
represented at the genus level. Negative value genera include *Neisseria* and
relatives *Eikenella*, *Haemophilus*, *Kingella*, and *Aggregatibacter.* In PC2,
the postive loading taxa are a mix of early colonizer aerobes and later
colonizer anaerobes, while the negative loading taxa are mostly later colinizer
anaerobes. This corresponds with the separation of plaque in positive PC2 values
(less mature community) and calculus on negative values (more mature community).
There is almost no overlap of species and genera in PC2 loadings. In PC3, the
positive loading taxa are unusual, rare, or poorly characterized in the oral
biofilm, while the negative loading taxa are mostly later colonizing anaerobes.
There is no overlap in the species and genera in PC3 loadings. The Yanomami
samples again separate from JAE and CMC along PC3, but overlap with HMP.
However, they separate from HMP along PC2. It is unexpected that none of the
genera with PC3 positive loadings are the aerobic taxa that characterized this
direction in the species analysis, however the HMP samples that pull this way
are all subgingival samples, not supragingival samples, so the main taxa
characterizing the highly aerotolerant community are not the ones pushing this
way.


```{r biplot_samples_no_yanomami_genus}

# remove the Yanomami samples from the input table
malt_genus.decontam_filtered_noblanks_noyanomami <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-contains("10M")) %>%
                                         select(-contains("SRR16460")) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered_noblanks_noyanomami) <- malt_genus.decontam_filtered_noblanks_noyanomami$SampleID

malt_genus.decontam_filtered_noblanks_noyanomami <- malt_genus.decontam_filtered_noblanks_noyanomami %>% select(-SampleID)

malt_genus.decontam_filtered_noblanks_noyanomami = malt_genus.decontam_filtered_noblanks_noyanomami+1

# perform PCA
malt_genus.decontam_filtered_noblanks_noyanomami.pca <- mixOmics::pca(malt_genus.decontam_filtered_noblanks_noyanomami, ncomp = 3, logratio = 'CLR')

# First plot by Description (do HMP supragingival and HMP subgingival plaque samples separate?) to see what it looks like before adding the biplot over the points
plot_pca_titles(malt_genus.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_genus.decontam_filtered_noblanks_noyanomami, "PC3", "PC2", "Description", "Description", "Description", 3)

# Then plot by Tooth_site (do anterior teeth and posterior teeth samples separate?) to see what it looks like before adding the biplot over the points
plot_pca_titles(malt_genus.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_noblanks_noyanomami, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)


# Now plot biplots, and make tables of the top 10 species in each loading
malt_genus_decontam_filtered_noyanomami_biplot_121 <- plot_pca_bi(malt_genus.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_genus_decontam_filtered_noyanomami_biplot_121

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_genus_decontam_filtered_noyanomami_biplot_122 <- plot_pca_bi(malt_genus.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_genus_decontam_filtered_noyanomami_biplot_122

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_genus_decontam_filtered_noyanomami_biplot_322 <- plot_pca_bi(malt_genus.decontam_filtered_noblanks_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_genus_decontam_filtered_noyanomami_biplot_322

malt_genus_decontam_filtered_noyanomami_biplot_323 <- plot_pca_bi(malt_genus.decontam_filtered_noblanks_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_genus_decontam_filtered_noyanomami_biplot_323

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
When excluding Yanomami samples, the patterns of taxa characterizing sample
group separation are still less clear at the genus level than they are at the
species level. In PC1, the positive loading taxa are mostly later colonizing
anaerobes, while the negative loading taxa are more aerobic and facultative
anaerobic genera. The taxa characterizing this direction corresond with the HMP
plaque samples, which would have a less mature biofilm community, mostly
plotting in negative PC1 values. The HMP samples furtheset to the left are
supragingival plaque, while the ones to the right are subgingival plaque. The
CMC samples do appeaer to separate by tooth site along PC1, with anterior
samples clustering to the left, and posterior samples clustering to the right,
however the JAE samples don't really follow that pattern. This also corresponds
with the genera loading in negative values as more aerotolerant and genera
loading in positive values more anaerobic.

In PC2, the positive loading taxa are mostly later colonizing anaerobes, which
corresponds with the JAE calculus samples plotting in this part of the graph.
The negative loading taxa are a mix of aerobic and facultative taxa, which
corresponds with the HMP plaque samples plotting in this part of the graph. In
PC3, the positive loading taxa are unusual, rare, or poorly characterized in the
oral biofilm, while the negative loading taxa are mostly later colonizer,
anaerobes. Several calculus samples plot far in PC3 positive values, which
suggests that there are some unusual, poorly characterized taxa in calculus that
are not seen in plaque. In PC3 the CMC samples again separate out a bit by
anterior (negative values)/posterior (posterior values) tooth site.

Run a permanova with function adonis to test if the clusters are significantly different.
```{r adonis_noblanks_noyanomami_genus}
# use the CLR-transformed matrix from the mixOmics PCA in the last section

# Euclidean distance
malt_genus.decontam_filtered_noblanks_noyanomami.euc <- vegdist(malt_genus.decontam_filtered_noblanks_noyanomami.pca$X, method = "euclidean", na.rm = T)
malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoa <- ape::pcoa(malt_genus.decontam_filtered_noblanks_noyanomami.euc)

malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoavectors <- as.data.frame(malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoa$vectors)
malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoavectors <- malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_genus.decontam_filtered_noblanks_noyanomami.euc.vecmet <- inner_join(malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoavectors, metadata, by = "SampleID")

malt_genus.decontam_filtered_noblanks_noyanomami.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Industry, shape = Industry)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Industry_colors) +
  scale_shape_manual(values = Industry_shapes) +
  stat_ellipse(aes(colour = Industry, group = Industry)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
# Sex can't be run b/c for JAE012 sex is unknown, and is NA
adonis(malt_genus.decontam_filtered_noblanks_noyanomami.euc ~ Industry + Tooth_site + Ethnic_Group + Age_group + Market_economy + Village, metadata_noblanks, perm=999)

```

Then run betadisper to see if the groups have different dispersion ("tight" vs. "loose" clustering of groups) - is one group more variable than the other(s)?
```{r betadisper_noblanks_noyanomami_genus}
# Test the 2 metadata categories that are significant (p < 0.05) based on adonis for significant differences in dispersion

# Tooth site
groups_tooth_site <- metadata_noblanks %>%
  select(Tooth_site) %>% 
  pull()

## Calculate multivariate dispersions
toothsite_betadisper_noblanks_noyanomami_genus <- betadisper(malt_genus.decontam_filtered_noblanks_noyanomami.euc, groups_tooth_site, bias.adjust =TRUE)
toothsite_betadisper_noblanks_noyanomami_genus

## Perform test
anova(toothsite_betadisper_noblanks_noyanomami_genus)

## Permutation test for F
permutest(toothsite_betadisper_noblanks_noyanomami_genus, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
toothsite_betadisper_noblanks_noyanomami_genus.HSD <- TukeyHSD(toothsite_betadisper_noblanks_noyanomami_genus)
plot(toothsite_betadisper_noblanks_noyanomami_genus.HSD)

## Plot the groups and distances to centroids on the
## first two PCoA axes
## with data ellipses instead of hulls
plot(toothsite_betadisper_noblanks_noyanomami_genus, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## can also specify which axes to plot, ordering respected
# plot(toothsite_betadisper_noblanks_noyanomami_genus, axes = c(3,1), seg.col = "forestgreen", seg.lty = "dashed")

## Draw a boxplot of the distances to centroid for each group
boxplot(toothsite_betadisper_noblanks_noyanomami_genus)


# Industry
groups_industry <- metadata_noblanks %>%
  select(Industry) %>% 
  pull()

## Calculate multivariate dispersions
industry_betadisper_noblanks_noyanomami_genus <- betadisper(malt_genus.decontam_filtered_noblanks_noyanomami.euc, groups_industry, bias.adjust =TRUE)
industry_betadisper_noblanks_noyanomami_genus

## Perform test
anova(industry_betadisper_noblanks_noyanomami_genus)

## Permutation test for F
permutest(industry_betadisper_noblanks_noyanomami_genus, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
industry_betadisper_noblanks_noyanomami_genus.HSD <- TukeyHSD(industry_betadisper_noblanks_noyanomami_genus)
plot(industry_betadisper_noblanks_noyanomami_genus.HSD)

## Plot with data ellipses instead of hulls
plot(industry_betadisper_noblanks_noyanomami_genus, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(industry_betadisper_noblanks_noyanomami_genus)

```



What is the relationship of Cameroon plaque samples to eachother without any
other sample types? Do they cluster by any of the metadata categories?
```{r pca_cmc_genus}
malt_genus.decontam_filtered_cmc <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-starts_with("SRR")) %>%
                                         select(-starts_with("JAE")) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered_cmc) <- malt_genus.decontam_filtered_cmc$SampleID

malt_genus.decontam_filtered_cmc <- malt_genus.decontam_filtered_cmc %>% select(-SampleID)

malt_genus.decontam_filtered_cmc = malt_genus.decontam_filtered_cmc+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_genus.decontam_filtered_cmc, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_genus.decontam_filtered_cmc.pca <- mixOmics::pca(malt_genus.decontam_filtered_cmc, ncomp = 3, logratio = 'CLR')
plot(malt_genus.decontam_filtered_cmc.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_genus.decontam_filtered_cmc.pca.variates <- as.data.frame(malt_genus.decontam_filtered_cmc.pca$variates$X)
malt_genus.decontam_filtered_cmc.pca.variates <- malt_genus.decontam_filtered_cmc.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_genus.decontam_filtered_cmc.pca.varmet <- inner_join(malt_genus.decontam_filtered_cmc.pca.variates, metadata, by = "SampleID")

pca_malt_genus_pc1pc2 <- ggplot(malt_genus.decontam_filtered_cmc.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_genus.decontam_filtered_cmc.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_genus.decontam_filtered_cmc.pca$explained_variance[2], 2)), "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_genus_pc1pc2


# ggsave("05-results.backup/pca_malt_genus_pc1pc2.pdf", plot = pca_malt_genus_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)
```

Run a permanova with function adonis to test if the clusters are significantly different.
```{r adonis_cmc_genus}
# use the CLR-transformed matrix from the mixOmics PCA in the last section

# Euclidean distance
malt_genus.decontam_filtered_cmc.euc <- vegdist(malt_genus.decontam_filtered_cmc.pca$X, method = "euclidean", na.rm = T)
malt_genus.decontam_filtered_cmc.euc.pcoa <- ape::pcoa(malt_genus.decontam_filtered_cmc.euc)

malt_genus.decontam_filtered_cmc.euc.pcoavectors <- as.data.frame(malt_genus.decontam_filtered_cmc.euc.pcoa$vectors)
malt_genus.decontam_filtered_cmc.euc.pcoavectors <- malt_genus.decontam_filtered_cmc.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_genus.decontam_filtered_cmc.euc.vecmet <- inner_join(malt_genus.decontam_filtered_cmc.euc.pcoavectors, metadata, by = "SampleID")

malt_genus.decontam_filtered_cmc.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis(malt_genus.decontam_filtered_cmc.euc ~ Tooth_site  + Ethnic_Group + Age_group + Sex + Village + Market_economy, metadata_cmc, perm=999)

```

Then run betadisper to see if the groups have different dispersion ("tight" vs. "loose" clustering of groups) - is one group more variable than the other(s)?
```{r betadisper_cmc_genus}
# Test the 3 metadata categories that are significant (p < 0.05) based on adonis for significant differences in dispersion

# Tooth site
groups_tooth_site <- metadata_cmc %>%
  select(Tooth_site) %>% 
  pull()

## Calculate multivariate dispersions
toothsite_betadisper_cmc_genus <- betadisper(malt_genus.decontam_filtered_cmc.euc, groups_tooth_site, bias.adjust =TRUE)
toothsite_betadisper_cmc_genus

## Perform test
anova(toothsite_betadisper_cmc_genus)

## Permutation test for F
permutest(toothsite_betadisper_cmc_genus, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
toothsite_betadisper_cmc_genus.HSD <- TukeyHSD(toothsite_betadisper_cmc_genus)
plot(toothsite_betadisper_cmc_genus.HSD)

## Plot the groups and distances to centroids on the
## first two PCoA axes
## with data ellipses instead of hulls
plot(toothsite_betadisper_cmc_genus, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## can also specify which axes to plot, ordering respected
# plot(toothsite_betadisper_cmc_genus, axes = c(3,1), seg.col = "forestgreen", seg.lty = "dashed")

## Draw a boxplot of the distances to centroid for each group
boxplot(toothsite_betadisper_cmc_genus)


# Sex
groups_sex <- metadata_cmc %>%
  select(Sex) %>% 
  pull()

## Calculate multivariate dispersions
sex_betadisper_cmc_genus <- betadisper(malt_genus.decontam_filtered_cmc.euc, groups_sex, bias.adjust =TRUE)
sex_betadisper_cmc_genus

## Perform test
anova(sex_betadisper_cmc_genus)

## Permutation test for F
permutest(sex_betadisper_cmc_genus, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
sex_betadisper_cmc_genus.HSD <- TukeyHSD(sex_betadisper_cmc_genus)
plot(sex_betadisper_cmc_genus.HSD)

## Plot with data ellipses instead of hulls
plot(sex_betadisper_cmc_genus, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(sex_betadisper_cmc_genus)


# Village
groups_village <- metadata_cmc %>%
  select(Village) %>% 
  pull()

## Calculate multivariate dispersions
village_betadisper_cmc_genus <- betadisper(malt_genus.decontam_filtered_cmc.euc, groups_village, bias.adjust =TRUE)
village_betadisper_cmc_genus

## Perform test
anova(village_betadisper_cmc_genus)

## Permutation test for F
permutest(village_betadisper_cmc_genus, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
village_betadisper_cmc_genus.HSD <- TukeyHSD(village_betadisper_cmc_genus)
plot(village_betadisper_cmc_genus.HSD)

## Plot with data ellipses instead of hulls
plot(village_betadisper_cmc_genus, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(village_betadisper_cmc_genus)



```



```{r biplots_cmc_genus}

malt_genus.decontam_filtered_nocontrols <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-starts_with("JAE")) %>%
                                         select(-starts_with("SRR")) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered_nocontrols) <- malt_genus.decontam_filtered_nocontrols$SampleID

malt_genus.decontam_filtered_nocontrols <- malt_genus.decontam_filtered_nocontrols %>% select(-SampleID)

malt_genus.decontam_filtered_nocontrols = malt_genus.decontam_filtered_nocontrols+1

malt_genus.decontam_filtered_nocontrols.pca <- mixOmics::pca(malt_genus.decontam_filtered_nocontrols, ncomp = 3, logratio = 'CLR')
plot(malt_genus.decontam_filtered_nocontrols.pca)

malt_genus_decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_genus_decontam_filtered_nocontrols_biplot_121

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")



malt_genus_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_genus_decontam_filtered_nocontrols_biplot_122

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_genus_decontam_filtered_nocontrols_biplot_322 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_genus_decontam_filtered_nocontrols_biplot_322

malt_genus_decontam_filtered_nocontrols_biplot_323 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_genus_decontam_filtered_nocontrols_biplot_323

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
Based on genus profiles, the Cameroon plaque samples separate by tooth site,
just like they do with the species profile. In PC1, the positive and negative
loading taxa are anaerobes or facultative anaerobes.This is in contrast to the
species that separated PC1 positive and negative loadings being distinguished by
different oxygen tolerance. In PC2, the positive loading taxa are mostly later
colonizing anaerobes, while the negative loading taxa are a mix of early
colonizing aerobes and facultative anaerobes. In PC3, the positive loading taxa
are mostly late colonizing anaerobes, while the negative loading taxa are a mix
of early colonizing aerobes and unusual, rare, or poorly characterized in the
oral biofilm.





