---
title: "Cameroon hunter-gatherer calculus/plaque HUMAnN2"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(decontam)
library(data.table)
library(vegan)
library(ape)
library(mixOmics)
library(janitor)
library(rstatix)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```


# HUMAnN2 assignment stats
```{r load_data}
load("./05-results.backup/CMC_humann2_tables.RData")
load("./05-results.backup/CMC_metadata.RData")

# remove unmapped and ungrouped reads
humann2_KOs <- humann2_KOs_full %>% filter(!str_detect(Ortholog, "UNMAPPED|UNGROUPED"))
humann2_path.decontam <- humann2_path.decontam %>% filter(!str_detect(Pathway, "UNMAPPED|UNINTEGRATED"))

```


```{r functions}

# add functions for:
# making the input table for PCA?

# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, color_group, shape_group, size_group) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))
    metadata_group_size <- get(paste(size_group, "_size", sep = ""))

    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_integration, Market_integration_split, Village, Tooth_site, Type), by = "SampleID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]
    size_group = df_X[[size_group]]
                      
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group, size = size_group)) +
                        geom_point() +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        scale_size_manual(values = metadata_group_size) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 14) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_integration, Market_integration_split, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                      # select the correct PC column and explained variance for PC1
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$SampleID), size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 14) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}

# this function plots the PCA with biplots to look at the categories with strongest loadings
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle, pathways = c("TRUE","FALSE")) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_integration, Market_integration_split, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }

    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Function") %>%
      top_n(10, !!arrow_pc)
   if (pathways == T) {
      pws_10 <- pws_10 %>%
        rename(pathway = Function) %>%
        mutate(Function = sapply(pathway, function(f) {
                                  unlist(str_split(f, ":"))[1]
                           })) 
    }

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Function") %>%
      top_n(-10, !!arrow_pc)
    if (pathways == T) {
      neg_10 <- neg_10 %>%
      rename(pathway = Function) %>%
       mutate(Function = sapply(pathway, function(f) {
                                  unlist(str_split(f, ":"))[1]
                           })) 
    }


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Function),
                   size = 1.5, colour = "grey20", label.padding = 0.2) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Function),
                   size = 1.5, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 14)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

```

Before we proceed with looking at differences in abundance of different enzymes,
let's compare the how many reads were identified by HUMAnN2 for each sample
type. With the first step we'll read in the default HUMAnN2 output table with
all UniRef90 assignments. We'll select out just the rows with no species
assignemtns so we can calculate total assignments. HUMAnN2 uses *reads per*
*kilobase* for the initial output, so we'll be calculating the percent of total
assigned/mapped reads per kilobase (RPK). Then write out the file with just the
unmapped and total reads assigned so we never have to load this huge file again.
This has to be done with a table for which the samples were renormed
individually and then joined, and not renormed after they were joined (with
HUMAnN2 scripts).


# Pathway analysis

Now plot some PCAs to see the relationship of the samples to eachother and between groups.
We don't need to look at PCAs with the Yanomamis samples b/c we're excluding them. 

Do the pathway profiles of the Cameroon plaque samples plot closer to industrial
plaque or industrial calculus?
```{r mr_paths}
# factor analysis MRs
humann2_path_noblanks.fa.df <- fread("./05-results.backup/minres_oblimin_noblanks.fa_path.tsv")
humann2_path_cmc.fa.df <- fread("./05-results.backup/minres_oblimin_cmc.fa_path.tsv")

```


```{r pca_paths_noyanomami_metadata}
# create the input matrix
humann2_path.decontam_noblanks_presence_more_30 <- humann2_path.decontam %>%
  select(-matches("EXB|LIB|10M|ERR|SRS|SRR164")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Pathway) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the pathways that are present in at least 1/3 of at least 2 groups
  ungroup() %>%
  # pathways present in > 1/3 of samples in each group
  filter(Percent_presence >= 33.33) %>%
  group_by(Pathway) %>%
  count() %>%
  # only pathways that are present in 1/3 of samples of 2 or more groups
  filter(n > 1) %>%
  # pull out these proteins to keep
  select(Pathway)

humann2_path_noblanks <- humann2_path.decontam %>%
  select(-matches("ERR|SRR164|SRS|EXB|LIB|10M")) %>%
  inner_join(., humann2_path.decontam_noblanks_presence_more_30, by = "Pathway") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_noblanks, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path_noblanks.pca <- pca(humann2_path_noblanks, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(humann2_path_noblanks.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

```

```{r mr1_paths_noblanks}
fa_paths_noblanks <- humann2_path_noblanks.fa.df %>% 
  filter(Residual %in% c("MR1")) %>%
  pull(Ortholog)

# sub-sample the gene family table for only those gene families
humann2_path_noblanks_mr1 <- humann2_path.decontam %>%
  select(-matches("ERR|SRR164|SRS|EXB|LIB|10M")) %>%
  filter(Pathway %in% fa_paths_noblanks) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.humann2_path_noblanks_mr1_pca <- tune.pca(humann2_path_noblanks_mr1, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path_noblanks_mr1.pca <- pca(humann2_path_noblanks_mr1, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(humann2_path_noblanks_mr1.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

```


```{r}
# plot the PCA
plot_pca(humann2_path_noblanks_mr1.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

plot_pca(humann2_path_noblanks_mr1.pca, "PC3", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

plot_pca(humann2_path_noblanks_mr1.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")


plot_pca(humann2_path_noblanks_mr1.pca, "PC1", "PC2", "Market_integration_split", "Market_integration_split", "Market_integration_split")

### there's a problem with this one about the size of the points. Using a different size_group lets it work
pca_S_plot <- plot_pca(humann2_path_noblanks_mr1.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
pca_S_plot

pca_TS_plot <- plot_pca(humann2_path_noblanks_mr1.pca, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group")
pca_TS_plot

# ggsave("./06-publication/supplemental_figures/SXX26/panel_parts/SXX26_humann2_path_ts_pc12.pdf", plot = pca_TS_plot, device = "pdf",
#        scale = 1, width = 7, height = 6, units = c("in"), dpi = 300)

pca_V_plot <- plot_pca(humann2_path_noblanks_mr1.pca, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group")
pca_V_plot

# ggsave("./06-publication/supplemental_figures/SXX26/panel_parts/SXX26_humann2_path_pc12.pdf", plot = pca_V_plot, device = "pdf",
#        scale = 1, width = 7, height = 6, units = c("in"), dpi = 300)

```

Are there significant differences in beta-diversity between any of the metadata groups?
```{r adonis_paths_noyanomami}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
humann2_path_noblanks_mr1.euc <- vegdist(humann2_path_noblanks_mr1.pca$X, method = "euclidean", na.rm = T)
humann2_path_noblanks_mr1.euc.pcoa <- ape::pcoa(humann2_path_noblanks_mr1.euc)

humann2_path_noblanks_mr1.euc.pcoavectors <- as.data.frame(humann2_path_noblanks_mr1.euc.pcoa$vectors)
humann2_path_noblanks_mr1.euc.pcoavectors <- humann2_path_noblanks_mr1.euc.pcoavectors %>%
  rownames_to_column("SampleID")

humann2_path_noblanks_mr1.euc.vecmet <- inner_join(humann2_path_noblanks_mr1.euc.pcoavectors, metadata, by = "SampleID")

humann2_path_noblanks_mr1.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 4) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis2(humann2_path_noblanks_mr1.euc ~ Industry +  Market_integration, humann2_path_noblanks_mr1.euc.vecmet, permutations = 999, by = "margin")


```

What are the relationships of the Cameroon plaque samples to eachother without
the industrial samples?
```{r pca_paths_cmc_metadata}
humann2_path.decontam_noblanks_presence_more_30 <- humann2_path.decontam %>%
  select(matches("Pathway|CMC")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Pathway) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the pathways that are present in at least 1/3 of at least 2 groups
  ungroup() %>%
  # pathways present in > 1/3 of samples in each group
  filter(Percent_presence >= 33.33) %>%
  group_by(Pathway) %>%
  count() %>%
  # only pathways that are present in 1/3 of samples of 2 or more groups
  filter(n > 1) %>%
  # pull out these proteins to keep
  select(Pathway)

humann2_path_cmc_mat <- humann2_path.decontam %>%
  select(matches("Pathway|CMC")) %>%
  inner_join(., humann2_path.decontam_noblanks_presence_more_30, by = "Pathway") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_cmc_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path_cmc.pca <- pca(humann2_path_cmc_mat, ncomp = 3, logratio = 'CLR')

plot_pca(humann2_path_cmc.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

```

```{r mr1_paths_noblanks}
fa_paths_cmc <- humann2_path_cmc.fa.df %>% 
  filter(Residual %in% c("MR1")) %>%
  pull(Ortholog)

# sub-sample the gene family table for only those gene families
humann2_path_cmc_mr1 <- humann2_path.decontam %>%
  select(matches("Pathway|CMC")) %>%
  filter(Pathway %in% fa_paths_cmc) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.humann2_path_cmc_mr1_pca <- tune.pca(humann2_path_cmc_mr1, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path_cmc_mr1.pca <- pca(humann2_path_cmc_mr1, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(humann2_path_cmc_mr1.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

```

```{r}
# plot the PCA
plot_pca(humann2_path_cmc_mr1.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

plot_pca(humann2_path_cmc_mr1.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")

plot_pca(humann2_path_cmc_mr1.pca, "PC1", "PC2", "Market_integration_split", "Market_integration_split", "Market_integration_split")

plot_pca(humann2_path_cmc_mr1.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
plot_pca(humann2_path_cmc_mr1.pca, "PC3", "PC1", "Sex", "Sex", "Sex")


pca_TS_plot <- plot_pca(humann2_path_cmc_mr1.pca, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group")
pca_TS_plot

# ggsave("./06-publication/supplemental_figures/SXX26/panel_parts/SXX26_humann2_cmc_path_ts_pc12.pdf", plot = pca_TS_plot, device = "pdf",
#        scale = 1, width = 7, height = 6, units = c("in"), dpi = 300)

pca_V_plot <- plot_pca(humann2_path_cmc_mr1.pca, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group")
pca_V_plot

# ggsave("./06-publication/supplemental_figures/SXX26/panel_parts/SXX26_humann2_cmc_path_pc12.pdf", plot = pca_V_plot, device = "pdf",
#        scale = 1, width = 7, height = 6, units = c("in"), dpi = 300)

```
Again the only sort-of clustering that we see is by tooth site (anterior or
posterior).  

Are there significant differences in beta-diversity between any of the metadata groups?
```{r adonis_paths_cmc}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
humann2_path_cmc_mr1_mat.euc <- vegdist(humann2_path_cmc_mr1.pca$X, method = "euclidean", na.rm = T)
humann2_path_cmc_mr1_mat.euc.pcoa <- ape::pcoa(humann2_path_cmc_mr1_mat.euc)

humann2_path_cmc_mr1_mat.euc.pcoavectors <- as.data.frame(humann2_path_cmc_mr1_mat.euc.pcoa$vectors)
humann2_path_cmc_mr1_mat.euc.pcoavectors <- humann2_path_cmc_mr1_mat.euc.pcoavectors %>%
  rownames_to_column("SampleID")

humann2_path_cmc_mr1_mat.euc.vecmet <- inner_join(humann2_path_cmc_mr1_mat.euc.pcoavectors, metadata, by = "SampleID")

humann2_path_cmc_mr1_mat.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 4) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis2(humann2_path_cmc_mr1_mat.euc ~ ExtractionBatch +  LibraryBatch, humann2_path_cmc_mr1_mat.euc.vecmet, permutations = 999, by = "margin")
adonis2(humann2_path_cmc_mr1_mat.euc ~ Tooth_site + Ethnic_Group + Market_integration, humann2_path_cmc_mr1_mat.euc.vecmet, permutations = 999, by = "margin")

# controlled for tooth site
perm <- how(nperm = 999)
setBlocks(perm) <- with(humann2_path_cmc_mr1_mat.euc.vecmet, Tooth_site)
adonis2(humann2_path_cmc_mr1_mat.euc ~  Market_integration, humann2_path_cmc_mr1_mat.euc.vecmet, permutations = perm, by = "margin")

```
Tooth site (p < 0.01) is significantly different

Now let's look at which pathways are driving separation of the sample groups.
Which of the pathways have the strongest loadings in PC1 and PC2?
```{r pca_path_biplot}
# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc1 <- plot_pca_bi(humann2_path_noblanks_mr1.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = T)
pca_EG_biplot_pc1

# ggsave("./05-results.backup/presentation_pdfs/pca_EG_bi_noyanomami_paths.pdf", plot = pca_EG_biplot_pc1, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# make a table of the pathways to save, to use again later in another R notebook
humann2_pathway_biplot_list <- pca_EG_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1+", Samples = "Industrial+CMC")
humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>%
  bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1-", Samples = "Industrial+CMC"))

# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc2 <- plot_pca_bi(humann2_path_noblanks_mr1.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = T)
pca_EG_biplot_pc2

# what are those pathways?
pandoc.table(pca_EG_biplot_pc2$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_biplot_pc2$plot_env$neg_10 %>% select(-Function) %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>%
  bind_rows(pca_EG_biplot_pc2$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2+", Samples = "Industrial+CMC"))
humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>%
  bind_rows(pca_EG_biplot_pc2$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2-", Samples = "Industrial+CMC"))


# which paths distinguish sample groups when only CMC samples are included?
pca_EG_cmc_biplot_pc1 <- plot_pca_bi(humann2_path_cmc_mr1.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = T)
pca_EG_cmc_biplot_pc1

# ggsave("./05-results.backup/presentation_pdfs/pca_EG_bi_cmc_paths.pdf", plot = pca_EG_cmc_biplot_pc1, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>%
  bind_rows(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1+", Samples = "CMC"))
humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>%
  bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1-", Samples = "CMC"))

pca_EG_cmc_biplot_pc2 <- plot_pca_bi(humann2_path_cmc_mr1.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = T)
pca_EG_cmc_biplot_pc2

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>%
  bind_rows(pca_EG_cmc_biplot_pc2$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2+", Samples = "CMC"))
humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>%
  bind_rows(pca_EG_biplot_pc2$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2-", Samples = "CMC"))
# 
fwrite(humann2_pathway_biplot_list, file = "./05-results.backup/humann2_pathway_biplot_list.tsv", quote = F, sep = "\t")

```



# Community-level functional analysis with individual gene families assigned UniRef90 identities
## Gene family distribution

```{r mr1_orthologs}
# Now keep only the orthologs that are in MR1 from the factor analysis
mr1_proteins <- fread("./05-results.backup/minres_oblimin_noblanks.fa_KO.tsv")

```

Now perform a PCA to see if there is more distinct clustering of the samples
based on any of the metadata categories than we see with the pathways.
```{r pca_KOs_with_controls}
# create the input matrix
humann2_KOs.decontam_noblanks_presence_more_30 <- humann2_KOs.decontam %>%
  select(-matches("10M|ERR|SRS|SRR164")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Ortholog) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the SEED proteins that are present in at least 1/3 of at least 2 groups
  ungroup() %>%
  # pathways present in > 1/3 of samples in each group
  filter(Percent_presence >= 33.33) %>%
  group_by(Ortholog) %>%
  count() %>%
  # only pathways that are present in 1/3 of samples of 2 or more groups
  filter(n > 1) %>%
  # pull out these proteins to keep
  select(Ortholog)

humann2_KO_mat <- humann2_KOs.decontam %>%
  select(-matches("ERR|SRR164|SRS|10M")) %>%
  inner_join(., humann2_KOs.decontam_noblanks_presence_more_30, by = "Ortholog") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Ortholog,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.humann2_KO_pca <- tune.pca(humann2_KO_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KO_allsamples.pca <- pca(humann2_KO_mat, ncomp = 3, logratio = 'CLR')

```

```{r}
# plot the PCA
plot_pca(humann2_KO_allsamples.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
plot_pca(humann2_KO_allsamples.pca, "PC3", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")


plot_pca_names(humann2_KO_allsamples.pca, "PC1", "PC2", "Ethnic_Group")
plot_pca_names(humann2_KO_allsamples.pca, "PC3", "PC2", "Ethnic_Group")

```
There are ExtractionGauze samples that plot with the HMP plaque samples, but
these probably had some spit on them as mentioned before. None of the CMC
samples plot with the extraction or library blanks, so none need to be removed
as outliers.

Plot a PCA without the blanks.
```{r pca_KOs_no_blanks}

# create the input matrix
humann2_KOs.decontam_noblanks_presence_more_30 <- humann2_KOs.decontam %>%
  select(-matches("EXB|LIB|10M|ERR|SRS|SRR164")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Ortholog) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the SEED proteins that are present in at least 1/3 of at least 2 groups
  ungroup() %>%
  # pathways present in > 1/3 of samples in each group
  filter(Percent_presence >= 33.33) %>%
  group_by(Ortholog) %>%
  count() %>%
  # only pathways that are present in 1/3 of samples of 2 or more groups
  filter(n > 1) %>%
  # pull out these proteins to keep
  select(Ortholog)

humann2_KO_noblanks_mat <- humann2_KOs.decontam %>%
  select(-matches("EXB|LIB|ERR|SRR164|SRS|10M")) %>%
  inner_join(., humann2_KOs.decontam_noblanks_presence_more_30, by = "Ortholog") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Ortholog,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.pca(humann2_KO_noblanks_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KO_noblanks.pca <- pca(humann2_KO_noblanks_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_EG_plot

```
The HMP and JAE samples that were subsampled to 10M reads each are plotting
exactly on top of their counterpart full samples, so we can keep using the full
samples. The difference in sequencing depth isn't responsible for the
differences we observe between the sample types.

```{r mr1_orthologs}

humann2_KOs.decontam_mr1 <- humann2_KOs.decontam %>%
  select(-matches("EXB|LIB|10M|ERR|SRS|SRR164")) %>%
  inner_join(., mr1_proteins %>%
               filter(Residual == "MR1") %>%
               select(Ortholog), by = "Ortholog") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Ortholog,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.pca(humann2_KOs.decontam_mr1, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KOs.decontam_mr1.pca <- pca(humann2_KOs.decontam_mr1, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(humann2_KOs.decontam_mr1.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
plot_pca(humann2_KOs.decontam_mr1.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
plot_pca(humann2_KOs.decontam_mr1.pca, "PC1", "PC2", "Market_integration_split", "Market_integration_split", "Market_integration_split")
plot_pca(humann2_KOs.decontam_mr1.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
h2_mr1_ts <- plot_pca(humann2_KOs.decontam_mr1.pca, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group")
h2_mr1_ts

ggsave("./06-publication/supplemental_figures/SXX27/panel_parts/SXX27_humann2_ko_ts_pc12.pdf", plot = h2_mr1_ts, device = "pdf",
       scale = 1, width = 7, height = 6, units = c("in"), dpi = 300)

h2_mr1_mi <- plot_pca(humann2_KOs.decontam_mr1.pca, "PC1", "PC2", "Market_integration_split", "Ethnic_Group", "Ethnic_Group")
h2_mr1_mi

ggsave("./06-publication/supplemental_figures/SXX27/panel_parts/SXX27_humann2_ko_pc12.pdf", plot = h2_mr1_mi, device = "pdf",
       scale = 1, width = 7, height = 6, units = c("in"), dpi = 300)

```


Are there significant differences between metadata groups?
```{r adonis_KOs_noblanks}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
humann2_KO_mat.euc <- vegdist(humann2_KOs.decontam_mr1.pca$X, method = "euclidean", na.rm = T)
humann2_KO_mat.euc.pcoa <- ape::pcoa(humann2_KO_mat.euc)

humann2_KO_mat.euc.pcoavectors <- as.data.frame(humann2_KO_mat.euc.pcoa$vectors)
humann2_KO_mat.euc.pcoavectors <- humann2_KO_mat.euc.pcoavectors %>%
  rownames_to_column("SampleID")

humann2_KO_mat.euc.vecmet <- inner_join(humann2_KO_mat.euc.pcoavectors, metadata, by = "SampleID")

humann2_KO_mat.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Ethnic_Group, shape = Ethnic_Group)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Ethnic_Group_colors) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  # stat_ellipse(aes(colour = Ethnic_Group, group = Ethnic_Group)) +
  theme_minimal(base_size = 7) +
  ggtitle("Ethnic Group")

# test for difference between different metadata categories 
adonis2(humann2_KO_mat.euc ~ Ethnic_Group +  Market_integration, humann2_KO_mat.euc.vecmet, permutations = 999, by = "margin")

# controlled for tooth site
# perm <- how(nperm = 999)
# setBlocks(perm) <- with(humann2_KO_mat.euc.vecmet, Ethnic_Group)
# adonis2(humann2_KO_mat.euc ~ Market_integration, humann2_KO_mat.euc.vecmet, permutations = perm, by = "margin")

```


What is the relationship of the Cameroon plaque samples without the industrial samples?
```{r pca_KOs_cmc_metadata}
# filter low-abundance orthologs
humann2_KOs.decontam_cmc_presence_more_30 <- humann2_KOs.decontam %>%
  select(matches("Ortholog|CMC")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Ortholog) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the SEED proteins that are present in at least 30% of at least one group
  ungroup() %>%
  filter(Percent_presence >= 33.33) %>%
  distinct(Ortholog) %>%
  # pull out these proteins to keep
  select(Ortholog)

humann2_KO_cmc_mat <- humann2_KOs.decontam %>%
  select(matches("Ortholog|CMC")) %>%
  inner_join(., humann2_KOs.decontam_cmc_presence_more_30, by = "Ortholog") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Ortholog,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.pca(humann2_KO_cmc_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KO_cmc.pca <- pca(humann2_KO_cmc_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(humann2_KO_cmc.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

# ggsave("./05-results.backup/presentation_pdfs/pca_EG_KOs_cmc.pdf", plot = pca_EG_plot, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

```

```{r mr1_orthologs_cmc}
# Now keep only the orthologs that are in MR1 from the factor analysis
mr1_orthologs_cmc <- fread("./05-results.backup/minres_oblimin_cmc.fa_KO.tsv")

humann2_KOs.decontam_cmc_mr1 <- humann2_KOs.decontam %>%
  select(matches("Ortholog|CMC")) %>%
  inner_join(., mr1_orthologs_cmc %>%
               filter(Residual == "MR1") %>%
               select(Ortholog), by = "Ortholog") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Ortholog,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.pca(humann2_KOs.decontam_cmc_mr1, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KOs.decontam_cmc_mr1.pca <- pca(humann2_KOs.decontam_cmc_mr1, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(humann2_KOs.decontam_cmc_mr1.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
plot_pca(humann2_KOs.decontam_cmc_mr1.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
plot_pca(humann2_KOs.decontam_cmc_mr1.pca, "PC1", "PC2", "Market_integration_split", "Market_integration_split", "Market_integration_split")
plot_pca(humann2_KOs.decontam_cmc_mr1.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
h2_mr1_ts <- plot_pca(humann2_KOs.decontam_cmc_mr1.pca, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group")
h2_mr1_ts
h2_mr1_mi <- plot_pca(humann2_KOs.decontam_cmc_mr1.pca, "PC1", "PC2", "Market_integration_split", "Ethnic_Group", "Ethnic_Group")
h2_mr1_mi

ggsave("./06-publication/supplemental_figures/SXX27/panel_parts/SXX27_humann2_ko_cmc_pc12.pdf", plot = h2_mr1_mi, device = "pdf",
       scale = 1, width = 7, height = 6, units = c("in"), dpi = 300)

ggsave("./06-publication/supplemental_figures/SXX27/panel_parts/SXX27_humann2_ko_cmc_ts_pc12.pdf", plot = h2_mr1_ts, device = "pdf",
       scale = 1, width = 7, height = 6, units = c("in"), dpi = 300)


```


We'll use the orthologs in MR1 from the factor analysis, not the abundance
cut-off or the prevalence cut-off. Are there significant differences between
metadata groups?
```{r adonis_KOs_cmc}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
humann2_KO_cmc_mat.euc <- vegdist(humann2_KOs.decontam_cmc_mr1.pca$X, method = "euclidean", na.rm = T)
humann2_KO_cmc_mat.euc.pcoa <- ape::pcoa(humann2_KO_cmc_mat.euc)

humann2_KO_cmc_mat.euc.pcoavectors <- as.data.frame(humann2_KO_cmc_mat.euc.pcoa$vectors)
humann2_KO_cmc_mat.euc.pcoavectors <- humann2_KO_cmc_mat.euc.pcoavectors %>%
  rownames_to_column("SampleID")

humann2_KO_cmc_mat.euc.vecmet <- inner_join(humann2_KO_cmc_mat.euc.pcoavectors, metadata, by = "SampleID")

humann2_KO_cmc_mat.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis2(humann2_KO_cmc_mat.euc ~ ExtractionBatch + LibraryBatch, humann2_KO_cmc_mat.euc.vecmet, permutations = 999, by = "margin")
adonis2(humann2_KO_cmc_mat.euc ~ Village + Sex + Tooth_site + Ethnic_Group + Age_group + Market_integration, humann2_KO_cmc_mat.euc.vecmet, permutations = 999, by = "margin")

# controlled for tooth site
perm <- how(nperm = 999)
setBlocks(perm) <- with(humann2_KO_cmc_mat.euc.vecmet, Tooth_site)
adonis2(humann2_KO_cmc_mat.euc ~ Tooth_site, humann2_KO_cmc_mat.euc.vecmet, permutations = 999, by = "margin")

```


Now to see which of the orthologs have the strongest loadings in PC1 and PC2,
using the MR1 orthologs
```{r pca_KO_biplot}
# which paths distinguish sample types when HMP and JAE are included?
pca_EG_KO_biplot_pc1 <- plot_pca_bi(humann2_KOs.decontam_mr1.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = F)
pca_EG_KO_biplot_pc1

# ggsave("./05-results.backup/presentation_pdfs/humann2_KO_bi_noblanks.pdf", plot = pca_EG_KO_biplot_pc1, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_KO_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_KO_biplot_pc1$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

humann2_KO_biplot_list <- pca_EG_KO_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)) %>% select(Function) %>% mutate(Direction = "PC1+", Samples = "Industrial+CMC")
humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
  bind_rows(pca_EG_KO_biplot_pc1$plot_env$neg_10 %>% arrange(desc(PC1)) %>% select(Function) %>% mutate(Direction = "PC1-", Samples = "Industrial+CMC"))

# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc2 <- plot_pca_bi(humann2_KOs.decontam_mr1.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = F)
pca_EG_biplot_pc2

# what are those pathways?
pandoc.table(pca_EG_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_biplot_pc2$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
  bind_rows(pca_EG_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)) %>% select(Function) %>% mutate(Direction = "PC2+", Samples = "Industrial+CMC"))
humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
  bind_rows(pca_EG_biplot_pc2$plot_env$neg_10 %>% arrange(desc(PC2)) %>% select(Function) %>% mutate(Direction = "PC2-", Samples = "Industrial+CMC"))

# which paths distinguish sample groups when only CMC samples are included?
pca_EG_cmc_biplot_pc1 <- plot_pca_bi(humann2_KOs.decontam_cmc_mr1.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = F)
pca_EG_cmc_biplot_pc1

# ggsave("./05-results.backup/presentation_pdfs/humann2_KO_bi_cmc.pdf", plot = pca_EG_cmc_biplot_pc1, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
  bind_rows(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)) %>% select(Function) %>% mutate(Direction = "PC1+", Samples = "CMC"))
humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
  bind_rows(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% arrange(desc(PC1)) %>% select(Function) %>% mutate(Direction = "PC1-", Samples = "CMC"))

pca_EG_cmc_biplot_pc2 <- plot_pca_bi(humann2_KOs.decontam_cmc_mr1.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = F)
pca_EG_cmc_biplot_pc2

pandoc.table(pca_EG_cmc_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc2$plot_env$neg_10  %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
  bind_rows(pca_EG_cmc_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)) %>% select(Function) %>% mutate(Direction = "PC2+", Samples = "CMC"))
humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
  bind_rows(pca_EG_cmc_biplot_pc2$plot_env$neg_10 %>% arrange(desc(PC2)) %>% select(Function) %>% mutate(Direction = "PC2-", Samples = "CMC"))

fwrite(humann2_KO_biplot_list, file = "./05-results.backup/humann2_KO_biplot_list.tsv", quote = F, sep = "\t")

```

```{r gf_info}

gf_info <- fread("./05-results.backup/gene_family_info.tsv")

gf_annotations <- humann2_KO_biplot_list %>%
  rename(`GeneFamily` = Function) %>%
  full_join(., gf_info %>%
              rename(`GeneFamily` = MetaCyc_reaction), by = "GeneFamily")

```















