---
title: "Cameroon hunter-gatherer calculus/plaque MALT beta diversity"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(psych)
library(ape)
library(mixOmics)
library(compositions)
library(vegan)
library(janitor)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

# MALT read stats
The number of reads assigned in each sample appears very low (~50%). We want to check this so we know how many were assinged in each sample. We used the cleaned and merged reads that didn't align to the human genome (hg19), with the following database: /projects1/malt/databases/indexed/index040/full-nt_2017-10. 
```{r load_data}

# load the taxonomy tables that have been decontaminated and the column headers are cleaned up
load("./05-results.backup/CMC_taxonomy_tables.RData")

# load the metadata file
load("./05-results.backup/CMC_metadata.RData")

# list only the anterior samples
anterior <- metadata %>%
  filter(str_detect(SampleID, "CMC")) %>%
  filter(Tooth_site == "Anterior") %>%
  pull(SampleID)

# list only the posterior samples
posterior <- metadata %>%
  filter(str_detect(SampleID, "CMC")) %>%
  filter(Tooth_site == "Posterior") %>%
  pull(SampleID)

```


```{r pca_plotting_functions}
# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type, ExtractionBatch, LibraryBatch), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group, shape = metadata_group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type, ExtractionBatch, LibraryBatch), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$SampleID), size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}


# plot PCA with colored dots and the title including the # of species or genera
plot_pca_titles <- function(df, pc1, pc2, color_group, shape_group, size_group, ncomps) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))
    metadata_group_size <- get(paste(size_group, "_size", sep = ""))

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$explained_variance * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type, ExtractionBatch, LibraryBatch), by = "SampleID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]
    size_group = df_X[[size_group]]
   
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    title_text <- df %>%
                     rownames_to_column("SampleID") %>%
                     gather("Species","Counts",2:ncol(.)) %>%
                     spread(SampleID, Counts) %>%
                     count
    
    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group, size = size_group)) +
                        geom_point() +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        scale_size_manual(values = metadata_group_size) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        ggtitle(paste(title_text))

    return(pca_plot)
}

```

```{r pca_biplot_functions}
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$explained_variance * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }
    
    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(10, !!arrow_pc)

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(-10, !!arrow_pc)


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 10)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

# pca_test_biplot <- plot_pca_bi(malt_species.pca, "PC3", "PC2", "Village", PC2)
# pca_test_biplot


```


List the outliers from the plot in section `raw_plot` in `Cameroon_h-g_taxonomy.Rmd` (above) and remove them from the taxonomy table. 
```{r outliers}
outliers <- c("CMC002.B0201","CMC032.B0101","CMC037.B0201","CMC047.A0201",
              "EXB059.A1901","EXB059.A1301","EXB059.A1101","EXB059.A0901",
              "SRR1646041","SRR1646032","SRR1646042","SRR1646045","SRR1646030","SRR1646047")

outliersF <- str_c(outliers, collapse = "|")

```

#Sample Clustering with PCA

## Species-level analyses

What is the relationship of the groups in a PCA without the blanks?
```{r pca_noblanks_species}
# make a table that has only samples from one or the other tooth site
malt_species.decontam_filtered_noblanks <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|10M")) %>%
                                         select(-one_of(posterior)) %>% # change this between anterior and posterior, depending on which samples you want to remove
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_noblanks) <- malt_species.decontam_filtered_noblanks$SampleID

malt_species.decontam_filtered_noblanks <- malt_species.decontam_filtered_noblanks %>% select(-SampleID)

malt_species.decontam_filtered_noblanks = malt_species.decontam_filtered_noblanks+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered_noblanks, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Description (do HMP supragingival and HMP subgingival plaque samples separate?)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Description", "Description", "Description", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_species_decontam_filtered_noblanks.pca <- mixOmics::pca(malt_species.decontam_filtered_noblanks, ncomp = 3, logratio = 'CLR')
plot(malt_species_decontam_filtered_noblanks.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_species_decontam_filtered_noblanks.pca.variates <- as.data.frame(malt_species_decontam_filtered_noblanks.pca$variates$X)
malt_species_decontam_filtered_noblanks.pca.variates <- malt_species_decontam_filtered_noblanks.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_species_decontam_filtered_noblanks.pca.varmet <- inner_join(malt_species_decontam_filtered_noblanks.pca.variates, metadata, by = "SampleID")

pca_malt_species_pc1pc2 <- ggplot(malt_species_decontam_filtered_noblanks.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[2], 2)), "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_species_pc1pc2

# ggsave("05-results.backup/pca_nt_species_pc1pc2.pdf", plot = pca_nt_species_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)

```
_Posterior samples only:_ The CMC posterior samples tend to separate out on PC1
from HMP and JAE samples. Plaque and calculus samples separate along PC2, which
is similar to the difference reported in Velsko, *et al.* 2019 *Microbiome*. The
CMC samples fall between the calculus and buccal swabs, but fall in line with
plaque, suggesting they are more similar to plaque than to calculus. Although
the Yanomami samples appear to fall within the variation of the HMP samples in
PC1/CP2 figures, they are clearly separated from all plaque/calculus in PC3. The
CMC samples do not plot with the Yanomami samples.


_Anterior samples only:_ The CM anterior samples are spread across the plot, but
generally plot closer to the JAE samples than the posterior samples do, and plot
between the HMP and JAE samples, rather than clustering away from both like the
posterior samples. Although the Yanomami samples appear to fall within the
variation of the HMP samples in PC1/CP2 figures, they are clearly separated from
all plaque/calculus in PC3. The CMC samples do not plot with the Yanomami
samples.

### Biplots
Now we will look at the species that have strongest loadings in PC1, PC2, and
PC3, to see what appears to drive differences between the sample groups. We will
visualize the loadings of these species with biplots. In the first set we
include the Yanomami samples, and in the second set we exclude them.
```{r biplot_noblanks_species}

malt_species_decontam_filtered_biplot_121 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_species_decontam_filtered_biplot_121

pandoc.table(malt_species_decontam_filtered_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_species_decontam_filtered_biplot_122 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_biplot_122

pandoc.table(malt_species_decontam_filtered_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_species_decontam_filtered_biplot_322 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_biplot_322

malt_species_decontam_filtered_biplot_323 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_species_decontam_filtered_biplot_323

pandoc.table(malt_species_decontam_filtered_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")


```
_Posterior samples only:_ There are some clear biological differences in the taxa with highest loadings in
each component, positive and negative. In PC1, the negative loading taxa are all
anaerobic, while the positive loading taxa are *Neisseria*, *Streptococcus*, and
*Rothia*, wich are aerobic. The HMP samples that fall farther to the right are
mostly supragingival plaque, which is consistent with the aerobic species
driving separation this direction. Supragingival plaque samples have higher
levels of aerobes than subgingival plaque samples. All Yanomami buccal swab
samples have negative PC2 values, and it makes sense that *Neisseria* and other
aerobic taxa characterize this direction, as they are more abundant on buccal
epithelia. In PC2, the positive loading taxa are anaerobic, later colonizers,
which corresponds with the calculus samples that plot in positice PC2 values.
The PC2 negative loading taxa are *Prevotella*, *Veillonella*, and *Haemophilus*
(which points towards the Yanomami samples). In PC3, the negative loading taxa
are mostly *Rothia*, with one *Actinomyces* & one *Actinobaculum*, while the
positive loading taxa are mostly anaerobic.

_Anterior samples only:_ n PC1, the negative loading taxa are all anaerobic,
while the positive loading taxa are *Neisseria*, *Streptococcus*, and *Rothia*,
wich are aerobic/facultative. The negative loading taxa are anaerobic. This
distinction doesn't clearly match any of the group plotting areas. In PC2
positive loadings, the taxa are mostly *Prevotella* and *Veillonella*, while in
the negative loadings the taxa are mostly anaerobes - this distinction matches
with the split of plaqe in PC2 positive values and calclus in PC2 negative
values. In PC3, the positivce loading taxa are mostly *Rothia* and
*Actinomyces*, while the negative values are mostly *Neisseria*. This
corresponds with the Yanomami buccal samples plotting in PC3 negative values, as
buccal samples have higher proportions of *Neisseria* than plaque.


```{r biplot_noblanks_noyanomami_species}
# remove the Yanomami samples from the input table
malt_species.decontam_filtered_noblanks_noyanomami <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|10M|SRR164")) %>%
                                         select(-one_of(posterior)) %>% # change this between anterior and posterior, depending on which samples you want to remove
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_noblanks_noyanomami) <- malt_species.decontam_filtered_noblanks_noyanomami$SampleID

malt_species.decontam_filtered_noblanks_noyanomami <- malt_species.decontam_filtered_noblanks_noyanomami %>% select(-SampleID)

malt_species.decontam_filtered_noblanks_noyanomami = malt_species.decontam_filtered_noblanks_noyanomami+1

# perform PCA
malt_species.decontam_filtered_noblanks_noyanomami.pca <- mixOmics::pca(malt_species.decontam_filtered_noblanks_noyanomami, ncomp = 3, logratio = 'CLR')

# First plot by Description (do HMP supragingival and HMP subgingival plaque samples separate?) to see what it looks like before adding the biplot over the points
plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC3", "PC2", "Description", "Description", "Description", 3)

# Then plot by Tooth_site (do anterior teeth and posterior teeth samples separate?) to see what it looks like before adding the biplot over the points
plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# Now plot biplots, and make tables of the top 10 species in each loading
malt_species_decontam_filtered_noyanomami_biplot_121 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_species_decontam_filtered_noyanomami_biplot_121

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_species_decontam_filtered_noyanomami_biplot_122 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_noyanomami_biplot_122

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_species_decontam_filtered_noyanomami_biplot_322 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_noyanomami_biplot_322

malt_species_decontam_filtered_noyanomami_biplot_323 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_species_decontam_filtered_noyanomami_biplot_323

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
_Posterior samples only:_  When we look only at samples from one tooth surface, we still see some biological
differences between the taxa in the different PCs with positive and negative
loadings. In PC1, the positive loading taxa are *Niesseria*, *Rothia*, and *Strep*,
suggesting these samples are less mature biofilms, or are exposed to higher
oxygen tension. The negative loading taxa are a mix of anaerobic species,
suggesting they are more mature or exposed to lower oxygen tension. The CMC posterior
samples separate from HMP along PC1, where HMP is in positive
loadings and CMC posterior is to the left (negative loadings), which corresponds with
the species loading in these directions. The HMP samples also separate this way,
with the supragingival plaque samples plotting farther to the right (more
positive) and subgingival plaque samples plotting more to the left (less
positive), which also corresponds with the species that load in positive and
negative values.

In PC2 positive loadings there is a mix of facultative anaerobes and anaerobic
taxa, while the negative loading taxa are mostly facultative anaerobes
(*Prevotella*, *Streptococcus*). This distinction corresponds to the separation
of plaque in positive values and calculus in negative values. The CMC posterior
samples plot in both positive and negative values in PC2, but mostly in PC3
negative values. PC3 positive values are characterized by a variety of species,
and PC3 negative values mostly by *Neisseria*.

_Anterior samples only:_ In PC1 positive values, the taxa with strongest
loadings are 9 *Rothio* and 1 *Streptococcus*, and the arrows point mostly up
into the PC1 positive/PC2 positive quadrant, where the HMP samples plot. The
taxa with strongest PC1 negative loading values are anaerobic. The taxa with
strongest loadings in PC2 positive values are mostly *Prevotella*, which point
into the PC1 negative/PC2 positive quadrant, away from most HMP samples, and
*Veillonella*, which point into the PC1 positive/PC2 positive quadrant, towards
the HMP samples. The PC2 negative loading taxa are mostly anaerobes. In PC3, the
taxa with strongest positive loadings are a combination of aerobes, facultative
anaerobes, and anaerobes, while the taxa with strongest negative loading values
are 7 *Neisseria* and the rest anaerobes.


```{r metadata_betadisper_noblanks}

metadata_noblanks <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata_noblanks <- metadata_noblanks %>%
  rename(SampleID = `#SampleID`)  %>%
  filter(!SampleID %in% outliers) %>%
  filter(!str_detect(SampleID, "SRR16460|EXB|LIB|10M")) %>%
  filter(!SampleID %in% posterior) %>% # this is the group to remove
  as_tibble() %>%
  mutate(Market_economy = str_replace(Market_economy, "HMP","Industrial"))%>%
  mutate(Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime","HMP","JAE"),
         Env = fct_relevel(Env, "HunterGatherer","HMP","JAE"),
         Industry = fct_relevel(Industry, "Non-industrial","Industrial"),
         Market_economy = fct_relevel(Market_economy, "Logging road","Forrest camp","Farming","Industrial"),
         Age_group = fct_relevel(Age_group, "18-29","30-39","40-49","50-59","60+","HMP"),
         Sex = fct_relevel(Sex, "Male","Female"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior","HMP"),
         Village = fct_relevel(Village, "202","205","204","203","206","HMP","JAE")) %>%
  select(SampleID, Age_group, Ethnic_Group, Env, Industry, Market_economy, Sex, Tooth_site, Village)

```

Run a permanova with function adonis to test if the clusters are significantly different.
```{r adonis_noblanks_noyanomami_species}
# use the CLR-transformed matrix from the mixOmics PCA in the last section

# Euclidean distance
malt_species.decontam_filtered_noblanks_noyanomami.euc <- vegdist(malt_species.decontam_filtered_noblanks_noyanomami.pca$X, method = "euclidean", na.rm = T)
malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoa <- ape::pcoa(malt_species.decontam_filtered_noblanks_noyanomami.euc)

malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoavectors <- as.data.frame(malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoa$vectors)
malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoavectors <- malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_species.decontam_filtered_noblanks_noyanomami.euc.vecmet <- inner_join(malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoavectors, metadata, by = "SampleID")

malt_species.decontam_filtered_noblanks_noyanomami.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Industry, shape = Industry)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Industry_colors) +
  scale_shape_manual(values = Industry_shapes) +
  stat_ellipse(aes(colour = Industry, group = Industry)) +
  theme_minimal(base_size = 7) +
  ggtitle("Industry")

# test for difference between different metadata categories 
# Sex can't be run b/c for JAE012 sex is unknown, and is NA
adonis(malt_species.decontam_filtered_noblanks_noyanomami.euc ~ Industry + Ethnic_Group + Age_group + Market_economy + Village, metadata_noblanks, perm=999)

```


Then run betadisper to see if the groups with significant differences in
clustering have different dispersion ("tight" vs. "loose" clustering of groups)
- is one group more variable than the other(s)?
```{r betadisper_noblanks_noyanomami_species}
# Test the metadata categories that are significant (p < 0.05) based on adonis for significant differences in dispersion
# When only including posterior samples, Industry and Ethnic group are significantly different
# When only including anterior samples, Industry and Ethnic group are significantly different

# Industry
groups_industry <- metadata_noblanks %>%
  select(Industry) %>% 
  pull()

## Calculate multivariate dispersions
industry_betadisper_noblanks_noyanomami <- betadisper(malt_species.decontam_filtered_noblanks_noyanomami.euc, groups_industry, bias.adjust =TRUE)
industry_betadisper_noblanks_noyanomami

## Perform test
anova(industry_betadisper_noblanks_noyanomami)

## Permutation test for F
permutest(industry_betadisper_noblanks_noyanomami, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
industry_betadisper_noblanks_noyanomami.HSD <- TukeyHSD(industry_betadisper_noblanks_noyanomami)
plot(industry_betadisper_noblanks_noyanomami.HSD)

## Plot with data ellipses instead of hulls
plot(industry_betadisper_noblanks_noyanomami, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(industry_betadisper_noblanks_noyanomami)


# Ethnic group
groups_ethnicity <- metadata_noblanks %>%
  select(Ethnic_Group) %>% 
  pull()

## Calculate multivariate dispersions
ethnicity_betadisper_noblanks_noyanomami <- betadisper(malt_species.decontam_filtered_noblanks_noyanomami.euc, groups_ethnicity, bias.adjust =TRUE)
ethnicity_betadisper_noblanks_noyanomami

## Perform test
anova(ethnicity_betadisper_noblanks_noyanomami)

## Permutation test for F
permutest(ethnicity_betadisper_noblanks_noyanomami, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
ethnicity_betadisper_noblanks_noyanomami.HSD <- TukeyHSD(ethnicity_betadisper_noblanks_noyanomami)
plot(ethnicity_betadisper_noblanks_noyanomami.HSD)

## Plot with data ellipses instead of hulls
plot(ethnicity_betadisper_noblanks_noyanomami, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(ethnicity_betadisper_noblanks_noyanomami)


```



What is the relationship of Cameroon plaque samples to eachother without any
other sample types? Do they cluster by any of the metadata categories?
```{r pca_cmc_species}
malt_species.decontam_filtered_cmc <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|SRR|JAE")) %>%
                                         select(-one_of(posterior)) %>% # change this between anterior and posterior, depending on which samples you want to remove
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_cmc) <- malt_species.decontam_filtered_cmc$SampleID

malt_species.decontam_filtered_cmc <- malt_species.decontam_filtered_cmc %>% select(-SampleID)

malt_species.decontam_filtered_cmc = malt_species.decontam_filtered_cmc+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered_cmc, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Extraction batch
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Library batch
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_species.decontam_filtered_cmc.pca <- mixOmics::pca(malt_species.decontam_filtered_cmc, ncomp = 3, logratio = 'CLR')
plot(malt_species.decontam_filtered_cmc.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_species.decontam_filtered_cmc.pca.variates <- as.data.frame(malt_species.decontam_filtered_cmc.pca$variates$X)
malt_species.decontam_filtered_cmc.pca.variates <- malt_species.decontam_filtered_cmc.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_species.decontam_filtered_cmc.pca.varmet <- inner_join(malt_species.decontam_filtered_cmc.pca.variates, metadata, by = "SampleID")

pca_malt_species_pc1pc2 <- ggplot(malt_species.decontam_filtered_cmc.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_species.decontam_filtered_cmc.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_species.decontam_filtered_cmc.pca$explained_variance[2], 2))< "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_species_pc1pc2

# ggsave("05-results.backup/pca_nt_species_pc1pc2.pdf", plot = pca_nt_species_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)
```
There is no obvious clustering of the CMC samples by village, sex, ethnicity,
age group, or market economy for either the posterior samples only or the
anterior samples only. Clustering also doesn't seem to be related to sequencing
depth, extraction batch, or library batch.

```{r metadata_betadisper_cmc}
# Need special metadata that has only the categories of CMC samples, b/c of issues w/ extra factors from removed groups
metadata_cmc <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata_cmc <- metadata_cmc %>%
  rename(SampleID = `#SampleID`) %>%
  filter(!SampleID %in% outliers) %>%
  filter(!SampleID %in% posterior) %>%
  filter(Description == "HunterGatherer") %>%
  select(SampleID, Age_group, Ethnic_Group, Market_economy, Sex, Tooth_site, Village) %>%
  as_tibble() %>%
  mutate(Age_group = fct_relevel(Age_group, "18-29","30-39","40-49","50-59","60+"),
         Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime"),
         Market_economy = fct_relevel(Market_economy, "Logging road","Forrest camp","Farming"),
         Sex = fct_relevel(Sex, "Male","Female"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior"),
         Village = fct_relevel(Village, "202","205","204","203","206"))
```


Run a permanova with function adonis to test if the clusters are significantly different.
```{r adonis_cmc_species}
# use the CLR-transformed matrix from the mixOmics PCA in the last section

# Euclidean distance
malt_species.decontam_filtered_cmc.euc <- vegdist(malt_species.decontam_filtered_cmc.pca$X, method = "euclidean", na.rm = T)
malt_species.decontam_filtered_cmc.euc.pcoa <- ape::pcoa(malt_species.decontam_filtered_cmc.euc)

malt_species.decontam_filtered_cmc.euc.pcoavectors <- as.data.frame(malt_species.decontam_filtered_cmc.euc.pcoa$vectors)
malt_species.decontam_filtered_cmc.euc.pcoavectors <- malt_species.decontam_filtered_cmc.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_species.decontam_filtered_cmc.euc.vecmet <- inner_join(malt_species.decontam_filtered_cmc.euc.pcoavectors, metadata, by = "SampleID")

malt_species.decontam_filtered_cmc.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis(malt_species.decontam_filtered_cmc.euc ~ Ethnic_Group + Market_economy + Sex + Village + Age_group, metadata_cmc, perm=999)

```

Then run betadisper to see if the groups with significant differences in
clustering have different dispersion ("tight" vs. "loose" clustering of groups)
- is one group more variable than the other(s)?
```{r betadisper_cmc_species}
# Test the 4 metadata categories that are significant (p < 0.05) based on adonis for significant differences in dispersion
# When including only posterior sampels, none of the groups are significantly different
# When including only anterior samples, Market economy and sex are significantly different

# Market economy
groups_market_economy <- metadata_cmc %>%
  select(Market_economy) %>% 
  pull()

## Calculate multivariate dispersions
market_economy_betadisper_cmc <- betadisper(malt_species.decontam_filtered_cmc.euc, groups_market_economy, bias.adjust =TRUE)
market_economy_betadisper_cmc

## Perform test
anova(market_economy_betadisper_cmc)

## Permutation test for F
permutest(market_economy_betadisper_cmc, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
market_economy_betadisper_cmc.HSD <- TukeyHSD(market_economy_betadisper_cmc)
plot(market_economy_betadisper_cmc.HSD)

## Plot with data ellipses instead of hulls
plot(market_economy_betadisper_cmc, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(market_economy_betadisper_cmc)


# Sex
groups_sex <- metadata_cmc %>%
  select(Sex) %>% 
  pull()

## Calculate multivariate dispersions
sex_betadisper_cmc <- betadisper(malt_species.decontam_filtered_cmc.euc, groups_sex, bias.adjust =TRUE)
sex_betadisper_cmc

## Perform test
anova(sex_betadisper_cmc)

## Permutation test for F
permutest(sex_betadisper_cmc, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
sex_betadisper_cmc.HSD <- TukeyHSD(sex_betadisper_cmc)
plot(sex_betadisper_cmc.HSD)

## Plot with data ellipses instead of hulls
plot(sex_betadisper_cmc, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(sex_betadisper_cmc)



```


```{r biplots_cmc_species}

malt_species.decontam_filtered_nocontrols <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|SRR|JAE")) %>%
                                         select(-one_of(posterior)) %>% # change this between anterior and posterior, depending on which samples you want to remove
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_nocontrols) <- malt_species.decontam_filtered_nocontrols$SampleID

malt_species.decontam_filtered_nocontrols <- malt_species.decontam_filtered_nocontrols %>% select(-SampleID)

malt_species.decontam_filtered_nocontrols = malt_species.decontam_filtered_nocontrols+1

malt_species.decontam_filtered_nocontrols.pca <- mixOmics::pca(malt_species.decontam_filtered_nocontrols, ncomp = 3, logratio = 'CLR')
plot(malt_species.decontam_filtered_nocontrols.pca)

malt_species.decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_species.decontam_filtered_nocontrols_biplot_121

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_species.decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_species.decontam_filtered_nocontrols_biplot_122

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_species.decontam_filtered_nocontrols_biplot_322 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_species.decontam_filtered_nocontrols_biplot_322

malt_species.decontam_filtered_nocontrols_biplot_323 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_species.decontam_filtered_nocontrols_biplot_323

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
_Posterior samples only:_ In PC1, the positive loading taxa are mostly anaerobic
taxa, while the negative loading taxa are a mix of aerobic, facultative, and
anaerobic taxa. In PC2, the positive loading taxa are largely aerobic, 8
*Neisseria*, while the negative loading taxa are mostly anaerobic. In PC3, the
positive loading taxa are mostly anaerobic taxa, while in negatice PC3 values
mostly facultative *Actinomyces*, along with some odd ones
(*Kytococcus*,*Cellulomonas*).

_Anterior samples only:_ In PC1, the taxa with strongest positive loadings are
mostly *Prevotella* and other anaerobes, and the taxa with strongest negative
loadings are mostly aerobes including *Neisseria* and *Rothia*. In PC2, the taxa
with strongest positive loadings are all *Rothia*, while the taxa with strongest
negative loadings are all anaerobes. In PC3, the taxa with the strongest
positive loadings are a variety of early colonizer aerobes/facultative anaerobes
and odd species (*Xylanimonas*, *Sanguibacter*), while the taxa with strongest
negative loadings are a variety of facultative anaerobes and anaerobes.


## Genus-level analyses

What is the relationship of the groups in a PCA without the blanks? Are
relationships similar to those observed using species data? Is there better
resolution with genus data?
```{r pca_noblanks_genus}
malt_genus.decontam_filtered <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|10M")) %>%
                                         select(-one_of(posterior)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.01) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered) <- malt_genus.decontam_filtered$SampleID

malt_genus.decontam_filtered <- malt_genus.decontam_filtered %>% select(-SampleID)

malt_genus.decontam_filtered = malt_genus.decontam_filtered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_genus.decontam_filtered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Description (do HMP supragingial and HMP subgingival plaque separate?)
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Description", "Description", "Description", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_genus.decontam_filtered.pca <- mixOmics::pca(malt_genus.decontam_filtered, ncomp = 3, logratio = 'CLR')
plot(malt_genus.decontam_filtered.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_genus.decontam_filtered.pca.variates <- as.data.frame(malt_genus.decontam_filtered.pca$variates$X)
malt_genus.decontam_filtered.pca.variates <- malt_genus.decontam_filtered.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_genus.decontam_filtered.pca.varmet <- inner_join(malt_genus.decontam_filtered.pca.variates, metadata, by = "SampleID")

pca_malt_genus_pc1pc2 <- ggplot(malt_genus.decontam_filtered.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_genus.decontam_filtered.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_genus.decontam_filtered.pca$explained_variance[2], 2)), "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_genus_pc1pc2


# ggsave("05-results.backup/pca_malt_genus_pc1pc2.pdf", plot = pca_malt_genus_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)


```
_Posterior samples only:_  Just like we saw in the species PCA plots, the CMC
posterior samples separate out from HMP along PC1, and from JAE along PC2.
Several CMC samples though do plot with HMP, or close to JAE. The Yanomami
samples separate out from the other group along PC3. Plaque and calculus samples
still separate along PC2.

_Anterior samples only:_ Similar to the plots using species assignments, the CMC
anterior samples plot more with JAE than with HMP. It's more of a distinction
here, where the CMC samples fall within the variation of the JAE sampels along
PC1, but only fall within the variation of subgingival plaque of the HMP
samples. Here the HMP supragingival plaque cluster farther away in PC1 negative
values. The CMC samples fall within the variation of both HMP and JAE in PC2. In
PC3, the Yanomami samples separate out from HMP along PC2.


### Biplots
Now we will look at the genera that have strongest loadings in PC1, PC2, and
PC3, to see what appears to drive differences between the sample groups. We will
visualize the loadings of these species with biplots. In the first set we
include the Yanomami samples, and in the second set we exclude them. Do the
genera with strongest loadings include the species with strongest loadings in
sections `biplots_samples_species` and `biplots_samples_species_no_yanomami`
above?
```{r biplot_noblanks_genus}

malt_genus.decontam_filtered_biplot_121 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_genus.decontam_filtered_biplot_121

pandoc.table(malt_genus.decontam_filtered_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")



malt_genus.decontam_filtered_biplot_122 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_genus.decontam_filtered_biplot_122

pandoc.table(malt_genus.decontam_filtered_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_genus.decontam_filtered_biplot_322 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_genus.decontam_filtered_biplot_322

malt_genus.decontam_filtered_biplot_323 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_genus.decontam_filtered_biplot_323

pandoc.table(malt_genus.decontam_filtered_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
_Posterior samples only:_ In PC1, the positive loading taxa are mostly
anaerobic, while the negative loading taxa are a mix of aerobic and anaerobic.
Negative value genera include *Neisseria* and relatives *Eikenella*,
*Haemophilus*, *Kingella*, and *Aggregatibacter*. In PC2, the postive loading
taxa are a mix of later colonizer anaerobes, while the negative loading taxa are
a mix of aerobes and anaerobes. This corresponds with the separation of plaque
in positive PC2 values (less mature community) and calculus on negative values
(more mature community). In PC3, the positive loading taxa are mostly later
colonizing anaerobes, while the negative loading taxa are unusual, rare, or
poorly characterized in the oral biofilm. The Yanomami samples again separate
from JAE and CMC along PC3, but separate from HMP on PC2.

_Anterior samples only:_ In PC1, the taxa with the strongest positive loadings
are mostly strict anaerobes, while the taxa with strongest negative loadings are
a mix of aerobes and facultative anaerobes, including *Neisseria* and relatives
*Eikenella*, *Haemophilus*, *Kingella*, and *Aggregatibacter*. In PC2, the taxa
with strongest positive loadings are split, mostly pointing either strongly into
the PC1 negative quadrant (anaerobes/unusual oral) or the PC1 positive quadrant
(anaerobes). In PC3, the taxa with the strongest postive loadings are anaerobes,
but also genera that are less common in the mouth, as are the taxa with
strongest negative loadings. The Yanomami samples again separate from JAE and
CMC along PC3, but separate from HMP on PC2.


```{r biplot_samples_no_yanomami_genus}

# remove the Yanomami samples from the input table
malt_genus.decontam_filtered_noblanks_noyanomami <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|10M|SRR164")) %>%
                                         select(-one_of(posterior)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered_noblanks_noyanomami) <- malt_genus.decontam_filtered_noblanks_noyanomami$SampleID

malt_genus.decontam_filtered_noblanks_noyanomami <- malt_genus.decontam_filtered_noblanks_noyanomami %>% select(-SampleID)

malt_genus.decontam_filtered_noblanks_noyanomami = malt_genus.decontam_filtered_noblanks_noyanomami+1

# perform PCA
malt_genus.decontam_filtered_noblanks_noyanomami.pca <- mixOmics::pca(malt_genus.decontam_filtered_noblanks_noyanomami, ncomp = 3, logratio = 'CLR')

# First plot by Description (do HMP supragingival and HMP subgingival plaque samples separate?) to see what it looks like before adding the biplot over the points
plot_pca_titles(malt_genus.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_genus.decontam_filtered_noblanks_noyanomami, "PC3", "PC2", "Description", "Description", "Description", 3)

# Then plot by Tooth_site (do anterior teeth and posterior teeth samples separate?) to see what it looks like before adding the biplot over the points
plot_pca_titles(malt_genus.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_noblanks_noyanomami, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)


# Now plot biplots, and make tables of the top 10 species in each loading
malt_genus_decontam_filtered_noyanomami_biplot_121 <- plot_pca_bi(malt_genus.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_genus_decontam_filtered_noyanomami_biplot_121

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_genus_decontam_filtered_noyanomami_biplot_122 <- plot_pca_bi(malt_genus.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_genus_decontam_filtered_noyanomami_biplot_122

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_genus_decontam_filtered_noyanomami_biplot_322 <- plot_pca_bi(malt_genus.decontam_filtered_noblanks_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_genus_decontam_filtered_noyanomami_biplot_322

malt_genus_decontam_filtered_noyanomami_biplot_323 <- plot_pca_bi(malt_genus.decontam_filtered_noblanks_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_genus_decontam_filtered_noyanomami_biplot_323

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
_Posterior samples only:_ When excluding Yanomami samples, the patterns of taxa
characterizing sample group separation are still less clear at the genus level
than they are at the species level. In PC1, the negative loading taxa are mostly
later colonizing anaerobes, while the positive loading taxa are more aerobic
(*Haemophilus, Kingella, Rothia, Neisseria, Eikenella*) and facultative
anaerobic genera. The taxa characterizing this direction corresond with the HMP
plaque samples, which would have a less mature biofilm community, mostly
plotting in positive PC1 values. The HMP samples furtheset to the left are
subgingival plaque, while the ones to the right are supragingival plaque. The
CMC posterior samples tend to plot in PC1 negative values. This also corresponds
with the genera loading in negative values as more aerobic and genera loading in
positive values more anaerotolerant.

In PC2, the positive loading taxa are a mix of aerobes and anaerobes. The
negative loading taxa are more anaerobes, but also some odd, not typiclly oral,
ones. In PC3, the positive loading taxa are mostly later colonizer, anaerobes,
while the negative loading taxa are unusual, rare, or poorly characterized in
the oral biofilm. In PC3 the CMC posterior samples again separate out from HMP.

_Anterior samples only:_ In PC1, the taxa with the strongest positive loadings
are mostly anaerobes, and the taxa with the strongest negative loadings are more
aerobes/facultative (*Haemophilus, Kingella, Rothia, Neisseria, Eikenella*).
Most of the arrows in PC1 negative point into the PC2 negative quadrant. In PC2,
the taxa with strongest positive loadings are mostly anaerobes with arrows
directed away from HMP, but *Veillonella* and *Actinobaculum* point towards the
HMP samples. The taxa with strongest PC2 negative loadings are mostly anaerobes.
In PC3, the taxa with the strongest positive loadings are mostly anaerobes,
while the taxa with strongest negative loadings are unusual, rare, or poorly
characterized in the oral biofilm.


Run a permanova with function adonis to test if the clusters are significantly different.
```{r adonis_noblanks_noyanomami_genus}
# use the CLR-transformed matrix from the mixOmics PCA in the last section

# Euclidean distance
malt_genus.decontam_filtered_noblanks_noyanomami.euc <- vegdist(malt_genus.decontam_filtered_noblanks_noyanomami.pca$X, method = "euclidean", na.rm = T)
malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoa <- ape::pcoa(malt_genus.decontam_filtered_noblanks_noyanomami.euc)

malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoavectors <- as.data.frame(malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoa$vectors)
malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoavectors <- malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_genus.decontam_filtered_noblanks_noyanomami.euc.vecmet <- inner_join(malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoavectors, metadata, by = "SampleID")

malt_genus.decontam_filtered_noblanks_noyanomami.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Industry, shape = Industry)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Industry_colors) +
  scale_shape_manual(values = Industry_shapes) +
  stat_ellipse(aes(colour = Industry, group = Industry)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
# Sex can't be run b/c for JAE012 sex is unknown, and is NA
adonis(malt_genus.decontam_filtered_noblanks_noyanomami.euc ~ Industry + Ethnic_Group + Age_group + Market_economy + Village, metadata_noblanks, perm=999)

```

Then run betadisper to see if the groups have different dispersion ("tight" vs. "loose" clustering of groups) - is one group more variable than the other(s)?
```{r betadisper_noblanks_noyanomami_genus}
# Test the 2 metadata categories that are significant (p < 0.05) based on adonis for significant differences in dispersion
# When including only posterior sampels, Industry and Ethnic Group significantly different
# When including only anterior samples, Industry, Ethnic Group, and Market economy are significantly different


# Industry
groups_industry <- metadata_noblanks %>%
  select(Industry) %>% 
  pull()

## Calculate multivariate dispersions
industry_betadisper_noblanks_noyanomami_genus <- betadisper(malt_genus.decontam_filtered_noblanks_noyanomami.euc, groups_industry, bias.adjust =TRUE)
industry_betadisper_noblanks_noyanomami_genus

## Perform test
anova(industry_betadisper_noblanks_noyanomami_genus)

## Permutation test for F
permutest(industry_betadisper_noblanks_noyanomami_genus, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
industry_betadisper_noblanks_noyanomami_genus.HSD <- TukeyHSD(industry_betadisper_noblanks_noyanomami_genus)
plot(industry_betadisper_noblanks_noyanomami_genus.HSD)

## Plot with data ellipses instead of hulls
plot(industry_betadisper_noblanks_noyanomami_genus, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(industry_betadisper_noblanks_noyanomami_genus)


# Ethnic group
groups_ethnicity <- metadata_noblanks %>%
  select(Ethnic_Group) %>% 
  pull()

## Calculate multivariate dispersions
ethnicity_betadisper_noblanks_noyanomami_genus <- betadisper(malt_genus.decontam_filtered_noblanks_noyanomami.euc, groups_ethnicity, bias.adjust =TRUE)
ethnicity_betadisper_noblanks_noyanomami_genus

## Perform test
anova(ethnicity_betadisper_noblanks_noyanomami_genus)

## Permutation test for F
permutest(ethnicity_betadisper_noblanks_noyanomami_genus, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
ethnicity_betadisper_noblanks_noyanomami_genus.HSD <- TukeyHSD(ethnicity_betadisper_noblanks_noyanomami_genus)
plot(ethnicity_betadisper_noblanks_noyanomami_genus.HSD)

## Plot with data ellipses instead of hulls
plot(ethnicity_betadisper_noblanks_noyanomami_genus, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(ethnicity_betadisper_noblanks_noyanomami_genus)


# Market economy
groups_marketecon <- metadata_noblanks %>%
  select(Market_economy) %>% 
  pull()

## Calculate multivariate dispersions
marketecon_betadisper_noblanks_noyanomami_genus <- betadisper(malt_genus.decontam_filtered_noblanks_noyanomami.euc, groups_marketecon, bias.adjust =TRUE)
marketecon_betadisper_noblanks_noyanomami_genus

## Perform test
anova(marketecon_betadisper_noblanks_noyanomami_genus)

## Permutation test for F
permutest(marketecon_betadisper_noblanks_noyanomami_genus, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
marketecon_betadisper_noblanks_noyanomami_genus.HSD <- TukeyHSD(marketecon_betadisper_noblanks_noyanomami_genus)
plot(marketecon_betadisper_noblanks_noyanomami_genus.HSD)

## Plot with data ellipses instead of hulls
plot(marketecon_betadisper_noblanks_noyanomami_genus, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(marketecon_betadisper_noblanks_noyanomami_genus)


```



What is the relationship of Cameroon plaque samples to eachother without any
other sample types? Do they cluster by any of the metadata categories?
```{r pca_cmc_genus}
malt_genus.decontam_filtered_cmc <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|SRR|JAE")) %>%
                                         select(-one_of(posterior)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered_cmc) <- malt_genus.decontam_filtered_cmc$SampleID

malt_genus.decontam_filtered_cmc <- malt_genus.decontam_filtered_cmc %>% select(-SampleID)

malt_genus.decontam_filtered_cmc = malt_genus.decontam_filtered_cmc+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_genus.decontam_filtered_cmc, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Extraction batch
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Library batch
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_genus.decontam_filtered_cmc.pca <- mixOmics::pca(malt_genus.decontam_filtered_cmc, ncomp = 3, logratio = 'CLR')
plot(malt_genus.decontam_filtered_cmc.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_genus.decontam_filtered_cmc.pca.variates <- as.data.frame(malt_genus.decontam_filtered_cmc.pca$variates$X)
malt_genus.decontam_filtered_cmc.pca.variates <- malt_genus.decontam_filtered_cmc.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_genus.decontam_filtered_cmc.pca.varmet <- inner_join(malt_genus.decontam_filtered_cmc.pca.variates, metadata, by = "SampleID")

pca_malt_genus_pc1pc2 <- ggplot(malt_genus.decontam_filtered_cmc.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_genus.decontam_filtered_cmc.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_genus.decontam_filtered_cmc.pca$explained_variance[2], 2)), "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_genus_pc1pc2


# ggsave("05-results.backup/pca_malt_genus_pc1pc2.pdf", plot = pca_malt_genus_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)
```
There is no obvious clustering of the CMC samples by village, sex, ethnicity,
age group, or market economy for either the posterior samples only or the
anterior samples only. Clustering also doesn't seem to be related to sequencing
depth, extraction batch, or library batch.



Run a permanova with function adonis to test if the clusters are significantly different.
```{r adonis_cmc_genus}
# use the CLR-transformed matrix from the mixOmics PCA in the last section

# Euclidean distance
malt_genus.decontam_filtered_cmc.euc <- vegdist(malt_genus.decontam_filtered_cmc.pca$X, method = "euclidean", na.rm = T)
malt_genus.decontam_filtered_cmc.euc.pcoa <- ape::pcoa(malt_genus.decontam_filtered_cmc.euc)

malt_genus.decontam_filtered_cmc.euc.pcoavectors <- as.data.frame(malt_genus.decontam_filtered_cmc.euc.pcoa$vectors)
malt_genus.decontam_filtered_cmc.euc.pcoavectors <- malt_genus.decontam_filtered_cmc.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_genus.decontam_filtered_cmc.euc.vecmet <- inner_join(malt_genus.decontam_filtered_cmc.euc.pcoavectors, metadata, by = "SampleID")

malt_genus.decontam_filtered_cmc.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis(malt_genus.decontam_filtered_cmc.euc ~ Ethnic_Group + Age_group + Sex + Village + Market_economy, metadata_cmc, perm=999)

```

Then run betadisper to see if the groups have different dispersion ("tight" vs.
"loose" clustering of groups) - is one group more variable than the other(s)?
```{r betadisper_cmc_genus}
# Test the 3 metadata categories that are significant (p < 0.05) based on adonis for significant differences in dispersion
# When including only posterior sampels, none of the groups are significantly different
# When including only anterior samples, Village is significantly different

# Village
groups_village <- metadata_cmc %>%
  select(Village) %>% 
  pull()

## Calculate multivariate dispersions
village_betadisper_cmc_genus <- betadisper(malt_genus.decontam_filtered_cmc.euc, groups_village, bias.adjust =TRUE)
village_betadisper_cmc_genus

## Perform test
anova(village_betadisper_cmc_genus)

## Permutation test for F
permutest(village_betadisper_cmc_genus, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
village_betadisper_cmc_genus.HSD <- TukeyHSD(village_betadisper_cmc_genus)
plot(village_betadisper_cmc_genus.HSD)

## Plot with data ellipses instead of hulls
plot(village_betadisper_cmc_genus, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(village_betadisper_cmc_genus)



```



```{r biplots_cmc_genus}

malt_genus.decontam_filtered_nocontrols <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|JAE|SRR")) %>%
                                         select(-one_of(posterior)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered_nocontrols) <- malt_genus.decontam_filtered_nocontrols$SampleID

malt_genus.decontam_filtered_nocontrols <- malt_genus.decontam_filtered_nocontrols %>% select(-SampleID)

malt_genus.decontam_filtered_nocontrols = malt_genus.decontam_filtered_nocontrols+1

malt_genus.decontam_filtered_nocontrols.pca <- mixOmics::pca(malt_genus.decontam_filtered_nocontrols, ncomp = 3, logratio = 'CLR')
plot(malt_genus.decontam_filtered_nocontrols.pca)

malt_genus_decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_genus_decontam_filtered_nocontrols_biplot_121

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")



malt_genus_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_genus_decontam_filtered_nocontrols_biplot_122

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_genus_decontam_filtered_nocontrols_biplot_322 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_genus_decontam_filtered_nocontrols_biplot_322

malt_genus_decontam_filtered_nocontrols_biplot_323 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_genus_decontam_filtered_nocontrols_biplot_323

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
_Posterior samples only:_ In PC1, the positive loading taxa are more aerobes and
facultative anaerobes, while the negative loading taxa are anaerobes. In PC2,
the positive loading taxa are mostly aerobes and facultative anaerobes, while
the  negative loading taxa are more anaerobes. In PC3, the positive loading taxa
are mostly late colonizing anaerobes, while the negative loading taxa are a mix
of genera that are unusual, rare, or poorly characterized in the oral biofilm.

_Anterior samples only:_ In PC1, the positive loading taxa are more unusual or
poorly characterized in the oral biofilm, and anaerobes, while the negative
loading taxa are a mix of anaerobes and aerobes. In PC2, the taxa with strongest
positive loadings are more anerobes (*Haemophilus, Kingella, Rothia, Neisseria,
Eikenella*), while the taxa with strongest negative loadings are more anaerobes.
In PC3, the taxa with strongest positive loadings are these unusual taxa again
(*Xylanimonas*, *Sanguibacter*, *Cellulomonas*, *Kytococus*) but also some
expected oral taxa, while the taxa with the strongest negative loadings are a
variety of anaerobes and other unusual oral taxa.



```{r uky_graphs, eval = F}

all_village <- plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backup/all_village_uky.pdf", plot = all_village, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

all_village_32 <- plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backup/all_village_32_uky.pdf", plot = all_village_32, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

noy_desc <- plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Description", "Description", "Description", 3)
ggsave("05-results.backup/noy_desc_uky.pdf", plot = noy_desc, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

noy_eg <- plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backup/noy_eg_uky.pdf", plot = noy_eg, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

noy_ts<- plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backup/noy_ts_uky.pdf", plot = noy_ts, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

malt_species_decontam_filtered_noyanomami_biplot_121 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Tooth_site", PC1)
malt_species_decontam_filtered_noyanomami_biplot_121
ggsave("05-results.backup/malt_species_decontam_filtered_noyanomami_biplot_121_uky.pdf", plot = malt_species_decontam_filtered_noyanomami_biplot_121, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

malt_species_decontam_filtered_noyanomami_biplot_122 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Tooth_site", PC2)
malt_species_decontam_filtered_noyanomami_biplot_122
ggsave("05-results.backup/malt_species_decontam_filtered_noyanomami_biplot_122_uky.pdf", plot = malt_species_decontam_filtered_noyanomami_biplot_122, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

cmc_village <- plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backupcmc_village_uky.pdf", plot = cmc_village, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

cmc_ts <- plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backup/cmc_ts_uky.pdf", plot = cmc_ts, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

malt_species.decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Tooth_site", PC1)
malt_species.decontam_filtered_nocontrols_biplot_121
ggsave("05-results.backup/malt_species.decontam_filtered_nocontrols_biplot_121_uky.pdf", plot = malt_species.decontam_filtered_nocontrols_biplot_121, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)


```




