---
title: "Cameroon hunter-gatherer calculus/plaque HUMAnN2"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F}
library(knitr)
library(decontam)
library(data.table)
library(mixOmics)
library(janitor)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

```{r colors_shapes_peachpear}
# lacroix_palette("PeachPear", n = 8, type = "continuous")
# "#FF3200" "#EF8158" "#E9BD8E" "#CBDDA7" "#38BCAD" "#0B91B5" "#065FA3" "#172869"

Village_colors <- c(`202` = "#FF3200", `204` = "#EF8158", `205` = "#E9BD8E", `203` = "#CBDDA7", `206` = "#38BCAD",
                    HMP = "#172869", HMP10M = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                     ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Village_shapes <- c(`202` = 15, `203` = 16, `204` = 17,`205` = 18, `206` = 25, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9,
                    ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Ethnic_Group_colors <- c(Baka = "#FF3200", Nzime = "#172869", HMP = "#969696", HMP10M = "#d9d9d9", JAE10M = "#ffeda0", JAE = "#feb24c",
                         ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Ethnic_Group_shapes <- c(Baka = 19, Nzime = 17,  HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9,
                         ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Env_colors <- c(HunterGatherer = "#FF3200", HMP = "#172869", HMP10M = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Env_shapes <- c(HunterGatherer = 19, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Market_economy_colors <- c(logging_road = "#FF3200", forrest_camp = "#172869", farming = "#82CDAA", Industrial = "#969696",
                           ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Market_economy_shapes <- c(logging_road = 19, forrest_camp = 17, farming = 18, HMP = 7, HMP10M = 9, Industrial = 15,
                           ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Age_group_colors <- c(`18-29` = "#FF3200", `30-39` = "#E9B186", `40-49` = "#82CDAA", `50-59` = "#0686B8", `60+` = "#172869",
                      HMP = "#969696", HMP10M = "#d9d9d9",
                      ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Age_group_shapes <- c(`18-29` = 15, `30-39` = 16, `40-49` = 17, `50-59` = 18, `60+` = 25, HMP = 7, HMP10M = 9,
                      ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Sex_colors <- c(Female = "#FF3200", Male = "#172869", HMP = "#969696", HMP10M = "#d9d9d9",
                ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Sex_shapes <- c(Female = 19, Male = 17,  HMP = 7, HMP10M = 9,ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Tooth_site_colors <- c(Anterior = "#FF3200", Posterior = "#172869", HMP = "#969696", HMP10M = "#d9d9d9",
                       ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Tooth_site_shapes <- c(Anterior = 19, Posterior = 17,  HMP = 7, HMP10M = 9,
                       ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

```

# HUMAnN2 assignment stats
*This section needs to be checked and re-written*The number of reads assigned in each sample appears very low (~50%). We want to check this so we know how many were assinged in each sample. We used the cleaned and merged reads that didn't align to the human genome (hg19), with the following database: /projects1/malt/databases/indexed/index040/full-nt_2017-10. 
```{r load_data}
## load the species and genus tables generated in MEGAN
humann2_path <- fread("./05-results.backup/humann2.pathabundance.all.tss.tsv")
humann2_path <- as_tibble(humann2_path)
humann2_KOs_full <- fread("./05-results.backup/humann2.genefamilies.all.i.tss.renorm.ko.tsv")
humann2_KOs_full <- as_tibble(humann2_KOs_full)

# clean the file names
humann2_KOs_full <- rename(humann2_KOs_full, Ortholog = `# Gene Family`)
colnames(humann2_KOs_full) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M_Abundance-RPKs","_10M", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","_10M", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","", colnames(humann2_KOs_full))

humann2_path <- rename(humann2_path, Pathway = `# Pathway`)
colnames(humann2_path) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M_Abundance","_10M", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","_10M", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","", colnames(humann2_path))

# remove unmapped and ungrouped reads
humann2_KOs <- humann2_KOs_full %>% filter(Ortholog != "UNMAPPED") %>% filter(!str_detect(Ortholog, "UNGROUPED"))

```

```{r load_metadata}

metadata <- fread("./00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata <- metadata %>%
  rename(SampleID = `#SampleID`) %>%
  as_tibble() %>%
  mutate(Description = fct_relevel(Description, "HunterGatherer", "HMP_supPlaque", "HMP_subPlaque",
                                   "HMP10M_supPlaque", "HMP10M_subPlaque","JAE","JAE10M",
                                   "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime","HMP","HMP10M","JAE","JAE10M",
                                     "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Market_economy = fct_relevel(Market_economy, "logging_road","forrest_camp","farming","Industrial",
                                      "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Age_group = fct_relevel(Age_group, "18-29","30-39","40-49","50-59","60+","HMP","HMP10M"),
         Sex = fct_relevel(Sex, "Male","Female","HMP","HMP10M"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior","HMP","HMP10M"),
         Type = fct_relevel(Type, "Calculus","Plaque","Blank"),
         Village = fct_relevel(Village, "202","204","205","203","206","ExtractionGauze","ExtractionBlank",
                               "LibraryBlank","HMP","HMP10M","JAE","JAE10M"))

# list out the full SRR (HMP) and JAE samples to separate them from the set subsampled to 10M reads
fullSRR <- metadata %>%
  select(SampleID) %>%
  filter(str_detect(SampleID, "SRR")) %>%
  filter(!str_detect(SampleID, "10M")) %>%
  pull

fullJAE <- metadata %>%
  select(SampleID) %>%
  filter(str_detect(SampleID, "JAE")) %>%
  filter(!str_detect(SampleID, "10M")) %>%
  pull

```

Before we proceed with looking at differences in abundance of different enzymes,
let's compare the how many reads were identified by HUMAnN2 for each sample
type. With the first step we'll read in the default HUMAnN2 output table with
all UniRef90 assignments. We'll select out just the rows with no species
assignemtns so we can calculate total assignments. HUMAnN2 uses *reads per
kilobase* for the initial output, so we'll be calculating the percent of total
assigned/mapped reads per kilobase (RPK). Then write out the file with just the
unmapped and total reads assigned so we never have to load this huge file again.
This has to be done with a table for which the samples were renormed
individually and then joined, and not renormed after they were joined (with
HUMAnN2 scripts).

```{r functions}

# add functions for:
# making the input table for PCA?

# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
                      
                      # select the correct PC column and explained variance for PC1
                      if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                       }
    
                      # select the correct PC column and explained variance for PC2
                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group, shape = metadata_group)) +
                        geom_point(size = 3) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(colnames(pc1), " - ", exp_var_pc1)) +
                        ylab(paste(colnames(pc2), " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                      # select the correct PC column and explained variance for PC1
                      if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                       }
    
                      # select the correct PC column and explained variance for PC2
                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$SampleID), size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(colnames(pc1), " - ", exp_var_pc1)) +
                        ylab(paste(colnames(pc2), " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}

# this function plots the PCA with biplots to look at the categories with strongest loadings
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle, pathways = c("TRUE","FALSE")) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$explained_variance * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }

    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "flag") %>%
      top_n(10, !!arrow_pc)
   if (pathways == T) {
      pws_10 <- pws_10 %>%
        rename(pathway = flag) %>%
        mutate(flag = sapply(pathway, function(f) {
                                  unlist(str_split(f, ":"))[1]
                           })) 
    }

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "flag") %>%
      top_n(-10, !!arrow_pc)
    if (pathways == T) {
      neg_10 <- neg_10 %>%
      rename(pathway = flag) %>%
       mutate(flag = sapply(pathway, function(f) {
                                  unlist(str_split(f, ":"))[1]
                           })) 
    }


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = flag),
                   size = 1.5, colour = "grey20", label.padding = 0.2) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = flag),
                   size = 1.5, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 10)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

# pca_test_biplot <- plot_pca_bi(humann2_path.pca, "PC1", "PC2", "Village", PC1, pathways = T)
# pca_test_biplot

```

Plot the total number of assigned reads in each sample group, and the
proportion of assigned reads in each group.
```{r}

```

# Pathway analysis
Start with a PCA to see if there are any outlier samples that should be removed,
other than CMC032.B0101, which has very few reads compared to the other samples
and will be removed before starting any analyses.
```{r pca_paths_with_controls}
# create the input matrix
humann2_path_mat <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Pathway, CPM)

rownames(humann2_path_mat) <- humann2_path_mat$SampleID
humann2_path_mat <- as.matrix(humann2_path_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path_allsamples.pca <- pca(humann2_path_mat, ncomp = 4, logratio = 'CLR')

# plot the PCA
pca_test_plot <- plot_pca(humann2_path_allsamples.pca, "PC1", "PC2", "Ethnic_Group")
pca_test_plot

# pca_test_plot_names <- plot_pca_names(humann2_path_allsamples.pca, "PC1", "PC3", "Ethnic_Group")
# pca_test_plot_names

```

Four of the gauze extraction blanks look like they are oral, and probably had
saliva or plaque that wasn't visible on the control strip that was cut. Now
repeat the PCA without any of the controls to compare the samples to known oral
plaque and calculus profiles. If the sub-sampled plaque (HMP) and calculus (JAE)
samples are similar (10M), then use just the full samples and not the
sub-sampled ones for all further analysis.

```{r pca_paths_no_blanks}

# create the input matrix
humann2_path_mat <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Pathway, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB"))

rownames(humann2_path_mat) <- humann2_path_mat$SampleID
humann2_path_mat <- as.matrix(humann2_path_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path.pca <- pca(humann2_path_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_path.pca, "PC1", "PC2", "Ethnic_Group")
pca_EG_plot

```
There is more overlap in functional profiles of CMC and JAE samples than with
HMP. The sub-sampled HMP and JAE samples are plotting in the same space as their
full samples, so we'll continue with the full samples.


```{r pca_paths_metadata}
# create the input matrix
humann2_path_mat <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Pathway, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M"))

rownames(humann2_path_mat) <- humann2_path_mat$SampleID
humann2_path_mat <- as.matrix(humann2_path_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path.pca <- pca(humann2_path_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_path.pca, "PC1", "PC2", "Ethnic_Group")
pca_EG_plot

pca_AG_plot <- plot_pca(humann2_path.pca, "PC1", "PC2", "Age_group")
pca_AG_plot

pca_ME_plot <- plot_pca(humann2_path.pca, "PC1", "PC2", "Market_economy")
pca_ME_plot

pca_S_plot <- plot_pca(humann2_path.pca, "PC1", "PC2", "Sex")
pca_S_plot

pca_TS_plot <- plot_pca(humann2_path.pca, "PC1", "PC2", "Tooth_site")
pca_TS_plot

pca_V_plot <- plot_pca(humann2_path.pca, "PC1", "PC2", "Village")
pca_V_plot

```
The plaque (HMP) and calculus (JAE) samples separate along PC2, but the CMC
samples have more overlap with JAE than HMP. There is much more spread of the
samples along PC1. Do any other metadata categories appear to drive separation
of the CMC samples along PC1?

Only the tooth site (anterior vs posterior) appears to be separating along PC1.
Let's also run these without the HMP and JAE samples to see how the CMC samples
cluster without industrialized samples included.

```{r pca_paths_cmc_metadata}
humann2_path_mat <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Pathway, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR|JAE"))

rownames(humann2_path_mat) <- humann2_path_mat$SampleID
humann2_path_mat <- as.matrix(humann2_path_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path_cmc.pca <- pca(humann2_path_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_path_cmc.pca, "PC1", "PC2", "Ethnic_Group")
pca_EG_plot

pca_AG_plot <- plot_pca(humann2_path_cmc.pca, "PC1", "PC2", "Age_group")
pca_AG_plot

pca_ME_plot <- plot_pca(humann2_path_cmc.pca, "PC1", "PC2", "Market_economy")
pca_ME_plot

pca_S_plot <- plot_pca(humann2_path_cmc.pca, "PC1", "PC2", "Sex")
pca_S_plot

pca_TS_plot <- plot_pca(humann2_path_cmc.pca, "PC1", "PC2", "Tooth_site")
pca_TS_plot

pca_TS_plot <- plot_pca(humann2_path_cmc.pca, "PC3", "PC2", "Tooth_site")
pca_TS_plot

pca_V_plot <- plot_pca(humann2_path_cmc.pca, "PC1", "PC2", "Village")
pca_V_plot


```
Again the only sort-of clustering that we see is by tooth site (anterior or
posterior). Now let's look at which pathways are driving separation of the
sample groups. 


**We should also run adonis to test for sample separation by the
different metadata categories to say if sample clustering is significant.**
```{r}

```


Now to see which of the pathways have the strongest loadings in PC1 and PC2
```{r pca_path_biplot}
# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc1 <- plot_pca_bi(humann2_path.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = T)
pca_EG_biplot_pc1

# what are those pathways?
pandoc.table(pca_EG_biplot_pc1$plot_env$pws_10 %>% select(-flag) %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-flag) %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc2 <- plot_pca_bi(humann2_path.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = T)
pca_EG_biplot_pc2

# what are those pathways?
pandoc.table(pca_EG_biplot_pc2$plot_env$pws_10 %>% select(-flag) %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_biplot_pc2$plot_env$neg_10 %>% select(-flag) %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

# which paths distinguish sample groups when only CMC samples are included?
pca_EG_cmc_biplot_pc1 <- plot_pca_bi(humann2_path_cmc.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = T)
pca_EG_cmc_biplot_pc1

# what are those pathways?
pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% select(-flag) %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% select(-flag) %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

pca_EG_cmc_biplot_pc2 <- plot_pca_bi(humann2_path_cmc.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = T)
pca_EG_cmc_biplot_pc2

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% select(-flag) %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% select(-flag) %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

```


```{r outliers, eval=F}
path_outliers <- c("CMC032.B0101")
```


# Community-level functional analysis with individual gene families assigned KEGG Orthology numbers
## KEGG ortholog distribution
We will start analysis of KEGG orthologs by looking at the total number of KOs assigned to each population to see if this may bias further analyses. Is the proportion of genefamilies that were re-grouped by HUMAnN2 script into KOs  consistent and high across all sample types? That would be good b/c we have no major group biases. 
```{r, eval = F}


```

Now perform a PCA to see if there is more distinct clustering of the samples based on any of the metadata categories than we see with the pathways. 
```{r pca_KOs_with_controls}
# create the input matrix
humann2_KO_mat <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Ortholog, CPM)

rownames(humann2_KO_mat) <- humann2_KO_mat$SampleID
humann2_KO_mat <- as.matrix(humann2_KO_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_KO_pca <- tune.pca(humann2_KO_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KO_allsamples.pca <- pca(humann2_KO_mat, ncomp = 4, logratio = 'CLR')

# plot the PCA
pca_test_plot <- plot_pca(humann2_KO_allsamples.pca, "PC1", "PC2", "Ethnic_Group")
pca_test_plot

# pca_test_plot_names <- plot_pca_names(humann2_KO_allsamples.pca, "PC1", "PC3", "Ethnic_Group")
# pca_test_plot_names

```


```{r pca_KOs_no_blanks}

# create the input matrix
humann2_KO_mat <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Ortholog, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB"))

rownames(humann2_KO_mat) <- humann2_KO_mat$SampleID
humann2_KO_mat <- as.matrix(humann2_KO_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_KO_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KO_noblanks.pca <- pca(humann2_KO_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Ethnic_Group")
pca_EG_plot

```


```{r pca_KOs_metadata}
# create the input matrix
humann2_KO_mat <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Ortholog, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M"))
# %>%
#   adorn_totals(where = "row", name = "Sum") %>%
#   gather("Ortholog","CPM", 2:ncol(.)) %>%
#   spread(SampleID, CPM) %>%
#   mutate(Percent = Sum/sum(Sum)) %>%
#   filter(Percent > 0.0005) %>%
#   gather("SampleID","CPM", 2:ncol(.)) %>%
#   spread(Ortholog, CPM)
# 
rownames(humann2_KO_mat) <- humann2_KO_mat$SampleID
humann2_KO_mat <- as.matrix(humann2_KO_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_KO_pca <- tune.pca(humann2_KO_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KO_noblanks.pca <- pca(humann2_KO_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Ethnic_Group")
pca_EG_plot

pca_AG_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Age_group")
pca_AG_plot

pca_ME_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Market_economy")
pca_ME_plot

pca_S_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Sex")
pca_S_plot

pca_TS_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Tooth_site")
pca_TS_plot

pca_V_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Village")
pca_V_plot

```


```{r pca_KOs_cmc_metadata}
# create the input matrix
humann2_KO_mat <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Ortholog, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR|JAE")) 
# %>% 
#   adorn_totals(where = "row", name = "Sum") %>%
#   gather("Ortholog","CPM", 2:ncol(.)) %>%
#   spread(SampleID, CPM) %>%
#   mutate(Percent = Sum/sum(Sum)) %>%
#   filter(Percent > 0.0005) %>%
#   gather("SampleID","CPM", 2:ncol(.)) %>%
#   spread(Ortholog, CPM)


rownames(humann2_KO_mat) <- humann2_KO_mat$SampleID
humann2_KO_mat <- as.matrix(humann2_KO_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_KO_pca <- tune.pca(humann2_KO_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KO_cmc.pca <- pca(humann2_KO_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_KO_cmc.pca, "PC1", "PC2", "Ethnic_Group")
pca_EG_plot

pca_AG_plot <- plot_pca(humann2_KO_cmc.pca, "PC1", "PC2", "Age_group")
pca_AG_plot

pca_ME_plot <- plot_pca(humann2_KO_cmc.pca, "PC1", "PC2", "Market_economy")
pca_ME_plot

pca_S_plot <- plot_pca(humann2_KO_cmc.pca, "PC1", "PC2", "Sex")
pca_S_plot

pca_TS_plot <- plot_pca(humann2_KO_cmc.pca, "PC1", "PC2", "Tooth_site")
pca_TS_plot

pca_V_plot <- plot_pca(humann2_KO_cmc.pca, "PC1", "PC2", "Village")
pca_V_plot

```


Now to see which of the orthologs have the strongest loadings in PC1 and PC2
```{r pca_KO_biplot}
# which paths distinguish sample types when HMP and JAE are included?
pca_EG_KO_biplot_pc1 <- plot_pca_bi(humann2_KO_noblanks.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = F)
pca_EG_KO_biplot_pc1

# what are those pathways?
pandoc.table(pca_EG_KO_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_KO_biplot_pc1$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc2 <- plot_pca_bi(humann2_KO_noblanks.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = F)
pca_EG_biplot_pc2

# what are those pathways?
pandoc.table(pca_EG_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_biplot_pc2$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

# which paths distinguish sample groups when only CMC samples are included?
pca_EG_cmc_biplot_pc1 <- plot_pca_bi(humann2_KO_cmc.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = F)
pca_EG_cmc_biplot_pc1


ggsave("~/Desktop/readflags.pdf", plot = pca_EG_cmc_biplot_pc1, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
        dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

pca_EG_cmc_biplot_pc2 <- plot_pca_bi(humann2_KO_cmc.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = F)
pca_EG_cmc_biplot_pc2

pandoc.table(pca_EG_cmc_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc2$plot_env$neg_10  %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

```

```{r}

```







