---
title: "Cameroon plaque stats analysis - Kraken RefSeq database w/ & w/o Pasolli MAGs"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---

```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(janitor)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

Load data files 
```{r load_data}
rso_readstats <- fread("./05-results.backup/readclassification_stats.rso.tsv")
rspm_readstats <- fread("./05-results.backup/readclassification_stats.rspm.tsv")

# shorten sample names
rso_readstats$SampleID <- gsub(".SG1.1_S0_L001_R1_001","", rso_readstats$SampleID)
rso_readstats$SampleID <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M.fastq.gz","_10M", rso_readstats$SampleID)
rso_readstats$SampleID <- gsub("_S0_L001_R1_000.10M","_10M", rso_readstats$SampleID)
rso_readstats$SampleID <- gsub("_S0_L001_R1_000","", rso_readstats$SampleID)
rso_readstats$SampleID <- gsub("_S0_L001_R1_001","", rso_readstats$SampleID)
rso_readstats$SampleID <- gsub(".fastq.truncated.prefixed.gz","", rso_readstats$SampleID)
rso_readstats$SampleID <- gsub(".kraken","", rso_readstats$SampleID)

rspm_readstats$SampleID <- gsub(".SG1.1_S0_L001_R1_001","", rspm_readstats$SampleID)
rspm_readstats$SampleID <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M.fastq.gz","_10M", rspm_readstats$SampleID)
rspm_readstats$SampleID <- gsub("_S0_L001_R1_000.10M","_10M", rspm_readstats$SampleID)
rspm_readstats$SampleID <- gsub("_S0_L001_R1_000","", rspm_readstats$SampleID)
rspm_readstats$SampleID <- gsub("_S0_L001_R1_001","", rspm_readstats$SampleID)
rspm_readstats$SampleID <- gsub(".fastq.truncated.prefixed.gz","", rspm_readstats$SampleID)
rspm_readstats$SampleID <- gsub(".kraken","", rspm_readstats$SampleID)

# # these are for the mpa-format output runs
# rso_readstats$SampleID <- gsub(".A0101.output",".A0101_10M.output", rso_readstats$SampleID)
# rso_readstats$SampleID <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.fastq.gz","", rso_readstats$SampleID)
# rso_readstats$SampleID <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam.fastq.gz","_10M", rso_readstats$SampleID)
# rso_readstats$SampleID <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam.fastq.gz","", rso_readstats$SampleID)
# rso_readstats$SampleID <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.fastq.gz","", rso_readstats$SampleID)
# rso_readstats$SampleID <- gsub(".kraken","", rso_readstats$SampleID)

# rspm_readstats$SampleID <- gsub(".A0101.output",".A0101_10M.output", rspm_readstats$SampleID)
# rspm_readstats$SampleID <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.fastq.gz","", rspm_readstats$SampleID)
# rspm_readstats$SampleID <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam.fastq.gz","_10M", rspm_readstats$SampleID)
# rspm_readstats$SampleID <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam.fastq.gz","", rspm_readstats$SampleID)
# rspm_readstats$SampleID <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.fastq.gz","", rspm_readstats$SampleID)
# rspm_readstats$SampleID <- gsub(".kraken","", rspm_readstats$SampleID)
# rspm_readstats$SampleID <- gsub(".SG1.1_S0_L001_R1_001","", rspm_readstats$SampleID)

# split the names into new columns
rso_readstats <- rso_readstats %>% 
  separate(col = "SampleID", into = c("SampleID","Run"), sep = "\\.output\\.") %>%
  separate(col = "Run", into =c("Run","Database"), sep = "\\.") %>%
  as_tibble(.) %>%
  mutate(Database = recode(Database, rso = "RefSeqOnly"))

rspm_readstats <- rspm_readstats %>% 
  separate(col = "SampleID", into = c("SampleID","Run"), sep = "\\.output\\.") %>%
  mutate(Database = "RefSeqPasolliMAGs") %>%
  select(SampleID,Run,Database,Classified,Unclassified) %>%
  as_tibble(.)
rspm_readstats$Run <- gsub(".rspm","", rspm_readstats$Run)


# merge the tables into 1
kraken_readstats <- bind_rows(rso_readstats,rspm_readstats)

# load the metadata file
load("./05-results.backup/CMC_metadata.RData")

Kraken_db_colors

# merge data and metadata
kraken_readstats <- kraken_readstats %>%
  mutate(Percent_classified = Classified/(Classified+Unclassified)*100) %>%
  inner_join(., metadata %>%
               select(SampleID,Env,Description,Ethnic_Group,Market_economy,Age_group,Sex,Tooth_site,Type,Village))

```


```{r plot_data}
kraken_readstats %>%
  # filter(!str_detect(SampleID, "EXB|LIB|10M")) %>%
  ggplot(., aes(x = Village, y = Percent_classified, fill = Database)) +
    geom_boxplot() +
    geom_jitter(size = 0.5, alpha = 0.5) +
    scale_fill_manual(values = Kraken_db_colors) +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    xlab("Village") +
    ylab("Percent of reads classified")

RSPM_kraken_compare <- kraken_readstats %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M")) %>%
  ggplot(., aes(x = Village, y = Percent_classified, fill = Database)) +
    geom_boxplot() +
    geom_jitter(size = 0.5, alpha = 0.5) +
    scale_fill_manual(values = Kraken_db_colors) +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
    xlab("Village") +
    ylab("Percent of reads classified")

RSPM_kraken_compare

ggsave("05-results.backup/RSPM_kraken_compare_UKY.pdf", plot = RSPM_kraken_compare, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

```

```{r stats}

wilcox_p <- kraken_readstats %>%
group_by(Village) %>%
summarise(pvalue = pairwise.wilcox.test(Classified, Database, p.adjust.method = "fdr")$p.value)

wilcox_p %>% filter(pvalue <= 0.05)

```

