---
title: "Cameroon plaque taxonomy analysis - Kraken RefSeq database w/ & w/o Pasolli MAGs clean up file"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---

```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(janitor)
library(taxize)
library(myTAI)
library(plyr)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```


```{r load_standard_report_data, eval = F}
# run this only once and then in the future read in the files that were saved, in the section below

# load the standard format output tables
rso_report_list <- list.files(path = "./04-analysis/kraken/output/RefSeqOnly/", pattern = ".+report.+", full.names = T)
rsmp_report_list <- list.files(path = "./04-analysis/kraken/output/RefSeqPasolliMAGs/", pattern = ".+report.+", full.names = T)

rso_reports <- map_dfr(rso_report_list, function(fn) {
  fread(fn, col.names = c("Percent", "Number_rooted", "Number_direct","Rank","NCBItaxID","SciName"), sep = "\t") %>%
  mutate(SampleID = basename(fn))
})

# mpa_short_list <- head(list.files(path = "./04-analysis/kraken/output/RefSeqPasolliMAGs/", pattern = ".+report.+", full.names = T), n = 3L)

#rso_reports$SciName <- gsub(" ","", rso_reports$SciName)
rso_reports$SampleID <- gsub(".SG1.1_S0_L001_R1_001","", rso_reports$SampleID)
rso_reports$SampleID <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M.fastq.gz","_10M", rso_reports$SampleID)
rso_reports$SampleID <- gsub(".fastq.truncated.prefixed.gz","", rso_reports$SampleID)
rso_reports$SampleID <- gsub("_S0_L001_R1_000.10M","_10M", rso_reports$SampleID)
rso_reports$SampleID <- gsub("_S0_L001_R1_000","", rso_reports$SampleID)
rso_reports$SampleID <- gsub("_S0_L001_R1_001","", rso_reports$SampleID)

# fwrite(rso_reports, file = "./00-documentation.backup/kraken_standard_rso_reports.tsv", quote = F, sep = "\t")

rso_reports %>% filter(str_detect(Rank, "U|R|D"))


rsmp_reports <- map_dfr(rsmp_report_list, function(fn) {
  fread(fn, col.names = c("Percent", "Number_rooted", "Number_direct","Rank","NCBItaxID","SciName"), sep = "\t") %>%
  mutate(SampleID = basename(fn))
})

#rsmp_reports$SciName <- gsub(" ","", rsmp_reports$SciName)
rsmp_reports$SampleID <- gsub(".SG1.1_S0_L001_R1_001","", rsmp_reports$SampleID)
rsmp_reports$SampleID <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M.fastq.gz","_10M", rsmp_reports$SampleID)
rsmp_reports$SampleID <- gsub(".fastq.truncated.prefixed.gz","", rsmp_reports$SampleID)
rsmp_reports$SampleID <- gsub("_S0_L001_R1_000.10M","_10M", rsmp_reports$SampleID)
rsmp_reports$SampleID <- gsub("_S0_L001_R1_000","", rsmp_reports$SampleID)
rsmp_reports$SampleID <- gsub("_S0_L001_R1_001","", rsmp_reports$SampleID)

# fwrite(rsmp_reports, file = "./00-documentation.backup/kraken_standard_rsmp_reports.tsv", quote = F, sep = "\t")

```


```{r load_raw_data_table, eval = F}
# run this once then after load the RData files this creates

# this takes several minutes, they're large files (~500MB)
rso_reports <- fread("./00-documentation.backup/over100MB/kraken_standard_rso_reports.tsv")
rso_reports$SampleID <- gsub(".kraken","", rso_reports$SampleID)

rsmp_reports <- fread("./00-documentation.backup/over100MB/kraken_standard_rsmp_reports.tsv")
rsmp_reports$SampleID <- gsub(".kraken","", rsmp_reports$SampleID)

rso_reports <- rso_reports %>% as_tibble() %>%
  separate(col = "SampleID", into = c("SampleID","Run"), sep = "\\.report\\.") %>%
  separate(col = "Run", into =c("Run","Database"), sep = "\\.")


rsmp_reports <- rsmp_reports %>% as_tibble() %>%
  separate(col = "SampleID", into = c("SampleID","Run"), sep = "\\.report\\.") %>%
  separate(col = "Run", into =c("Run","Database"), sep = "\\.")


# save(rso_reports, rsmp_reports, file = "./05-results.backup/over100MB/kraken_refseq_reports.RData")
# save(rso_reports, file = "./05-results.backup/kraken_refseqO_reports.RData")
# save(rsmp_reports, file = "./05-results.backup/kraken_refseqPM_reports.RData")
# save(rso_reports, rsmp_reports, file = "~/Downloads/kraken_refseq_reports.RData")

```

```{r load_data}

# load("./05-results.backup/over100MB/kraken_refseq_reports.RData")
load("./05-results.backup/kraken_refseqO_reports.RData")
load("./05-results.backup/kraken_refseqPM_reports.RData")

# this has the taxonomy list from the mpa-style reports
taxonomy_list_raw <- fread("./00-documentation.backup/taxonomy.list.uniq.split")
# make sure to remove all trailing spaces at the ends of the taxonomy, and add root| to the begining of each
taxonomy_list <- taxonomy_list_raw %>%
  mutate(Taxonomy = str_squish(Taxonomy),
         Root = "root|") %>%
  select(SciName, Root, Taxonomy) %>%
  unite("TaxonomyFull", Root:Taxonomy) %>%
  rename(Taxonomy = TaxonomyFull)

```

This will put the full taxonomy onto the Kraken report files by the species
names. The text needs to be manipulated for this to work, and still not all of
them will have full taxonomy assigned because of differences in the text.
Following this, export a list of the NCBI taxIDs that don't have the full
taxonomy added, and this will need to be gotten with taxize in a chunk below,
followed by hand-checking. Also export a list of the full taxonomy with the
NCBItaxID, for easier adding to the report files later along with the taxonomy
that had to be retrieved with taxize.
```{r taxonomy_na, eval = F}

# add the full taxonomy series in taxonomy_list to the samples by their species name
# this requires a lot of manipulation b/c of inconsistencies in naming
# there are still species that don't have a full taxonomy b/c of inconsistencies in naming
rso_reports_tax_mod <- rso_reports %>%
  mutate(SciNameFull = SciName) %>% # create a new column called SciNameFull by copying column SciName (keep this column untouched for later)
  rename(SciNameSplit = SciName) %>% # rename column SciName to SciNameSplit (this will be modified, split, and re-pasted together)
  mutate(SciNameSplit = str_replace(SciNameSplit, " sp\\. ", "_sp\\._"),
         SciNameSplit = str_replace(SciNameSplit, "oral taxon ", "oral_taxon_"),
         SciNameSplit = str_replace(SciNameSplit, "\\'", ""),
         SciNameSplit = str_replace(SciNameSplit, "\\'", ""),
         SciNameSplit = str_replace(SciNameSplit, "Candidatus ", "Candidatus_"),
         SciNameSplit = str_replace(SciNameSplit, " bacterium ", "_bacterium_"),
         SciNameSplit = str_replace(SciNameSplit, " phage ", "_phage_"),
         SciNameSplit = str_replace(SciNameSplit, "incertae sedis", ""),
         SciNameSplit = str_replace(SciNameSplit, " sensu stricto", ""),
         SciNameSplit = str_replace(SciNameSplit, " = ", "_=_"),
         SciNameSplit = str_replace(SciNameSplit, " genomosp\\. ", "_genomosp\\._"),
         SciNameSplit = str_replace(SciNameSplit, " archaeon ", "_archaeon_"),
         SciNameSplit = str_replace(SciNameSplit, " virus ", "_virus_"),
         SciNameSplit = str_replace(SciNameSplit, "axon 44", "axon44"),
         SciNameSplit = str_replace(SciNameSplit, "archaeon ", "archaeon_"),
         SciNameSplit = str_replace(SciNameSplit, "Clostridiales Family", "Clostridiales_Family"),
         SciNameSplit = str_replace(SciNameSplit, "Muribaculaceae ", "Muribaculaceae_"),
         SciNameSplit = str_replace(SciNameSplit, "Arthrobacter virus ", "Arthrobacter_virus_"),
         SciNameSplit = str_replace(SciNameSplit, "alpha proteobacterium", "alpha_proteobacterium"),
         SciNameSplit = str_replace(SciNameSplit, "beta proteobacterium", "beta_proteobacterium"),
         SciNameSplit = str_replace(SciNameSplit, "gamma proteobacterium", "gamma_proteobacterium"),
         SciNameSplit = str_replace(SciNameSplit, "delta proteobacterium", "delta_proteobacterium"),
         SciNameSplit = str_replace(SciNameSplit, "Bacillus virus", "Bacillus_virus"),
         SciNameSplit = str_replace(SciNameSplit, "Group II \\'CF-1\\'", "Group_II_\\'CF-1\\'"),
         SciNameSplit = str_replace(SciNameSplit, "[Scytonema hofmanni]_UTEX ", "[Scytonema_hofmanni]_UTEX_"),
         SciNameSplit = str_replace(SciNameSplit, "marine gamma proteobacterium", "marine_gamma_proteobacterium"),
         SciNameSplit = str_replace(SciNameSplit, "Paramecium bursaria Chlorella ", "Paramecium_bursaria_Chlorella_"),
         SciNameSplit = str_replace(SciNameSplit, "filamentous cyanobacterium CCP", "filamentous_cyanobacterium_CCP"),
         SciNameSplit = str_replace(SciNameSplit, "Legionella endosymbiont of Polyplax serrata", "Legionella_endosymbiont_of_Polyplax_serrata"),
         SciNameSplit = str_replace(SciNameSplit, "endosymbiont of unidentified scaly snail isolate Monju", "endosymbiont_of_unidentified_scaly_snail_isolate_Monju"),
         SciNameSplit = str_replace(SciNameSplit, "endosymbiont GvMRE of Glomus versiforme", "endosymbiont_GvMRE_of_Glomus_versiforme"),
         SciNameSplit = str_replace(SciNameSplit, "unclassified ", "")) %>%
  select(Percent, Number_rooted, Number_direct, Rank, NCBItaxID, SciNameFull, SampleID, Run, Database, SciNameSplit) %>%
  separate(col = "SciNameSplit", into = c("genus","species","strain"), sep = " ", convert = TRUE) %>% # SciNameSplit is now the last column. Split it
  select(-strain) %>% # remove the strains
  mutate(species = replace_na(species, "lll")) %>%
  unite(SciNameShort, genus, species, sep = " ") %>% # repaste genus and specis names from SciNameSplit into SciNameShort, then put the text back to the original
  mutate(SciNameShort = str_replace(SciNameShort, "_sp\\._", " sp\\. "),
         SciNameShort = str_replace(SciNameShort,  "oral_taxon_", "oral taxon "),
         SciNameShort = str_replace(SciNameShort,  "Candidatus_", "Candidatus "),
         SciNameShort = str_replace(SciNameShort,  "_bacterium_", " bacterium "),
         SciNameShort = str_replace(SciNameShort, "_=_", " = "),
         SciNameShort = str_replace(SciNameShort, "_genomosp\\._", " genomosp\\. "),
         SciNameShort = str_replace(SciNameShort, "_archaeon_", " archaeon "),
         SciNameShort = str_replace(SciNameShort, "archaeon_", "archaeon "),
         SciNameShort = str_replace(SciNameShort, "_virus_", " virus "),
         SciNameShort = str_replace(SciNameShort, "axon44", "axon 44"),
         SciNameShort = str_replace(SciNameShort, "taxon 44", "Taxon 44"),
         SciNameShort = str_replace(SciNameShort, "Clostridiales_Family", "Clostridiales Family"),
         SciNameShort = str_replace(SciNameShort, "Muribaculaceae_", "Muribaculaceae "),
         SciNameShort = str_replace(SciNameShort, "Arthrobacter_virus_", "Arthrobacter virus "),
         SciNameShort = str_replace(SciNameShort, "alpha_proteobacterium", "alpha proteobacterium"),
         SciNameShort = str_replace(SciNameShort, "beta_proteobacterium", "beta proteobacterium"),
         SciNameShort = str_replace(SciNameShort, "gamma_proteobacterium", "gamma proteobacterium"),
         SciNameShort = str_replace(SciNameShort, "delta_proteobacterium", "delta proteobacterium"),
         SciNameShort = str_replace(SciNameShort, "Bacillus_virus", "Bacillus virus"),
         SciNameShort = str_replace(SciNameShort, "Group_II_\\'CF-1\\'", "Group II \\'CF-1\\'"),
         SciNameShort = str_replace(SciNameShort, "[Scytonema_hofmanni]_UTEX_", "[Scytonema hofmanni]_UTEX "),
         SciNameShort = str_replace(SciNameShort, "marine_gamma_proteobacterium", "marine gamma proteobacterium"),
         SciNameShort = str_replace(SciNameShort, "Paramecium_bursaria_Chlorella_", "Paramecium bursaria Chlorella "),
         SciNameShort = str_replace(SciNameShort, "filamentous_cyanobacterium_CCP", "filamentous cyanobacterium CCP"),
         SciNameShort = str_replace(SciNameShort, "Legionella_endosymbiont_of_Polyplax_serrata", "Legionella endosymbiont of Polyplax serrata"),
         SciNameShort = str_replace(SciNameShort, "filamentous_cyanobacterium_CCP", "filamentous cyanobacterium CCP"),
         SciNameShort = str_replace(SciNameShort, "endosymbiont_of_unidentified_scaly_snail_isolate_Monju", "endosymbiont of unidentified scaly snail isolate Monju"),
         SciNameShort = str_replace(SciNameShort, "endosymbiont_GvMRE_of_Glomus_versiforme", "endosymbiont GvMRE of Glomus versiforme"),
         SciNameShort = str_replace(SciNameShort, "_phage_", " phage "),
         SciNameShort = str_replace(SciNameShort, " str\\.", ""),
         SciNameShort = str_replace(SciNameShort,  " lll", "")) %>%
  rename(SciName = SciNameShort)%>% # rename SciNameShort to SciName to match the column in taxonomy_list
  full_join(., taxonomy_list, by = "SciName") # add taxonomy to the table with taxonomy_list by merging the columns "SciName"


# add the full taxonomy series in taxonomy_list to the samples by their species name
# this requires a lot of manipulation b/c of inconsistencies in naming
# there are still species that don't have a full taxonomy b/c of inconsistencies in naming
rsmp_reports_tax_mod <- rsmp_reports %>%
  mutate(SciNameFull = SciName) %>%
  rename(SciNameSplit = SciName) %>%
  mutate(SciNameSplit = str_replace(SciNameSplit, " sp\\. ", "_sp\\._"),
         SciNameSplit = str_replace(SciNameSplit, "oral taxon ", "oral_taxon_"),
         SciNameSplit = str_replace(SciNameSplit, "\\'", ""),
         SciNameSplit = str_replace(SciNameSplit, "\\'", ""),
         SciNameSplit = str_replace(SciNameSplit, "Candidatus ", "Candidatus_"),
         SciNameSplit = str_replace(SciNameSplit, " bacterium ", "_bacterium_"),
         SciNameSplit = str_replace(SciNameSplit, " phage ", "_phage_"),
         SciNameSplit = str_replace(SciNameSplit, "incertae sedis", ""),
         SciNameSplit = str_replace(SciNameSplit, " sensu stricto", ""),
         SciNameSplit = str_replace(SciNameSplit, " = ", "_=_"),
         SciNameSplit = str_replace(SciNameSplit, " genomosp\\. ", "_genomosp\\._"),
         SciNameSplit = str_replace(SciNameSplit, " archaeon ", "_archaeon_"),
         SciNameSplit = str_replace(SciNameSplit, "axon 44", "axon44"),
         SciNameSplit = str_replace(SciNameSplit, " virus ", "_virus_"),
         SciNameSplit = str_replace(SciNameSplit, "archaeon ", "archaeon_"),
         SciNameSplit = str_replace(SciNameSplit, "Clostridiales Family", "Clostridiales_Family"),
         SciNameSplit = str_replace(SciNameSplit, "Muribaculaceae ", "Muribaculaceae_"),
         SciNameSplit = str_replace(SciNameSplit, "Arthrobacter virus ", "Arthrobacter_virus_"),
         SciNameSplit = str_replace(SciNameSplit, "alpha proteobacterium", "alpha_proteobacterium"),
         SciNameSplit = str_replace(SciNameSplit, "beta proteobacterium", "beta_proteobacterium"),
         SciNameSplit = str_replace(SciNameSplit, "gamma proteobacterium", "gamma_proteobacterium"),
         SciNameSplit = str_replace(SciNameSplit, "delta proteobacterium", "delta_proteobacterium"),
         SciNameSplit = str_replace(SciNameSplit, "Bacillus virus", "Bacillus_virus"),
         SciNameSplit = str_replace(SciNameSplit, "Group II \\'CF-1\\'", "Group_II_\\'CF-1\\'"),
         SciNameSplit = str_replace(SciNameSplit, "[Scytonema hofmanni]_UTEX ", "[Scytonema_hofmanni]_UTEX_"),
         SciNameSplit = str_replace(SciNameSplit, "marine gamma proteobacterium", "marine_gamma_proteobacterium"),
         SciNameSplit = str_replace(SciNameSplit, "Paramecium bursaria Chlorella ", "Paramecium_bursaria_Chlorella_"),
         SciNameSplit = str_replace(SciNameSplit, "filamentous cyanobacterium CCP", "filamentous_cyanobacterium_CCP"),
         SciNameSplit = str_replace(SciNameSplit, "Legionella endosymbiont of Polyplax serrata", "Legionella_endosymbiont_of_Polyplax_serrata"),
         SciNameSplit = str_replace(SciNameSplit, "endosymbiont of unidentified scaly snail isolate Monju", "endosymbiont_of_unidentified_scaly_snail_isolate_Monju"),
         SciNameSplit = str_replace(SciNameSplit, "endosymbiont GvMRE of Glomus versiforme", "endosymbiont_GvMRE_of_Glomus_versiforme"),
         SciNameSplit = str_replace(SciNameSplit, "unclassified ", "")) %>%
  select(Percent, Number_rooted, Number_direct, Rank, NCBItaxID, SciNameFull, SampleID, Run, Database, SciNameSplit) %>%
  separate(col = "SciNameSplit", into = c("genus","species","strain"), sep = " ", convert = TRUE) %>%
  select(-strain) %>%
  mutate(species = replace_na(species, "lll")) %>%
  unite(SciNameShort, genus, species, sep = " ") %>%
  mutate(SciNameShort = str_replace(SciNameShort, "_sp\\._", " sp\\. "),
         SciNameShort = str_replace(SciNameShort,  "oral_taxon_", "oral taxon "),
         SciNameShort = str_replace(SciNameShort,  "Candidatus_", "Candidatus "),
         SciNameShort = str_replace(SciNameShort,  "_bacterium_", " bacterium "),
         SciNameShort = str_replace(SciNameShort, "_=_", " = "),
         SciNameShort = str_replace(SciNameShort, "_genomosp\\._", " genomosp\\. "),
         SciNameShort = str_replace(SciNameShort, "_archaeon_", " archaeon "),
         SciNameShort = str_replace(SciNameShort, "archaeon_", "archaeon "),
         SciNameShort = str_replace(SciNameShort, "_virus_", " virus "),
         SciNameShort = str_replace(SciNameShort, "axon44", "axon 44"),
         SciNameShort = str_replace(SciNameShort, "taxon 44", "Taxon 44"),
         SciNameShort = str_replace(SciNameShort, "Clostridiales_Family", "Clostridiales Family"),
         SciNameShort = str_replace(SciNameShort, "Muribaculaceae_", "Muribaculaceae "),
         SciNameShort = str_replace(SciNameShort, "Arthrobacter_virus_", "Arthrobacter virus "),
         SciNameShort = str_replace(SciNameShort, "alpha_proteobacterium", "alpha proteobacterium"),
         SciNameShort = str_replace(SciNameShort, "beta_proteobacterium", "beta proteobacterium"),
         SciNameShort = str_replace(SciNameShort, "gamma_proteobacterium", "gamma proteobacterium"),
         SciNameShort = str_replace(SciNameShort, "delta_proteobacterium", "delta proteobacterium"),
         SciNameShort = str_replace(SciNameShort, "Bacillus_virus", "Bacillus virus"),
         SciNameShort = str_replace(SciNameShort, "Group_II_\\'CF-1\\'", "Group II \\'CF-1\\'"),
         SciNameShort = str_replace(SciNameShort, "[Scytonema_hofmanni]_UTEX_", "[Scytonema hofmanni]_UTEX "),
         SciNameShort = str_replace(SciNameShort, "marine_gamma_proteobacterium", "marine gamma proteobacterium"),
         SciNameShort = str_replace(SciNameShort, "Paramecium_bursaria_Chlorella_", "Paramecium bursaria Chlorella "),
         SciNameShort = str_replace(SciNameShort, "filamentous_cyanobacterium_CCP", "filamentous cyanobacterium CCP"),
         SciNameShort = str_replace(SciNameShort, "Legionella_endosymbiont_of_Polyplax_serrata", "Legionella endosymbiont of Polyplax serrata"),
         SciNameShort = str_replace(SciNameShort, "filamentous_cyanobacterium_CCP", "filamentous cyanobacterium CCP"),
         SciNameShort = str_replace(SciNameShort, "endosymbiont_of_unidentified_scaly_snail_isolate_Monju", "endosymbiont of unidentified scaly snail isolate Monju"),
         SciNameShort = str_replace(SciNameShort, "endosymbiont_GvMRE_of_Glomus_versiforme", "endosymbiont GvMRE of Glomus versiforme"),
         SciNameShort = str_replace(SciNameShort, " str\\.", ""),
         SciNameShort = str_replace(SciNameShort, "_phage_", " phage "),
         SciNameShort = str_replace(SciNameShort,  " lll", "")) %>%
  rename(SciName = SciNameShort)%>%
  full_join(., taxonomy_list, by = "SciName")


# now combine the RefSeqOnly and RefSeq+PasolliMAGs tables and pull out only the taxa that don't have full taxonomy
taxonomy_na_mod <- rso_reports_tax_mod %>%
  bind_rows(., rsmp_reports_tax_mod) %>%
  filter(is.na(Taxonomy)) %>%
  arrange(SciName) %>%
  select(Rank, NCBItaxID, SciNameFull, SciName) %>%
  unique()

# fwrite(taxonomy_na_mod, file = "./00-documentation.backup/taxonomy_na.tsv", quote = F, sep = "\t")


# now get the taxonomy with NCBItaxIDs in a single file for later
taxonomy_IDs <- rso_reports_tax_mod %>%
  bind_rows(., rsmp_reports_tax_mod) %>%
  filter(!is.na(Taxonomy)) %>%
  filter(!is.na(NCBItaxID)) %>%
  arrange(NCBItaxID) %>%
  select(NCBItaxID, Taxonomy) %>%
  unique()

# fwrite(taxonomy_IDs, file = "./00-documentation.backup/taxonomy_IDs.tsv", quote = F, sep = "\t")

```

Read in the file with NCBItaxIDs that didn't have full taxonomy after the chunk
above. We'll try to get these through the package taxize.
```{r taxid_list_na, eval = F}
# read in the list of taxonomy IDs that don't have a full taxonomy series from the chunk above, if starting here
taxonomy_na_mod <- fread("./00-documentation.backup/taxonomy_na.tsv")

# pull out the list of NCBI taxonomy IDs to feed into taxize to get the full taxonomy series
taxidlist <- taxonomy_na_mod %>%
  select(NCBItaxID) %>%
  unique() %>%
  pull()


```

Use taxize to get the full taxonomy in a data frame. 
```{r get_taxids_na, eval = F}
# use this chunk to get the full taxonomy series of the NCBItaxIDs that didn't work
# this comes from this website: https://reslp.github.io/blog/Download-Taxinfo-with-R/

# use the IDs to get the taxonomy from NCBI with taxize
taxon_summary <- ncbi_get_taxon_summary(id = taxidlist, key = "fc17e8c944d5eb3217cce3d5e7e41af83708")

# make a df list with the taxonomy of all IDs separated into individual columns by level (phylum, superfamily, etc)
df_list <- list()
for (i in 1:nrow(taxon_summary)){
tax  <- taxonomy(organism = taxon_summary[i,]$name, db = "ncbi",output = "classification")
df <- data.frame(lapply(tax$name, function(x) data.frame(x)))
colnames(df) <- tax$rank
df_list[[i]] <- df
Sys.sleep(0.5)
}

# put all the entries together into one data frame
# this doesn't include the taxID for each entry, but we need to have the taxID for each
combined_df <- do.call(rbind.fill, df_list)

# select only the levels we want from the full data frame (superkingdom, phylum, class, order, family, genus, species)
combined_df <- combined_df %>%
  select(superkingdom, phylum, class, order, family, genus, species)

# fwrite(combined_df, file = "./00-documentation.backup/taxonomy_na_found.tsv", quote = F, sep = "\t")

```

Create a full, non-redundant taxonomy file with the NCBItaxID and the full taxonomy string
```{r full_taxonomy_list}
# this has the taxonomy and NCBItaxIDs that worked from the chunk taxonomy_na
taxonomy_IDs_raw <- fread("./00-documentation.backup/taxonomy_IDs.tsv")

# this file contains duplicate NCBItaxIDs that need to be filtered out
tax_IDs_dupes <- taxonomy_IDs %>%
  get_dupes(NCBItaxID)

# we'll select and get rid of the even rows
tax_IDs_dupes <- tax_IDs_dupes %>% filter(row_number() %% 2 == 0)

taxonomy_IDs <- taxonomy_IDs_raw %>%
  anti_join(., tax_IDs_dupes)

# this has the missing taxonomy from the chunk taxonomy_na, that was gotten with taxize and then checked and filled in by hand in ./00-documentation.backup/taxonomy_na.tsv
missing_taxonomy_raw <- fread("./00-documentation.backup/missing_taxonomy.csv")

# make sure to remove all trailing spaces at the ends of the taxonomy, and add root| to the begining of each
missing_taxonomy <- missing_taxonomy_raw %>%
  as_tibble(.) %>%
  mutate(Taxonomy = str_squish(Taxonomy),
         Root = "root|") %>%
  select(NCBItaxID, Root, Taxonomy) %>%
  unite("TaxonomyFull", Root:Taxonomy, sep = "") %>%
  rename(Taxonomy = TaxonomyFull) %>%
  mutate(Taxonomy = str_replace(Taxonomy, "root\\|root", "root"))

# these files both have some duplicated NCBItaxIDs, so we need to remove them
missing_tax_dupes <- missing_taxonomy %>%
  get_dupes(NCBItaxID)

# we'll select and get rid of the even rows
missing_tax_dupes <- missing_tax_dupes %>% filter(row_number() %% 2 == 0)

missing_taxonomy <- missing_taxonomy %>%
  anti_join(., missing_tax_dupes)

# combine both taxonomy files to make one big complete one
all_taxonomy <- taxonomy_IDs %>%
  bind_rows(., missing_taxonomy) %>%
  as_tibble(.) %>%
  arrange(NCBItaxID)

# there are 3 taxIDs that are now duplicated, and one copy of each must be removed
duplicate_taxIds <- all_taxonomy %>%
  arrange(NCBItaxID) %>%
  get_dupes(NCBItaxID)

# we'll select and get rid of the even rows
duplicate_taxIds <- duplicate_taxIds %>% filter(row_number() %% 2 == 0)

all_taxonomy <- all_taxonomy %>%
  anti_join(., duplicate_taxIds)

```

Now add all of the full taxonomy to the Kraken report files. 
```{r add_taxonomy}

# now add the taxonomy to the kraken RefSeqOnly report file
rso_reports_tax <- rso_reports %>%
  full_join(., all_taxonomy) 

# the number of rows is bigger for rso_reports_tax than for rso_reports by 802. Check for NAs to explain this
# check if there are still NAs in the taxonomy column (nope, that's good)
rso_reports_tax %>%
  filter(is.na(NCBItaxID))

# check if there are  NAs in the other columns (yes, there are 802)
rso_reports_tax %>%
  filter(is.na(Database))

# get rid of those 802 rows with NAs (these taxa must be only in the rsmp_reports file)
rso_reports_tax <- rso_reports_tax %>%
  filter(!is.na(Database))


# now add the taxonomy to the kraken RefSeq+PasolliMAGs report file
rsmp_reports_tax <- rsmp_reports %>%
  full_join(., all_taxonomy)

# the number of rows is bigger for rso_reports_tax than for rso_reports by 1322. Check for NAs to explain this
# check if there are still NAs in the taxonomy column (nope, that's good)
rsmp_reports_tax %>%
  filter(is.na(NCBItaxID))

# check if there are  NAs in the other columns (yes, there are 1322)
rsmp_reports_tax %>%
  filter(is.na(Database))

 # get rid of those 1322 rows with NAs (these taxa must be only in the rso_reports file)
rsmp_reports_tax <- rsmp_reports_tax %>%
  filter(!is.na(Database))

```


























