---
title: "Cameroon hunter-gatherer calculus/plaque HUMAnN2"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(decontam)
library(data.table)
library(vegan)
library(ape)
library(mixOmics)
library(janitor)
library(rstatix)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```


# HUMAnN2 assignment stats
```{r load_data, eval = F}
## load the species and genus tables generated with HUMAnN2
humann2_path <- fread("./05-results.backup/humann2.pathabundance.all.tss.tsv")
humann2_path <- as_tibble(humann2_path)
humann2_KOs_full <- fread("./05-results.backup/humann2.genefamilies.all.i.tss.renorm.ko.tsv")
humann2_KOs_full <- as_tibble(humann2_KOs_full)

# clean the file names
humann2_KOs_full <- rename(humann2_KOs_full, Ortholog = `# Gene Family`)
colnames(humann2_KOs_full) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M_Abundance-RPKs","_10M", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","_10M", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub(".fastq.truncated_Abundance-RPKs","", colnames(humann2_KOs_full))

humann2_path <- rename(humann2_path, Pathway = `# Pathway`)
colnames(humann2_path) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M_Abundance","_10M", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","_10M", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","", colnames(humann2_path))
colnames(humann2_path) <- gsub(".fastq.truncated_Abundance","", colnames(humann2_path))


```

```{r load_metadata}
load("./05-results.backup/CMC_humann2_tables.RData")
load("./05-results.backup/CMC_metadata.RData")

# remove unmapped and ungrouped reads
humann2_KOs <- humann2_KOs_full %>% filter(Ortholog != "UNMAPPED") %>% filter(!str_detect(Ortholog, "UNGROUPED"))

```


```{r functions}

# add functions for:
# making the input table for PCA?

# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, color_group, shape_group, size_group) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))
    metadata_group_size <- get(paste(size_group, "_size", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]
    size_group = df_X[[size_group]]
                      
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group, size = size_group)) +
                        geom_point() +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        scale_size_manual(values = metadata_group_size) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                      # select the correct PC column and explained variance for PC1
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$SampleID), size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}

# this function plots the PCA with biplots to look at the categories with strongest loadings
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle, pathways = c("TRUE","FALSE")) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$explained_variance * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }

    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Function") %>%
      top_n(10, !!arrow_pc)
   if (pathways == T) {
      pws_10 <- pws_10 %>%
        rename(pathway = Function) %>%
        mutate(Function = sapply(pathway, function(f) {
                                  unlist(str_split(f, ":"))[1]
                           })) 
    }

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Function") %>%
      top_n(-10, !!arrow_pc)
    if (pathways == T) {
      neg_10 <- neg_10 %>%
      rename(pathway = Function) %>%
       mutate(Function = sapply(pathway, function(f) {
                                  unlist(str_split(f, ":"))[1]
                           })) 
    }


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Function),
                   size = 1.5, colour = "grey20", label.padding = 0.2) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Function),
                   size = 1.5, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 10)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

```

Before we proceed with looking at differences in abundance of different enzymes,
let's compare the how many reads were identified by HUMAnN2 for each sample
type. With the first step we'll read in the default HUMAnN2 output table with
all UniRef90 assignments. We'll select out just the rows with no species
assignemtns so we can calculate total assignments. HUMAnN2 uses *reads per*
*kilobase* for the initial output, so we'll be calculating the percent of total
assigned/mapped reads per kilobase (RPK). Then write out the file with just the
unmapped and total reads assigned so we never have to load this huge file again.
This has to be done with a table for which the samples were renormed
individually and then joined, and not renormed after they were joined (with
HUMAnN2 scripts).

Plot the proportion of assigned reads in each group.
```{r read_assignment_stats}

humann2_alignment_stats <- fread("./05-results.backup/humann2_alignment_stats.tsv")
humann2_alignment_stats$SampleID <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M.fastq.gz","_10M", humann2_alignment_stats$SampleID)
humann2_alignment_stats$SampleID <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam.fastq.gz","_10M", humann2_alignment_stats$SampleID)
humann2_alignment_stats$SampleID <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam.fastq.gz","", humann2_alignment_stats$SampleID)
humann2_alignment_stats$SampleID <- gsub(".fastq.truncated.prefixed.gz","", humann2_alignment_stats$SampleID)
humann2_alignment_stats$SampleID <- gsub(".SG1.1","", humann2_alignment_stats$SampleID)

humann2_alignment_stats <- humann2_alignment_stats %>%
  arrange(SampleID) %>%
  inner_join(., metadata %>% select(SampleID, Individual_Seq_Depth_NonHuman, Ethnic_Group, Village, Description), by = "SampleID") %>%
  mutate(UnalignedReads = round(Individual_Seq_Depth_NonHuman*(UnalignedPct/100)),
         AlignedReads = Individual_Seq_Depth_NonHuman - UnalignedReads, 
         AlignedPct = 100 - UnalignedPct) %>%
  select(SampleID, Individual_Seq_Depth_NonHuman, UnalignedReads, AlignedReads, UnalignedPct, AlignedPct, everything())

humann2_alignment_stats_box <- humann2_alignment_stats %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR164|10M")) %>%
  ggplot(., aes(Village, AlignedPct, fill = Village)) +
    geom_boxplot() +
    scale_fill_manual(values = Village_colors) +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ylab("Percent") +
    ggtitle("Percent of reads aligned by HUMAnN2")
humann2_alignment_stats_box

# ggsave("./05-results.backup/humann2_alignment_stats_box.pdf", plot = humann2_alignment_stats_box, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

humann2_alignment_stats_w <- humann2_alignment_stats %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR164|10M")) 
pairwise.wilcox.test(humann2_alignment_stats_w$AlignedPct, humann2_alignment_stats_w$Village, p.adjust.method = "fdr")

```

# Pathway analysis
Start with a PCA to see if there are any outlier samples that should be removed,
other than CMC032.B0101, which has very few reads compared to the other samples
and will be removed before starting any analyses.
```{r pca_paths_with_controls}
# create the input matrix
humann2_path_mat <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Pathway, CPM)

rownames(humann2_path_mat) <- humann2_path_mat$SampleID
humann2_path_mat <- as.matrix(humann2_path_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path_allsamples.pca <- pca(humann2_path_mat, ncomp = 4, logratio = 'CLR')

# plot the PCA
pca_test_plot <- plot_pca(humann2_path_allsamples.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_test_plot

pca_test_plot_names <- plot_pca_names(humann2_path_allsamples.pca, "PC1", "PC2", "Ethnic_Group")
pca_test_plot_names

```

Four of the gauze extraction blanks look like they are oral, and probably had
saliva or plaque that wasn't visible on the control strip that was cut. Now
repeat the PCA without any of the controls to compare the samples to known oral
plaque and calculus profiles. If the sub-sampled plaque (HMP) and calculus (JAE)
samples are similar (10M), then use just the full samples and not the
sub-sampled ones for all further analysis.

```{r pca_paths_no_blanks}

# create the input matrix
humann2_path_mat <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-c("SRR1646032","SRR1646041")) %>% # these plot with EXB and LIB controls in the PCA above
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Pathway, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB"))

rownames(humann2_path_mat) <- humann2_path_mat$SampleID
humann2_path_mat <- as.matrix(humann2_path_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path.pca <- pca(humann2_path_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_path.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_EG_plot

```
There is more overlap in functional profiles of CMC and JAE samples than with
HMP. The sub-sampled HMP and JAE samples are plotting in the same space as their
full samples, so we'll continue with the full samples. The Yanomami samples are
plotting by themselves - they were processed differently (MDA treatment before
sequencing) and that's probably why.

```{r pathways_bars}
humann2_path_noblanks_long <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-c("SRR1646032","SRR1646041")) %>% # these plot with EXB and LIB controls in the PCA above
  gather("SampleID","CPM", 2:ncol(.)) %>%
  mutate_if(is.numeric, ~1 * (. != 0)) %>%
  group_by(SampleID) %>%
  summarize(Paths = sum(CPM)) %>%
  inner_join(., metadata %>% select(SampleID, Ethnic_Group, Village), by = "SampleID")

humann2_path_noblanks_long_box <- humann2_path_noblanks_long %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR164|10M")) %>%
  ggplot(., aes(Village, Paths, fill = Village)) +
    geom_boxplot() +
    scale_fill_manual(values = Village_colors) +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ylab("Number of pathways detected") +
    ggtitle("Total number of pathways ID'd by HUMAnN2")
humann2_path_noblanks_long_box

# ggsave("./05-results.backup/humann2_path_noblanks_box.pdf", plot = humann2_path_noblanks_long_box, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

humann2_path_noblanks_long_w <- humann2_path_noblanks_long %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR164|10M")) 
pairwise.wilcox.test(humann2_path_noblanks_long_w$Paths, humann2_path_noblanks_long_w$Village, p.adjust.method = "fdr")

h2_path_test <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-c("SRR1646032","SRR1646041")) %>% # these plot with EXB and LIB controls in the PCA above
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","CPM",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Pathway) %>%
  summarize(Percent_presence = sum(CPM)/length(SampleID)*100) %>%
  # select the SEED proteins that are present in at least 30% of at least one group
  ungroup() %>%
  filter(Percent_presence >= 33.33) %>%
  distinct(Pathway) %>%
  # pull out these proteins to keep
  select(Pathway)

humann2_path_noblanks_long <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-c("SRR1646032","SRR1646041")) %>% # these plot with EXB and LIB controls in the PCA above
  inner_join(., h2_path_test, by = "Pathway") %>%
  gather("SampleID","CPM", 2:ncol(.)) %>%
  mutate_if(is.numeric, ~1 * (. != 0)) %>%
  group_by(SampleID) %>%
  summarize(Paths = sum(CPM)) %>%
  inner_join(., metadata %>% select(SampleID, Ethnic_Group, Village), by = "SampleID")

humann2_path_noblanks_long_box <- humann2_path_noblanks_long %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR164|10M")) %>%
  ggplot(., aes(Village, Paths, fill = Village)) +
    geom_boxplot() +
    scale_fill_manual(values = Village_colors) +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ylab("Number of pathways detected") +
    ggtitle("Total number of pathways ID'd by HUMAnN2")
humann2_path_noblanks_long_box

# ggsave("./05-results.backup/humann2_path_noblanks_box.pdf", plot = humann2_path_noblanks_long_box, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

humann2_path_noblanks_long_w <- humann2_path_noblanks_long %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR164|10M")) 
pairwise.wilcox.test(humann2_path_noblanks_long_w$Paths, humann2_path_noblanks_long_w$Village, p.adjust.method = "fdr")


```
HMP and JAE both have significantly more pathways identified than each of the
villages. There are no differences between the villages.

```{r pca_paths_noblanks_metadata, eval = F}
# create the input matrix
humann2_path_noblanks_mat <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-c("SRR1646032","SRR1646041")) %>% # these plot with EXB and LIB controls in the PCA above
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Pathway, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M"))

rownames(humann2_path_noblanks_mat) <- humann2_path_noblanks_mat$SampleID
humann2_path_noblanks_mat <- as.matrix(humann2_path_noblanks_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_noblanks_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path_noblanks.pca <- pca(humann2_path_noblanks_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_path_noblanks.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_EG_plot
plot_pca(humann2_path_noblanks.pca, "PC3", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

pca_AG_plot <- plot_pca(humann2_path_noblanks.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
pca_AG_plot

pca_ME_plot <- plot_pca(humann2_path_noblanks.pca, "PC1", "PC2", "Market_economy", "Market_economy", "Market_economy")
pca_ME_plot

pca_S_plot <- plot_pca(humann2_path_noblanks.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
pca_S_plot

pca_TS_plot <- plot_pca(humann2_path_noblanks.pca, "PC1", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
pca_TS_plot

pca_V_plot <- plot_pca(humann2_path_noblanks.pca, "PC1", "PC2", "Village", "Village", "Village")
pca_V_plot

```
The plaque (HMP) and calculus (JAE) samples separate along PC2, but the CMC
samples have more overlap with JAE than HMP. There is much more spread of the
samples along PC1. Do any other metadata categories appear to drive separation
of the CMC samples along PC1?

Only the tooth site (anterior vs posterior) appears to be separating along PC1.
Let's also run these without the HMP and JAE samples to see how the CMC samples
cluster without industrialized samples included.

```{r adonis_paths_samples, eval = F}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
humann2_path_noblanks_mat.euc <- vegdist(humann2_path_noblanks.pca$X, method = "euclidean", na.rm = T)
humann2_path_noblanks_mat.euc.pcoa <- ape::pcoa(humann2_path_noblanks_mat.euc)

humann2_path_noblanks_mat.euc.pcoavectors <- as.data.frame(humann2_path_noblanks_mat.euc.pcoa$vectors)
humann2_path_noblanks_mat.euc.pcoavectors <- humann2_path_noblanks_mat.euc.pcoavectors %>%
  rownames_to_column("SampleID")

humann2_path_noblanks_mat.euc.vecmet <- inner_join(humann2_path_noblanks_mat.euc.pcoavectors, metadata, by = "SampleID")

humann2_path_noblanks_mat.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 4) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis(humann2_path_noblanks_mat.euc ~ Tooth_site, humann2_path_noblanks_mat.euc.vecmet, perm=999)

adonis2(humann2_path_noblanks_mat.euc ~ ExtractionBatch + Village + Sex + Industry + Tooth_site + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, humann2_path_noblanks_mat.euc.vecmet, permutations = 999, by = "margin")

# controlled for tooth site
adonis2(humann2_path_noblanks_mat.euc ~ ExtractionBatch + Village + Sex + Industry + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, metadata_noblanks, permutations = 999, by = "margin", strata = humann2_path_noblanks_mat.euc.vecmet$Tooth_site)

```


```{r pca_paths_noyanomami_metadata}
# create the input matrix
humann2_path_noyanomami_mat <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-c("SRR1646032","SRR1646041")) %>% # these plot with EXB and LIB controls in the PCA above
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Pathway, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M|SRR164"))

rownames(humann2_path_noyanomami_mat) <- humann2_path_noyanomami_mat$SampleID
humann2_path_noyanomami_mat <- as.matrix(humann2_path_noyanomami_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_noyanomami_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path_noyanomami.pca <- pca(humann2_path_noyanomami_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_path_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_EG_plot
plot_pca(humann2_path_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

pca_AG_plot <- plot_pca(humann2_path_noyanomami.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
pca_AG_plot

pca_ME_plot <- plot_pca(humann2_path_noyanomami.pca, "PC1", "PC2", "Market_economy", "Market_economy", "Market_economy")
pca_ME_plot

### there's a problem with this one about the size of the points. Using a different size_group lets it work
pca_S_plot <- plot_pca(humann2_path_noyanomami.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
pca_S_plot

pca_TS_plot <- plot_pca(humann2_path_noyanomami.pca, "PC1", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
pca_TS_plot

pca_V_plot <- plot_pca(humann2_path_noyanomami.pca, "PC1", "PC2", "Village", "Village", "Village")
pca_V_plot

# ggsave("./05-results.backup/pca_EG_noyanomami_paths.pdf", plot = pca_EG_plot, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

```

```{r adonis_paths_noyanomami}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
humann2_path_noyanomami_mat.euc <- vegdist(humann2_path_noyanomami.pca$X, method = "euclidean", na.rm = T)
humann2_path_noyanomami_mat.euc.pcoa <- ape::pcoa(humann2_path_noyanomami_mat.euc)

humann2_path_noyanomami_mat.euc.pcoavectors <- as.data.frame(humann2_path_noyanomami_mat.euc.pcoa$vectors)
humann2_path_noyanomami_mat.euc.pcoavectors <- humann2_path_noyanomami_mat.euc.pcoavectors %>%
  rownames_to_column("SampleID")

humann2_path_noyanomami_mat.euc.vecmet <- inner_join(humann2_path_noyanomami_mat.euc.pcoavectors, metadata, by = "SampleID")

humann2_path_noyanomami_mat.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 4) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis2(humann2_path_noyanomami_mat.euc ~ ExtractionBatch + Village + Sex + Industry + Tooth_site + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, humann2_path_noyanomami_mat.euc.vecmet, permutations = 999, by = "margin")

# controlled for tooth site
adonis2(humann2_path_noyanomami_mat.euc ~ ExtractionBatch + Village + Sex + Industry + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, humann2_path_noyanomami_mat.euc.vecmet, permutations = 999, by = "margin", strata = metadata_noblanks$Tooth_site)

```
Tooth site (p < 0.05) and village (p < 0.01) are significantly different. When
controlling for tooth site, village is significant (p < 0.05).

```{r pca_paths_cmc_metadata}
humann2_path_cmc_mat <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-c("SRR1646032","SRR1646041")) %>% # these plot with EXB and LIB controls in the PCA above
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Pathway, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR|JAE"))

rownames(humann2_path_cmc_mat) <- humann2_path_cmc_mat$SampleID
humann2_path_cmc_mat <- as.matrix(humann2_path_cmc_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_cmc_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path_cmc.pca <- pca(humann2_path_cmc_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_path_cmc.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_EG_plot

pca_AG_plot <- plot_pca(humann2_path_cmc.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
pca_AG_plot

pca_ME_plot <- plot_pca(humann2_path_cmc.pca, "PC1", "PC2", "Market_economy", "Market_economy", "Market_economy")
pca_ME_plot

pca_S_plot <- plot_pca(humann2_path_cmc.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
pca_S_plot

pca_TS_plot <- plot_pca(humann2_path_cmc.pca, "PC1", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
pca_TS_plot

pca_TS_plot <- plot_pca(humann2_path_cmc.pca, "PC3", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
pca_TS_plot

pca_V_plot <- plot_pca(humann2_path_cmc.pca, "PC1", "PC2", "Village", "Village", "Village")
pca_V_plot

# ggsave("./05-results.backup/pca_EG_cmc_paths.pdf", plot = pca_EG_plot, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

```
Again the only sort-of clustering that we see is by tooth site (anterior or
posterior). Now let's look at which pathways are driving separation of the
sample groups. 

```{r adonis_paths_cmc}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
humann2_path_cmc_mat.euc <- vegdist(humann2_path_cmc.pca$X, method = "euclidean", na.rm = T)
humann2_path_cmc_mat.euc.pcoa <- ape::pcoa(humann2_path_cmc_mat.euc)

humann2_path_cmc_mat.euc.pcoavectors <- as.data.frame(humann2_path_cmc_mat.euc.pcoa$vectors)
humann2_path_cmc_mat.euc.pcoavectors <- humann2_path_cmc_mat.euc.pcoavectors %>%
  rownames_to_column("SampleID")

humann2_path_cmc_mat.euc.vecmet <- inner_join(humann2_path_cmc_mat.euc.pcoavectors, metadata, by = "SampleID")

humann2_path_cmc_mat.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 4) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis(humann2_path_cmc_mat.euc ~ Tooth_site + Ethnic_Group + Age_group + Sex + Village + Market_economy, humann2_path_cmc_mat.euc.vecmet, perm=999)

adonis2(humann2_path_cmc_mat.euc ~ ExtractionBatch + Village + Sex + Tooth_site + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, humann2_path_cmc_mat.euc.vecmet, permutations = 999, by = "margin")

# controlled for tooth site
adonis2(humann2_path_cmc_mat.euc ~ ExtractionBatch + Village + Sex + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, humann2_path_cmc_mat.euc.vecmet, permutations = 999, by = "margin", strata = humann2_path_cmc_mat.euc$Tooth_site)

```
Tooth site (p < 0.01) is significantly different, but when controlling for tooth
site none are significantly different.

Now to see which of the pathways have the strongest loadings in PC1 and PC2
```{r pca_path_biplot}
# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc1 <- plot_pca_bi(humann2_path_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = T)
pca_EG_biplot_pc1

# ggsave("./05-results.backup/pca_EG_bi_noyanomami_paths.pdf", plot = pca_EG_biplot_pc1, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# make a table of the pathways to save, to use again later in another R notebook
# humann2_pathway_biplot_list <- pca_EG_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1+", Samples = "Industrial+CMC") 
# humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1-", Samples = "Industrial+CMC"))

# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc2 <- plot_pca_bi(humann2_path_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = T)
pca_EG_biplot_pc2

# what are those pathways?
pandoc.table(pca_EG_biplot_pc2$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_biplot_pc2$plot_env$neg_10 %>% select(-Function) %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

# humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc2$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2+", Samples = "Industrial+CMC"))
# humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc2$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2-", Samples = "Industrial+CMC"))


# which paths distinguish sample groups when only CMC samples are included?
pca_EG_cmc_biplot_pc1 <- plot_pca_bi(humann2_path_cmc.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = T)
pca_EG_cmc_biplot_pc1

# ggsave("./05-results.backup/pca_EG_bi_cmc_paths.pdf", plot = pca_EG_cmc_biplot_pc1, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>% 
#   bind_rows(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1+", Samples = "CMC"))
# humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1-", Samples = "CMC"))

pca_EG_cmc_biplot_pc2 <- plot_pca_bi(humann2_path_cmc.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = T)
pca_EG_cmc_biplot_pc2

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>% 
#   bind_rows(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2+", Samples = "CMC"))
# humann2_pathway_biplot_list <- humann2_pathway_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2-", Samples = "CMC"))
# 
# fwrite(humann2_pathway_biplot_list, file = "./05-results.backup/humann2_pathway_biplot_list.tsv", quote = F, sep = "\t")

```
_*All samples*_
_PC1+:_
GLYCOCAT-PWY: glycogen degradation I (bacterial) - breaks (internally stored?) long chain carbohydrates into short, processible ones that are further broken down into maltoses and glucose for glycolysis
GALACTUROCAT-PWY: D-galacturonate degradation I - breaks down D-galacturonate (common in plant cell walls and the environment) to a modified glucose that enters the Entner-Doudoroff shunt
PWY-7204: pyridoxal 5'-phosphate salvage II (plants) - vitamin B6 salvage pathway; plants and bacteria have a similar system; roles in sphingolipid biosynthesis and carbohydrate metabolism
PWY-5104: L-isoleucine biosynthesis I - common alternate pathway to synthesize L-isoleucine; known to exist in *Methanobrevibacter* and *Desulfovibrio*.
PWY-6630: superpathway of L-tyrosine biosynthesis - standard pathway
PWY-5005: biotin biosynthesis II - biotin is a cofactor that facilitates transfer of CO2 in fatty acid and carbohydrate metabolism
PWY-5941: glycogen degradation II (eukaryotic) - same idea as GLYCOCAT-PWY, but found in fungi and animals
GALACT-GLUCUROCAT-PWY: superpathway of hexuronide and hexuronate degradation - D-glucuronate, D-fructuronate and D-galacturonate (main monomer of pectin); shuffles these till they can enter the Entner-Doudoroff pathway
PWY-7371: 1,4-dihydroxy-6-naphthoate  biosynthesis II - this is converted to menaquinone; menaquinones and demethylmenaquinones (vitamin K2) are reversible redox component of the electron transfer chain; Gram +s, and anaerobic G+ and G- have either; comes from chorismate (intermediate of aromatic amino acid biosynthesis);  
PWY1G-0: mycothiol biosynthesis - major thiol found in Actinobacteria (opposed to glutathione); detoxifies stress-inducing factors, maintains reducing envt within the cell; 

_PC1-:_
PWY-6285: superpathway of fatty acids biosynthesis (E. coli) - standard fatty acid biosynthesis
PWY-5971: palmitate biosynthesis II (bacteria and plants) - common saturated fatty acid
PWY-7391: isoprene biosynthesis II (engineered) - volatile compound produced by plants, incorporated into cell membranes in response to heat stress
PWY-6803: phosphatidylcholine acyl editing - production of triacylglycerols
PWY-6113: superpathway of mycolate biosynthesis - found in mycobacterial cell walls
PWY-5675: nitrate reduction V (assimilatory) - environmental nitrogen-containing compounds (nitrate, nitrite) must be reduced to be used by bacteria in producing their own N-containing compounds such as amino acids
P122-PWY: heterolactic fermentation - found in Firmicutes, particularly lactic acid bacteria; this fermentation pathway produces substantial amounts of volatile compounds and C02 along with lactic acid; this produces lactate, EtOH, CO2 and trace acetate
PWY-5121: superpathway of geranylgeranyl diphosphate biosynthesis II (via MEP) - crucial compound involved in biosynthesis of a variety of terpenes and terpenoids, including central compounds such as ubiquinones and menaquinones
PWY-6859: all-trans-farnesol biosynthesis - found in eukaryia such as yeast; part of terpenoid synthesis
PWY-5910: superpathway of geranylgeranyldiphosphate  biosynthesis I (via mevalonate) - crucial compound involved in biosynthesis of a variety of terpenes and terpenoids, including central compounds such as ubiquinones and menaquinones

_PC2+:_
P23-PWY: reductive TCA cycle I - carbon dioxide fixation pathway found in autotrophic eubacteria and archaea; considered to be a primordial pathway for production of starting organic molecules for biosynthesis of sugars, lipids, amino acids, pyrimidines and pyrroles
PWY-5104: L-isoleucine biosynthesis IV - synthesis from propanoate
PWY-7003: glycerol degradation to butanol - butanol biosynthesis pathway from glycerol
PWY4LZ-257: superpathway of fermentation (Chlamydomonas reinhardtii) - fermentation from pyruvate
PWY-6588: pyruvate fermentation to acetone - anaerobic
PWY66-398: TCA cycle III (animals) - energy metabolism
PWY-6269: adenosylcobalamin salvage from cobinamide II - production of cobamides, which are enzyme cofactors; found in bacteria and archaea
PWY-6549: L-glutamine biosynthesis III - amino acid biosynthesis in plants
PWY-7254: TCA cycle VII (acetate-producers) - used by acetic acid bacteria are Gram-negative acidophilic α-proteobacteria that oxidize ethanol to acetate
REDCITCYC: TCA cycle VIII (helicobacter) - specific to H. pylori

_PC2-:_
PWY0-1241: ADP-L-glycero-&beta;-D-manno-heptose biosynthesis - glycero-manno-heptoses are found in the cell surface polymers of many bacteria, particularly LPS
PWY-6895: superpathway of thiamin diphosphate biosynthesis II - vitamin B1, an essential cofactor for enzymes involved in energy metabolism
THISYN-PWY: superpathway of thiamin diphosphate biosynthesis I - vitamin B1, an essential cofactor for enzymes involved in energy metabolism; different version from above, carried by different organisms
PWY-5083: NAD/NADH phosphorylation and dephosphorylation - very common cofactors
PWY-5838: superpathway of menaquinol-8 biosynthesis I - menaquinones and demethylmenaquinones (vitamin K2) are reversible redox component of the electron transfer chain; Gram +s, and anaerobic G+ and G- have either; menaquinol-8 is common in facultatively anaerobic Gram - organisms
PWY-5861: superpathway of demethylmenaquinol-8 biosynthesis - menaquinones and demethylmenaquinones (vitamin K2) are reversible redox component of the electron transfer chain; Gram +s, and anaerobic G+ and G- have either; dimethylmenaquinol-8 is common in facultatively anaerobic Gram - organisms
PWY-7210: pyrimidine deoxyribonucleotides biosynthesis from CTP - DNA synthesis 
PWY-6892: thiazole biosynthesis I (E. coli) - only the synthesis of the thiazole moiety of thiamin; part of the thiamin phosphate synthesis pathway (vitamin B1) (E. coli version)
PWY-6891: thiazole biosynthesis II (Bacillus) - only the synthesis of the thiazole moiety of thiamin; part of the thiamin phosphate synthesis pathway (vitamin B1) (Bacillus version)
PWY-5675: nitrate reduction V (assimilatory) - environmental nitrogen-containing compounds (nitrate, nitrite) must be reduced to be used by bacteria in producing their own N-containing compounds such as amino acids


_*CMC samples*_
_PC1+:_
GLYCOCAT-PWY: glycogen degradation I (bacterial) - breaks (internally stored?) long chain carbohydrates into short, processible ones that are further broken down into maltoses and glucose for glycolysis
PWY-7204: pyridoxal 5'-phosphate salvage II (plants) - vitamin B6 salvage pathway; plants and bacteria have a similar system; roles in sphingolipid biosynthesis and carbohydrate metabolism
PWY-5005: biotin biosynthesis II - biotin is a cofactor that facilitates transfer of CO2 in fatty acid and carbohydrate metabolism
GALACTUROCAT-PWY: D-galacturonate degradation I - breaks down D-galacturonate (common in plant cell walls and the environment) to a modified glucose that enters the Entner-Doudoroff shunt
PWY-6630: superpathway of L-tyrosine biosynthesis - standard pathway
GALACT-GLUCUROCAT-PWY: superpathway of hexuronide and hexuronate degradation - D-glucuronate, D-fructuronate and D-galacturonate (main monomer of pectin); shuffles these till they can enter the Entner-Doudoroff pathway
RHAMCAT-PWY: L-rhamnose degradation I - common bacterial cell wall component, also used for energy
PWY-5941: glycogen degradation II (eukaryotic) - same idea as GLYCOCAT-PWY, but found in fungi and animals
PWY-5101: L-isoleucine biosynthesis II - synthesis from pyruvate
PWY-7209: superpathway of pyrimidine ribonucleosides degradation - pyrimidine salvage pathway; uridine broken down to uracil and a ribose or to beta-alanine

_PC1-:_
PWY-6803: phosphatidylcholine acyl editing - production of triacylglycerols
PWY-7431: aromatic biogenic amine degradation (bacteria) - synthesis of aromatic and aliphatic biogenic amines by decarboxylation of amino acids; biogenic amines have physiologic roles in prokaryotes (not listed)
PWY-5910: superpathway of geranylgeranyldiphosphate biosynthesis I (via mevalonate) - crucial compound involved in biosynthesis of a variety of terpenes and terpenoids, including central compounds such as ubiquinones and menaquinones
PWY-7391: isoprene biosynthesis II (engineered) - volatile compound produced by plants, incorporated into cell membranes in response to heat stress
PWY-7392: taxadiene biosynthesis (engineered) - taxadiene is an intermediate in taxadiol biosynthesis (from Pacific yew tree, used in chemotherapy)
PWY-5121: superpathway of geranylgeranyl diphosphate biosynthesis II (via MEP) - crucial compound involved in biosynthesis of a variety of terpenes and terpenoids, including central compounds such as ubiquinones and menaquinones
GLYCOLYSIS-TCA-GLYOX-BYPASS: superpathway of glycolysis, pyruvate dehydrogenase, TCA, and glyoxylate bypass - energy metabolism
P122-PWY: heterolactic fermentation - found in Firmicutes, particularly lactic acid bacteria; this fermentation pathway produces substantial amounts of volatile compounds and C02 along with lactic acid; this produces lactate, EtOH, CO2 and trace acetate
PWY-6353: purine nucleotides degradation II (aerobic) - degradation of purine nucleotides to purine nucleosides, purine bases; nitrogen metabolism for nitrogen utilization
TCA-GLYOX-BYPASS: superpathway of glyoxylate bypass and TCA - bypasses steps that lead to loss of CO2, saving carbon

_PC2+:_
PWY-5101: L-isoleucine biosynthesis II - synthesis from pyruvate
GALACTUROCAT-PWY: D-galacturonate degradation I - breaks down D-galacturonate (common in plant cell walls and the environment) to a modified glucose that enters the Entner-Doudoroff shunt
GALACT-GLUCUROCAT-PWY: superpathway of hexuronide and hexuronate degradation - D-glucuronate, D-fructuronate and D-galacturonate (main monomer of pectin); shuffles these till they can enter the Entner-Doudoroff pathway
GLYCOCAT-PWY: glycogen degradation I (bacterial) - breaks (internally stored?) long chain carbohydrates into short, processible ones that are further broken down into maltoses and glucose for glycolysis
PWY-5005: biotin biosynthesis II - biotin is a cofactor that facilitates transfer of CO2 in fatty acid and carbohydrate metabolism
PWY-6630: superpathway of L-tyrosine biosynthesis - standard pathway
PWY-5941: glycogen degradation II (eukaryotic) - same idea as GLYCOCAT-PWY, but found in fungi and animals
RHAMCAT-PWY: L-rhamnose degradation I - common bacterial cell wall component, also used for energy
PWY-7204: pyridoxal 5'-phosphate salvage II (plants) - vitamin B6 salvage pathway; plants and bacteria have a similar system; roles in sphingolipid biosynthesis and carbohydrate metabolism
PWY-7209: superpathway of pyrimidine ribonucleosides degradation - pyrimidine salvage pathway; uridine broken down to uracil and a ribose or to beta-alanine

_PC2-:_
TCA-GLYOX-BYPASS: superpathway of glyoxylate bypass and TCA - bypasses steps that lead to loss of CO2, saving carbon
GLYCOLYSIS-TCA-GLYOX-BYPASS: superpathway of glycolysis, pyruvate dehydrogenase, TCA, and glyoxylate bypass - energy metabolism
PWY-7391: isoprene biosynthesis II (engineered) - volatile compound produced by plants, incorporated into cell membranes in response to heat stress
PWY-7431: aromatic biogenic amine degradation (bacteria) - synthesis of aromatic and aliphatic biogenic amines by decarboxylation of amino acids; biogenic amines have physiologic roles in prokaryotes (not listed)
P122-PWY: heterolactic fermentation - found in Firmicutes, particularly lactic acid bacteria; this fermentation pathway produces substantial amounts of volatile compounds and C02 along with lactic acid; this produces lactate, EtOH, CO2 and trace acetate
PWY-6803: phosphatidylcholine acyl editing - production of triacylglycerols
PWY-5910: superpathway of geranylgeranyldiphosphate biosynthesis I (via mevalonate) - crucial compound involved in biosynthesis of a variety of terpenes and terpenoids, including central compounds such as ubiquinones and menaquinones
PWY-5121: superpathway of geranylgeranyl diphosphate biosynthesis II (via MEP) - crucial compound involved in biosynthesis of a variety of terpenes and terpenoids, including central compounds such as ubiquinones and menaquinones
PWY-7392: taxadiene biosynthesis (engineered) - taxadiene is an intermediate in taxadiol biosynthesis (from Pacific yew tree, used in chemotherapy)
PWY-6353: purine nucleotides degradation II (aerobic) - degradation of purine nucleotides to purine nucleosides, purine bases; nitrogen metabolism for nitrogen utilization


```{r outliers, eval=F}
path_outliers <- c("CMC032.B0101")
```


# Community-level functional analysis with individual gene families assigned KEGG Orthology numbers
## KEGG ortholog distribution
We will start analysis of KEGG orthologs by looking at the total number of KOs assigned to each population to see if this may bias further analyses. Is the proportion of genefamilies that were re-grouped by HUMAnN2 script into KOs  consistent and high across all sample types? That would be good b/c we have no major group biases. 
```{r orthologs_bars}

humann2_KO_long <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-c("SRR1646032","SRR1646041")) %>% # these plot with EXB and LIB controls in the PCA above
  gather("SampleID","CPM", 2:ncol(.)) %>%
  mutate_if(is.numeric, ~1 * (. != 0)) %>%
  group_by(SampleID) %>%
  summarize(Paths = sum(CPM)) %>%
  inner_join(., metadata %>% select(SampleID, Ethnic_Group, Village), by = "SampleID")

humann2_KO_box <- humann2_KO_long %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR164|10M")) %>%
  ggplot(., aes(Village, Paths, fill = Village)) +
    geom_boxplot() +
    scale_fill_manual(values = Village_colors) +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ylab("Number of KEGG orthologs detected")
    ggtitle("Total number of KEGG orthologs ID'd by HUMAnN2 (no filtering)")
humann2_KO_box

# ggsave("./05-results.backup/humann2_KO_box.pdf", plot = humann2_KO_box, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

humann2_KO_long_w <- humann2_KO_long %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR164|10M")) 
pairwise.wilcox.test(humann2_KO_long_w$Paths, humann2_KO_long_w$Village, p.adjust.method = "fdr")

```

Now perform a PCA to see if there is more distinct clustering of the samples based on any of the metadata categories than we see with the pathways. 
```{r pca_KOs_with_controls}
# create the input matrix
humann2_KO_mat <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Ortholog, CPM)

rownames(humann2_KO_mat) <- humann2_KO_mat$SampleID
humann2_KO_mat <- as.matrix(humann2_KO_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_KO_pca <- tune.pca(humann2_KO_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KO_allsamples.pca <- pca(humann2_KO_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_test_plot <- plot_pca(humann2_KO_allsamples.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_test_plot
# pca_test_plot <- plot_pca(humann2_KO_allsamples.pca, "PC1", "PC3", "Ethnic_Group")
# pca_test_plot

pca_test_plot_names <- plot_pca_names(humann2_KO_allsamples.pca, "PC1", "PC2", "Ethnic_Group")
pca_test_plot_names
# pca_test_plot_names <- plot_pca_names(humann2_KO_allsamples.pca, "PC1", "PC3", "Ethnic_Group")
# pca_test_plot_names

```


```{r pca_KOs_no_blanks}

# create the input matrix
humann2_KO_noblanks_mat <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-c("SRR1646032","SRR1646041")) %>% # these plot with EXB and LIB controls in the PCA above
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Ortholog, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB"))

rownames(humann2_KO_noblanks_mat) <- humann2_KO_noblanks_mat$SampleID
humann2_KO_noblanks_mat <- as.matrix(humann2_KO_noblanks_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_KO_noblanks_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KO_noblanks.pca <- pca(humann2_KO_noblanks_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_EG_plot

```


```{r pca_KOs_metadata}
# create the input matrix
humann2_KO_mat <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-c("SRR1646032","SRR1646041")) %>% # these plot with EXB and LIB controls in the PCA above
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Ortholog, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M"))%>%
  adorn_totals(where = "row", name = "Sum") %>%
  gather("Ortholog","CPM", 2:ncol(.)) %>%
  spread(SampleID, CPM) %>%
  mutate(Percent = Sum/sum(Sum)) %>%
  filter(Percent > 0.0005) %>%
  select(-c("Sum","Percent")) %>%
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Ortholog, CPM)

rownames(humann2_KO_mat) <- humann2_KO_mat$SampleID
humann2_KO_mat <- as.matrix(humann2_KO_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_KO_pca <- tune.pca(humann2_KO_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KO_noblanks.pca <- pca(humann2_KO_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_EG_plot

pca_AG_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
pca_AG_plot

pca_ME_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Market_economy", "Market_economy", "Market_economy")
pca_ME_plot

pca_S_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
pca_S_plot

pca_TS_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
pca_TS_plot

pca_V_plot <- plot_pca(humann2_KO_noblanks.pca, "PC1", "PC2", "Village", "Village", "Village")
pca_V_plot

```

```{r pca_KOs_noyanomami_metadata}
# create the input matrix
humann2_KO_mat <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-contains("SRR164")) %>% # these plot with EXB and LIB controls in the PCA above
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Ortholog, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M"))%>%
  adorn_totals(where = "row", name = "Sum") %>%
  gather("Ortholog","CPM", 2:ncol(.)) %>%
  spread(SampleID, CPM) %>%
  mutate(Percent = Sum/sum(Sum)) %>%
  filter(Percent > 0.0005) %>%
  select(-c("Sum","Percent")) %>%
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Ortholog, CPM)

rownames(humann2_KO_mat) <- humann2_KO_mat$SampleID
humann2_KO_mat <- as.matrix(humann2_KO_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_KO_pca <- tune.pca(humann2_KO_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KO_noyanomami.pca <- pca(humann2_KO_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_KO_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_EG_plot
pca_EG_plot23 <- plot_pca(humann2_KO_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_EG_plot23

pca_AG_plot <- plot_pca(humann2_KO_noyanomami.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
pca_AG_plot

pca_ME_plot <- plot_pca(humann2_KO_noyanomami.pca, "PC1", "PC2", "Market_economy", "Market_economy", "Market_economy")
pca_ME_plot

pca_S_plot <- plot_pca(humann2_KO_noyanomami.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
pca_S_plot

pca_TS_plot <- plot_pca(humann2_KO_noyanomami.pca, "PC1", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
pca_TS_plot

pca_V_plot <- plot_pca(humann2_KO_noyanomami.pca, "PC1", "PC2", "Village", "Village", "Village")
pca_V_plot


# ggsave("./05-results.backup/pca_EG_KOs_noyanomami.pdf", plot = pca_EG_plot, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)


```

```{r adonis_KOs_noyanomami}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
humann2_KO_mat.euc <- vegdist(humann2_KO_noyanomami.pca$X, method = "euclidean", na.rm = T)
humann2_KO_mat.euc.pcoa <- ape::pcoa(humann2_KO_mat.euc)

humann2_KO_mat.euc.pcoavectors <- as.data.frame(humann2_KO_mat.euc.pcoa$vectors)
humann2_KO_mat.euc.pcoavectors <- humann2_KO_mat.euc.pcoavectors %>%
  rownames_to_column("SampleID")

humann2_KO_mat.euc.vecmet <- inner_join(humann2_KO_mat.euc.pcoavectors, metadata, by = "SampleID")

humann2_KO_mat.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis(humann2_KO_mat.euc ~ Tooth_site + Ethnic_Group + Age_group + Sex + Village + Market_economy, humann2_KO_mat.euc.vecmet, perm=999)

adonis2(humann2_KO_mat.euc ~ ExtractionBatch + Village + Sex + Industry + Tooth_site + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, humann2_KO_mat.euc.vecmet, permutations = 999, by = "margin")

# controlled for tooth site
adonis2(humann2_KO_mat.euc ~ ExtractionBatch + Village + Sex + Industry + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, humann2_KO_mat.euc.vecmet, permutations = 999, by = "margin", strata = humann2_KO_mat.euc.vecmet$Tooth_site)

```


```{r pca_KOs_cmc_metadata}
# create the input matrix
humann2_KO_mat <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-c("SRR1646032","SRR1646041")) %>% # these plot with EXB and LIB controls in the PCA above
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Ortholog, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR|JAE")) %>%
  adorn_totals(where = "row", name = "Sum") %>%
  gather("Ortholog","CPM", 2:ncol(.)) %>%
  spread(SampleID, CPM) %>%
  mutate(Percent = Sum/sum(Sum)) %>%
  filter(Percent > 0.0005) %>%
  select(-c("Sum","Percent")) %>%
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Ortholog, CPM)


rownames(humann2_KO_mat) <- humann2_KO_mat$SampleID
humann2_KO_mat <- as.matrix(humann2_KO_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_KO_pca <- tune.pca(humann2_KO_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KO_cmc.pca <- pca(humann2_KO_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_EG_plot <- plot_pca(humann2_KO_cmc.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_EG_plot

pca_AG_plot <- plot_pca(humann2_KO_cmc.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
pca_AG_plot

pca_ME_plot <- plot_pca(humann2_KO_cmc.pca, "PC1", "PC2", "Market_economy", "Market_economy", "Market_economy")
pca_ME_plot

pca_S_plot <- plot_pca(humann2_KO_cmc.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
pca_S_plot

pca_TS_plot <- plot_pca(humann2_KO_cmc.pca, "PC1", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
pca_TS_plot

pca_V_plot <- plot_pca(humann2_KO_cmc.pca, "PC1", "PC2", "Village", "Village", "Village")
pca_V_plot

# ggsave("./05-results.backup/pca_EG_KOs_cmc.pdf", plot = pca_EG_plot, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

```

```{r adonis_KOs_cmc}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
humann2_KO_mat.euc <- vegdist(humann2_KO_cmc.pca$X, method = "euclidean", na.rm = T)
humann2_KO_mat.euc.pcoa <- ape::pcoa(humann2_KO_mat.euc)

humann2_KO_mat.euc.pcoavectors <- as.data.frame(humann2_KO_mat.euc.pcoa$vectors)
humann2_KO_mat.euc.pcoavectors <- humann2_KO_mat.euc.pcoavectors %>%
  rownames_to_column("SampleID")

humann2_KO_mat.euc.vecmet <- inner_join(humann2_KO_mat.euc.pcoavectors, metadata, by = "SampleID")

humann2_KO_mat.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis(humann2_KO_mat.euc ~ Tooth_site + Ethnic_Group + Age_group + Sex + Village + Market_economy, humann2_KO_mat.euc.vecmet, perm=999)

adonis2(humann2_KO_mat.euc ~ ExtractionBatch + Village + Sex + Tooth_site + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, humann2_KO_mat.euc.vecmet, permutations = 999, by = "margin")

# controlled for tooth site
adonis2(humann2_KO_mat.euc ~ ExtractionBatch + Village + Sex + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, humann2_KO_mat.euc.vecmet, permutations = 999, by = "margin", strata = humann2_KO_mat.euc.vecmet$Tooth_site)

```


Now to see which of the orthologs have the strongest loadings in PC1 and PC2
```{r pca_KO_biplot}
# which paths distinguish sample types when HMP and JAE are included?
pca_EG_KO_biplot_pc1 <- plot_pca_bi(humann2_KO_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = F)
pca_EG_KO_biplot_pc1

# ggsave("./05-results.backup/humann2_KO_bi_noyanomami.pdf", plot = pca_EG_KO_biplot_pc1, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_KO_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_KO_biplot_pc1$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# humann2_KO_biplot_list <- pca_EG_KO_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)) %>% select(Function) %>% mutate(Direction = "PC1+", Samples = "Industrial+CMC") 
# humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
#   bind_rows(pca_EG_KO_biplot_pc1$plot_env$neg_10 %>% arrange(desc(PC1)) %>% select(Function) %>% mutate(Direction = "PC1-", Samples = "Industrial+CMC"))

# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc2 <- plot_pca_bi(humann2_KO_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = F)
pca_EG_biplot_pc2

# what are those pathways?
pandoc.table(pca_EG_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_biplot_pc2$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

# humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)) %>% select(Function) %>% mutate(Direction = "PC2+", Samples = "Industrial+CMC"))
# humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc2$plot_env$neg_10 %>% arrange(desc(PC2)) %>% select(Function) %>% mutate(Direction = "PC2-", Samples = "Industrial+CMC"))

# which paths distinguish sample groups when only CMC samples are included?
pca_EG_cmc_biplot_pc1 <- plot_pca_bi(humann2_KO_cmc.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = F)
pca_EG_cmc_biplot_pc1

# ggsave("./05-results.backup/humann2_KO_bi_cmc.pdf", plot = pca_EG_cmc_biplot_pc1, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
#   bind_rows(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)) %>% select(Function) %>% mutate(Direction = "PC1+", Samples = "CMC"))
# humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
#   bind_rows(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% arrange(desc(PC1)) %>% select(Function) %>% mutate(Direction = "PC1-", Samples = "CMC"))

pca_EG_cmc_biplot_pc2 <- plot_pca_bi(humann2_KO_cmc.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = F)
pca_EG_cmc_biplot_pc2

pandoc.table(pca_EG_cmc_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc2$plot_env$neg_10  %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

# humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
#   bind_rows(pca_EG_cmc_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)) %>% select(Function) %>% mutate(Direction = "PC2+", Samples = "CMC"))
# humann2_KO_biplot_list <- humann2_KO_biplot_list %>%
#   bind_rows(pca_EG_cmc_biplot_pc2$plot_env$neg_10 %>% arrange(desc(PC2)) %>% select(Function) %>% mutate(Direction = "PC2-", Samples = "CMC"))
# 
# fwrite(humann2_KO_biplot_list, file = "./05-results.backup/humann2_KO_biplot_list.tsv", quote = F, sep = "\t")

```
_*All samples*_
_PC1+:_
K01839 - phosphopentomutase [EC:5.4.2.7]; carbohydrate metabolism
K01989 - putative ABC transport system substrate-binding protein; membrane transport
K01819 - galactose-6-phosphate isomerase [EC:5.3.1.26]; carbohydrate metabolism
K07088 - uncharacterized protein
K01223 - 6-phospho-beta-glucosidase [EC:3.2.1.86]; carbohydrate metabolism
K03839 - flavodoxin I; energy metabolism
K15986 - manganese-dependent inorganic pyrophosphatase [EC:3.6.1.1]; energy metabolism
K03650 - tRNA modification GTPase [EC:3.6.-.-]; tRNA biosynthesis
K03569 - rod shape-determining protein MreB and related proteins
K15580 - oligopeptide transport system substrate-binding protein; membrane transport

_PC1-:_
K09791 - uncharacterized protein (ycaR)
K07481 - transposase, IS5 family
K01494 - dCTP deaminase [EC:3.5.4.13]; nucleotide metabolism
K03801 - lipoyl(octanoyl) transferase [EC:2.3.1.181]; metabolism of cofactors and vitamins
K00033 - 6-phosphogluconate dehydrogenase [EC:1.1.1.44 1.1.1.343]; carbohydrate metabolism
K07054 - uncharacterized protein (ypfJ)
K02897 - large subunit ribosomal protein L25; translation
K02050 - NitT/TauT family transport system permease protein; membrane transport
K00340 - NADH-quinone oxidoreductase subunit K [EC:7.1.1.2]; energy metabolism
K02437 - glycine cleavage system H protein; carbohydrate metabolism

_PC2+:_
K02032 - peptide/nickel transport system ATP-binding protein; membrane transport
K07114 - Ca-activated chloride channel homolog; membrane transport
K07133 - uncharacterized protein
K11710 - manganese/zinc/iron transport system ATP- binding protein [EC:7.2.2.5]; membrane transport
K02031 - peptide/nickel transport system ATP-binding protein; membrane transport
K10441 - ribose transport system ATP-binding protein [EC:7.5.2.7]; membrane transport
K03924 - MoxR-like ATPase [EC:3.6.3.-]
K02034 - peptide/nickel transport system permease protein; membrane transport
K02033 - peptide/nickel transport system permease protein; membrane transport
K02026 - multiple sugar transport system permease protein; membrane transport

_PC2-:_
K07729 - putative transcriptional regulator; transcription factor
K06199 - fluoride exporter; membrane transport
K02825 - pyrimidine operon attenuation protein / uracil phosphoribosyltransferase [EC:2.4.2.9]; nucleotide metabolism
K03621 - phosphate acyltransferase [EC:2.3.1.274]; lipid metabolism
K00991 - 2-C-methyl-D-erythritol 4-phosphate cytidylyltransferase [EC:2.7.7.60]; biosynthesis of secondary metabolites
K02793 - mannose PTS system EIIA component [EC:2.7.1.191]; carbohydrate metabolism
K02794 - mannose PTS system EIIAB component [EC:2.7.1.191]; carbohydrate metabolism
K07166 - ACT domain-containing protein; signaling protein
K01091 - phosphoglycolate phosphatase [EC:3.1.3.18]; carbohydrate metabolism
K07058 - membrane protein (yhjD, yihH, rbn)

_*CMC samples only*_
_PC1+:_
K00260 - glutamate dehydrogenase [EC:1.4.1.2]; amino acid metabolism, arginine biosynthesis
K01819 - galactose-6-phosphate isomerase [EC:5.3.1.26]; carbohydrate metabolism
K01839 - phosphopentomutase [EC:5.4.2.7]; carbohydrate metabolism
K03856 - 3-deoxy-7-phosphoheptulonate synthase [EC:2.5.1.54]; amino acid metabolism
K01989 - putative ABC transport system substrate-binding protein
K01223 - 6-phospho-beta-glucosidase [EC:3.2.1.86]; carbohydrate metabolism
K07088 - uncharacterized protein
K16957 - L-cystine transport system substrate-binding protein; membrane transport
K15986 - manganese-dependent inorganic pyrophosphatase [EC:3.6.1.1]; energy metabolism
K04486 - histidinol-phosphatase (PHP family) [EC:3.1.3.15]; amino acid metabolism

_PC1-:_
K09791 - uncharacterized protein (ycaR)
K07481 - transposase, IS5 family
K07054 - uncharacterized protein (ypfJ)
K01494 - dCTP deaminase [EC:3.5.4.13]; nucleotide metabolism
K04759 - ferrous iron transport protein B; membrane transport
K02050 - NitT/TauT family transport system permease protein
K10441 - ribose transport system ATP-binding protein [EC:7.5.2.7]; membrane transport
K02027 - multiple sugar transport system substrate-binding protein (ycjN); membrane transport
K02897 - large subunit ribosomal protein L25
K00340 - NADH-quinone oxidoreductase subunit K [EC:7.1.1.2]; energy metabolism

_PC2+:_
K00260 - glutamate dehydrogenase [EC:1.4.1.2]; energy metabolism, amino acid metabolism
K04031 - ethanolamine utilization protein EutS
K01623 - fructose-bisphosphate aldolase, class I [EC:4.1.2.13]; carbohydrate metabolism
K11708 - manganese/zinc/iron transport system permease protein; membrane transport
K02032 - peptide/nickel transport system ATP-binding protein; membrane transport
K02009 - cobalt/nickel transport protein; membrane transport
K01995 - branched-chain amino acid transport system ATP-binding protein; membrane transport
K03839 - flavodoxin I; eneregy metabolism
K11707 - manganese/zinc/iron transport system substrate-binding protein; membrane transport
K03856 - 3-deoxy-7-phosphoheptulonate synthase [EC:2.5.1.54]; amino acid metabolism

_PC2-:_
K03060 - DNA-directed RNA polymerase subunit omega [EC:2.7.7.6]; transcription
K02036 - phosphate transport system ATP-binding protein [EC:7.3.2.1]; membrane transport
K03402 - transcriptional regulator of arginine metabolism; transcription factor
K01626 - 3-deoxy-7-phosphoheptulonate synthase [EC:2.5.1.54]; amino acid metabolism
K00969 - nicotinate-nucleotide adenylyltransferase [EC:2.7.7.18]; metabolism of cofactors and vitamins
K07166 - ACT domain-containing protein; signaling protein (gvcR)
K02529 - LacI family transcriptional regulator; transcription factor
K02970 - small subunit ribosomal protein S21
K05896 - segregation and condensation protein A; DNA replication
K00016 - L-lactate dehydrogenase [EC:1.1.1.27]; carbohydrate metabolism

```{r}

```







