---
title: "Cameroon hunter-gatherer calculus/plaque HUMAnN2"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries}
library(knitr)
library(decontam)
library(data.table)
library(mixOmics)
library(janitor)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

```{r colors_shapes_peachpear}
# lacroix_palette("PeachPear", n = 8, type = "continuous")
# "#FF3200" "#EF8158" "#E9BD8E" "#CBDDA7" "#38BCAD" "#0B91B5" "#065FA3" "#172869"

Village_colors <- c(`202` = "#FF3200", `204` = "#EF8158", `205` = "#E9BD8E", `203` = "#CBDDA7", `206` = "#38BCAD",
                    HMP = "#172869", HMP10M = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                     ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Village_shapes <- c(`202` = 15, `203` = 16, `204` = 17,`205` = 18, `206` = 25, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9,
                    ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Ethnic_Group_colors <- c(Baka = "#FF3200", Nzime = "#172869", HMP = "#969696", HMP10M = "#d9d9d9", JAE10M = "#ffeda0", JAE = "#feb24c",
                         ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Ethnic_Group_shapes <- c(Baka = 19, Nzime = 17,  HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9,
                         ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Env_colors <- c(HunterGatherer = "#FF3200", HMP = "#172869", HMP10M = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Env_shapes <- c(HunterGatherer = 19, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Market_economy_colors <- c(logging_road = "#FF3200", forrest_camp = "#172869", farming = "#82CDAA", Industrial = "#969696",
                           ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Market_economy_shapes <- c(logging_road = 19, forrest_camp = 17, farming = 18, HMP = 7, HMP10M = 9, Industrial = 15,
                           ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Age_group_colors <- c(`18-29` = "#FF3200", `30-39` = "#E9B186", `40-49` = "#82CDAA", `50-59` = "#0686B8", `60+` = "#172869",
                      HMP = "#969696", HMP10M = "#d9d9d9",
                      ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Age_group_shapes <- c(`18-29` = 15, `30-39` = 16, `40-49` = 17, `50-59` = 18, `60+` = 25, HMP = 7, HMP10M = 9,
                      ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Sex_colors <- c(Female = "#FF3200", Male = "#172869", HMP = "#969696", HMP10M = "#d9d9d9",
                ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Sex_shapes <- c(Female = 19, Male = 17,  HMP = 7, HMP10M = 9,ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Tooth_site_colors <- c(Anterior = "#FF3200", Posterior = "#172869", HMP = "#969696", HMP10M = "#d9d9d9",
                       ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Tooth_site_shapes <- c(Anterior = 19, Posterior = 17,  HMP = 7, HMP10M = 9,
                       ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

```

# HUMAnN2 assignment stats
The number of reads assigned in each sample appears very low (~50%). We want to check this so we know how many were assinged in each sample. We used the cleaned and merged reads that didn't align to the human genome (hg19), with the following database: /projects1/malt/databases/indexed/index040/full-nt_2017-10. 
```{r load_data}
## load the species and genus tables generated in MEGAN
humann2_path <- fread("./05-results.backup/humann2.pathabundance.all.tss.tsv")
humann2_path <- as_tibble(humann2_path)
humann2_KOs_full <- fread("./05-results.backup/humann2.genefamilies.all.i.tss.renorm.ko.tsv")
humann2_KOs_full <- as_tibble(humann2_KOs_full)

# clean the file names
humann2_KOs_full <- rename(humann2_KOs_full, Ortholog = `# Gene Family`)
colnames(humann2_KOs_full) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M_Abundance-RPKs","_10M", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","_10M", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance-RPKs","", colnames(humann2_KOs_full))

humann2_path <- rename(humann2_path, Pathway = `# Pathway`)
colnames(humann2_path) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M_Abundance","_10M", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","_10M", colnames(humann2_path))
colnames(humann2_path) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam_Abundance","", colnames(humann2_path))

# remove unmapped and ungrouped reads
humann2_KOs <- humann2_KOs_full %>% filter(!str_detect(Ortholog, "UNMAPPED"))
humann2_KOs <- humann2_KOs_full %>% filter(!str_detect(Ortholog, "UNGROUPED"))

```

```{r load_metadata}

metadata <- fread("./00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata <- metadata %>%
  rename(SampleID = `#SampleID`) %>%
  as_tibble() %>%
  mutate(Description = fct_relevel(Description, "HunterGatherer", "HMP_supPlaque", "HMP_subPlaque",
                                   "HMP10M_supPlaque", "HMP10M_subPlaque","JAE","JAE10M",
                                   "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime","HMP","HMP10M","JAE","JAE10M",
                                     "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Market_economy = fct_relevel(Market_economy, "logging_road","forrest_camp","farming","Industrial",
                                      "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Age_group = fct_relevel(Age_group, "18-29","30-39","40-49","50-59","60+","HMP","HMP10M"),
         Sex = fct_relevel(Sex, "Male","Female","HMP","HMP10M"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior","HMP","HMP10M"),
         Type = fct_relevel(Type, "Calculus","Plaque","Blank"),
         Village = fct_relevel(Village, "202","204","205","203","206","ExtractionGauze","ExtractionBlank",
                               "LibraryBlank","HMP","HMP10M","JAE","JAE10M"))

# list out the full SRR (HMP) and JAE samples to separate them from the set subsampled to 10M reads
fullSRR <- metadata %>%
  select(SampleID) %>%
  filter(str_detect(SampleID, "SRR")) %>%
  filter(!str_detect(SampleID, "10M")) %>%
  pull

fullJAE <- metadata %>%
  select(SampleID) %>%
  filter(str_detect(SampleID, "JAE")) %>%
  filter(!str_detect(SampleID, "10M")) %>%
  pull

```

Before we proceed with looking at differences in abundance of different enzymes, let's compare the how many reads were identified by HUMAnN2 for each sample type. With the first step we'll read in the default HUMAnN2 output table with all UniRef90 assignments. We'll select out just the rows with no species assignemtns so we can calculate total assignments. HUMAnN2 uses *reads per kilobase* for the initial output, so we'll be calculating the percent of total assigned/mapped reads per kilobase (RPK). Then write out the file with just the unmapped and total reads assigned so we never have to load this huge file again. This has to be done with a table for which the samples were renormed individually and then joined, and not renormed after they were joined (with HUMAnN2 scripts). 

```{r functions}

# add functions for:
# making the input table for PCA?

# plotting PCA
plot_pca <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group, shape = metadata_group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(pc1, " - ", exp_var[1])) +
                        ylab(paste(pc2, " - ", exp_var[2])) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

```

Plot the total number of assigned reads in each subsistence strategy, and the proportion of assigned reads in each group. 
```{r}

```

# Pathway analysis
Start with a PCA to see if there are any outlier samples that should be removed
```{r}

humann2_path_mat <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Pathway, CPM)

rownames(humann2_path_mat) <- humann2_path_mat$SampleID
humann2_path_mat <- as.matrix(humann2_path_mat %>%
  select(-SampleID)) + 1

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path.pca <- pca(humann2_path_mat, ncomp = 4, logratio = 'CLR')


pca_test_plot <- plot_pca(humann2_path.pca, "PC1", "PC2", "Ethnic_Group")
pca_test_plot
```


```{r}

```


# Community-level functional analysis with individual gene families assigned KEGG Orthology numbers
## KEGG ortholog distribution
We will start analysis of KEGG orthologs by looking at the total number of KOs assigned to each population to see if this may bias further analyses. The proportion of genefamilies that were re-grouped by HUMAnN2 script into KOs is consistent and high across all sample types, which is good b/c we have no major population/subsistence strategy biases. 
```{r, eval = F}
plot_pca <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group, shape = metadata_group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(pc1, " - ", exp_var[1])) +
                        ylab(paste(pc2, " - ", exp_var[2])) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

pca_test_plot <- plot_pca(humann2_path.pca, "PC1", "PC2", "Village")
pca_test_plot

```





```{r}

exp_var <- paste0(round(humann2_path.pca$explained_variance * 100, 2), "%")
df_X <- humann2_path.pca$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")


pca_plot <- ggplot(df_X, aes(PC1, PC2, colour = Ethnic_Group, shape = Ethnic_Group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = Ethnic_Group_colors) +
                        scale_shape_manual(values = Ethnic_Group_shapes) +
                        # stat_ellipse() +
                        xlab(paste("PC1 - ", exp_var[1])) +
                        ylab(paste("PC2 - ", exp_var[2])) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())


pca_plot
```








