---
title: "Cameroon hunter-gatherer calculus/plaque HUMAnN2"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---

```{r load_libraries, echo = F, message = F}
library(knitr)
library(decontam)
library(data.table)
library(mixOmics)
library(janitor)
library(tidyverse)
library(gplots)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

```{r functions}

# add functions for:
# making the input table for PCA?

# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, color_group, shape_group, size_group) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))
    metadata_group_size <- get(paste(size_group, "_size", sep = ""))

    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_integration, Village, Tooth_site, Type), by = "SampleID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]
    size_group = df_X[[size_group]]
                      
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group, size = size_group)) +
                        geom_point() +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        scale_size_manual(values = metadata_group_size) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

```

```{r pca_names}
# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_integration, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                      # select the correct PC column and explained variance for PC1
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$SampleID), size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}


```

```{r biplots}
# this function plots the PCA with biplots to look at the categories with strongest loadings
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle, pathways = c("TRUE","FALSE")) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_integration, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }

    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Function") %>%
      top_n(10, !!arrow_pc)
   if (pathways == T) {
      pws_10 <- pws_10 %>%
        rename(pathway = Function) %>%
        mutate(Function = sapply(pathway, function(f) {
                                  unlist(str_split(f, ":"))[1]
                           })) 
    }

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Function") %>%
      top_n(-10, !!arrow_pc)
    if (pathways == T) {
      neg_10 <- neg_10 %>%
      rename(pathway = Function) %>%
       mutate(Function = sapply(pathway, function(f) {
                                  unlist(str_split(f, ":"))[1]
                           })) 
    }


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Function),
                   size = 1.5, colour = "grey20", label.padding = 0.2) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Function),
                   size = 1.5, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 10)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

```

```{r load_data}
## load the species and genus tables generated with HUMAnN2
humann2_path <- fread("./05-results.backup/pathabundance_joined_cpm.tsv")
humann2_path <- as_tibble(humann2_path)
humann2_KOs_full <- fread("./05-results.backup/genefamilies_joined_cpm_ur90rxn.tsv.gz")
humann2_KOs_full <- as_tibble(humann2_KOs_full)

# clean the file names
humann2_KOs_full <- rename(humann2_KOs_full, Ortholog = `# Gene Family`)
colnames(humann2_KOs_full) <- gsub(".unmapped_Abundance-RPKs","", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub(".SG1","", colnames(humann2_KOs_full))

humann2_path <- rename(humann2_path, Pathway = `# Pathway`)
colnames(humann2_path) <- gsub(".unmapped_Abundance","", colnames(humann2_path))
colnames(humann2_path) <- gsub(".SG1","", colnames(humann2_path))

# remove unmapped and ungrouped reads
humann2_KOs <- humann2_KOs_full %>% filter(Ortholog != "UNMAPPED") %>% filter(!str_detect(Ortholog, "UNGROUPED"))

```


```{r load_metadata}

load("./05-results.backup/CMC_metadata.RData")

```

# Pathways

First perform a PCA to see if there are any samples that plot with the controls
and should be removed as outliers.
```{r pca_paths_with_controls}
# create the input matrix
humann2_path_mat <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-matches("SRR164|ERR|SRS")) %>%
  filter(!str_detect(Pathway, "UNMAPPED|UNINTEGRATED")) %>%
  gather("SampleID","CPM", 2:ncol(.)) %>%
  mutate(CPM = CPM + 1) %>%
  spread(Pathway, CPM) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path_allsamples.pca <- pca(humann2_path_mat, ncomp = 4, logratio = 'CLR')

# plot the PCA
h3plot <- plot_pca(humann2_path_allsamples.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
h3plot

plot_pca_names(humann2_path_allsamples.pca, "PC1", "PC2", "Ethnic_Group")

```
None of the samples plot with controls, so there are no outliers to remove. 
Four of the gauze extraction blanks look like they are oral, and probably had
saliva or plaque that wasn't visible on the control strip that was cut. Now
repeat the PCA without any of the controls to compare the samples to known oral
plaque and calculus profiles. If the sub-sampled plaque (HMP) and calculus (JAE)
samples are similar (10M), then use just the full samples and not the
sub-sampled ones for all further analysis.

```{r pca_paths_no_blanks}

# create the input matrix
humann2_path_mat <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-matches("SRR164|ERR|SRS|EXB|LIB")) %>%
  filter(!str_detect(Pathway, "UNMAPPED|UNINTEGRATED")) %>%
  gather("SampleID","CPM", 2:ncol(.)) %>%
  mutate(CPM = CPM + 1) %>%
  spread(Pathway, CPM) %>%
  filter(!str_detect(SampleID, "EXB|LIB")) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.humann2_path_pca <- tune.pca(humann2_path_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_path.pca <- pca(humann2_path_mat, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(humann2_path.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

plot_pca(humann2_path.pca, "PC1", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")


```

We want to know if there are any pathways that can be removed b/c they appear to
be contaminants. Use the package decontam to determine which species may be
contaminants. Prepare the matrix for decontam.
```{r decontam_path_data_prep}
# list out the samples that have all 0 or NA values to remove from the input matrix
NA_filter_list <- metadata %>% filter((is.na(SumLibraryConcentration_copies_ul) | SumLibraryConcentration_copies_ul <= 0)) %>% select(SampleID)

# remove NA value samples from the input data frame
humann2_path_mat <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-matches("SRR164|ERR|SRS")) %>%
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Pathway, CPM) %>%
  anti_join(., NA_filter_list)

# make a matrix for decontam
humann2_path_mat.matrix <- data.matrix(humann2_path_mat[2:ncol(humann2_path_mat)])
rownames(humann2_path_mat.matrix) <- humann2_path_mat$SampleID

# make a metadata dataframe with the same NA-containing samples removed
metadata.decontam <- metadata %>%
  anti_join(., NA_filter_list, by = "SampleID") %>%
  filter(SampleID != "CMC032.B0101") %>%
  filter(!str_detect(SampleID, "SRR164|ERR|SRS"))

metadata.decontam %>% select(SampleID) %>% anti_join(., humann2_path_mat %>% select(SampleID))

```

Make a protein abundance curve for the samples based on Env to get an idea of
the number of proteins identified in each prior to filtering, and how long the
tail is.
```{r path_rank_abundance_curves}

humann2_path_matL <- humann2_path_mat %>%
  filter(str_detect(SampleID, "CMC|EXB|LIB")) %>%
  gather("Pathway","CPM",2:ncol(.)) %>%
  inner_join(., metadata.decontam %>%
               select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_integration, Village, Tooth_site, Type))

Envmt <- unique(humann2_path_matL$Description)

pathway_orders <- lapply(Envmt, function(envmt) {
humann2_path_matLL_check <- humann2_path_matL %>%
  filter(Description == envmt) %>%
  group_by(Pathway) %>%
  summarise(mean_cpm = mean(CPM)) %>%
  arrange(desc(mean_cpm)) %>%
  filter(mean_cpm > 0) %>%
  rownames_to_column("order") %>%
  mutate(order = as.numeric(order)) %>%
  mutate(Description = envmt)
}) %>%
  bind_rows(.)

# define a function to plot the species abundance curves  
pathway_abundance_curves <- function(df, envt, title_text) {
      options(scipen = 10)
      
    species_abundance_plot <- pathway_orders %>%
        filter(Description == envt) %>%
           ggplot(., aes(x=order, y=mean_cpm)) +
           geom_point() +
           theme_minimal() +
           scale_y_log10() +
           geom_hline(yintercept=10, linetype="dashed", color = "red") +
           # xlim(NA, 600) +
           # ylim(0.0001, NA) +
           # coord_cartesian(xlim = c(0, 600), ylim = c(0.0001, 1)) +
           ylab("Mean copies per million") +
           xlab("protein order") +
           ggtitle(title_text)
  
    return(species_abundance_plot)
}

path_curve_HG <- pathway_abundance_curves(species_orders, "CMC", "CMC - HUMAnN2 Paths")
path_curve_EG <- pathway_abundance_curves(species_orders, "ExtractionGauze", "Extraction Gauze - HUMAnN2 Paths")
path_curve_EB <- pathway_abundance_curves(species_orders, "ExtractionBlank", "Extraction Blank - HUMAnN2 Paths")
path_curve_LB <- pathway_abundance_curves(species_orders, "LibraryBlank", "Library Blank - HUMAnN2 Paths")

plot_grid(path_curve_HG, path_curve_EG,
          path_curve_EB, path_curve_LB,
          nrow=2, labels = c("A", "B", "C", "D"), rel_widths = c(1, 1))


```

Now we're ready to run Decontam
USe the prevalence method.
```{r decontam_path_prevalence}
# Assign sample-type status to identify control samples
metadata.decontam$is.control <- metadata.decontam$Sample_or_Control=="Control"

#Try changing the probability threshold cut-off to see how many more or less are ID'd as contaminants
contam.prev.species <- isContaminant(humann2_path_mat.matrix, method="prevalence", 
                                 neg=metadata.decontam$is.control, 
                                 threshold = 0.2)

# Check number of contaminants
table(contam.prev.species$contaminant)

# List of contaminants
cont_prev_path_list <- contam.prev.species[which(contam.prev.species$contaminant=="TRUE"), ]

# Look at the histogram - does the probability threshold need to be changed?
ggplot(contam.prev.species, aes(x=p)) +
  geom_histogram(bins=100) +
  scale_y_log10() +
  geom_hline(yintercept=10, linetype="dashed", color = "red") +
  geom_hline(yintercept=2, linetype="dashed", color = "red") +
  geom_vline(xintercept=0.2, linetype="dashed", color = "red") +
  ggtitle('Prevalence - HUMAnN2 pathways')

```


Check the rank abundance curves to see how changing the cut-off threshold
changes the tail. This set of plots removes the proteins determined to be
"contaminants" by decontam, using the list created from which ever method is
selected.
```{r decontam_path_rank_abund_curve}

contaminants_h2_path <- cont_prev_path_list %>% rownames_to_column("Pathway") %>% select(Pathway) # this uses the prevalence method contaminant list
humann2_path.decontam <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(!contains("SRR164")) %>%
  anti_join(., contaminants_h2_path)

# write.table(malt_species.decontam, file = "05-results.backup/malt_species.decontam.tsv", sep="\t", quote=F, row.names = F)
# write.table(malt_species, "05-results.backup/malt_species.tsv", sep="\t", quote=F, row.names = F)

humann2_path.decontamL <- humann2_path.decontam %>%
  gather("SampleID","CPM",2:ncol(.)) %>%
  inner_join(., metadata %>%
              select(SampleID, Env, Description), by = "SampleID")

Envmt_decontam <- unique(humann2_path.decontamL$Description)

h2path_decontam_orders <- lapply(Envmt_decontam, function(envmt) {
humann2_path.decontamL_check <- humann2_path.decontamL %>%
  filter(Description == envmt) %>%
  group_by(Pathway) %>%
  summarise(mean_cpm = mean(CPM)) %>%
  arrange(desc(mean_cpm)) %>%
  filter(mean_cpm > 0) %>%
  rownames_to_column("order") %>%
  mutate(order = as.numeric(order)) %>%
  mutate(Description = envmt)
}) %>%
  bind_rows(.)
  
h2path_decontam_abundance_curves <- function(df, envt, title_text) {
      options(scipen = 10)
      
    path_abundance_plot <- h2path_decontam_orders %>%
        filter(Description == envt) %>%
           ggplot(., aes(x=order, y=mean_cpm)) +
           geom_point() +
           theme_minimal() +
           scale_y_log10() +
           geom_hline(yintercept=10, linetype="dashed", color = "red") +
           ylab("mean copies per million") +
           xlab("protein order") +
           ggtitle(title_text) + 
           theme(plot.title = element_text(size = 8))
  
    return(path_abundance_plot)
}

h2path_decontam_curve_HG <- h2path_decontam_abundance_curves(h2path_decontam_orders, "CMC", "CMC, decontam")
h2path_decontam_curve_EG <- h2path_decontam_abundance_curves(h2path_decontam_orders, "ExtractionGauze", "EG, decontam")
h2path_decontam_curve_EB <- h2path_decontam_abundance_curves(h2path_decontam_orders, "ExtractionBlank", "EB, decontam")
h2path_decontam_curve_LB <- h2path_decontam_abundance_curves(h2path_decontam_orders, "LibraryBlank", "LB, decontam")
h2path_decontam_curve_HMPsup <- h2path_decontam_abundance_curves(h2path_decontam_orders, "HMP_supPlaque", "HMPsup, decontam")
h2path_decontam_curve_HMPsub <- h2path_decontam_abundance_curves(h2path_decontam_orders, "HMP_subPlaque", "HMPsub, decontam")
h2path_decontam_curve_JAE <- h2path_decontam_abundance_curves(h2path_decontam_orders, "JAE", "JAE, decontam")

plot_grid(h2path_decontam_curve_HG, h2path_decontam_curve_EG, h2path_decontam_curve_EB, h2path_decontam_curve_LB, 
          h2path_decontam_curve_HMPsup, h2path_decontam_curve_HMPsub, h2path_decontam_curve_JAE,
          nrow=2, labels = c("A", "B", "C", "D", "E", "F", "G"), rel_widths = c(1, 1))

```

# KEGG Orthologs
```{r pca_KOs_with_controls}
# create the input matrix
humann2_KO_mat <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-matches("SRR164|ERR|SRS")) %>%
  gather("SampleID","CPM", 2:ncol(.)) %>%
  mutate(CPM = CPM + 1) %>%
  spread(Ortholog, CPM) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.humann2_KO_pca <- tune.pca(humann2_KO_mat, logratio = 'CLR')

# perform a PCA to see how the data cluster
humann2_KO_allsamples.pca <- pca(humann2_KO_mat, ncomp = 4, logratio = 'CLR')

# plot the PCA
plot_pca(humann2_KO_allsamples.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

plot_pca_names(humann2_KO_allsamples.pca, "PC1", "PC2", "Ethnic_Group")


```

We want to know if there are any species that can be removed b/c they appear to be contaminants. Use the package decontam to determine which species may be contaminants. Prepare the matrix for decontam. 
```{r decontam_KO_data_prep}
# list out the samples that have all 0 or NA values to remove from the input matrix
NA_filter_list <- metadata %>% filter((is.na(SumLibraryConcentration_copies_ul) | SumLibraryConcentration_copies_ul <= 0)) %>% select(SampleID)

# remove NA value samples from the input data frame
humann2_KO_mat <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(-matches("SRR164|ERR|SRS")) %>%
  gather("SampleID","CPM", 2:ncol(.)) %>%
  spread(Ortholog, CPM) %>%
  anti_join(., NA_filter_list)

# make a matrix for decontam
humann2_KO_mat.matrix <- data.matrix(humann2_KO_mat[2:ncol(humann2_KO_mat)])
rownames(humann2_KO_mat.matrix) <- humann2_KO_mat$SampleID

# make a metadata dataframe with the same NA-containing samples removed
metadata.decontam <- metadata %>%
  anti_join(., NA_filter_list, by = "SampleID") %>%
  filter(SampleID != "CMC032.B0101") %>%
  select(-matches("SRR164|ERR|SRS"))

```

Make a protein abundance curve for the samples based on Env to get an idea of
the number of proteins identified in each prior to filtering, and how long the
tail is.
```{r KO_rank_abundance_curves}

humann2_KO_matL <- humann2_KO_mat %>%
  filter(!str_detect(SampleID, "SRR|JAE")) %>%
  gather("Ortholog","CPM",2:ncol(.)) %>%
  inner_join(., metadata.decontam %>%
               select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_integration, Village, Tooth_site, Type))

Envmt <- unique(humann2_KO_matL$Description)

pathway_orders <- lapply(Envmt, function(envmt) {
humann2_KO_matLL_check <- humann2_KO_matL %>%
  filter(Description == envmt) %>%
  group_by(Ortholog) %>%
  summarise(mean_cpm = mean(CPM)) %>%
  arrange(desc(mean_cpm)) %>%
  filter(mean_cpm > 0) %>%
  rownames_to_column("order") %>%
  mutate(order = as.numeric(order)) %>%
  mutate(Description = envmt)
}) %>%
  bind_rows(.)

# define a function to plot the species abundance curves  
pathway_abundance_curves <- function(df, envt, title_text) {
      options(scipen = 10)
      
    species_abundance_plot <- pathway_orders %>%
        filter(Description == envt) %>%
           ggplot(., aes(x=order, y=mean_cpm)) +
           geom_point() +
           theme_minimal() +
           scale_y_log10() +
           geom_hline(yintercept=0.1, linetype="dashed", color = "red") +
           # xlim(NA, 600) +
           # ylim(0.0001, NA) +
           # coord_cartesian(xlim = c(0, 600), ylim = c(0.0001, 1)) +
           ylab("Mean copies per million") +
           xlab("protein order") +
           ggtitle(title_text)
  
    return(species_abundance_plot)
}

path_curve_HG <- pathway_abundance_curves(species_orders, "CMC", "CMC - KEGG Orthologs")
path_curve_EG <- pathway_abundance_curves(species_orders, "ExtractionGauze", "Extraction Gauze - KEGG Orthologs")
path_curve_EB <- pathway_abundance_curves(species_orders, "ExtractionBlank", "Extraction Blank - KEGG Orthologs")
path_curve_LB <- pathway_abundance_curves(species_orders, "LibraryBlank", "Library Blank - KEGG Orthologs")

plot_grid(path_curve_HG, path_curve_EG,
          path_curve_EB, path_curve_LB,
          nrow=2, labels = c("A", "B", "C", "D"), rel_widths = c(1, 1))


```

Now we're ready to run Decontam
USe the prevalence method.
```{r decontam_KO_prevalence}
# Assign sample-type status to identify control samples
metadata.decontam$is.control <- metadata.decontam$Sample_or_Control=="Control"

#Try changing the probability threshold cut-off to see how many more or less are ID'd as contaminants
contam.prev.species <- isContaminant(humann2_KO_mat.matrix, method="prevalence", 
                                 neg=metadata.decontam$is.control, 
                                 threshold = 0.5)

# Check number of contaminants
table(contam.prev.species$contaminant)

# List of contaminants
cont_prev_KO_list <- contam.prev.species[which(contam.prev.species$contaminant=="TRUE"), ]

# Look at the histogram - does the probability threshold need to be changed?
ggplot(contam.prev.species, aes(x=p)) +
  geom_histogram(bins=100) +
  scale_y_log10() +
  geom_hline(yintercept=10, linetype="dashed", color = "red") +
  geom_hline(yintercept=2, linetype="dashed", color = "red") +
  geom_vline(xintercept=0.5, linetype="dashed", color = "red") +
  ggtitle('Prevalence - HUMAnN2 pathways')

```


Check the rank abundance curves to see how changing the cut-off threshold
changes the tail. This set of plots removes the proteins determined to be
"contaminants" by decontam, using the list created from which ever method is
selected.
```{r decontam_KO_rank_abund_curve}

contaminants_h2_KO <- cont_prev_KO_list %>% rownames_to_column("Ortholog") %>% select(Ortholog) # this uses the prevalence method contaminant list
humann2_KO.decontam <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(!contains("SRR164")) %>%
  anti_join(., contaminants_h2_KO)

# write.table(malt_species.decontam, file = "05-results.backup/malt_species.decontam.tsv", sep="\t", quote=F, row.names = F)
# write.table(malt_species, "05-results.backup/malt_species.tsv", sep="\t", quote=F, row.names = F)

humann2_KO.decontamL <- humann2_KO.decontam %>%
  gather("SampleID","CPM",2:ncol(.)) %>%
  inner_join(., metadata %>%
              select(SampleID, Env, Description), by = "SampleID")

Envmt_decontam <- unique(humann2_KO.decontamL$Description)

h2path_decontam_orders <- lapply(Envmt_decontam, function(envmt) {
humann2_KO.decontamL_check <- humann2_KO.decontamL %>%
  filter(Description == envmt) %>%
  group_by(Ortholog) %>%
  summarise(mean_cpm = mean(CPM)) %>%
  arrange(desc(mean_cpm)) %>%
  filter(mean_cpm > 0) %>%
  rownames_to_column("order") %>%
  mutate(order = as.numeric(order)) %>%
  mutate(Description = envmt)
}) %>%
  bind_rows(.)
  
h2path_decontam_abundance_curves <- function(df, envt, title_text) {
      options(scipen = 10)
      
    path_abundance_plot <- h2path_decontam_orders %>%
        filter(Description == envt) %>%
           ggplot(., aes(x=order, y=mean_cpm)) +
           geom_point() +
           theme_minimal() +
           scale_y_log10() +
           geom_hline(yintercept=1, linetype="dashed", color = "red") +
           ylab("mean copies per million") +
           xlab("protein order") +
           ggtitle(title_text) + 
           theme(plot.title = element_text(size = 8))
  
    return(path_abundance_plot)
}

h2path_decontam_curve_HG <- h2path_decontam_abundance_curves(h2path_decontam_orders, "CMC", "CMC, decontam")
h2path_decontam_curve_EG <- h2path_decontam_abundance_curves(h2path_decontam_orders, "ExtractionGauze", "EG, decontam")
h2path_decontam_curve_EB <- h2path_decontam_abundance_curves(h2path_decontam_orders, "ExtractionBlank", "EB, decontam")
h2path_decontam_curve_LB <- h2path_decontam_abundance_curves(h2path_decontam_orders, "LibraryBlank", "LB, decontam")
h2path_decontam_curve_HMPsup <- h2path_decontam_abundance_curves(h2path_decontam_orders, "HMP_supPlaque", "HMPsup, decontam")
h2path_decontam_curve_HMPsub <- h2path_decontam_abundance_curves(h2path_decontam_orders, "HMP_subPlaque", "HMPsub, decontam")
h2path_decontam_curve_JAE <- h2path_decontam_abundance_curves(h2path_decontam_orders, "JAE", "JAE, decontam")

plot_grid(h2path_decontam_curve_HG, h2path_decontam_curve_EG, h2path_decontam_curve_EB, h2path_decontam_curve_LB, 
          h2path_decontam_curve_HMPsup, h2path_decontam_curve_HMPsub, h2path_decontam_curve_JAE,
          nrow=2, labels = c("A", "B", "C", "D", "E", "F", "G"), rel_widths = c(1, 1))

```


```{r decontam_objects}
contaminants_h2_path <- cont_prev_path_list %>% rownames_to_column("Pathway") %>% select(Pathway) # this uses the prevalence method contaminant list
humann2_path.decontam <- humann2_path %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(!contains("SRR164")) %>%
  anti_join(., contaminants_h2_path)

contaminants_h2_KO <- cont_prev_KO_list %>% rownames_to_column("Ortholog") %>% select(Ortholog) # this uses the prevalence method contaminant list
humann2_KOs.decontam <- humann2_KOs %>%
  filter(!str_detect(Ortholog, "\\|")) %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  select(!contains("SRR164")) %>%
  anti_join(., contaminants_h2_KO)


save(humann2_path, humann2_path.decontam, humann2_KOs_full, humann2_KOs.decontam,
     file = "./05-results.backup/CMC_humann2_tables.RData")

```





































