---
title: "Cameroon hunter-gatherer calculus/plaque MALT beta diversity"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(decontam)
library(data.table)
library(psych)
library(ape)
library(mixOmics)
library(compositions)
library(vegan)
library(janitor)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

# MALT read stats
The number of reads assigned in each sample appears very low (~50%). We want to check this so we know how many were assinged in each sample. We used the cleaned and merged reads that didn't align to the human genome (hg19), with the following database: /projects1/malt/databases/indexed/index040/full-nt_2017-10. 
```{r load_data}
## load the species and genus tables generated in MEGAN
# malt_species <- fread("05-results.backup/CMC_MALT_nt_species_bacteria_archaea.txt")
# malt_genus <- fread("05-results.backup/CMC_MALT_nt_genus_bacteria_archaea.txt")

# malt_species <- fread("05-results.backup/CMC_MALT_RSC_species_bacteria_archaea.txt")
# malt_genus <- fread("05-results.backup/CMC_MALT_RSC_genus_bacteria_archaea.txt")

# malt_species <- fread("05-results.backup/CMC_someHMP_MALT_RSC_species_bacteria_archaea.txt")
# malt_genus <- fread("05-results.backup/CMC_someHMP_MALT_RSC_genus_bacteria_archaea.txt")

# malt_species <- fread("./05-results.backup/CMC_HMP_MALT_RSC_species_bacteria_archaea.txt")
# malt_genus <- fread("./05-results.backup/CMC_HMP_MALT_RSC_genus_bacteria_archaea.txt")

# malt_species <- fread("./05-results.backup/CMC_HMP_JAE_MALT_RSC_species_bacteria_archaea.txt")
# malt_genus <- fread("./05-results.backup/CMC_HMP_JAE_MALT_RSC_genus_bacteria_archaea.txt")

malt_species <- fread("./05-results.backup/CMC_HMP_JAE_Yano_MALT_RSC_species_bacteria_archaea.txt")
malt_genus <- fread("./05-results.backup/CMC_HMP_JAE_Yano_MALT_RSC_genus_bacteria_archaea.txt")

# clean the file names
malt_species <- rename(malt_species, Species = `#Datasets`)
colnames(malt_species) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M","_10M", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam","_10M", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub(".fastq.truncated","", colnames(malt_species))


# remove human and unassigned reads
malt_species <- malt_species %>% filter(!str_detect(Species, "Homo sapiens"))
malt_species <- malt_species %>% filter(!str_detect(Species, "Not assigned"))
# get rid of the extra column
malt_species <- malt_species %>% select(-Comparison_CMC_HMP_JAE_MALT_RSC)

malt_genus <- rename(malt_genus, Species = `#Datasets`)
colnames(malt_genus) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam","_10M", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M","_10M", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub(".fastq.truncated","", colnames(malt_genus))

# remove human and unassigned reads
malt_genus <- malt_genus %>% filter(!str_detect(Species, "Homo sapiens"))
malt_genus <- malt_genus %>% filter(!str_detect(Species, "Not assigned"))
# get rid of the extra column at the begining
malt_genus <- malt_genus %>% select(-Comparison_CMC_HMP_JAE_MALT_RSC)

# load the metadata file
metadata <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata <- metadata %>%
  rename(SampleID = `#SampleID`)  %>%
  as_tibble() %>%
  mutate(Description = fct_relevel(Description, "HunterGatherer", "Yanomami", "HMP_supPlaque", "HMP_subPlaque",
                                   "HMP10M_supPlaque", "HMP10M_subPlaque","JAE","JAE10M",
                                   "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime","Yanomami","HMP","HMP10M","JAE","JAE10M",
                                     "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Env = fct_relevel(Env, "HunterGatherer","HMP","HMP10M","JAE","JAE10M",
                                     "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Market_economy = fct_relevel(Market_economy, "logging_road","forrest_camp","farming","Yanomami",
                                      "Industrial","ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Age_group = fct_relevel(Age_group, "10-18","18-29","30-39","40-49","50-59","60+","HMP","HMP10M"),
         Sex = fct_relevel(Sex, "Male","Female","HMP","HMP10M"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior","Buccal_mucosa","HMP","HMP10M"),
         Type = fct_relevel(Type, "Calculus","Plaque","Blank"),
         Village = fct_relevel(Village, "202","205","204","203","206","Yanomami","HMP","HMP10M","JAE","JAE10M",
                               "ExtractionGauze","ExtractionBlank","LibraryBlank"))

fullSRR <- metadata %>%
  select(SampleID) %>%
  filter(str_detect(SampleID, "SRR")) %>%
  filter(!str_detect(SampleID, "10M")) %>%
  pull
```

```{r colors_shapes_passionfruit, eval = F}
Village_colors <- c(`202` = "#C70E7B", `204` = "#60CB57", `205` = "#577EA1", `203` = "#E68561", `206` = "#172869",
                     ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
village_shapes <- c(`202` = 15, `205` = 18, `204` = 17, `203` = 16, `206` = 25,
                    ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Ethnic_Group_colors <- c(Baka = "#C70E7B", Nzime = "#172869",
                         ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Ethnic_Group_shapes <- c(Baka = 19, Nzime = 17, 
                         ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Env_colors <- c(HunterGatherer = "#C70E7B", ExtractionGauze = "#881C00",
                ExtractionBlank = "#A34D1A", LibraryBlank = "#CC986A")
Env_shapes <- c(HunterGatherer = 19, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

market_economy_colors <- c(logging_road = "#C70E7B", forrest_camp = "#172869", farming = "#60CB57",
                           ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
market_economy_shapes <- c(logging_road = 19, forrest_camp = 17, other = 18,
                           ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

age_group_colors <- c(`18-29` = "#C70E7B", `30-39` = "#E68561", `40-49` = "#60CB57", `50-59` = "#577EA1", `60+` = "#172869",
                      ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
age_group_shapes <- c(`18-29` = 15, `30-39` = 16, `40-49` = 17, `50-59` = 18, `60+` = 25,
                      ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

sex_colors <- c(Female = "#C70E7B", Male = "#172869", ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
sex_shapes <- c(Female = 19, Male = 17, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

tooth_site_colors <- c(Anterior = "#C70E7B", Posterior = "#172869", 
                       ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
tooth_site_shapes <- c(Anterior = 19, Posterior = 17, 
                       ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

```

```{r colors_shapes_peachpear}
# lacroix_palette("PeachPear", n = 8, type = "continuous")
# "#FF3200" "#EF8158" "#E9BD8E" "#CBDDA7" "#38BCAD" "#0B91B5" "#065FA3" "#172869"

Village_colors <- c(`202` = "#FF3200", `205` = "#EF8158", `204` = "#E9BD8E", `203` = "#CBDDA7", `206` = "#38BCAD",
                    Yanomami = "#C70E7B", HMP = "#172869", HMP10M = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                     ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Village_shapes <- c(`202` = 15, `203` = 16, `205` = 17,`204` = 18, `206` = 25, Yanomami = 14, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9,
                    ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Ethnic_Group_colors <- c(Baka = "#FF3200", Nzime = "#172869", Yanomami = "#C70E7B", HMP = "#969696", HMP10M = "#d9d9d9", 
                         JAE10M = "#ffeda0", JAE = "#feb24c",ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Ethnic_Group_shapes <- c(Baka = 19, Nzime = 17,  Yanomami = 14, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9,
                         ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Env_colors <- c(HunterGatherer = "#FF3200", HMP = "#172869", HMP10M = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Env_shapes <- c(HunterGatherer = 19, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Description_colors <- c(HunterGatherer = "#FF3200", Yanomami = "#C70E7B", HMP_supPlaque = "#172869", HMP_subPlaque = "#172869", 
                        HMP10M_supPlaque = "#065FA3", HMP10M_subPlaque = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                        ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Description_shapes <- c(HunterGatherer = 19, Yanomami = 14, HMP_supPlaque = 7, HMP_subPlaque = 7, HMP10M_supPlaque = 9, 
                        HMP10M_subPlaque = 9, JAE = 7, JAE10M = 9, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Market_economy_colors <- c(logging_road = "#FF3200", forrest_camp = "#172869", farming = "#82CDAA", Yanomami = "#C70E7B", Industrial = "#969696",
                           ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Market_economy_shapes <- c(logging_road = 19, forrest_camp = 17, farming = 18, Yanomami = 14, HMP = 7, HMP10M = 9, Industrial = 15,
                           ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Age_group_colors <- c(`18-29` = "#FF3200", `30-39` = "#E9B186", `40-49` = "#82CDAA", `50-59` = "#0686B8", `60+` = "#172869",
                      `10-18` = "#C70E7B", HMP = "#969696", HMP10M = "#d9d9d9",
                      ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Age_group_shapes <- c(`18-29` = 15, `30-39` = 16, `40-49` = 17, `50-59` = 18, `60+` = 25, `10-18` = 14, HMP = 7, HMP10M = 9,
                      ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Sex_colors <- c(Female = "#FF3200", Male = "#172869", HMP = "#969696", HMP10M = "#d9d9d9",
                ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Sex_shapes <- c(Female = 19, Male = 17,  HMP = 7, HMP10M = 9,ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Tooth_site_colors <- c(Anterior = "#FF3200", Posterior = "#172869", Buccal_mucosa = "#C70E7B", HMP = "#969696", HMP10M = "#d9d9d9",
                       ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Tooth_site_shapes <- c(Anterior = 19, Posterior = 17,  Buccal_mucosa = 14, HMP = 7, HMP10M = 9,
                       ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

```

```{r pca_plotting_functions}
# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group, shape = metadata_group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$SampleID), size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}


# plot PCA with colored dots and the title including the # of species or genera
plot_pca_titles <- function(df, pc1, pc2, color_group, shape_group, ncomps) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$explained_variance * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]
   
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    title_text <- df %>%
                     rownames_to_column("SampleID") %>%
                     gather("Species","Counts",2:ncol(.)) %>%
                     spread(SampleID, Counts) %>%
                     count
    
    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        ggtitle(paste(title_text))

    return(pca_plot)
}


```

```{r pca_biplot_functions}
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$explained_variance * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }
    
    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(10, !!arrow_pc)

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(-10, !!arrow_pc)


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 10)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

# pca_test_biplot <- plot_pca_bi(malt_species.pca, "PC3", "PC2", "Village", PC2)
# pca_test_biplot


```


Before running decontam, let's plot the samples in a PCA to see if there are any that are clearly not good-quality
```{r raw_plot}

malt_species_raw <- malt_species %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species_raw) <- malt_species_raw$SampleID

malt_species_raw <- malt_species_raw %>% select(-SampleID)

malt_species_raw = malt_species_raw+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.malt_species_pca <- tune.pca(malt_species_raw, logratio = 'CLR')
tune.malt_species_pca

# perform a PCA to see how the data cluster
malt_species.pca <- mixOmics::pca(malt_species_raw, ncomp = 3, logratio = 'CLR')
plot(malt_species.pca)


plot_pca(malt_species.pca, "PC1", "PC2", "Description")
plot_pca_names(malt_species.pca,  "PC1", "PC2", "Description")

```

```{r sipca_full_data}
# use the clr-transformed input matrix from the PCA above
malt_species_raw.sipca <- sipca(malt_species.pca$X, scale = F, ncomp = 3, keepX = c(10,10,10)) # 50,50,50 10,10,10 100,100,100

malt_species_raw.sipca.table <- as.data.frame(malt_species_raw.sipca$x)
malt_species_raw.sipca.table <- malt_species_raw.sipca.table %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>% select(SampleID, Ethnic_Group, Description, Age_group, Market_economy, Sex, Tooth_site, Village))
  
malt_species_raw.sipca.table %>%
  ggplot(., aes(`IPC 1`, `IPC 2`, colour = Ethnic_Group, shape = Ethnic_Group)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Ethnic_Group_colors) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("sIPCA")

head(selectVar(malt_species_raw.sipca, comp = 1)$value)
malt_species_raw.sipca$loadings$X %>% as.data.frame(.) %>% rownames_to_column("species") %>% filter(str_detect(species, "Prevotella")) %>% filter(V1 > 0)

```

```{r plsda_full_data}
malt_species_raw.plsda <- plsda(malt_species_raw, metadata$Ethnic_Group, ncomp = 3, logratio = "CLR")

malt_species_raw.plsda.table <- as.data.frame(malt_species_raw.plsda$variates$X)
malt_species_raw.plsda.table <- malt_species_raw.plsda.table %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>% select(SampleID, Ethnic_Group, Description, Age_group, Market_economy, Sex, Tooth_site, Village))

malt_species_raw.plsda.table %>%
  ggplot(., aes(comp1, comp2, colour = Ethnic_Group, shape = Ethnic_Group)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Ethnic_Group_colors) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("PLS-DA")

malt_species_raw.plsda$loadings$X %>%
  as.data.frame(.) %>%
  bind_cols(., malt_species_raw.plsda$names$colnames$X %>%
              as.data.frame(.)) %>%
  rename(Species = ".") %>%
  select(Species, everything()) %>%
  arrange((comp2))

```


List the outliers from the plot in section `raw_plot` in `Cameroon_h-g_taxonomy.Rmd` and remove them from the taxonomy table. 
```{r outliers}
outliers <- c("CMC002.B0201","CMC032.B0101","CMC037.B0201","CMC047.A0201",
              "EXB059.A1901","EXB059.A1301","EXB059.A1101","EXB059.A0901",
              "SRR1646041","SRR1646032","SRR1646042","SRR1646045","SRR1646030","SRR1646047")

outliersF <- str_c(outliers, collapse = "|")

```

Read in the contaminant list from `Cameroon_taxonomy_decontam.Rmd` and remove them from the taxonomy table. 
```{r contaminants}

contaminants_species <- fread("05-results.backup/malt_contaminants_species.tsv", header = T, sep = "\t")
contaminants_genus <- fread("05-results.backup/malt_contaminants_genus.tsv")

malt_species.decontam <- malt_species %>%
  anti_join(., contaminants_species)

malt_genus.decontam <- malt_genus %>%
  anti_join(., contaminants_genus)


```

#Sample Clustering with PCA

Let's do a quick and not at all enough data filtering so we can run a quick PCA to see if there are trends
```{r pca_all_species}
# unfiltered and not decontaminated
malt_species_unfiltered <- malt_species %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("10M")) %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species_unfiltered) <- malt_species_unfiltered$SampleID

malt_species_unfiltered <- malt_species_unfiltered %>% select(-SampleID)

malt_species_unfiltered = malt_species_unfiltered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.species_pca <- tune.pca(malt_species_unfiltered, logratio = 'CLR')
tune.species_pca

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_species_unfiltered, "PC1", "PC2", "Description", "Env", 3)
plot_pca_titles(malt_species_unfiltered, "PC3", "PC2", "Description", "Env", 3)


# ggsave("05-results.backup/pca_species_unfiltered_pc1pc2.pdf", plot = pca_species_unfiltered_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)


# filtered only
malt_species_filtered <- malt_species %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("10M")) %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.005) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species_filtered) <- malt_species_filtered$SampleID

malt_species_filtered <- malt_species_filtered %>% select(-SampleID)

malt_species_filtered = malt_species_filtered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.species_pca <- tune.pca(malt_species_filtered, logratio = 'CLR')
tune.species_pca

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_species_filtered, "PC1", "PC2", "Description", "Env", 3)
plot_pca_titles(malt_species_filtered, "PC3", "PC2", "Description", "Env", 3)


# decontaminated only
malt_species.decontam_unfilt <- malt_species.decontam %>%
                                         adorn_totals(where = "col", name = "Sum_species.decontam") %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("10M")) %>%
                                         mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
                                         select(-one_of(c("Sum_species.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_unfilt) <- malt_species.decontam_unfilt$SampleID

malt_species.decontam_unfilt <- malt_species.decontam_unfilt %>% select(-SampleID)

malt_species.decontam_unfilt = malt_species.decontam_unfilt+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.species.decontam_pca <- tune.pca(malt_species.decontam_unfilt, logratio = 'CLR')
tune.species.decontam_pca

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_species.decontam_unfilt, "PC1", "PC2", "Description", "Env", 3)
plot_pca_titles(malt_species.decontam_unfilt, "PC3", "PC2", "Description", "Env", 3)

# decontaminated and filtered
malt_species.decontam_filtered <- malt_species.decontam %>%
                                         adorn_totals(where = "col", name = "Sum_species.decontam_filtered") %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("10M")) %>%
                                         mutate(Percent = Sum_species.decontam_filtered/sum(Sum_species.decontam_filtered)*100) %>%
                                         filter(Percent > 0.005) %>%
                                         select(-one_of(c("Sum_species.decontam_filtered","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered) <- malt_species.decontam_filtered$SampleID

malt_species.decontam_filtered <- malt_species.decontam_filtered %>% select(-SampleID)

malt_species.decontam_filtered = malt_species.decontam_filtered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.species.decontam_filtered_pca <- tune.pca(malt_species.decontam_filtered, logratio = 'CLR')
tune.species.decontam_filtered_pca

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_species.decontam_filtered, "PC1", "PC2", "Description", "Env", 3)
plot_pca_titles(malt_species.decontam_filtered, "PC3", "PC2", "Description", "Env", 3)

```


```{r pca_all_genus}
# unfiltered and not decontaminated
malt_genus_unfiltered <- malt_genus %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         select(-one_of(outliers)) %>%
                                         select(-contains("10M")) %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus_unfiltered) <- malt_genus_unfiltered$SampleID

malt_genus_unfiltered <- malt_genus_unfiltered %>% select(-SampleID)

malt_genus_unfiltered = malt_genus_unfiltered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.nt_genus_pca <- tune.pca(malt_genus_unfiltered, logratio = 'CLR')
tune.nt_genus_pca

# perform a PCA to see how the data cluster
nt_genus.pca <- mixOmics::pca(malt_genus_unfiltered, ncomp = 3, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_genus_unfiltered, "PC1", "PC2", "Description", "Env", 3)
plot_pca_titles(malt_genus_unfiltered, "PC3", "PC2", "Description", "Env", 3)


# filtered only
malt_genus_filtered <- malt_genus %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         select(-one_of(outliers)) %>%
                                         select(-contains("10M")) %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.01) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus_filtered) <- malt_genus_filtered$SampleID

malt_genus_filtered <- malt_genus_filtered %>% select(-SampleID)

malt_genus_filtered = malt_genus_filtered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.nt_genus_pca <- tune.pca(malt_genus_filtered, logratio = 'CLR')
tune.nt_genus_pca

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_genus_filtered, "PC1", "PC2", "Description", "Env", 3)
plot_pca_titles(malt_genus_filtered, "PC3", "PC2", "Description", "Env", 3)


# ggsave("05-results.backup/pca_nt_genus_pc1pc2.pdf", plot = pca_nt_genus_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)


# decontaminated only
malt_genus.decontam_unfiltered <- malt_genus.decontam %>%
                                         adorn_totals(where = "col", name = "Sum_genus.decontam") %>%
                                         select(-one_of(outliers)) %>%
                                         select(-contains("10M")) %>%
                                         mutate(Percent = Sum_genus.decontam/sum(Sum_genus.decontam)*100) %>%
                                         select(-one_of(c("Sum_genus.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_unfiltered) <- malt_genus.decontam_unfiltered$SampleID

malt_genus.decontam_unfiltered <- malt_genus.decontam_unfiltered %>% select(-SampleID)

malt_genus.decontam_unfiltered = malt_genus.decontam_unfiltered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.nt_genus.decontam_pca <- tune.pca(malt_genus.decontam_unfiltered, logratio = 'CLR')
tune.nt_genus.decontam_pca

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_genus.decontam_unfiltered, "PC1", "PC2", "Description", "Env", 3)
plot_pca_titles(malt_genus.decontam_unfiltered, "PC3", "PC2", "Description", "Env", 3)


# decontaminated and filtered
malt_genus.decontam_filtered <- malt_genus.decontam %>%
                                         adorn_totals(where = "col", name = "Sum_genus.decontam") %>%
                                         select(-one_of(outliers)) %>%
                                         select(-contains("10M")) %>%
                                         mutate(Percent = Sum_genus.decontam/sum(Sum_genus.decontam)*100) %>%
                                         filter(Percent > 0.01) %>%
                                         select(-one_of(c("Sum_genus.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered) <- malt_genus.decontam_filtered$SampleID

malt_genus.decontam_filtered <- malt_genus.decontam_filtered %>% select(-SampleID)

malt_genus.decontam_filtered = malt_genus.decontam_filtered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.nt_genus.decontam_pca <- tune.pca(malt_genus.decontam_filtered, logratio = 'CLR')
tune.nt_genus.decontam_pca

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Description", "Env", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Description", "Env", 3)

```

```{r pca_samples_species}
malt_species.decontam_filtered_noblanks <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         # select(-contains("10M")) %>%
                                         # select(-one_of(fullSRR)) %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_noblanks) <- malt_species.decontam_filtered_noblanks$SampleID

malt_species.decontam_filtered_noblanks <- malt_species.decontam_filtered_noblanks %>% select(-SampleID)

malt_species.decontam_filtered_noblanks = malt_species.decontam_filtered_noblanks+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.nt_species_pca <- tune.pca(malt_species.decontam_filtered_noblanks, logratio = 'CLR')
tune.nt_species_pca

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Ethnic_Group", "Village", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Ethnic_Group", "Village", 3)

# by Sex
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Ethnic_Group", "Sex", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Ethnic_Group", "Sex", 3)

# by Age group
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Ethnic_Group", "Age_group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Ethnic_Group", "Age_group", 3)

# by Market economy
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Ethnic_Group", "Market_economy", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Ethnic_Group", "Market_economy", 3)

# by Tooth site
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Ethnic_Group", "Tooth_site", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Ethnic_Group", "Tooth_site", 3)


# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_species_decontam_filtered_noblanks.pca <- mixOmics::pca(malt_species.decontam_filtered_noblanks, ncomp = 3, logratio = 'CLR')
plot(malt_species_decontam_filtered_noblanks.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_species_decontam_filtered_noblanks.pca.variates <- as.data.frame(malt_species_decontam_filtered_noblanks.pca$variates$X)
malt_species_decontam_filtered_noblanks.pca.variates <- malt_species_decontam_filtered_noblanks.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_species_decontam_filtered_noblanks.pca.varmet <- inner_join(malt_species_decontam_filtered_noblanks.pca.variates, metadata, by = "SampleID")

pca_malt_species_pc1pc2 <- ggplot(malt_species_decontam_filtered_noblanks.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  xlab(paste("PC1 - ", round(malt_species_decontam_filtered_noblanks.pca$explained_variance[1], 2), sep = "")) +
  ylab(paste("PC2 - ", round(malt_species_decontam_filtered_noblanks.pca$explained_variance[2], 2), sep = "")) +
  theme_minimal(base_size = 14) 

pca_malt_species_pc1pc2

# ggsave("05-results.backup/pca_nt_species_pc1pc2.pdf", plot = pca_nt_species_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)

```


```{r biplot_samples_species}

malt_species_decontam_filtered_biplot_121 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC1", "PC2", "Village", PC1)
malt_species_decontam_filtered_biplot_121

malt_species_decontam_filtered_biplot_122 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC1", "PC2", "Village", PC2)
malt_species_decontam_filtered_biplot_122

malt_species_decontam_filtered_biplot_322 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC3", "PC2", "Village", PC2)
malt_species_decontam_filtered_biplot_322

malt_species_decontam_filtered_biplot_323 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC3", "PC2", "Village", PC3)
malt_species_decontam_filtered_biplot_323


# what are those pathways?
pandoc.table(malt_species_decontam_filtered_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

pandoc.table(malt_species_decontam_filtered_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

pandoc.table(malt_species_decontam_filtered_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")


```

```{r sipca_species_decontam_noblanks}
malt_species_decontam_filtered_noblanks.sipca <- sipca(malt_species_decontam_filtered_noblanks.pca$X, scale = F, ncomp = 3, keepX = c(100,100,100)) # 10,10,10 50,50,50 100,100,100

malt_species_decontam_filtered_noblanks.sipca.table <- as.data.frame(malt_species_decontam_filtered_noblanks.sipca$x)
malt_species_decontam_filtered_noblanks.sipca.table <- malt_species_decontam_filtered_noblanks.sipca.table %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>% select(SampleID, Ethnic_Group, Description, Age_group, Market_economy, Sex, Tooth_site, Village))
  
malt_species_decontam_filtered_noblanks.sipca.table %>%
  ggplot(., aes(`IPC 1`, `IPC 2`, colour = Ethnic_Group, shape = Ethnic_Group)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Ethnic_Group_colors) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("sIPCA")

head(selectVar(malt_species_decontam_filtered_noblanks.sipca, comp = 1)$value)
malt_species_decontam_filtered_noblanks.sipca$loadings$X %>% as.data.frame(.) %>% rownames_to_column("species") %>% filter(str_detect(species, "Rothia")) %>% filter(V1 > 0)

```


```{r plsda_species_decontam_noblanks}
metadata_noblanks <- metadata %>%
                       filter(!str_detect(SampleID, outliersF)) %>%
                       filter(!str_detect(SampleID, "EXB")) %>%
                       filter(!str_detect(SampleID, "LIB"))


malt_species_decontam_filtered_noblanks.plsda <- plsda(malt_species.decontam_filtered_noblanks, metadata_noblanks$Ethnic_Group, ncomp = 3, logratio = "CLR")

malt_species_decontam_filtered_noblanks.plsda.table <- as.data.frame(malt_species_decontam_filtered_noblanks.plsda$variates$X)
malt_species_decontam_filtered_noblanks.plsda.table <- malt_species_decontam_filtered_noblanks.plsda.table %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>% select(SampleID, Ethnic_Group, Description, Age_group, Market_economy, Sex, Tooth_site, Village))

malt_species_decontam_filtered_noblanks.plsda.table %>%
  ggplot(., aes(comp1, comp2, colour = Ethnic_Group, shape = Ethnic_Group)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Ethnic_Group_colors) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("PLS-DA")

malt_species_decontam_filtered_noblanks.plsda$loadings$X %>%
  as.data.frame(.) %>%
  bind_cols(., malt_species_decontam_filtered_noblanks.plsda$names$colnames$X %>%
              as.data.frame(.)) %>%
  rename(Species = ".") %>%
  select(Species, everything()) %>%
  arrange((comp1))

```

```{r fa_da_species_decontam_noblanks}
# fa.parallel(malt_species.decontam_filtered_noblanks, fm = "minres", n.iter = 19)

malt_species.decontam_filtered_noblanks.fa <- fa(malt_species.decontam_filtered_noblanks, nfactors = 5, rotate = "varimax", fm = "minres") # fm = "ols"
print(malt_species.decontam_filtered_noblanks.fa)
print(malt_species.decontam_filtered_noblanks.fa$loadings,cutoff = 0.7)
print(malt_species.decontam_filtered_noblanks.fa,sort=TRUE)

malt_species.decontam_filtered_noblanks.fa$values %>%
  as.data.frame(.) %>%
  rename(Eigenvalues = ".") %>%
  arrange(desc(Eigenvalues))%>%
  rownames_to_column("xvalue") %>%
  mutate(xvalue = as.numeric(xvalue)) %>%
  filter(xvalue <= 20) %>%
  ggplot(., aes(xvalue, Eigenvalues)) +
         geom_point() +
         theme_minimal() +
         geom_hline(yintercept = 10, linetype = "dashed", color = "red")

# get the minimum residual values into a datatable
graphinfo <- as.data.frame(factor2cluster(malt_species.decontam_filtered_noblanks.fa))
graphinfo <- graphinfo %>%
  rownames_to_column("Species") %>%
  as_tibble(.)

# now make loadings into a table and designate each species in the residual with the highest absolute value
malt_species.decontam_filtered_noblanks.fa.df <- as.data.frame.matrix(malt_species.decontam_filtered_noblanks.fa$loadings)
malt_species.decontam_filtered_noblanks.fa.df <- malt_species.decontam_filtered_noblanks.fa.df %>%
  as_tibble(.) %>%
  bind_cols(., graphinfo) %>%
  mutate(MR11 = as.character(MR11),
         MR21 = as.character(MR21),
         MR31 = as.character(MR31),
         MR41 = as.character(MR41),
         MR51 = as.character(MR51)) %>%
  mutate(MR11 = replace(MR11, MR11 != "0", "MR1"),
         MR21 = replace(MR21, MR21 != "0", "MR2"),
         MR31 = replace(MR31, MR31 != "0", "MR3"),
         MR41 = replace(MR41, MR41 != "0", "MR4"),
         MR51 = replace(MR51, MR51 != "0", "MR5")) %>%
  select(Species,everything(.)) %>%
  gather("Group","Residual", 7:ncol(.)) %>%
  filter(Residual != "0") %>%
  select(Residual, Species, MR1, MR2, MR3, MR4, MR5)

# now run a discriminant analysis

```



```{r pca_cmc_species}
malt_species.decontam_filtered_cmc <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-starts_with("SRR")) %>%
                                         select(-starts_with("JAE")) %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_cmc) <- malt_species.decontam_filtered_cmc$SampleID

malt_species.decontam_filtered_cmc <- malt_species.decontam_filtered_cmc %>% select(-SampleID)

malt_species.decontam_filtered_cmc = malt_species.decontam_filtered_cmc+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.nt_species_pca <- tune.pca(malt_species.decontam_filtered_cmc, logratio = 'CLR')
tune.nt_species_pca

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Ethnic_Group", "Village", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Ethnic_Group", "Village", 3)

# by Sex
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Ethnic_Group", "Sex", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Ethnic_Group", "Sex", 3)

# by Age group
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Ethnic_Group", "Age_group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Ethnic_Group", "Age_group", 3)

# by Market economy
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Ethnic_Group", "Market_economy", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Ethnic_Group", "Market_economy", 3)

# by Tooth site
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Ethnic_Group", "Tooth_site", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Ethnic_Group", "Tooth_site", 3)


# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_species.decontam_filtered_cmc.pca <- mixOmics::pca(malt_species.decontam_filtered_cmc, ncomp = 3, logratio = 'CLR')
plot(malt_species.decontam_filtered_cmc.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_species.decontam_filtered_cmc.pca.variates <- as.data.frame(malt_species.decontam_filtered_cmc.pca$variates$X)
malt_species.decontam_filtered_cmc.pca.variates <- malt_species.decontam_filtered_cmc.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_species.decontam_filtered_cmc.pca.varmet <- inner_join(malt_species.decontam_filtered_cmc.pca.variates, metadata, by = "SampleID")

pca_malt_species_pc1pc2 <- ggplot(malt_species.decontam_filtered_cmc.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  xlab(paste("PC1 - ", round(malt_species.decontam_filtered_cmc.pca$explained_variance[1], 2), sep = "")) +
  ylab(paste("PC2 - ", round(malt_species.decontam_filtered_cmc.pca$explained_variance[2], 2), sep = "")) +
  theme_minimal(base_size = 14) 

pca_malt_species_pc1pc2

# ggsave("05-results.backup/pca_nt_species_pc1pc2.pdf", plot = pca_nt_species_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)
```


```{r adonis_cmc_species}
# use the CLR-transformed matrix from the mixOmics PCA in the last section

# Euclidean distance
malt_species.decontam_filtered_cmc.euc <- vegdist(malt_species.decontam_filtered_cmc.pca$X, method = "euclidean", na.rm = T)
malt_species.decontam_filtered_cmc.euc.pcoa <- ape::pcoa(malt_species.decontam_filtered_cmc.euc)

malt_species.decontam_filtered_cmc.euc.pcoavectors <- as.data.frame(malt_species.decontam_filtered_cmc.euc.pcoa$vectors)
malt_species.decontam_filtered_cmc.euc.pcoavectors <- malt_species.decontam_filtered_cmc.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_species.decontam_filtered_cmc.euc.vecmet <- inner_join(malt_species.decontam_filtered_cmc.euc.pcoavectors, metadata, by = "SampleID")

malt_species.decontam_filtered_cmc.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis(malt_species.decontam_filtered_cmc.euc ~ Tooth_site + Ethnic_Group + Age_group + Sex + Village + Market_economy, malt_species.decontam_filtered_cmc.euc.vecmet, perm=999)

```


```{r biplots_cmc_species}

malt_species.decontam_filtered_nocontrols <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-starts_with("JAE")) %>%
                                         select(-starts_with("SRR")) %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_nocontrols) <- malt_species.decontam_filtered_nocontrols$SampleID

malt_species.decontam_filtered_nocontrols <- malt_species.decontam_filtered_nocontrols %>% select(-SampleID)

malt_species.decontam_filtered_nocontrols = malt_species.decontam_filtered_nocontrols+1

malt_species.decontam_filtered_nocontrols.pca <- mixOmics::pca(malt_species.decontam_filtered_nocontrols, ncomp = 3, logratio = 'CLR')
plot(malt_species.decontam_filtered_nocontrols.pca)

malt_species.decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Tooth_site", PC1)
malt_species.decontam_filtered_nocontrols_biplot_121

malt_species.decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Tooth_site", PC2)
malt_species.decontam_filtered_nocontrols_biplot_122

malt_species.decontam_filtered_nocontrols_biplot_322 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Tooth_site", PC2)
malt_species.decontam_filtered_nocontrols_biplot_322

malt_species.decontam_filtered_nocontrols_biplot_323 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Tooth_site", PC3)
malt_species.decontam_filtered_nocontrols_biplot_323

# what are those pathways?
pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```


```{r sipca_species_decontam_cmc}
malt_species.decontam_filtered_cmc.sipca <- sipca(malt_species.decontam_filtered_cmc.pca$X, scale = F, ncomp = 3, keepX = c(100,100,100)) # 10,10,10 50,50,50 100,100,100

malt_species.decontam_filtered_cmc.sipca.table <- as.data.frame(malt_species.decontam_filtered_cmc.sipca$x)
malt_species.decontam_filtered_cmc.sipca.table <- malt_species.decontam_filtered_cmc.sipca.table %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>% select(SampleID, Ethnic_Group, Description, Age_group, Market_economy, Sex, Tooth_site, Village))
  
malt_species.decontam_filtered_cmc.sipca.table %>%
  ggplot(., aes(`IPC 1`, `IPC 2`, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("sIPCA")

head(selectVar(malt_species.decontam_filtered_cmc.sipca, comp = 1)$value)
malt_species.decontam_filtered_cmc.sipca$loadings$X %>% as.data.frame(.) %>% rownames_to_column("species") %>% filter(V1 > 0) %>% arrange(desc(V1))

```

```{r plsda_species_decontam_cmc}
metadata_cmc <- metadata %>%
                       filter(!str_detect(SampleID, outliersF)) %>%
                       filter(str_detect(SampleID, "CMC"))

malt_species.decontam_filtered_cmc.plsda <- plsda(malt_species.decontam_filtered_cmc, metadata_cmc$Tooth_site, ncomp = 3, logratio = "CLR")

malt_species.decontam_filtered_cmc.plsda.table <- as.data.frame(malt_species.decontam_filtered_cmc.plsda$variates$X)
malt_species.decontam_filtered_cmc.plsda.table <- malt_species.decontam_filtered_cmc.plsda.table %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>% select(SampleID, Ethnic_Group, Description, Age_group, Market_economy, Sex, Tooth_site, Village))

malt_species.decontam_filtered_cmc.plsda.table %>%
  ggplot(., aes(comp1, comp2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("PLS-DA")

malt_species.decontam_filtered_cmc.plsda$loadings$X %>%
  as.data.frame(.) %>%
  bind_cols(., malt_species.decontam_filtered_cmc.plsda$names$colnames$X %>%
              as.data.frame(.)) %>%
  rename(Species = ".") %>%
  select(Species, everything()) %>%
  arrange((comp1))

```

```{r fa_da_species_decontam_noblanks}
# fa.parallel(malt_species.decontam_filtered_cmc, fm = "minres", n.iter = 19)

malt_species.decontam_filtered_cmc.fa <- fa(malt_species.decontam_filtered_cmc, nfactors = 3, rotate = "varimax", fm = "minres") # fm = "ols"
print(malt_species.decontam_filtered_cmc.fa)
print(malt_species.decontam_filtered_cmc.fa$loadings,cutoff = 0.7)
print(malt_species.decontam_filtered_cmc.fa,sort=TRUE)

malt_species.decontam_filtered_cmc.fa$values %>%
  as.data.frame(.) %>%
  rename(Eigenvalues = ".") %>%
  arrange(desc(Eigenvalues))%>%
  rownames_to_column("xvalue") %>%
  mutate(xvalue = as.numeric(xvalue)) %>%
  filter(xvalue <= 20) %>%
  ggplot(., aes(xvalue, Eigenvalues)) +
         geom_point() +
         theme_minimal() +
         geom_hline(yintercept = 10, linetype = "dashed", color = "red")

# get the minimum residual values into a datatable
graphinfo <- as.data.frame(factor2cluster(malt_species.decontam_filtered_cmc.fa))
graphinfo <- graphinfo %>%
  rownames_to_column("Species") %>%
  as_tibble(.)

# now make loadings into a table and designate each species in the residual with the highest absolute value
malt_species.decontam_filtered_cmc.fa.df <- as.data.frame.matrix(malt_species.decontam_filtered_cmc.fa$loadings)
malt_species.decontam_filtered_cmc.fa.df <- malt_species.decontam_filtered_cmc.fa.df %>%
  as_tibble(.) %>%
  bind_cols(., graphinfo) %>%
  mutate(MR11 = as.character(MR11),
         MR21 = as.character(MR21),
         MR31 = as.character(MR31)) %>%
  mutate(MR11 = replace(MR11, MR11 != "0", "MR1"),
         MR21 = replace(MR21, MR21 != "0", "MR2"),
         MR31 = replace(MR31, MR31 != "0", "MR3")) %>%
  select(Species,everything(.)) %>%
  gather("Group","Residual", 7:ncol(.)) %>%
  filter(Residual != "0") %>%
  select(Residual, Species, MR1, MR2, MR3)

# now run a discriminant analysis

```


```{r pca_samples_genus}
malt_genus.decontam_filtered <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-contains("10M")) %>%
                                         # select(-one_of(fullSRR)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.01) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered) <- malt_genus.decontam_filtered$SampleID

malt_genus.decontam_filtered <- malt_genus.decontam_filtered %>% select(-SampleID)

malt_genus.decontam_filtered = malt_genus.decontam_filtered+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.nt_genus_pca <- tune.pca(malt_genus.decontam_filtered, logratio = 'CLR')
tune.nt_genus_pca

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Ethnic_Group", "Village", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Ethnic_Group", "Village", 3)

# by Sex
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Ethnic_Group", "Sex", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Ethnic_Group", "Sex", 3)

# by Age group
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Ethnic_Group", "Age_group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Ethnic_Group", "Age_group", 3)

# by Market economy
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Ethnic_Group", "Market_economy", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Ethnic_Group", "Market_economy", 3)

# by Tooth site
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Ethnic_Group", "Tooth_site", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Ethnic_Group", "Tooth_site", 3)


# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_genus.decontam_filtered.pca <- mixOmics::pca(malt_genus.decontam_filtered, ncomp = 3, logratio = 'CLR')
plot(malt_genus.decontam_filtered.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_genus.decontam_filtered.pca.variates <- as.data.frame(malt_genus.decontam_filtered.pca$variates$X)
malt_genus.decontam_filtered.pca.variates <- malt_genus.decontam_filtered.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_genus.decontam_filtered.pca.varmet <- inner_join(malt_genus.decontam_filtered.pca.variates, metadata, by = "SampleID")

pca_malt_genus_pc1pc2 <- ggplot(malt_genus.decontam_filtered.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  # scale_colour_manual(values = common_colors) +
  # scale_shape_manual(values = village_shapes) +
  xlab(paste("PC1 - ", round(malt_genus.decontam_filtered.pca$explained_variance[1], 2), sep = "")) +
  ylab(paste("PC2 - ", round(malt_genus.decontam_filtered.pca$explained_variance[2], 2), sep = "")) +
  theme_minimal(base_size = 14) 

pca_malt_genus_pc1pc2


# ggsave("05-results.backup/pca_malt_genus_pc1pc2.pdf", plot = pca_malt_genus_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)

```


```{r biplot_samples_genus}

malt_genus.decontam_filtered_biplot_121 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC1", "PC2", "Village", PC1)
malt_genus.decontam_filtered_biplot_121

malt_genus.decontam_filtered_biplot_122 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC1", "PC2", "Village", PC2)
malt_genus.decontam_filtered_biplot_122

malt_genus.decontam_filtered_biplot_322 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC3", "PC2", "Village", PC2)
malt_genus.decontam_filtered_biplot_322

malt_genus.decontam_filtered_biplot_323 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC3", "PC2", "Village", PC3)
malt_genus.decontam_filtered_biplot_323

# what are those pathways?
pandoc.table(malt_genus.decontam_filtered_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```


```{r pca_cmc_genus}
malt_genus.decontam_filtered_cmc <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-starts_with("SRR")) %>%
                                         select(-starts_with("JAE")) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered_cmc) <- malt_genus.decontam_filtered_cmc$SampleID

malt_genus.decontam_filtered_cmc <- malt_genus.decontam_filtered_cmc %>% select(-SampleID)

malt_genus.decontam_filtered_cmc = malt_genus.decontam_filtered_cmc+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.nt_genus_pca <- tune.pca(malt_genus.decontam_filtered_cmc, logratio = 'CLR')
tune.nt_genus_pca

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Ethnic_Group", "Village", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Ethnic_Group", "Village", 3)

# by Sex
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Ethnic_Group", "Sex", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Ethnic_Group", "Sex", 3)

# by Age group
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Ethnic_Group", "Age_group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Ethnic_Group", "Age_group", 3)

# by Market economy
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Ethnic_Group", "Market_economy", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Ethnic_Group", "Market_economy", 3)

# by Tooth site
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Ethnic_Group", "Tooth_site", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Ethnic_Group", "Tooth_site", 3)


# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_genus.decontam_filtered_cmc.pca <- mixOmics::pca(malt_genus.decontam_filtered_cmc, ncomp = 3, logratio = 'CLR')
plot(malt_genus.decontam_filtered_cmc.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_genus.decontam_filtered_cmc.pca.variates <- as.data.frame(malt_genus.decontam_filtered_cmc.pca$variates$X)
malt_genus.decontam_filtered_cmc.pca.variates <- malt_genus.decontam_filtered_cmc.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_genus.decontam_filtered_cmc.pca.varmet <- inner_join(malt_genus.decontam_filtered_cmc.pca.variates, metadata, by = "SampleID")

pca_malt_genus_pc1pc2 <- ggplot(malt_genus.decontam_filtered_cmc.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  # scale_colour_manual(values = common_colors) +
  # scale_shape_manual(values = village_shapes) +
  xlab(paste("PC1 - ", round(malt_genus.decontam_filtered_cmc.pca$explained_variance[1], 2), sep = "")) +
  ylab(paste("PC2 - ", round(malt_genus.decontam_filtered_cmc.pca$explained_variance[2], 2), sep = "")) +
  theme_minimal(base_size = 14) 

pca_malt_genus_pc1pc2


# ggsave("05-results.backup/pca_malt_genus_pc1pc2.pdf", plot = pca_malt_genus_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)
```


```{r adonis_cmc_genus}
# use the CLR-transformed matrix from the mixOmics PCA in the last section

# Euclidean distance
malt_genus.decontam_filtered_cmc.euc <- vegdist(malt_genus.decontam_filtered_cmc.pca$X, method = "euclidean", na.rm = T)
malt_genus.decontam_filtered_cmc.euc.pcoa <- ape::pcoa(malt_genus.decontam_filtered_cmc.euc)

malt_genus.decontam_filtered_cmc.euc.pcoavectors <- as.data.frame(malt_genus.decontam_filtered_cmc.euc.pcoa$vectors)
malt_genus.decontam_filtered_cmc.euc.pcoavectors <- malt_genus.decontam_filtered_cmc.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_genus.decontam_filtered_cmc.euc.vecmet <- inner_join(malt_genus.decontam_filtered_cmc.euc.pcoavectors, metadata, by = "SampleID")

malt_genus.decontam_filtered_cmc.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis(malt_genus.decontam_filtered_cmc.euc ~ Tooth_site  + Ethnic_Group + Age_group + Sex + Village + Market_economy, malt_genus.decontam_filtered_cmc.euc.vecmet, perm=999)

```


```{r biplots_cmc_genus}

malt_genus.decontam_filtered_nocontrols <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-starts_with("JAE")) %>%
                                         select(-starts_with("SRR")) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered_nocontrols) <- malt_genus.decontam_filtered_nocontrols$SampleID

malt_genus.decontam_filtered_nocontrols <- malt_genus.decontam_filtered_nocontrols %>% select(-SampleID)

malt_genus.decontam_filtered_nocontrols = malt_genus.decontam_filtered_nocontrols+1

malt_genus.decontam_filtered_nocontrols.pca <- mixOmics::pca(malt_genus.decontam_filtered_nocontrols, ncomp = 3, logratio = 'CLR')
plot(malt_genus.decontam_filtered_nocontrols.pca)

malt_genus_decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Tooth_site", PC1)
malt_genus_decontam_filtered_nocontrols_biplot_121

malt_genus_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Tooth_site", PC2)
malt_genus_decontam_filtered_nocontrols_biplot_122

malt_genus_decontam_filtered_nocontrols_biplot_322 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Tooth_site", PC2)
malt_genus_decontam_filtered_nocontrols_biplot_322

malt_genus_decontam_filtered_nocontrols_biplot_323 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Tooth_site", PC3)
malt_genus_decontam_filtered_nocontrols_biplot_323

# what are those pathways?
pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```







