---
title: "Cameroon hunter-gatherer calculus/plaque MALT decontam"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---

```{r load_libraries, echo = F, message = F}
library(knitr)
library(decontam)
library(data.table)
library(janitor)
library(tidyverse)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

```{r load_data}
malt_species <- fread("./05-results.backup/CMC_HMP_JAE_Yano_MALT_RSC_species_bacteria_archaea.txt")
malt_genus <- fread("./05-results.backup/CMC_HMP_JAE_Yano_MALT_RSC_genus_bacteria_archaea.txt")

# clean the file names
malt_species <- rename(malt_species, Species = `#Datasets`)
colnames(malt_species) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M","_10M", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam","_10M", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub(".fastq.truncated","", colnames(malt_species))


# remove human and unassigned reads
malt_species <- malt_species %>% filter(!str_detect(Species, "Homo sapiens"))
malt_species <- malt_species %>% filter(!str_detect(Species, "Not assigned"))
# get rid of the extra column
malt_species <- malt_species %>% select(-Comparison_CMC_HMP_JAE_MALT_RSC)

malt_genus <- rename(malt_genus, Species = `#Datasets`)
colnames(malt_genus) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam","_10M", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M","_10M", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub(".fastq.truncated","", colnames(malt_genus))

# remove human and unassigned reads
malt_genus <- malt_genus %>% filter(!str_detect(Species, "Homo"))
malt_genus <- malt_genus %>% filter(!str_detect(Species, "Not assigned"))
# get rid of the extra column at the begining
malt_genus <- malt_genus %>% select(-Comparison_CMC_HMP_JAE_MALT_RSC)

# load the metadata file
load("./05-results.backup/CMC_metadata.RData")

```


```{r pca_plotting_functions}
# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group, shape = metadata_group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$SampleID), size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}

```


Before running decontam, let's plot the samples in a PCA to see if there are any that are clearly not good-quality
```{r raw_plot}

malt_species_raw <- malt_species %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species_raw) <- malt_species_raw$SampleID

malt_species_raw <- malt_species_raw %>% select(-SampleID)

malt_species_raw = malt_species_raw+1

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.malt_species_pca <- tune.pca(malt_species_raw, logratio = 'CLR')
tune.malt_species_pca

# perform a PCA to see how the data cluster
malt_species.pca <- mixOmics::pca(malt_species_raw, ncomp = 3, logratio = 'CLR')
plot(malt_species.pca)


plot_pca(malt_species.pca, "PC1", "PC2", "Description")
plot_pca_names(malt_species.pca,  "PC1", "PC2", "Description")

```



# Decontamination with Decontam
We want to know if there are any species that can be removed b/c they appear to be contaminants. Use the package decontam to determine which species may be contaminants. Prepare the matrix for decontam. 
```{r decontam_data_prep}
# list out the samples that have all 0 or NA values to remove from the input matrix
NA_filter_list <- metadata %>% filter((is.na(SumLibraryConcentration_copies_ul) | SumLibraryConcentration_copies_ul <= 0)) %>% select(SampleID)

# remove NA value samples from the input data frame
malt_species.dataframe <- malt_species %>%
  filter(!str_detect(Species, "Not assigned")) %>%
  select(-one_of(outliers)) %>%
  gather("SampleID","Counts", 2:ncol(.)) %>%
  spread(Species,Counts) %>%
  anti_join(., NA_filter_list)

# make a matrix for decontam
malt_species.matrix <- data.matrix(malt_species.dataframe[2:ncol(malt_species.dataframe)])
rownames(malt_species.matrix) <- malt_species.dataframe$SampleID

# remove NA value samples from the input data frame
malt_genus.dataframe <- malt_genus %>%
  filter(!str_detect(Species, "Not assigned")) %>%
  select(-one_of(outliers)) %>%
  gather("SampleID","Counts", 2:ncol(.)) %>%
  spread(Species,Counts) %>%
  anti_join(., NA_filter_list)

# make a matrix for decontam
malt_genus.matrix <- data.matrix(malt_genus.dataframe[2:ncol(malt_genus.dataframe)])
rownames(malt_genus.matrix) <- malt_genus.dataframe$SampleID


# make a metadata dataframe with the same NA-containing samples removed
metadata.decontam <- metadata %>%
  anti_join(., NA_filter_list, by = "SampleID") %>%
  filter(!str_detect(SampleID, outliers %>% str_c(collapse = "|")))

```

Make a species abundance curve for the samples based on Env to get an idea of the number of species identified in each prior to filtering, and how long the tail is. 
```{r rank_abundance_curves}

malt_species.dataframeL <- malt_species.dataframe %>%
  adorn_percentages(denominator = "row", na.rm=F) %>%
  gather("Species","Relab",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type))

Envmt <- unique(malt_species.dataframeL$Description)

species_orders <- lapply(Envmt, function(envmt) {
malt_species.dataframeL_check <- malt_species.dataframeL %>%
  filter(Description == envmt) %>%
  group_by(Species) %>%
  summarise(mean_relab = mean(Relab)) %>%
  arrange(desc(mean_relab)) %>%
  filter(mean_relab > 0) %>%
  rownames_to_column("order") %>%
  mutate(order = as.numeric(order)) %>%
  mutate(Description = envmt)
}) %>%
  bind_rows(.)
  
species_abundance_curves <- function(df, envt, title_text) {
      options(scipen = 10)
      
    species_abundance_plot <- species_orders %>%
        filter(Description == envt) %>%
           ggplot(., aes(x=order, y=mean_relab)) +
           geom_point() +
           theme_minimal() +
           scale_y_log10() +
           geom_hline(yintercept=0.01, linetype="dashed", color = "red") +
           xlim(NA, 600) +
           ylab("mean relative abundance") +
           xlab("species order") +
           ggtitle(title_text)
  
    return(species_abundance_plot)
}

sp_curve_HG <- species_abundance_curves(species_orders, "HunterGatherer", "Hunter Gatherer - species")
sp_curve_EG <- species_abundance_curves(species_orders, "ExtractionGauze", "Extraction Gauze - species")
sp_curve_EB <- species_abundance_curves(species_orders, "ExtractionBlank", "Extraction Blank - species")
sp_curve_LB <- species_abundance_curves(species_orders, "LibraryBlank", "Library Blank - species")

plot_grid(sp_curve_HG, sp_curve_EG,
          sp_curve_EB, sp_curve_LB,
          nrow=2, labels = c("A", "B", "C", "D"), rel_widths = c(1, 1))


# repeat for genus level
malt_genus.dataframeL <- malt_genus.dataframe %>%
  adorn_percentages(denominator = "row", na.rm=F) %>%
  gather("genus","Relab",2:ncol(.)) %>%
  inner_join(., metadata)

Envmt <- unique(malt_genus.dataframeL$Description)

genus_orders <- lapply(Envmt, function(envmt) {
malt_genus.dataframeL_check <- malt_genus.dataframeL %>%
  filter(Description == envmt) %>%
  group_by(genus) %>%
  summarise(mean_relab = mean(Relab)) %>%
  arrange(desc(mean_relab)) %>%
  filter(mean_relab > 0) %>%
  rownames_to_column("order") %>%
  mutate(order = as.numeric(order)) %>%
  mutate(Description = envmt)
}) %>%
  bind_rows(.)
  
genus_abundance_curves <- function(df, envt, title_text) {
      options(scipen = 10)
      
    genus_abundance_plot <- genus_orders %>%
        filter(Description == envt) %>%
           ggplot(., aes(x=order, y=mean_relab)) +
           geom_point() +
           theme_minimal() +
           scale_y_log10() +
           geom_hline(yintercept=0.01, linetype="dashed", color = "red") +
           xlim(NA, 250) +
           ylab("mean relative abundance") +
           xlab("genus order") +
           ggtitle(title_text)
  
    return(genus_abundance_plot)
}

gn_curve_HG <- genus_abundance_curves(genus_orders, "HunterGatherer", "Hunter Gatherer - genus")
gn_curve_EG <- genus_abundance_curves(genus_orders, "ExtractionGauze", "Extraction Gauze - genus")
gn_curve_EB <- genus_abundance_curves(genus_orders, "ExtractionBlank", "Extraction Blank - genus")
gn_curve_LB <- genus_abundance_curves(genus_orders, "LibraryBlank", "Library Blank - genus")

plot_grid(gn_curve_HG, gn_curve_EG,
          gn_curve_EB, gn_curve_LB,
          nrow=2, labels = c("A", "B", "C", "D"), rel_widths = c(1, 1))

```



Now we're ready to run Decontam. Start with the frequency method. 
```{r decontam_frequency, eval = F}
# Check for contamination by fequency
contam.freq.species <- isContaminant(malt_species.matrix, method="frequency", 
                                 conc=metadata.decontam$SumLibraryConcentration_copies_ul, 
                                 threshold = 0.9)

# Print the # of species identified as contaminants by frequency
table(contam.freq.species$contaminant)

cont_freq_species_list <- contam.freq.species[which(contam.freq.species$contaminant=="TRUE"), ]

# Make a histogram to look at the distribution of the p-value (probability for classifying contaminants)

ggplot(contam.freq.species, aes(x=p)) +
  geom_histogram(bins=100) +
  scale_y_log10() +
  ggtitle('Frequency - Species')

# Check for contamination by fequency
contam.freq.genus <- isContaminant(malt_genus.matrix, method="frequency", 
                                 conc=metadata.decontam$SumLibraryConcentration_copies_ul, 
                                 threshold = 0.9)

# Print the # of genus identified as contaminants by frequency
table(contam.freq.genus$contaminant)

cont_freq_genus_list <- contam.freq.genus[which(contam.freq.genus$contaminant=="TRUE"), ]

# Make a histogram to look at the distribution of the p-value (probability for classifying contaminants)

ggplot(contam.freq.genus, aes(x=p)) +
  geom_histogram(bins=100) +
  scale_y_log10() +
  ggtitle('Frequency - Genus')

```

Try the prevalence method to compare. 
```{r decontam_prevalence}
# Assign sample-type status to identify control samples
metadata.decontam$is.control <- metadata.decontam$Sample_or_Control=="Control"

#Try changing the probability threshold cut-off to see how many more or less are ID'd as contaminants
contam.prev.species <- isContaminant(malt_species.matrix, method="prevalence", 
                                 neg=metadata.decontam$is.control, 
                                 threshold = 0.6)

# Check number of contaminants
table(contam.prev.species$contaminant)

# List of contaminants
cont_prev_species_list <- contam.prev.species[which(contam.prev.species$contaminant=="TRUE"), ]

# Look at the histogram - does the probability threshold need to be changed?
ggplot(contam.prev.species, aes(x=p)) +
  geom_histogram(bins=100) +
  scale_y_log10() +
  ggtitle('Prevalence - Species')

#Try changing the probability threshold cut-off to see how many more or less are ID'd as contaminants
contam.prev.genus <- isContaminant(malt_genus.matrix, method="prevalence", 
                                 neg=metadata.decontam$is.control, 
                                 threshold = 0.6)

# Check number of contaminants
table(contam.prev.genus$contaminant)

# List of contaminants
cont_prev_genus_list <- contam.prev.genus[which(contam.prev.genus$contaminant=="TRUE"), ]

# Look at the histogram - does the probability threshold need to be changed?
ggplot(contam.prev.genus, aes(x=p)) +
  geom_histogram(bins=100) +
  scale_y_log10() +
  ggtitle('Prevalence -Genus')

```

Check the rank abundance curves to see how changing the cut-off threshold changes the tail.
```{r decontam_rank_abund_curve}

contaminants_species <- cont_prev_species_list %>% rownames_to_column("Species") %>% select(Species)
malt_species.decontam <- malt_species %>%
  anti_join(., contaminants_species)

# write.table(malt_species.decontam, file = "05-results.backup/malt_species.decontam.tsv", sep="\t", quote=F, row.names = F)
# write.table(malt_species, "05-results.backup/malt_species.tsv", sep="\t", quote=F, row.names = F)

malt_species.decontamL <- malt_species.decontam %>%
  adorn_percentages(denominator = "col", na.rm=F) %>%
  gather("SampleID","Relab",2:ncol(.)) %>%
  inner_join(., metadata %>%
              select(SampleID, Env, Description), by = "SampleID")

Envmt_decontam <- unique(malt_species.decontamL$Description)

species_decontam_orders <- lapply(Envmt_decontam, function(envmt) {
malt_species.dataframeL_check <- malt_species.decontamL %>%
  filter(Description == envmt) %>%
  group_by(Species) %>%
  summarise(mean_relab = mean(Relab)) %>%
  arrange(desc(mean_relab)) %>%
  filter(mean_relab > 0) %>%
  rownames_to_column("order") %>%
  mutate(order = as.numeric(order)) %>%
  mutate(Description = envmt)
}) %>%
  bind_rows(.)
  
species_decontam_abundance_curves <- function(df, envt, title_text) {
      options(scipen = 10)
      
    species_abundance_plot <- species_decontam_orders %>%
        filter(Description == envt) %>%
           ggplot(., aes(x=order, y=mean_relab)) +
           geom_point() +
           theme_minimal() +
           scale_y_log10() +
           geom_hline(yintercept=0.01, linetype="dashed", color = "red") +
           xlim(NA, 500) +
           ylab("mean relative abundance") +
           xlab("species order") +
           ggtitle(title_text) + 
           theme(plot.title = element_text(size = 8))
  
    return(species_abundance_plot)
}

sp_decontam_curve_HG <- species_decontam_abundance_curves(species_decontam_orders, "HunterGatherer", "HG, decontam")
sp_decontam_curve_EG <- species_decontam_abundance_curves(species_decontam_orders, "ExtractionGauze", "EG, decontam")
sp_decontam_curve_EB <- species_decontam_abundance_curves(species_decontam_orders, "ExtractionBlank", "EB, decontam")
sp_decontam_curve_LB <- species_decontam_abundance_curves(species_decontam_orders, "LibraryBlank", "LB, decontam")
sp_decontam_curve_Yano <- species_decontam_abundance_curves(species_decontam_orders, "Yanomami", "Yanomami, decontam")
sp_decontam_curve_HMPsup <- species_decontam_abundance_curves(species_decontam_orders, "HMP_supPlaque", "HMPsup, decontam")
sp_decontam_curve_HMPsub <- species_decontam_abundance_curves(species_decontam_orders, "HMP_subPlaque", "HMPsub, decontam")
sp_decontam_curve_HMP10sup <- species_decontam_abundance_curves(species_decontam_orders, "HMP10M_supPlaque", "HMP10Msup, decontam")
sp_decontam_curve_HMP10sub <- species_decontam_abundance_curves(species_decontam_orders, "HMP10M_subPlaque", "HMP10Msub, decontam")
sp_decontam_curve_JAE <- species_decontam_abundance_curves(species_decontam_orders, "JAE", "JAE, decontam")
sp_decontam_curve_JAE <- species_decontam_abundance_curves(species_decontam_orders, "JAE", "JAE, decontam")
sp_decontam_curve_JAE10 <- species_decontam_abundance_curves(species_decontam_orders, "JAE10M", "JAE10M, decontam")

plot_grid(sp_decontam_curve_HG, sp_decontam_curve_EG, sp_decontam_curve_EB, sp_decontam_curve_LB, sp_decontam_curve_Yano, sp_decontam_curve_HMPsup,
          sp_decontam_curve_HMPsub, sp_decontam_curve_HMP10sup, sp_decontam_curve_HMP10sup, sp_decontam_curve_JAE, sp_decontam_curve_JAE10,
          nrow=2, labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K"), rel_widths = c(1, 1))

# Now repeatw with genus
contaminants_genus <- cont_prev_genus_list %>% rownames_to_column("Species") %>% select(Species)
malt_genus.decontam <- malt_genus %>%
  anti_join(., contaminants_genus)

malt_genus.decontamL <- malt_genus.decontam %>%
  adorn_percentages(denominator = "col", na.rm=F) %>%
  gather("SampleID","Relab",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Env, Description), by = "SampleID")


Envmt_decontam <- unique(malt_genus.decontamL$Description)

genus_decontam_orders <- lapply(Envmt_decontam, function(envmt) {
malt_genus.dataframeL_check <- malt_genus.decontamL %>%
  filter(Description == envmt) %>%
  group_by(Species) %>%
  summarise(mean_relab = mean(Relab)) %>%
  arrange(desc(mean_relab)) %>%
  filter(mean_relab > 0) %>%
  rownames_to_column("order") %>%
  mutate(order = as.numeric(order)) %>%
  mutate(Description = envmt)
}) %>%
  bind_rows(.)
  
genus_decontam_abundance_curves <- function(df, envt, title_text) {
      options(scipen = 10)
      
    genus_abundance_plot <- genus_decontam_orders %>%
        filter(Description == envt) %>%
           ggplot(., aes(x=order, y=mean_relab)) +
           geom_point() +
           theme_minimal() +
           scale_y_log10() +
           geom_hline(yintercept=0.01, linetype="dashed", color = "red") +
           xlim(NA, 150) +
           ylab("mean relative abundance") +
           xlab("genus order") +
           ggtitle(title_text) + 
           theme(plot.title = element_text(size = 8))
  
    return(genus_abundance_plot)
}

gn_decontam_curve_HG <- genus_decontam_abundance_curves(genus_decontam_orders, "HunterGatherer", "HG, decontam")
gn_decontam_curve_EG <- genus_decontam_abundance_curves(genus_decontam_orders, "ExtractionGauze", "EG, decontam")
gn_decontam_curve_EB <- genus_decontam_abundance_curves(genus_decontam_orders, "ExtractionBlank", "EB, decontam")
gn_decontam_curve_LB <- genus_decontam_abundance_curves(genus_decontam_orders, "LibraryBlank", "LB, decontam")
gn_decontam_curve_Yano <- genus_decontam_abundance_curves(genus_decontam_orders, "Yanomami", "Yanomami, decontam")
gn_decontam_curve_HMPsup <- genus_decontam_abundance_curves(genus_decontam_orders, "HMP_supPlaque", "HMPsup, decontam")
gn_decontam_curve_HMPsub <- genus_decontam_abundance_curves(genus_decontam_orders, "HMP_subPlaque", "HMPsub, decontam")
gn_decontam_curve_HMP10sup <- genus_decontam_abundance_curves(genus_decontam_orders, "HMP10M_supPlaque", "HMP10Msup, decontam")
gn_decontam_curve_HMP10sub <- genus_decontam_abundance_curves(genus_decontam_orders, "HMP10M_subPlaque", "HMP10Msub, decontam")
gn_decontam_curve_JAE <- genus_decontam_abundance_curves(genus_decontam_orders, "JAE", "JAE, decontam")
gn_decontam_curve_JAE10M <- genus_decontam_abundance_curves(genus_decontam_orders, "JAE10M", "JAE10M, decontam")

plot_grid(gn_decontam_curve_HG, gn_decontam_curve_EG, gn_decontam_curve_EB, gn_decontam_curve_LB, gn_decontam_curve_Yano,
          gn_decontam_curve_HMPsup, gn_decontam_curve_HMPsub, gn_decontam_curve_HMP10sup, gn_decontam_curve_HMP10sub, gn_decontam_curve_JAE, gn_decontam_curve_JAE10M,
          nrow=2, labels = c("A", "B", "C", "D", "E", "F","G","H", "I", "J", "K"), rel_widths = c(1, 1))

```


Check the combined prevalence and frequency method to see how it differs from the individual methods. 
```{r decontam_combined, eval=F}
# Assign sample-type status to identify control samples (same as was done for prevalence)
metadata.decontam$is.control <- metadata.decontam$Sample_or_Control=="Control"

# Assign contaminant status
contam.comb.species <- isContaminant(malt_species.matrix, method="combined", 
                                 conc=metadata.decontam$SumLibraryConcentration_copies_ul, 
                                 neg=metadata.decontam$is.control,
                                 threshold = 0.9)

## Check how many contaminants
table(contam.comb.species$contaminant)

## List of contaminants
cont_comb_species_list <- contam.comb.species[which(contam.comb.species$contaminant=="TRUE"), ]

# Graph it to look at the histogram and pick a suitable probability-values cut-off
ggplot(contam.comb.species, aes(x=p)) +
  geom_histogram(bins=100) +
  scale_y_log10() +
  geom_hline(yintercept=10, linetype="dashed", color = "red") +
  geom_hline(yintercept=2, linetype="dashed", color = "red") +
  ggtitle('Combined - Species')

# Assign contaminant status
contam.comb.genus <- isContaminant(malt_genus.matrix, method="combined", 
                                 conc=metadata.decontam$SumLibraryConcentration_copies_ul, 
                                 neg=metadata.decontam$is.control,
                                 threshold = 0.9)

## Check how many contaminants
table(contam.comb.genus$contaminant)

## List of contaminants
cont_comb_genus_list <- contam.comb.genus[which(contam.comb.genus$contaminant=="TRUE"), ]

# Graph it to look at the histogram and pick a suitable probability-values cut-off
ggplot(contam.comb.genus, aes(x=p)) +
  geom_histogram(bins=100) +
  scale_y_log10() +
  geom_hline(yintercept=10, linetype="dashed", color = "red") +
  geom_hline(yintercept=2, linetype="dashed", color = "red") +
  ggtitle('Combined - Genus')

```

We'll use the prevalence method, because this doesn't include known oral taxa in the contaminant list (frequency method does), and it picks up a lot more than the combined method does (but is that important if we cut off the tail of low-abundance taxa (say <0.01% abundance) anyway?). But which cut-off to use? How does changing the cut-off affect the tail of low-abundance taxa?
List out the pathways that are considered contaminants, to be removed from the table prior to all analyses. Then remove them from the table to make the decontaminated input table for analysis. *Note that the genus list includes Actinomyces, and this needs to be removed from the list b/c Actinomyces are a well known oral genus*
```{r contaminants}
contaminants_species <- cont_prev_species_list %>% rownames_to_column("Species") %>% select(Species)
malt_species.decontam <- malt_species %>%
  anti_join(., contaminants_species)

# write.table(malt_species.decontam, file = "05-results.backup/malt_species.decontam.tsv", sep="\t", quote=F, row.names = F)
# write.table(malt_species, "05-results.backup/malt_species.tsv", sep="\t", quote=F, row.names = F)
# write.table(contaminants_species, "05-results.backup/malt_contaminants_species.tsv", sep="\t", quote=F, row.names = F)

contaminants_genus <- cont_prev_genus_list %>% rownames_to_column("Species") %>% select(Species) %>% filter(Species != "Actinomyces")
malt_genus.decontam <- malt_genus %>%
  anti_join(., contaminants_genus)

# write.table(contaminants_genus, "05-results.backup/malt_contaminants_genus.tsv", sep="\t", quote=F, row.names = F)

```

