---
title: "Cameroon hunter-gatherer calculus/plaque SEED"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(decontam)
library(data.table)
library(vegan)
library(ape)
library(mixOmics)
library(janitor)
library(rstatix)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```


# SEED category assignment stats
```{r load_data}
## load the species and genus tables generated with HUMAnN2
load("./05-results.backup/CMC_seed_tables.RData")

```

```{r load_metadata}

load("./05-results.backup/CMC_metadata.RData")

# try metadata w/o factors for pairwise.adonis2
metadata_noblanks <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")

# make a separate table for adonis2 stats later
metadata_noblanks <- metadata_noblanks %>%
  rename(SampleID = `#SampleID`)  %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR164|10M")) %>%
  filter(SampleID != "CMC032.B0101") %>%  # this sample has very few reads
  select(SampleID, ExtractionBatch, LibraryBatch, Village, Age_group, Market_economy, Market_economy_split, Ethnic_Group, Sex, Industry, Tooth_site) %>%
  as_tibble() %>%
  mutate(Village = fct_relevel(Village, "202","205","204","203","206","HMP","JAE"),
         ExtractionBatch = fct_relevel(ExtractionBatch, "Ex01_VE_2019-01-23","Ex04_VE_2019-02-18","Ex05_VE_2019-02-18","Ex06_VE_2019-02-21","Ex07_VE_2019-02-21","Ex08_VE_2019-02-25","Ex09_VE_2019-03-13"),
         LibraryBatch = fct_relevel(LibraryBatch, "Li03_VE_2019-04-17","Li04_VE_2019-04-17","Li05_VE_2019-04-17","Li06_VE_2019-05-02","Li07_VE_2019-05-06","Li08_VE_2019-05-06","Li09_VE_2019-05-06"),
         Market_economy = fct_relevel(Market_economy, "Logging road","Forrest camp","Farming","Industrial"),
         Market_economy_split = fct_relevel(Market_economy_split, "Logging road","Forrest camp","Farming","Industrial plaque","Industrial calculus"),
         Age_group = fct_relevel(Age_group, "18-29","30-39","40-49","50-59","60+","HMP"),
         Sex = fct_relevel(Sex, "Male","Female"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior","HMP"))

metadata_cmc <- metadata %>%
  filter(!str_detect(SampleID, "EXB|LIB|SRR|JAE")) %>%
  filter(SampleID != "CMC032.B0101") %>%  # this sample has very few reads
  select(SampleID, ExtractionBatch, LibraryBatch, Village, Age_group, Market_economy, Ethnic_Group, Sex, Tooth_site) %>%
  as_tibble()



```

```{r pca_functions}
# add functions for:
# making the input table for PCA?

# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, color_group, shape_group, size_group) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))
    metadata_group_size <- get(paste(size_group, "_size", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]
    size_group = df_X[[size_group]]
                      
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group, size = size_group)) +
                        geom_point() +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        scale_size_manual(values = metadata_group_size) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

# example: 
# pca_L4_plot <- plot_pca(seed_full_L4.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")



# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                      # select the correct PC column and explained variance for PC1
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$SampleID), size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}

```

```{r pca_biplot_function}
# this function plots the PCA with biplots to look at the categories with strongest loadings
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle, pathways = c("TRUE","FALSE")) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$explained_variance * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }

    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Function") %>%
      top_n(10, !!arrow_pc)
   if (pathways == T) {
      pws_10 <- pws_10 %>%
        rename(pathway = Function) %>%
        mutate(Function = sapply(pathway, function(f) {
                                  unlist(str_split(f, ";"))[4]
                           })) 
    }

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Function") %>%
      top_n(-10, !!arrow_pc)
    if (pathways == T) {
      neg_10 <- neg_10 %>%
      rename(pathway = Function) %>%
       mutate(Function = sapply(pathway, function(f) {
                                  unlist(str_split(f, ";"))[4]
                           })) 
    }


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Function),
                   size = 1.5, colour = "grey20", label.padding = 0.2) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Function),
                   size = 1.5, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 10)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

```

First look at the total number of reads assigned to a SEED category
```{r percent_assigned}

seed_assignments <- seed_full %>%
  filter(Pathway == "SEED;" | Pathway == "SEED;Not assigned;") %>%
  mutate(Pathway = str_replace(Pathway, "SEED;Not assigned;","Not assigned")) %>%
  adorn_totals(where = "row", name = "Assigned") %>%
  filter(!Pathway == "SEED;") %>%
  gather("SampleID","Counts", 2:ncol(.)) %>%
  spread("Pathway","Counts") %>%
  adorn_percentages(denominator = "row") %>%
  mutate(Assigned = Assigned*100,
         "Not assigned" = `Not assigned`*100) %>%
  # merge with the metadata for plotting
  inner_join(., metadata %>% 
               select(SampleID, Individual_Seq_Depth_NonHuman, Ethnic_Group, Village, Description))


seed_assignments_box <- seed_assignments %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M")) %>%
  ggplot(., aes(Village, Assigned, fill = Village)) +
    geom_boxplot() +
    scale_fill_manual(values = Village_colors) +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ylab("Percent") +
    ggtitle("Percent of reads assigned to SEED proteins")
seed_assignments_box

# ggsave("./05-results.backup/humann2_alignment_stats_box.pdf", plot = humann2_alignment_stats_box, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

seed_assignments_w <- seed_assignments %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M")) 
pairwise.wilcox.test(seed_assignments_w$Assigned, seed_assignments_w$Village, p.adjust.method = "fdr")

```

SEED pathways are separated into 4 levels: SEED, general biomolecule processing
pathway, specific pathway, protein. I use L2, L3, and L4 to indicated the
general pathway, specific pathway, and protein assignments, respectively. All
assignments at levels higher than L4 are summs of the levels directly underneath
(to get the total counts for a specific L3 path, sum all L4 proteins under that
path, to get the total counts for a specific L2 path, sum all L3 paths under
that L2 path). All analysis here is on the protein level, L4, and this is the
abbreviation I use throughout. 

```{r number_proteins}
# start with the full table and pull out only the level 4 entries
seed_full_L4 <- seed_full[str_count(seed_full$Pathway, ";") == 4, ]

seed_full_L4 <- seed_full_L4 %>%
  select(-CMC032.B0101)  # this sample has very few reads

seed_counts <- seed_full_L4 %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  group_by(SampleID) %>%
  summarize(Total_detected = sum(Counts)) %>%
  inner_join(., metadata %>%
               select(SampleID, Village), by = "SampleID")

# make a box plot of the total number of proteins assigned in each group
seed_counts_box <- seed_counts %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M")) %>%
  ggplot(., aes(Village, Total_detected, fill = Village)) +
    geom_boxplot() +
    scale_fill_manual(values = Village_colors) +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ylab("Total number of proteins") +
    ggtitle("Total number of SEED proteins identified in samples")
seed_counts_box

# ggsave("./05-results.backup/humann2_alignment_stats_box.pdf", plot = humann2_alignment_stats_box, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

seed_counts_w <- seed_counts %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M")) 
pairwise.wilcox.test(seed_counts_w$Total_detected, seed_counts_w$Village, p.adjust.method = "fdr")

```
HMP samples have more proteins assigned than all villages, and JAE samples have
more proteins assigned than all villages and HMP. There are no significant
differences in the number of proteins assigned between any of the villages.


```{r number_proteins_decontam}
# start with the decontaminated table that has only the level 4 entries
seed_counts_decontam <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>% # this sample has very few reads
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  group_by(SampleID) %>%
  summarize(Total_detected = sum(Counts)) %>%
  inner_join(., metadata %>%
               select(SampleID, Village), by = "SampleID")

# make a box plot of the total number of proteins assigned in each group
seed_counts_decontam_box <- seed_counts_decontam %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M")) %>%
  ggplot(., aes(Village, Total_detected, fill = Village)) +
    geom_boxplot() +
    scale_fill_manual(values = Village_colors) +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
    ylab("Total number of proteins") +
    ggtitle("Number of SEED proteins remaining after filtering contaminants")
seed_counts_decontam_box

# ggsave("./05-results.backup/humann2_alignment_stats_box.pdf", plot = humann2_alignment_stats_box, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

seed_counts_decontam_w <- seed_counts_decontam %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M")) 
pairwise.wilcox.test(seed_counts_decontam_w$Total_detected, seed_counts_decontam_w$Village, p.adjust.method = "fdr")

```
HMP samples have significantly more proteins assigned than all villages after
removing potential contaminants identified by decontam. JAE samples have
significantly more proteins assigned than all villages and HMP after removing
potential contaminants identified by decontam. There are no significant
differences in the number of proteins assigned between any of the villages.


Start with a PCA to see if any of the samples fall with the controls based on
functional profile.
```{r pca_L4_with_controls}
# create the input matrix of only the protein-level assignments
# start with the full table and pull out only the level 4 entries
seed_full_L4 <- seed_full[str_count(seed_full$Pathway, ";") == 4, ]

# add +1 to all count cells and transpose
seed_full_L4 <- seed_full_L4 %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  gather("SampleID","Counts", 2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway, Counts) %>%
  column_to_rownames("SampleID") 

# check the number of components to retain by tuning the PCA
tune.seed_full_L4_pca <- tune.pca(seed_full_L4, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_full_L4.pca <- pca(seed_full_L4, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_L4_plot <- plot_pca(seed_full_L4.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_L4_plot

pca_L4_plot_names <- plot_pca_names(seed_full_L4.pca, "PC1", "PC2", "Ethnic_Group")
pca_L4_plot_names

# Now with the decontaminated table
seed_L4.decontam_filt <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  gather("SampleID","Counts", 2:ncol(.)) %>%
  spread(Pathway, Counts)

rownames(seed_L4.decontam_filt) <- seed_L4.decontam_filt$SampleID
seed_L4.decontam_filt <- as.matrix(seed_L4.decontam_filt %>% 
  select(-SampleID)) + 1 # add +1 to all values to be able to do a CLR transformation before running PCA

# check the number of components to retain by tuning the PCA
tune.seed_L4.decontam_filt_pca <- tune.pca(seed_L4.decontam_filt, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_L4.decontam_filt.pca <- pca(seed_L4.decontam_filt, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_L4_decontam_plot <- plot_pca(seed_L4.decontam_filt.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

L4_pca_controls <- plot_grid(pca_L4_plot, pca_L4_decontam_plot,
          nrow=1, labels = c("A", "B"), rel_widths = c(1, 1))

L4_pca_controls

```
None of the samples appear to be outliers. The extraction gauze plots with the
samples, but is very likely contaminated with spit. This is what we saw in the
taxonomy. The decontaminated L4 plot doesn't look all that different from the
full L4 plot. Clearly in both plots, the subsampled HMP and JAE samples are
plotting closer to the CMC samples than their full counterparts. This suggests
that there is a stronger effect of read count on the functional profiles than
there is on the taxonomy. We will continue to use the full HMP and JAE samples
however, since that's what we used for taxonomy and HUMAnN2 analyses.


How do the samples plot when the controls are removed? What's the relationship
of the samples to eachother? Let's also look at how filtering and
decontaminating affects the relationships of the samples.
```{r pca_L4_full_filt}
# Start with a full table that is not filtered or decontaminated
seed_full_L4 <- seed_full[str_count(seed_full$Pathway, ";") == 4, ]

seed_full_L4 <- seed_full_L4 %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  gather("SampleID","Counts", 2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway, Counts) %>%
  column_to_rownames("SampleID") 

# check the number of components to retain by tuning the PCA
tune.pca(seed_full_L4, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_full_L4.pca <- pca(seed_full_L4, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_L4_plot <- plot_pca(seed_full_L4.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")


# Now filter the full table, but don't decontaminate it
seed_full_L4 <- seed_full[str_count(seed_full$Pathway, ";") == 4, ]

seed_full_L4 <- seed_full_L4 %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  adorn_totals(where = "col", name = "Total") %>%
  mutate(Percent = Total/(sum(Total))*100) %>%
  filter(Percent >= 0.0001) %>%
  select(-c("Total","Percent")) %>%
  gather("SampleID","Counts", 2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway, Counts) %>%
  column_to_rownames("SampleID") 

# check the number of components to retain by tuning the PCA
tune.pca(seed_full_L4, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_full_L4.pca <- pca(seed_full_L4, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_L4_filt_plot <- plot_pca(seed_full_L4.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")


# Now filter the full table for only proteins present in at least 1/3 of samples from any ethnic group, but don't decontaminate it
# Determine the prevalence of SEED proteins in each group
seed_full_L4 <- seed_full[str_count(seed_full$Pathway, ";") == 4, ]

seed_full_L4_more_33 <- seed_full_L4 %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Pathway) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the SEED proteins that are present in at least 30% of at least one group
  ungroup() %>%
  filter(Percent_presence >= 33.33) %>%
  distinct(Pathway) %>%
  # pull out these proteins to keep
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(Pathway)

# keep only the proteins in at least 1/3 of at least one ethnic group 
seed_full_L4_33 <- seed_full_L4 %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(-c("SEED","L2","L3")) %>%
  inner_join(., seed_full_L4_more_33, by = "Pathway") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID") 

# check the number of components to retain by tuning the PCA
tune.pca(seed_full_L4_33, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_full_L4_33.pca <- pca(seed_full_L4_33, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_L4_33_plot <- plot_pca(seed_full_L4_33.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")


# Now keep only the proteins that are in MR1 from the factor analysis
mr1_proteins <- fread("./05-results.backup/minres_oblimin_noblanks.fa_seed.tsv")

seed_full_L4 <- seed_full[str_count(seed_full$Pathway, ";") == 4, ]

seed_full_L4_mr1 <- seed_full_L4 %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(-c("SEED","L2","L3")) %>%
  inner_join(., mr1_proteins %>%
               select(Pathway), by = "Pathway") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID") 

# check the number of components to retain by tuning the PCA
tune.pca(seed_full_L4_mr1, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_full_L4_mr1.pca <- pca(seed_full_L4_mr1, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_L4_mr1_plot <- plot_pca(seed_full_L4_mr1.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")


```

```{r pca_L4_decontam_filt}
# Now with the decontaminated table
seed_L4.decontam_filt <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  gather("SampleID","Counts", 2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway, Counts) %>%
  column_to_rownames("SampleID") 

# check the number of components to retain by tuning the PCA
tune.seed_L4.decontam_filt_pca <- tune.pca(seed_L4.decontam_filt, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_L4.decontam_filt.pca <- pca(seed_L4.decontam_filt, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_L4_decontam_plot <- plot_pca(seed_L4.decontam_filt.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")


# Now filter the decontaminated table
seed_L4.decontam_filt <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  adorn_totals(where = "col", name = "Total") %>%
  mutate(Percent = Total/(sum(Total))*100) %>%
  filter(Percent >= 0.0001) %>%
  select(-c("Total","Percent")) %>%
  gather("SampleID","Counts", 2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway, Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.pca(seed_L4.decontam_filt, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_L4.decontam_filt.pca <- pca(seed_L4.decontam_filt, ncomp = 3, logratio = 'CLR')

# plot the PCAs
pca_L4_decontam_filt_plot <- plot_pca(seed_L4.decontam_filt.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")


# Now filter the full table for only proteins present in at least 1/3 of samples from any ethnic group, but don't decontaminate it
# Determine the prevalence of SEED proteins in each group
seed_L4.decontam_noblanks_presence_more_30 <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Pathway) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the SEED proteins that are present in at least 30% of at least one group
  ungroup() %>%
  filter(Percent_presence >= 33.33) %>%
  distinct(Pathway) %>%
  # pull out these proteins to keep
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(Pathway)

# keep only the proteins in at least 1/3 of at least one ethnic group 
seed_L4.decontam_noblanks_33 <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(-c("SEED","L2","L3")) %>%
  inner_join(., seed_L4.decontam_noblanks_presence_more_30, by = "Pathway") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.pca(seed_L4.decontam_noblanks_33, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_L4.decontam_noblanks_33.pca <- pca(seed_L4.decontam_noblanks_33, ncomp = 3, logratio = 'CLR')

# plot the PCAs
pca_L4_decontam_33_plot <- plot_pca(seed_L4.decontam_noblanks_33.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")


# now filter for only the proteins in MR1
mr1_proteins <- fread("./05-results.backup/minres_oblimin_noblanks.fa_seed.tsv")

# keep only the proteins in at least 1/3 of at least one ethnic group 
seed_L4.decontam_noblanks_mr1 <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(-c("SEED","L2","L3")) %>%
  inner_join(., mr1_proteins %>%
               select(Pathway), by = "Pathway") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.pca(seed_L4.decontam_noblanks_mr1, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_L4.decontam_noblanks_mr1.pca <- pca(seed_L4.decontam_noblanks_mr1, ncomp = 3, logratio = 'CLR')

# plot the PCAs
pca_L4_decontam_mr1_plot <- plot_pca(seed_L4.decontam_noblanks_mr1.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")

```


```{r filteringPCA_plots}

L4_pca_filtering <- plot_grid(pca_L4_plot, pca_L4_filt_plot, pca_L4_33_plot, pca_L4_mr1_plot,
                              pca_L4_decontam_plot, pca_L4_decontam_filt_plot, pca_L4_decontam_33_plot, pca_L4_decontam_mr1_plot,
                              nrow=2, labels = c("Full", "Full-filtered", "Full-33", "Full-MR1", "Decontam", "Decontam-filtered", "Decontam-33", "Decontam-MR1"), rel_widths = c(1, 1))

L4_pca_filtering

```
The full table and the decontaminated table plots look very similar, as do the
full-filtered and decontaminated-filtered plots. Clearly removing the
"contaminant" proteins a doesn't remove proteins that strongly influence the
relationship of the industrial and non-industrial samples to eachother.
Filtering out the low abundance proteins (<0.0001%, <0.00005%) does cause the
CMC and industrial samples to plot more closely, and in some cases overlap,
suggesting that the low abundance proteins are what are distinct between the two
sample types. Is this related to the pan-genome/accessory genome differences
that may exist between strains of species found in each sample group? The HMP
and JAE samples separate from eachother on PC2, which is more visible in the
PC3-PC2 plots than in the PC1-PC2 plots. The PCAs with only proteins present in
at least 1/3 of at least one ethnic group look similar to the filtered PCAs,
with overlap of JAE and HMP with CMC, although the JAE/HMP look a little more
separated than in the abundance filtered plot. The PCAs using only proteins in
MR1 from factor analysis maintain more separation between the HMP/JAE and CMC
samples.

So which table to use, the decontaminated and prevalence-filtered table
(Decontam-33 plot above), or the decontaminated and MR1 protein-filtered table
(Decontam-MR1 plot above)? I'lll try both to see what the differences are.
Additional filtering proteins based on abundance might be important for the
biplot visualizations to be able to see the arrows, otherwise, they'll all
cluster tightly around the center of the plot.

# Decontaminated and prevalence-filtered table

## Cameroon and industrial samples
```{r pca_decontam_33_noblanks_metadata}

# Determine the prevalence of SEED proteins in each group
seed_L4.decontam_noblanks_presence_more_30 <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Pathway) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the SEED proteins that are present in at least 30% of at least one group
  ungroup() %>%
  filter(Percent_presence >= 33.33) %>%
  distinct(Pathway) %>%
  # pull out these proteins to keep
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(Pathway)

# keep only the proteins in at least 1/3 of at least one ethnic group 
seed_L4.decontam_noblanks_33 <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(-c("SEED","L2","L3")) %>%
  inner_join(., seed_L4.decontam_noblanks_presence_more_30, by = "Pathway") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.pca(seed_L4.decontam_noblanks_33, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_L4.decontam_noblanks_33.pca <- pca(seed_L4.decontam_noblanks_33, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(seed_L4.decontam_noblanks_33.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
plot_pca(seed_L4.decontam_noblanks_33.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
plot_pca(seed_L4.decontam_noblanks_33.pca, "PC1", "PC2", "Market_economy", "Market_economy", "Market_economy")
plot_pca(seed_L4.decontam_noblanks_33.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
plot_pca(seed_L4.decontam_noblanks_33.pca, "PC1", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
plot_pca(seed_L4.decontam_noblanks_33.pca, "PC1", "PC2", "Village", "Village", "Village")


```

```{r adonis_decontam_33_noblanks}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
seed_L4.decontam_noblanks_33.euc <- vegdist(seed_L4.decontam_noblanks_33.pca$X, method = "euclidean", na.rm = T)
seed_L4.decontam_noblanks_33.euc.pcoa <- ape::pcoa(seed_L4.decontam_noblanks_33.euc)

seed_L4.decontam_noblanks_33.euc.pcoavectors <- as.data.frame(seed_L4.decontam_noblanks_33.euc.pcoa$vectors)
seed_L4.decontam_noblanks_33.euc.pcoavectors <- seed_L4.decontam_noblanks_33.euc.pcoavectors %>%
  rownames_to_column("SampleID")

# seed_L4.decontam_noblanks_33.euc.vecmet <- seed_L4.decontam_noblanks_33.euc.pcoavectors %>%
#   select(1:11) %>%
#   inner_join(., metadata, by = "SampleID")

seed_L4.decontam_noblanks_33.euc.pcoavectors %>%
  select(1:11) %>%
  inner_join(., metadata, by = "SampleID")  %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
    geom_point(size = 4) +
    scale_colour_manual(values = Tooth_site_colors) +
    scale_shape_manual(values = Tooth_site_shapes) +
    stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
    theme_minimal(base_size = 7) +
    ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis2(seed_L4.decontam_noblanks_33.euc ~ ExtractionBatch +  LibraryBatch, metadata_noblanks, permutations = 999, by = "margin")
adonis2(seed_L4.decontam_noblanks_33.euc ~ Age_group, metadata_noblanks, permutations = 999, by = "margin")
adonis2(seed_L4.decontam_noblanks_33.euc ~ Ethnic_Group, metadata_noblanks, permutations = 999, by = "margin")
adonis2(seed_L4.decontam_noblanks_33.euc ~ Tooth_site, metadata_noblanks, permutations = 999, by = "margin")
adonis2(seed_L4.decontam_noblanks_33.euc ~ Sex, metadata_noblanks, permutations = 999, by = "margin")
adonis2(seed_L4.decontam_noblanks_33.euc ~ Village, metadata_noblanks, permutations = 999, by = "margin")
adonis2(seed_L4.decontam_noblanks_33.euc ~ Industry, metadata_noblanks, permutations = 999, by = "margin")
adonis2(seed_L4.decontam_noblanks_33.euc ~ Market_economy, metadata_noblanks, permutations = 999, by = "margin")
adonis2(seed_L4.decontam_noblanks_33.euc ~ Village + Sex + Industry + Ethnic_Group + Age_group +  Market_economy, metadata_noblanks, permutations = 999, by = "margin")
adonis2(seed_L4.decontam_noblanks_33.euc ~ Village + Sex + Industry + Ethnic_Group + Age_group +  Market_economy, metadata_noblanks, permutations = 999)

adonis2(seed_L4.decontam_noblanks_33.euc ~ Village + Sex + Industry + Ethnic_Group + Age_group +  Market_economy, data = metadata_noblanks, permutations = 999, by = "margin")

# controlled for tooth site
adonis2(seed_L4.decontam_noblanks_33.euc ~ Village + Sex + Industry + Ethnic_Group + Age_group +  Market_economy, metadata_noblanks, permutations = 999, by = "margin", strata = metadata_noblanks$Tooth_site)

adonis2(seed_L4.decontam_noblanks_33.euc ~ Village, metadata_noblanks, permutations = 999, by = "margin", strata = metadata_noblanks$Tooth_site)

pairwise.adonis2(seed_L4.decontam_noblanks_33.euc ~ Village, metadata_noblanks, nperm = 999, by = "margin", strata = metadata_noblanks$Tooth_site)
# Error in check(sn, control = control, quietly = quietly) : Number of observations and length of Block 'strata' do not match.

```
Tooth site is the only category that is significantly different (p < 0.01). When
controlling for tooth site, sex is the only category that is significantly
different (p < 0.05).


```{r biplots_decontam_33_noblanks_33}
# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc1 <- plot_pca_bi(seed_L4.decontam_noblanks_33.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = F)
pca_EG_biplot_pc1

ggsave("./05-results.backup/pca_EG_bi_noyanomami_paths.pdf", plot = pca_EG_biplot_pc1, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
        dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_biplot_pc1$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# make a table of the pathways to save, to use again later in another R notebook
# seed_protein_biplot_list <- pca_EG_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1+", Samples = "Industrial+CMC") 
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1-", Samples = "Industrial+CMC"))

# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc2 <- plot_pca_bi(seed_L4.decontam_noblanks_33.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = F)
pca_EG_biplot_pc2

# what are those pathways?
pandoc.table(pca_EG_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_biplot_pc2$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc2$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2+", Samples = "Industrial+CMC"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc2$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2-", Samples = "Industrial+CMC"))


# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc3 <- plot_pca_bi(seed_L4.decontam_noblanks_33.pca, "PC3", "PC2", "Ethnic_Group", PC2, pathways = F)
pca_EG_biplot_pc3

# what are those pathways?
pandoc.table(pca_EG_biplot_pc3$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_biplot_pc3$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc2$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC3)) %>% select(pathway) %>% mutate(Direction = "PC3+", Samples = "Industrial+CMC"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc2$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC3)) %>% select(pathway) %>% mutate(Direction = "PC3-", Samples = "Industrial+CMC"))


```
_PC1+_


_PC1-_


_PC2+_


_PC2-_



## Cameroon samples only
```{r pca_decontam_33_cmc_categories}

# Determine the prevalence of SEED proteins in each group
seed_L4.decontam_cmc_presence_more_30 <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|JAE|SRR")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Pathway) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the SEED proteins that are present in at least 30% of at least one group
  ungroup() %>%
  filter(Percent_presence >= 33.33) %>%
  distinct(Pathway) %>%
  # pull out these proteins to keep
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(Pathway)

# keep only the proteins in at least 1/3 of at least one ethnic group 
seed_L4.decontam_cmc_33 <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|JAE|SRR")) %>%
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(-c("SEED","L2","L3")) %>%
  inner_join(., seed_L4.decontam_cmc_presence_more_30, by = "Pathway") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.seed_L4.decontam_cmc_33_pca <- tune.pca(seed_L4.decontam_cmc_33, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_L4.decontam_cmc_33.pca <- pca(seed_L4.decontam_cmc_33, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(seed_L4.decontam_cmc_33.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
plot_pca(seed_L4.decontam_cmc_33.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
plot_pca(seed_L4.decontam_cmc_33.pca, "PC1", "PC2", "Market_economy", "Market_economy", "Market_economy")
plot_pca(seed_L4.decontam_cmc_33.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
plot_pca(seed_L4.decontam_cmc_33.pca, "PC1", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
plot_pca(seed_L4.decontam_cmc_33.pca, "PC1", "PC2", "Village", "Village", "Village")

plot_pca(seed_L4.decontam_cmc_33.pca, "PC3", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
plot_pca(seed_L4.decontam_cmc_33.pca, "PC3", "PC2", "Age_group", "Age_group", "Age_group")
plot_pca(seed_L4.decontam_cmc_33.pca, "PC3", "PC2", "Market_economy", "Market_economy", "Market_economy")
plot_pca(seed_L4.decontam_cmc_33.pca, "PC3", "PC2", "Sex", "Sex", "Sex")
plot_pca(seed_L4.decontam_cmc_33.pca, "PC3", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
plot_pca(seed_L4.decontam_cmc_33.pca, "PC3", "PC2", "Village", "Village", "Village")

```

```{r adonis_decontam_33_cmc}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
seed_L4.decontam_cmc_33.euc <- vegdist(seed_L4.decontam_cmc_33.pca$X, method = "euclidean", na.rm = T)
seed_L4.decontam_cmc_33.euc.pcoa <- ape::pcoa(seed_L4.decontam_cmc_33.euc)

seed_L4.decontam_cmc_33.euc.pcoavectors <- as.data.frame(seed_L4.decontam_cmc_33.euc.pcoa$vectors)
seed_L4.decontam_cmc_33.euc.pcoavectors <- seed_L4.decontam_cmc_33.euc.pcoavectors %>%
  rownames_to_column("SampleID")

seed_L4.decontam_cmc_33.euc.vecmet <- seed_L4.decontam_cmc_33.euc.pcoavectors %>%
  select(1:11) %>%
  inner_join(., metadata, by = "SampleID")

seed_L4.decontam_cmc_33.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 4) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis2(seed_L4.decontam_cmc_33.euc ~ ExtractionBatch +  LibraryBatch, seed_L4.decontam_cmc_33.euc.vecmet, permutations = 999, by = "margin")

adonis2(seed_L4.decontam_cmc_33.euc ~ Village + Sex + Tooth_site + Ethnic_Group + Age_group +  Market_economy, metadata_cmc, permutations = 999, by = "margin")

# controlled for tooth site
adonis2(seed_L4.decontam_cmc_33.euc ~ Village + Sex + Ethnic_Group + Age_group + Market_economy, metadata_cmc, permutations = 999, by = "margin", strata = metadata_cmc$Tooth_site)

```
Only tooth site is significantly different (p < 0.001). When controlling for
tooth site, only sex is significantly different (p < 0.05).


```{r biplots_decontam_33_cmc}
# which paths distinguish sample groups when only CMC samples are included?
pca_EG_cmc_33_biplot_pc1 <- plot_pca_bi(seed_L4.decontam_cmc_33.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = F)
pca_EG_cmc_33_biplot_pc1

ggsave("./05-results.backup/pca_EG_bi_cmc_33_paths.pdf", plot = pca_EG_cmc_33_biplot_pc1, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_cmc_33_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_33_biplot_pc1$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>% 
#   bind_rows(pca_EG_cmc_33_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1+", Samples = "cmc_33"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1-", Samples = "cmc_33"))

pca_EG_cmc_33_biplot_pc2 <- plot_pca_bi(seed_L4.decontam_cmc_33.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = F)
pca_EG_cmc_33_biplot_pc2

pandoc.table(pca_EG_cmc_33_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_33_biplot_pc2$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>% 
#   bind_rows(pca_EG_cmc_33_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2+", Samples = "cmc_33"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2-", Samples = "cmc_33"))
# 
# fwrite(seed_protein_biplot_list, file = "./05-results.backup/seed_protein_biplot_list.tsv", quote = F, sep = "\t")

pca_EG_cmc_33_biplot_pc3 <- plot_pca_bi(seed_L4.decontam_cmc_33.pca, "PC3", "PC2", "Ethnic_Group", PC3, pathways = F)
pca_EG_cmc_33_biplot_pc3

pandoc.table(pca_EG_cmc_33_biplot_pc3$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_33_biplot_pc3$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>% 
#   bind_rows(pca_EG_cmc_33_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC3)) %>% select(pathway) %>% mutate(Direction = "PC3+", Samples = "cmc_33"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC3)) %>% select(pathway) %>% mutate(Direction = "PC3-", Samples = "cmc_33"))
# 
# fwrite(seed_protein_biplot_list, file = "./05-results.backup/seed_protein_biplot_list.tsv", quote = F, sep = "\t")


```
_PC1+_


_PC1-_


_PC2+_


_PC2-_




# Decontaminated and MR1 protein only-filtered table

## Cameroon and idustrial samples
```{r pca_decontam_mr1_noblanks_metadata}

# now filter for only the proteins in MR1
mr1_proteins <- fread("./05-results.backup/minres_oblimin_noblanks.fa_seed.tsv")

# keep only the proteins in at least 1/3 of at least one ethnic group 
seed_L4.decontam_noblanks_mr1 <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(-c("SEED","L2","L3")) %>%
  inner_join(., mr1_proteins %>%
               filter(Residual == "MR1") %>%
               select(Pathway), by = "Pathway") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.seed_L4.decontam_noblanks_mr1_pca <- tune.pca(seed_L4.decontam_noblanks_mr1, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_L4.decontam_noblanks_mr1.pca <- pca(seed_L4.decontam_noblanks_mr1, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(seed_L4.decontam_noblanks_mr1.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
plot_pca(seed_L4.decontam_noblanks_mr1.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
plot_pca(seed_L4.decontam_noblanks_mr1.pca, "PC1", "PC2", "Market_economy", "Market_economy", "Market_economy")
plot_pca(seed_L4.decontam_noblanks_mr1.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
plot_pca(seed_L4.decontam_noblanks_mr1.pca, "PC1", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
plot_pca(seed_L4.decontam_noblanks_mr1.pca, "PC1", "PC2", "Village", "Village", "Village")

plot_pca(seed_L4.decontam_noblanks_mr1.pca, "PC3", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
plot_pca(seed_L4.decontam_noblanks_mr1.pca, "PC3", "PC2", "Age_group", "Age_group", "Age_group")
plot_pca(seed_L4.decontam_noblanks_mr1.pca, "PC3", "PC2", "Market_economy", "Market_economy", "Market_economy")
plot_pca(seed_L4.decontam_noblanks_mr1.pca, "PC3", "PC2", "Sex", "Sex", "Sex")
plot_pca(seed_L4.decontam_noblanks_mr1.pca, "PC3", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
plot_pca(seed_L4.decontam_noblanks_mr1.pca, "PC3", "PC2", "Village", "Village", "Village")


```


```{r adonis_decontam_mr1_noblanks}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
seed_L4.decontam_noblanks_mr1.euc <- vegdist(seed_L4.decontam_noblanks_mr1.pca$X, method = "euclidean", na.rm = T)
seed_L4.decontam_noblanks_mr1.euc.pcoa <- ape::pcoa(seed_L4.decontam_noblanks_mr1.euc)

seed_L4.decontam_noblanks_mr1.euc.pcoavectors <- as.data.frame(seed_L4.decontam_noblanks_mr1.euc.pcoa$vectors)
seed_L4.decontam_noblanks_mr1.euc.pcoavectors <- seed_L4.decontam_noblanks_mr1.euc.pcoavectors %>%
  rownames_to_column("SampleID")

seed_L4.decontam_noblanks_mr1.euc.pcoavectors %>%
  select(1:11) %>%
  inner_join(., metadata, by = "SampleID") %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
    geom_point(size = 4) +
    scale_colour_manual(values = Tooth_site_colors) +
    scale_shape_manual(values = Tooth_site_shapes) +
    stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
    theme_minimal(base_size = 7) +
    ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis2(seed_L4.decontam_noblanks_mr1.euc ~ ExtractionBatch +  LibraryBatch, metadata_noblanks, permutations = 999, by = "margin")

adonis2(seed_L4.decontam_noblanks_mr1.euc ~ Village + Sex + Industry + Tooth_site + Ethnic_Group + Age_group + Market_economy, metadata_noblanks, permutations = 999, by = "margin")

# controlled for tooth site
adonis2(seed_L4.decontam_noblanks_mr1.euc ~ Village + Sex + Industry + Ethnic_Group + Age_group + Market_economy, metadata_noblanks, permutations = 999, by = "margin", strata = metadata_cmc$Tooth_site)

```
When using proteins from both MR1 and MR2, only tooth site is significantly
different (p < 0.05), and further controlling for tooth site shows no categories
are significantly different. When using only proteins in MR1, no categories are
significantly different, but further controlling for tooth site shows that sex
is significantly different (p < 0.05).


```{r biplots_decontam_mr1_noblanks}
# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc1 <- plot_pca_bi(seed_L4.decontam_noblanks_mr1.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = F)
pca_EG_biplot_pc1

# ggsave("./05-results.backup/pca_EG_bi_noyanomami_paths.pdf", plot = pca_EG_biplot_pc1, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#         dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_biplot_pc1$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# make a table of the pathways to save, to use again later in another R notebook
# seed_protein_biplot_list <- pca_EG_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1+", Samples = "Industrial+CMC") 
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1-", Samples = "Industrial+CMC"))

# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc2 <- plot_pca_bi(seed_L4.decontam_noblanks_mr1.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = F)
pca_EG_biplot_pc2

# what are those pathways?
pandoc.table(pca_EG_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_biplot_pc2$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc2$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2+", Samples = "Industrial+CMC"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc2$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2-", Samples = "Industrial+CMC"))

# which paths distinguish sample types when HMP and JAE are included?
pca_EG_biplot_pc3 <- plot_pca_bi(seed_L4.decontam_noblanks_mr1.pca, "PC3", "PC2", "Ethnic_Group", PC3, pathways = F)
pca_EG_biplot_pc3

# what are those pathways?
pandoc.table(pca_EG_biplot_pc3$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(pca_EG_biplot_pc3$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc2$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2+", Samples = "Industrial+CMC"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc2$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2-", Samples = "Industrial+CMC"))

```
_PC1+_


_PC1-_


_PC2+_


_PC2-_



## Cameroon samples only

```{r pca_decontam_mr1_cmc_categories}

# now filter for only the proteins in MR1
mr1_proteins <- fread("./05-results.backup/minres_oblimin_noblanks.fa_seed.tsv")

# keep only the proteins in at least 1/3 of at least one ethnic group 
seed_L4.decontam_cmc_mr1 <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|JAE|SRR")) %>%
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(-c("SEED","L2","L3")) %>%
  inner_join(., mr1_proteins %>%
               filter(Residual == "MR1") %>%
               select(Pathway), by = "Pathway") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.seed_L4.decontam_cmc_mr1_pca <- tune.pca(seed_L4.decontam_cmc_mr1, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_L4.decontam_cmc_mr1.pca <- pca(seed_L4.decontam_cmc_mr1, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(seed_L4.decontam_cmc_mr1.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
plot_pca(seed_L4.decontam_cmc_mr1.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
plot_pca(seed_L4.decontam_cmc_mr1.pca, "PC1", "PC2", "Market_economy", "Market_economy", "Market_economy")
plot_pca(seed_L4.decontam_cmc_mr1.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
plot_pca(seed_L4.decontam_cmc_mr1.pca, "PC1", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
plot_pca(seed_L4.decontam_cmc_mr1.pca, "PC1", "PC2", "Village", "Village", "Village")

plot_pca(seed_L4.decontam_cmc_mr1.pca, "PC3", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
plot_pca(seed_L4.decontam_cmc_mr1.pca, "PC3", "PC2", "Age_group", "Age_group", "Age_group")
plot_pca(seed_L4.decontam_cmc_mr1.pca, "PC3", "PC2", "Market_economy", "Market_economy", "Market_economy")
plot_pca(seed_L4.decontam_cmc_mr1.pca, "PC3", "PC2", "Sex", "Sex", "Sex")
plot_pca(seed_L4.decontam_cmc_mr1.pca, "PC3", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
plot_pca(seed_L4.decontam_cmc_mr1.pca, "PC3", "PC2", "Village", "Village", "Village")

```
PC1 separates by tooth site (anterior/posterior) and PC3 separates by sex
(male/female). It's not clear to me what PC2 is separating by.


```{r adonis_decontam_mr1_cmc}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
seed_L4.decontam_cmc_mr1.euc <- vegdist(seed_L4.decontam_cmc_mr1.pca$X, method = "euclidean", na.rm = T)
seed_L4.decontam_cmc_mr1.euc.pcoa <- ape::pcoa(seed_L4.decontam_cmc_mr1.euc)

seed_L4.decontam_cmc_mr1.euc.pcoavectors <- as.data.frame(seed_L4.decontam_cmc_mr1.euc.pcoa$vectors)
seed_L4.decontam_cmc_mr1.euc.pcoavectors <- seed_L4.decontam_cmc_mr1.euc.pcoavectors %>%
  rownames_to_column("SampleID")

seed_L4.decontam_cmc_mr1.euc.pcoavectors %>%
  select(1:11) %>%
  inner_join(., metadata, by = "SampleID") %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
    geom_point(size = 4) +
    scale_colour_manual(values = Tooth_site_colors) +
    scale_shape_manual(values = Tooth_site_shapes) +
    stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
    theme_minimal(base_size = 7) +
    ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis2(seed_L4.decontam_cmc_mr1.euc ~ ExtractionBatch + Village + Sex + Tooth_site + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, metadata_cmc, permutations = 999, by = "margin")

adonis2(seed_L4.decontam_cmc_mr1.euc ~ ExtractionBatch +  LibraryBatch, metadata_cmc, permutations = 999, by = "margin")

adonis2(seed_L4.decontam_cmc_mr1.euc ~ Village + Sex + Tooth_site + Ethnic_Group + Age_group + Market_economy, metadata_cmc, permutations = 999, by = "margin")

adonis2(seed_L4.decontam_cmc_mr1.euc ~ Ethnic_Group, metadata_cmc, permutations = 999, by = "margin")

# controlled for tooth site
adonis2(seed_L4.decontam_cmc_mr1.euc ~ ExtractionBatch + Village + Sex + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, metadata_cmc, permutations = 999, by = "margin", strata = metadata_cmc$Tooth_site)

```
When including proteins from both MR1 and MR2, only tooth site is significantly
different (p < 0.01), and further controlling for tooth site shows none of the
categories are significantly different. When including proteins from only MR1,
both tooth site (p < 0.01) and sex (p < 0.05) are significantly different, and
further controlling for tooth site shows sex (p < 0.05) is significantly
different.


```{r biplots_decontam_mr1_noblanks}
# which paths distinguish sample groups when only CMC samples are included?
pca_EG_cmc_biplot_pc1 <- plot_pca_bi(seed_L4.decontam_cmc_mr1.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = F)
pca_EG_cmc_biplot_pc1

# ggsave("./05-results.backup/pca_EG_bi_cmc_paths.pdf", plot = pca_EG_cmc_biplot_pc1, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>% 
#   bind_rows(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1+", Samples = "CMC"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1-", Samples = "CMC"))

pca_EG_cmc_biplot_pc2 <- plot_pca_bi(seed_L4.decontam_cmc_mr1.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = F)
pca_EG_cmc_biplot_pc2

pandoc.table(pca_EG_cmc_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc2$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>% 
#   bind_rows(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2+", Samples = "CMC"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2-", Samples = "CMC"))
# 
# fwrite(seed_protein_biplot_list, file = "./05-results.backup/seed_protein_biplot_list.tsv", quote = F, sep = "\t")

pca_EG_cmc_biplot_pc3 <- plot_pca_bi(seed_L4.decontam_cmc_mr1.pca, "PC3", "PC2", "Sex", PC3, pathways = F)
pca_EG_cmc_biplot_pc3

pandoc.table(pca_EG_cmc_biplot_pc3$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc3$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>% 
#   bind_rows(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC3+", Samples = "CMC"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC3-", Samples = "CMC"))
# 
# fwrite(seed_protein_biplot_list, file = "./05-results.backup/seed_protein_biplot_list.tsv", quote = F, sep = "\t")

```
_PC1+_


_PC1-_


_PC2+_


_PC2-_

# MR1 protein pathway check
```{r protein_difference_stats}
mr1_proteins <- fread("./05-results.backup/minres_oblimin_noblanks.fa_seed.tsv")

# keep only the proteins in at least 1/3 of at least one ethnic group 
seed_L4_mr1_proteins <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  inner_join(., mr1_proteins %>%
               filter(Residual == "MR1") %>%
               select(Pathway), by = "Pathway")

# how many proteins in each metabolic pathway
L3_counts <- seed_L4_mr1_proteins %>%
  group_by(L3) %>%
  summarize(ProteinCount = length(Pathway))

# how many proteins in each higher-level processing pathway?  
L2_counts <- seed_L4_mr1_proteins %>%
  group_by(L2) %>%
  summarize(ProteinCount = length(Pathway))


# convert to a long table
seed_L4_mr1_proteinsL <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  select(-matches("EXB|LIB|10M")) %>%
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  inner_join(., mr1_proteins %>%
               filter(Residual == "MR1") %>%
               select(Pathway), by = "Pathway") %>%
  gather("SampleID","Counts", 5:ncol(.)) %>%
  full_join(., metadata %>%
              select(SampleID, Ethnic_Group), by = "SampleID")

seed_L4_mr1_proteinsL %>% group_by(Ethnic_Group, L3) %>% summarize(Totals = sum(Counts)) # total counts at L3 by Ethnic group
seed_L4_mr1_proteinsL %>% group_by(Ethnic_Group, L3) %>% filter(Counts > 0) %>% summarize(Totals = length(unique(SampleID))) # number of samples in each ethnic group with each L3 pathway
seed_L4_mr1_proteinsL %>% group_by(Ethnic_Group, L3) %>% filter(Counts > 0) %>% summarize(Totals = length(unique(Pathway))) # total number of proteins in each L3 by ethnic group
seed_L4_mr1_proteinsL %>% group_by(Ethnic_Group, L2) %>% filter(Counts > 0) %>% summarize(Totals = length(unique(Pathway))) # total number of proteins in each L2 by ethnic group

```


# Carbohydrate pathway proteins only
Now we'll look at whether there are differences in the sample groups based only
on proteins in carbohydrate processing pathways. There are differences in the
carbohydrates consumed by each group, and bacteria are able to process a wide
variety of carbohydrates. 
```{r carb_proteins}

# select only proteins in the Carbohydrates pathway
seed_L4_carbohydrates <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%
  select(-matches("EXB|LIB|SRR164|10M")) %>%
  filter(str_detect(Pathway, "Carbohydrates")) %>%
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(-c("SEED","L2","L3")) %>%
  adorn_totals(where = "col") %>%
  filter(Total > 0) %>% # remove any that are not found in the remaining samples (HMP, JAE, CMC)
  select(-Total) %>%
  gather("SampleID","Counts", 2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.seed_L4_carbohydrates_pca <- tune.pca(seed_L4_carbohydrates, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_L4_carbohydrates.pca <- pca(seed_L4_carbohydrates, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(seed_L4_carbohydrates.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
plot_pca(seed_L4_carbohydrates.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
plot_pca(seed_L4_carbohydrates.pca, "PC1", "PC2", "Market_economy", "Market_economy", "Market_economy")
plot_pca(seed_L4_carbohydrates.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
plot_pca(seed_L4_carbohydrates.pca, "PC1", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
plot_pca(seed_L4_carbohydrates.pca, "PC1", "PC2", "Village", "Village", "Village")

plot_pca(seed_L4_carbohydrates.pca, "PC3", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
plot_pca(seed_L4_carbohydrates.pca, "PC3", "PC2", "Age_group", "Age_group", "Age_group")
plot_pca(seed_L4_carbohydrates.pca, "PC3", "PC2", "Market_economy", "Market_economy", "Market_economy")
plot_pca(seed_L4_carbohydrates.pca, "PC3", "PC2", "Sex", "Sex", "Sex")
plot_pca(seed_L4_carbohydrates.pca, "PC3", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
plot_pca(seed_L4_carbohydrates.pca, "PC3", "PC2", "Village", "Village", "Village")

```
There is a clear separation between the Cameroon plaque samples and the
industrial plaque and calculus samples along PC1. The CMC samples separate by
tooth site along PC2, with posterior samples plotting in more positive values
and anterior samples plotting in more negative values. This is a little
unexpected, b/c the HMP samples plot in more positice PC2 values and the JAE
samples plot in more negative PC2 values, and I'd expect the posterior samples
to plot more with the JAE samples.

Are there significant differences between the groups?
```{r adonis_carbohydrates_noblanks}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
seed_L4_carbohydrates_noblanks.euc <- vegdist(seed_L4_carbohydrates.pca$X, method = "euclidean", na.rm = T)
seed_L4_carbohydrates_noblanks.euc.pcoa <- ape::pcoa(seed_L4_carbohydrates_noblanks.euc)

seed_L4_carbohydrates_noblanks.euc.pcoavectors <- as.data.frame(seed_L4_carbohydrates_noblanks.euc.pcoa$vectors)
seed_L4_carbohydrates_noblanks.euc.pcoavectors <- seed_L4_carbohydrates_noblanks.euc.pcoavectors %>%
  rownames_to_column("SampleID")

seed_L4_carbohydrates_noblanks.euc.vecmet <- seed_L4_carbohydrates_noblanks.euc.pcoavectors %>%
  select(1:11) %>%
  inner_join(., metadata, by = "SampleID")

seed_L4_carbohydrates_noblanks.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 4) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis2(seed_L4_carbohydrates_noblanks.euc ~ ExtractionBatch +  LibraryBatch, metadata_noblanks, permutations = 999, by = "margin")

adonis2(seed_L4_carbohydrates_noblanks.euc ~ Village + Sex + Industry + Tooth_site + Ethnic_Group + Age_group + Market_economy, metadata_noblanks, permutations = 999, by = "margin")

# controlled for tooth site
adonis2(seed_L4_carbohydrates_noblanks.euc ~ Village + Sex + Industry + Ethnic_Group + Age_group + Market_economy, metadata_noblanks, permutations = 999, by = "margin", strata = metadata_noblanks$Tooth_site)

```
Only tooth site is significantly different (p < 0.01). When controlling for
tooth site, none of the groups are significantly different.

What are the differences if we look only within the Cameroon plaque samples?
```{r carb_proteins_cmc}
# select only proteins in the Carbohydrates pathway
seed_L4_carbohydrates_cmc <- seed_L4.decontam %>%
  select(-CMC032.B0101) %>%
  select(-matches("EXB|LIB|SRR|JAE")) %>%
  filter(str_detect(Pathway, "Carbohydrates")) %>%
  separate(., Pathway, into = c("SEED","L2","L3","Pathway"), sep = ";") %>%
  select(-c("SEED","L2","L3")) %>%
  adorn_totals(where = "col") %>%
  filter(Total > 0) %>% # remove any that are not found in the remaining samples (HMP, JAE, CMC)
  select(-Total) %>%
  gather("SampleID","Counts", 2:ncol(.)) %>%
  mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")

# check the number of components to retain by tuning the PCA
tune.seed_L4_carbohydrates_cmc_pca <- tune.pca(seed_L4_carbohydrates_cmc, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_L4_carbohydrates_cmc.pca <- pca(seed_L4_carbohydrates_cmc, ncomp = 3, logratio = 'CLR')

# plot the PCA
plot_pca(seed_L4_carbohydrates_cmc.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
plot_pca(seed_L4_carbohydrates_cmc.pca, "PC1", "PC2", "Age_group", "Age_group", "Age_group")
plot_pca(seed_L4_carbohydrates_cmc.pca, "PC1", "PC2", "Market_economy", "Market_economy", "Market_economy")
plot_pca(seed_L4_carbohydrates_cmc.pca, "PC1", "PC2", "Sex", "Sex", "Sex")
plot_pca(seed_L4_carbohydrates_cmc.pca, "PC1", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
plot_pca(seed_L4_carbohydrates_cmc.pca, "PC1", "PC2", "Village", "Village", "Village")

# plot_pca(seed_L4_carbohydrates_cmc.pca, "PC3", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
# plot_pca(seed_L4_carbohydrates_cmc.pca, "PC3", "PC2", "Age_group", "Age_group", "Age_group")
# plot_pca(seed_L4_carbohydrates_cmc.pca, "PC3", "PC2", "Market_economy", "Market_economy", "Market_economy")
# plot_pca(seed_L4_carbohydrates_cmc.pca, "PC3", "PC2", "Sex", "Sex", "Sex")
# plot_pca(seed_L4_carbohydrates_cmc.pca, "PC3", "PC2", "Tooth_site", "Tooth_site", "Tooth_site")
# plot_pca(seed_L4_carbohydrates_cmc.pca, "PC3", "PC2", "Village", "Village", "Village")

```
The samples separate by tooth site along PC1, but no other potential clusterings are clear. 

Are there significant differences between the groups?
```{r adonis_carbohydrates_cmc}
# Use the CLR-transformed matrix from the mixOmics PCA in the section above
# Euclidean distance
seed_L4_carbohydrates_cmc.euc <- vegdist(seed_L4_carbohydrates_cmc.pca$X, method = "euclidean", na.rm = T)
seed_L4_carbohydrates_cmc.euc.pcoa <- ape::pcoa(seed_L4_carbohydrates_cmc.euc)

seed_L4_carbohydrates_cmc.euc.pcoavectors <- as.data.frame(seed_L4_carbohydrates_cmc.euc.pcoa$vectors)
seed_L4_carbohydrates_cmc.euc.pcoavectors <- seed_L4_carbohydrates_cmc.euc.pcoavectors %>%
  rownames_to_column("SampleID")

seed_L4_carbohydrates_cmc.euc.pcoavectors %>%
  select(1:11) %>%
  inner_join(., metadata, by = "SampleID")%>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
    geom_point(size = 4) +
    scale_colour_manual(values = Tooth_site_colors) +
    scale_shape_manual(values = Tooth_site_shapes) +
    stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
    theme_minimal(base_size = 7) +
    ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis2(seed_L4_carbohydrates_cmc.euc ~ ExtractionBatch + Village + Sex + Tooth_site + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, metadata_cmc, permutations = 999, by = "margin")

# controlled for tooth site
adonis2(seed_L4_carbohydrates_cmc.euc ~ Village + Sex + Ethnic_Group + Age_group +  Market_economy, metadata_cmc, permutations = 999, by = "margin", strata = metadata_cmc$Tooth_site)

```
Only tooth site is significantly different (p < 0.01). When controlling for
tooth site, sex is significantly different (p < 0.05).


Which carbohydrate processing pathway proteins are contributing most to the
separation of samples?
```{r biplots_decontam_mr1_cmc}
# which paths distinguish sample groups when only CMC samples are included?
pca_EG_noblanks_biplot_pc1 <- plot_pca_bi(seed_L4_carbohydrates.pca, "PC1", "PC2", "Ethnic_Group", PC1, pathways = F)
pca_EG_noblanks_biplot_pc1

# ggsave("./05-results.backup/pca_EG_bi_noblanks_paths.pdf", plot = pca_EG_noblanks_biplot_pc1, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# what are those pathways?
pandoc.table(pca_EG_noblanks_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_noblanks_biplot_pc1$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>% 
#   bind_rows(pca_EG_noblanks_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1+", Samples = "CMC"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_noblanks_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC1)) %>% select(pathway) %>% mutate(Direction = "PC1-", Samples = "CMC"))

pca_EG_noblanks_biplot_pc2 <- plot_pca_bi(seed_L4_carbohydrates.pca, "PC1", "PC2", "Ethnic_Group", PC2, pathways = F)
pca_EG_noblanks_biplot_pc2

pandoc.table(pca_EG_noblanks_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_noblanks_biplot_pc2$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>% 
#   bind_rows(pca_EG_noblanks_biplot_pc2$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2+", Samples = "CMC"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_noblanks_biplot_pc2$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC2-", Samples = "CMC"))
# 
# fwrite(seed_protein_biplot_list, file = "./05-results.backup/seed_protein_biplot_list.tsv", quote = F, sep = "\t")

pca_EG_cmc_biplot_pc1 <- plot_pca_bi(seed_L4_carbohydrates_cmc.pca, "PC1", "PC2", "Sex", PC1, pathways = F)
pca_EG_cmc_biplot_pc1

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC1 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>% 
#   bind_rows(pca_EG_cmc_biplot_pc1$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC3+", Samples = "CMC"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_cmc_biplot_pc1$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC3-", Samples = "CMC"))
# 
# fwrite(seed_protein_biplot_list, file = "./05-results.backup/seed_protein_biplot_list.tsv", quote = F, sep = "\t")


pca_EG_cmc_biplot_pc2 <- plot_pca_bi(seed_L4_carbohydrates_cmc.pca, "PC1", "PC2", "Sex", PC2, pathways = F)
pca_EG_cmc_biplot_pc2

pandoc.table(pca_EG_cmc_biplot_pc2$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(pca_EG_cmc_biplot_pc2$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lrrr",
             split.tables = Inf,
             caption = "PC2 negative loadings")

# seed_protein_biplot_list <- seed_protein_biplot_list %>% 
#   bind_rows(pca_EG_cmc_biplot_pc2$plot_env$pws_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC3+", Samples = "CMC"))
# seed_protein_biplot_list <- seed_protein_biplot_list %>%
#   bind_rows(pca_EG_cmc_biplot_pc2$plot_env$neg_10 %>% select(-Function) %>% arrange(desc(PC2)) %>% select(pathway) %>% mutate(Direction = "PC3-", Samples = "CMC"))
# 
# fwrite(seed_protein_biplot_list, file = "./05-results.backup/seed_protein_biplot_list.tsv", quote = F, sep = "\t")

```
















