---
title: "Cameroon hunter-gatherer calculus/plaque MALT discriminant taxa"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---

```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(psych)
library(ape)
library(mixOmics)
library(compositions)
library(vegan)
library(MASS)
library(janitor)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

```{r load_data}

malt_species <- fread("./05-results.backup/CMC_HMP_JAE_Yano_MALT_RSC_species_bacteria_archaea.txt")
malt_genus <- fread("./05-results.backup/CMC_HMP_JAE_Yano_MALT_RSC_genus_bacteria_archaea.txt")

# clean the file names
malt_species <- rename(malt_species, Species = `#Datasets`)
colnames(malt_species) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M","_10M", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam","_10M", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub(".fastq.truncated","", colnames(malt_species))


# remove human and unassigned reads
malt_species <- malt_species %>% filter(!str_detect(Species, "Homo sapiens"))
malt_species <- malt_species %>% filter(!str_detect(Species, "Not assigned"))
# get rid of the extra column
malt_species <- malt_species %>% select(-Comparison_CMC_HMP_JAE_MALT_RSC)

malt_genus <- rename(malt_genus, Species = `#Datasets`)
colnames(malt_genus) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam","_10M", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M","_10M", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub(".fastq.truncated","", colnames(malt_genus))

# remove human and unassigned reads
malt_genus <- malt_genus %>% filter(!str_detect(Species, "Homo"))
malt_genus <- malt_genus %>% filter(!str_detect(Species, "Not assigned"))
# get rid of the extra column at the begining
malt_genus <- malt_genus %>% select(-Comparison_CMC_HMP_JAE_MALT_RSC)

# load the metadata file
metadata <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata <- metadata %>%
  rename(SampleID = `#SampleID`)  %>%
  as_tibble() %>%
  mutate(Description = fct_relevel(Description, "HunterGatherer", "Yanomami", "HMP_supPlaque", "HMP_subPlaque",
                                   "HMP10M_supPlaque", "HMP10M_subPlaque","JAE","JAE10M",
                                   "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime","Yanomami","HMP","HMP10M","JAE","JAE10M",
                                     "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Env = fct_relevel(Env, "HunterGatherer","HMP","HMP10M","JAE","JAE10M",
                                     "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Market_economy = fct_relevel(Market_economy, "Logging road","Forrest camp","Farming","Yanomami",
                                      "Industrial","ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Age_group = fct_relevel(Age_group, "10-18","18-29","30-39","40-49","50-59","60+","HMP","HMP10M"),
         Sex = fct_relevel(Sex, "Male","Female", "ExtractionGauze","ExtractionBlank","LibraryBlank","Unknown"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior","Buccal_mucosa","HMP","HMP10M"),
         Type = fct_relevel(Type, "Calculus","Plaque","Blank"),
         Village = fct_relevel(Village, "202","205","204","203","206","Yanomami","HMP","HMP10M","JAE","JAE10M",
                               "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Industry = fct_relevel(Industry, "Non-industrial","Industrial"))

fullSRR <- metadata %>%
  select(SampleID) %>%
  filter(str_detect(SampleID, "SRR")) %>%
  filter(!str_detect(SampleID, "10M")) %>%
  pull
```


```{r colors_shapes_peachpear}
# lacroix_palette("PeachPear", n = 8, type = "continuous")
# "#FF3200" "#EF8158" "#E9BD8E" "#CBDDA7" "#38BCAD" "#0B91B5" "#065FA3" "#172869"

Village_colors <- c(`202` = "#FF3200", `205` = "#EF8158", `204` = "#E9BD8E", `203` = "#CBDDA7", `206` = "#38BCAD",
                    Yanomami = "#C70E7B", HMP = "#172869", HMP10M = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                     ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Village_shapes <- c(`202` = 15, `203` = 16, `205` = 17,`204` = 18, `206` = 25, Yanomami = 14, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9,
                    ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Village_size <- c(`202` = 4.5, `203` = 4, `205` = 4,`204` = 6, `206` = 4, Yanomami = 4, HMP = 4, HMP10M = 4, JAE = 4, JAE10M = 4,
                    ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Ethnic_Group_colors <- c(Baka = "#FF3200", Nzime = "#172869", Yanomami = "#C70E7B", HMP = "#969696", HMP10M = "#d9d9d9", 
                         JAE10M = "#ffeda0", JAE = "#feb24c",ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Ethnic_Group_shapes <- c(Baka = 19, Nzime = 17,  Yanomami = 14, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9,
                         ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Ethnic_Group_size <- c(Baka = 4, Nzime = 4,  Yanomami = 4, HMP = 4, HMP10M = 4, JAE = 4, JAE10M = 4,
                         ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Env_colors <- c(HunterGatherer = "#FF3200", HMP = "#172869", HMP10M = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Env_shapes <- c(HunterGatherer = 19, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Env_size <- c(HunterGatherer = 4, HMP = 4, HMP10M = 4, JAE = 4, JAE10M = 4, ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Description_colors <- c(HunterGatherer = "#FF3200", Yanomami = "#C70E7B", HMP_supPlaque = "#969696", HMP_subPlaque = "#969696", # HMP_supPlaque = "#172869", HMP_subPlaque = "#172869", 
                        HMP10M_supPlaque = "#065FA3", HMP10M_subPlaque = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                        ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Description_shapes <- c(HunterGatherer = 19, Yanomami = 14, HMP_supPlaque = 10, HMP_subPlaque = 13, HMP10M_supPlaque = 10, 
                        HMP10M_subPlaque = 13, JAE = 7, JAE10M = 9, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Description_size <- c(HunterGatherer = 4, Yanomami = 4, HMP_supPlaque = 4, HMP_subPlaque = 4, HMP10M_supPlaque = 4, 
                        HMP10M_subPlaque = 4, JAE = 4, JAE10M = 4, ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Market_economy_colors <- c(`Logging road` = "#FF3200", `Forrest camp` = "#172869", Farming = "#82CDAA", Yanomami = "#C70E7B", Industrial = "#969696",
                           ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Market_economy_shapes <- c(`Logging road` = 19, `Forrest camp` = 17, Farming = 18, Yanomami = 14, Industrial = 15,
                           ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Market_economy_size <- c(`Logging road` = 4, `Forrest camp` = 4, Farming = 6, Yanomami = 4, Industrial = 4,
                           ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Age_group_colors <- c(`18-29` = "#FF3200", `30-39` = "#E9B186", `40-49` = "#82CDAA", `50-59` = "#0686B8", `60+` = "#172869",
                      `10-18` = "#C70E7B", HMP = "#969696", HMP10M = "#d9d9d9",
                      ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Age_group_shapes <- c(`18-29` = 15, `30-39` = 16, `40-49` = 17, `50-59` = 18, `60+` = 25, `10-18` = 14, HMP = 7, HMP10M = 9,
                      ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Age_group_size <- c(`18-29` = 4.5, `30-39` = 4, `40-49` = 4, `50-59` = 6, `60+` = 4, `10-18` = 4, HMP = 4, HMP10M = 4,
                      ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Sex_colors <- c(Female = "#FF3200", Male = "#172869", Unknown = "#bdbdbd",
                ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Sex_shapes <- c(Female = 19, Male = 17, Unknown = 16, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Sex_size <- c(Female = 4, Male = 4, Unknown = 4, ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Tooth_site_colors <- c(Anterior = "#FF3200", Posterior = "#172869", Buccal_mucosa = "#C70E7B", HMP = "#969696", HMP10M = "#d9d9d9",
                       ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Tooth_site_shapes <- c(Anterior = 19, Posterior = 17,  Buccal_mucosa = 14, HMP = 7, HMP10M = 9,
                       ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Tooth_site_size <- c(Anterior = 4, Posterior = 4,  Buccal_mucosa = 4, HMP = 4, HMP10M = 4,
                       ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

Industry_colors <- c(`Non-industrial` = "#FF3200", Industrial = "#969696",
                           ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Industry_shapes <- c(`Non-industrial` = 19, Industrial = 17,
                           ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)
Industry_size <- c(`Non-industrial` = 4, Industrial = 4,
                           ExtractionGauze = 4, ExtractionBlank = 4, LibraryBlank = 4)

```

```{r pca_plotting_functions}
# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Industry, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group, shape = metadata_group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Industry, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$SampleID), size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}

```

List the outliers from the plot in section `raw_plot` in `Cameroon_h-g_taxonomy.Rmd` and remove them from the taxonomy table. 
```{r outliers}
outliers <- c("CMC002.B0201","CMC032.B0101","CMC037.B0201","CMC047.A0201",
              "EXB059.A1901","EXB059.A1301","EXB059.A1101","EXB059.A0901",
              "SRR1646041","SRR1646032","SRR1646042","SRR1646045","SRR1646030","SRR1646047")

outliersF <- str_c(outliers, collapse = "|")

```

Read in the contaminant list from `Cameroon_taxonomy_decontam.Rmd` and remove them from the taxonomy table. 
```{r contaminants}

contaminants_species <- fread("05-results.backup/malt_contaminants_species.tsv", header = T, sep = "\t")
contaminants_genus <- fread("05-results.backup/malt_contaminants_genus.tsv")

malt_species.decontam <- malt_species %>%
  anti_join(., contaminants_species)

malt_genus.decontam <- malt_genus %>%
  anti_join(., contaminants_genus)


```

We'll need separate metadata files later b/c the factors stick around even when the samples with those factors are removed.
```{r metadata_for_factors}
metadata_original <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")

metadata_noblanks <- metadata_original %>%
  rename(SampleID = `#SampleID`)  %>%
  filter(!SampleID %in% outliers) %>%
  filter(!str_detect(SampleID, "EXB|LIB|10M")) %>%
  as_tibble() %>%
  mutate(Market_economy = str_replace(Market_economy, "HMP","Industrial")) %>%
  mutate(Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime","Yanomami","HMP","JAE"),
         Env = fct_relevel(Env, "HunterGatherer","HMP","JAE"),
         Market_economy = fct_relevel(Market_economy, "Logging road","Forrest camp","Farming","Yanomami","Industrial"),
         Age_group = fct_relevel(Age_group, "10-18","18-29","30-39","40-49","50-59","60+","HMP"),
         Sex = fct_relevel(Sex, "Male","Female"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior","Buccal_mucosa","HMP"),
         Village = fct_relevel(Village, "202","205","204","203","206","Yanomami","HMP","JAE"),
         Industry = fct_relevel(Industry, "Non-industrial","Industrial")) %>%
  select(SampleID, Age_group, Ethnic_Group, Env, Industry, Market_economy, Sex, Tooth_site, Village)

metadata_noblanks_noyanomami <- metadata_original %>%
  rename(SampleID = `#SampleID`)  %>%
  filter(!SampleID %in% outliers) %>%
  filter(!str_detect(SampleID, "SRR16460|EXB|LIB|10M")) %>%
  as_tibble() %>%
  mutate(Market_economy = str_replace(Market_economy, "HMP","Industrial")) %>%
  mutate(Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime","HMP","JAE"),
         Env = fct_relevel(Env, "HunterGatherer","HMP","JAE"),
         Market_economy = fct_relevel(Market_economy, "Logging road","Forrest camp","Farming","Industrial"),
         Age_group = fct_relevel(Age_group, "18-29","30-39","40-49","50-59","60+","HMP"),
         Sex = fct_relevel(Sex, "Male","Female"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior","HMP"),
         Village = fct_relevel(Village, "202","205","204","203","206","HMP","JAE"),
         Industry = fct_relevel(Industry, "Non-industrial","Industrial")) %>%
  select(SampleID, Age_group, Ethnic_Group, Env, Industry, Market_economy, Sex, Tooth_site, Village)

metadata_cmc <- metadata_original %>%
  rename(SampleID = `#SampleID`) %>%
  filter(!SampleID %in% outliers) %>%
  filter(Description == "HunterGatherer") %>%
  select(SampleID, Age_group, Ethnic_Group, Market_economy, Sex, Tooth_site, Village) %>%
  as_tibble() %>%
  mutate(Age_group = fct_relevel(Age_group, "18-29","30-39","40-49","50-59","60+"),
         Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime"),
         Market_economy = fct_relevel(Market_economy, "Logging road","Forrest camp","Farming"),
         Sex = fct_relevel(Sex, "Male","Female"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior"),
         Village = fct_relevel(Village, "202","205","204","203","206"))

```


```{r input_tables}
# species, no blanks 
malt_species.decontam_filtered_noblanks <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-contains("10M")) %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_noblanks) <- malt_species.decontam_filtered_noblanks$SampleID

malt_species.decontam_filtered_noblanks <- malt_species.decontam_filtered_noblanks %>% select(-SampleID)

malt_species.decontam_filtered_noblanks = malt_species.decontam_filtered_noblanks+1

# perform PCA to get a clr-transformed species table
malt_species.decontam_filtered_noblanks.pca <- mixOmics::pca(malt_species.decontam_filtered_noblanks, ncomp = 3, logratio = 'CLR')
# save the clr-transformed species table
malt_species.decontam_filtered_noblanks_clr <- as.data.frame.matrix(malt_species.decontam_filtered_noblanks.pca$X)


# species, no blanks, no Yanomami
malt_species.decontam_filtered_noblanks_noyanomami <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-contains("10M")) %>%
                                         select(-starts_with("SRR16460")) %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_noblanks_noyanomami) <- malt_species.decontam_filtered_noblanks_noyanomami$SampleID

malt_species.decontam_filtered_noblanks_noyanomami <- malt_species.decontam_filtered_noblanks_noyanomami %>% select(-SampleID)

malt_species.decontam_filtered_noblanks_noyanomami = malt_species.decontam_filtered_noblanks_noyanomami+1

# perform PCA to get a clr-transformed species table
malt_species.decontam_filtered_noblanks_noyanomami.pca <- mixOmics::pca(malt_species.decontam_filtered_noblanks_noyanomami, ncomp = 3, logratio = 'CLR')
# save the clr-transformed species table
malt_species.decontam_filtered_noblanks_noyanomami_clr <- as.data.frame.matrix(malt_species.decontam_filtered_noblanks_noyanomami.pca$X)

# species, CMC samples only
malt_species.decontam_filtered_cmc <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-starts_with("SRR")) %>%
                                         select(-starts_with("JAE")) %>%
                                         select(-contains("10M")) %>%
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_filtered_cmc) <- malt_species.decontam_filtered_cmc$SampleID

malt_species.decontam_filtered_cmc <- malt_species.decontam_filtered_cmc %>% select(-SampleID)

malt_species.decontam_filtered_cmc = malt_species.decontam_filtered_cmc+1

# perform PCA to get a clr-transformed species table
malt_species.decontam_filtered_cmc.pca <- mixOmics::pca(malt_species.decontam_filtered_cmc, ncomp = 3, logratio = 'CLR')
# save the clr-transformed species table
malt_species.decontam_filtered_cmc_clr <- as.data.frame.matrix(malt_species.decontam_filtered_cmc.pca$X)


# genus, no blanks
malt_genus.decontam_filtered <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-contains("10M")) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.01) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered) <- malt_genus.decontam_filtered$SampleID

malt_genus.decontam_filtered <- malt_genus.decontam_filtered %>% select(-SampleID)

malt_genus.decontam_filtered = malt_genus.decontam_filtered+1

# perform PCA to get a clr-transformed species table
malt_genus.decontam_filtered.pca <- mixOmics::pca(malt_genus.decontam_filtered, ncomp = 3, logratio = 'CLR')
# save the clr-transformed species table
malt_genus.decontam_filtered_clr <- as.data.frame.matrix(malt_genus.decontam_filtered.pca$X)


# genus, no blanks, no Yanomami
malt_genus.decontam_filtered_noyanomami <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-contains("10M")) %>%
                                         # select(-one_of(fullSRR)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.01) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered_noyanomami) <- malt_genus.decontam_filtered_noyanomami$SampleID

malt_genus.decontam_filtered_noyanomami <- malt_genus.decontam_filtered_noyanomami %>% select(-SampleID)

malt_genus.decontam_filtered_noyanomami = malt_genus.decontam_filtered_noyanomami+1

# perform PCA to get a clr-transformed species table
malt_genus.decontam_filtered_noyanomami.pca <- mixOmics::pca(malt_genus.decontam_filtered_noyanomami, ncomp = 3, logratio = 'CLR')
# save the clr-transformed species table
malt_genus.decontam_filtered_noyanomami_clr <- as.data.frame.matrix(malt_genus.decontam_filtered_noyanomami.pca$X)


# genus, CMC samples only
malt_genus.decontam_filtered_cmc <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-starts_with("EXB")) %>%
                                         select(-starts_with("LIB")) %>%
                                         select(-starts_with("SRR")) %>%
                                         select(-starts_with("JAE")) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_filtered_cmc) <- malt_genus.decontam_filtered_cmc$SampleID

malt_genus.decontam_filtered_cmc <- malt_genus.decontam_filtered_cmc %>% select(-SampleID)

malt_genus.decontam_filtered_cmc = malt_genus.decontam_filtered_cmc+1

# perform PCA to get a clr-transformed species table
malt_genus.decontam_filtered_cmc.pca <- mixOmics::pca(malt_genus.decontam_filtered_cmc, ncomp = 3, logratio = 'CLR')
# save the clr-transformed species table
malt_genus.decontam_filtered_cmc_clr <- as.data.frame.matrix(malt_genus.decontam_filtered_cmc.pca$X)


```


```{r sipca_species_decontam_noblanks, eval = F}
# first run a pca to get the clr-transformed matrix
malt_species_decontam_filtered_noblanks.pca <- mixOmics::pca(malt_species.decontam_filtered_noblanks, ncomp = 3, logratio = 'CLR')

# then run sipca with the clr-transformed matrix
malt_species_decontam_filtered_noblanks.sipca <- sipca(malt_species_decontam_filtered_noblanks.pca$X, scale = F, ncomp = 3, keepX = c(100,100,100)) # 10,10,10 50,50,50 100,100,100

malt_species_decontam_filtered_noblanks.sipca.table <- as.data.frame(malt_species_decontam_filtered_noblanks.sipca$x)
malt_species_decontam_filtered_noblanks.sipca.table <- malt_species_decontam_filtered_noblanks.sipca.table %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>% select(SampleID, Ethnic_Group, Description, Age_group, Market_economy, Sex, Tooth_site, Village))
  
malt_species_decontam_filtered_noblanks.sipca.table %>%
  ggplot(., aes(`IPC 1`, `IPC 2`, colour = Ethnic_Group, shape = Ethnic_Group)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Ethnic_Group_colors) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("sIPCA")

head(selectVar(malt_species_decontam_filtered_noblanks.sipca, comp = 1)$value)
malt_species_decontam_filtered_noblanks.sipca$loadings$X %>% as.data.frame(.) %>% rownames_to_column("species") %>% filter(str_detect(species, "Rothia")) %>% filter(V1 > 0)

```


```{r plsda_species_decontam_noblanks, eval = F}
metadata_noblanks <- metadata %>%
                       filter(!str_detect(SampleID, outliersF)) %>%
                       filter(!str_detect(SampleID, "EXB")) %>%
                       filter(!str_detect(SampleID, "LIB"))


malt_species_decontam_filtered_noblanks.plsda <- plsda(malt_species.decontam_filtered_noblanks, metadata_noblanks$Ethnic_Group, ncomp = 3, logratio = "CLR")

malt_species_decontam_filtered_noblanks.plsda.table <- as.data.frame(malt_species_decontam_filtered_noblanks.plsda$variates$X)
malt_species_decontam_filtered_noblanks.plsda.table <- malt_species_decontam_filtered_noblanks.plsda.table %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>% select(SampleID, Ethnic_Group, Description, Age_group, Market_economy, Sex, Tooth_site, Village))

malt_species_decontam_filtered_noblanks.plsda.table %>%
  ggplot(., aes(comp1, comp2, colour = Ethnic_Group, shape = Ethnic_Group)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Ethnic_Group_colors) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("PLS-DA")

malt_species_decontam_filtered_noblanks.plsda$loadings$X %>%
  as.data.frame(.) %>%
  bind_cols(., malt_species_decontam_filtered_noblanks.plsda$names$colnames$X %>%
              as.data.frame(.)) %>%
  rename(Species = ".") %>%
  select(Species, everything()) %>%
  arrange((comp1))

```

```{r fa_species_decontam_noblanks}
# fa.parallel(malt_species.decontam_filtered_noblanks, fm = "minres", n.iter = 19)

malt_species.decontam_filtered_noblanks.fa <- fa(malt_species.decontam_filtered_noblanks_clr, nfactors = 6, rotate = "varimax", fm = "minres") # rotate = "Varimax" "oblimin" # fm = "ols" "minres"
# if using fm = "ols", the column titles will be 1, 2, etc and not M1, M2, etc as you get from using fm = "minres"

print(malt_species.decontam_filtered_noblanks.fa)
# print(malt_species.decontam_filtered_noblanks.fa$loadings,cutoff = 0.7)
# print(malt_species.decontam_filtered_noblanks.fa,sort=TRUE)

malt_species.decontam_filtered_noblanks.fa$values %>%
  as.data.frame(.) %>%
  rename(Eigenvalues = ".") %>%
  arrange(desc(Eigenvalues))%>%
  rownames_to_column("xvalue") %>%
  mutate(xvalue = as.numeric(xvalue)) %>%
  filter(xvalue <= 20) %>%
  ggplot(., aes(xvalue, Eigenvalues)) +
         geom_point() +
         theme_minimal() +
         geom_hline(yintercept = 10, linetype = "dashed", color = "red")

# get the minimum residual values into a datatable
graphinfo <- as.data.frame(factor2cluster(malt_species.decontam_filtered_noblanks.fa))
graphinfo <- graphinfo %>%
  rownames_to_column("Species") %>%
  as_tibble(.)

# now make loadings into a table and designate each species in the residual with the highest absolute value
malt_species.decontam_filtered_noblanks.fa.df <- as.data.frame.matrix(malt_species.decontam_filtered_noblanks.fa$loadings)
# b/c of rotation, the MRs may not be in numerial order. Change the column titles to be in numerical order for downstream processing
colnames(malt_species.decontam_filtered_noblanks.fa.df) <- c("MR1","MR2","MR3","MR4","MR5","MR6")
malt_species.decontam_filtered_noblanks.fa.df <- malt_species.decontam_filtered_noblanks.fa.df %>%
  as_tibble(.) %>%
  bind_cols(., graphinfo) %>%
  mutate(MR11 = as.character(MR11),
         MR21 = as.character(MR21),
         MR31 = as.character(MR31),
         MR41 = as.character(MR41),
         MR51 = as.character(MR51)) %>%
  mutate(MR11 = replace(MR11, MR11 != "0", "MR1"),
         MR21 = replace(MR21, MR21 != "0", "MR2"),
         MR31 = replace(MR31, MR31 != "0", "MR3"),
         MR41 = replace(MR41, MR41 != "0", "MR4"),
         MR51 = replace(MR51, MR51 != "0", "MR5")) %>%
  select(Species,everything(.)) %>%
  gather("Group","Residual", 7:ncol(.)) %>%
  filter(Residual != "0") %>%
  select(Residual, Species, MR1, MR2, MR3, MR4, MR5)

head(malt_species.decontam_filtered_noblanks.fa.df)

# now run a discriminant analysis

# save the species using in downstream discriminant analyses as a table
malt_species.decontam_filtered_noblanks.fa_species <- malt_species.decontam_filtered_noblanks.fa.df %>% 
  filter(Residual %in% c("MR1","MR2","MR3")) %>%
  select(Species)

fwrite(malt_species.decontam_filtered_noblanks.fa_species, file = "./05-results.backup/malt_species.decontam_filtered_noblanks.fa_species.tsv", quote = F, sep = "\t")

```

```{r lda_species_decontam_noblanks_eg}

# select only the species that make up the 1st 3 residuals, based on the scree plot above
fa_species_noblanks <- malt_species.decontam_filtered_noblanks.fa.df %>% 
  filter(Residual %in% c("MR1","MR2","MR3")) %>%
  pull(Species)

# sub-sample the species table for only those species
malt_species.decontam_filtered_noblanks_lda_table <- malt_species.decontam_filtered_noblanks_clr %>%
  select(one_of(fa_species_noblanks)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata_noblanks %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_economy, Sex, Tooth_site, Village
               select(SampleID, Village)) %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_economy, Sex, Tooth_site, Village
  select(SampleID, Village, everything())

rownames(malt_species.decontam_filtered_noblanks_lda_table) <- malt_species.decontam_filtered_noblanks_lda_table$SampleID
malt_species.decontam_filtered_noblanks_lda_table <- malt_species.decontam_filtered_noblanks_lda_table %>% select(-SampleID)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_economy, Sex, Tooth_site, Village
malt_species.decontam_filtered_noblanks_lda <- lda(Village ~ ., malt_species.decontam_filtered_noblanks_lda_table)
# malt_species.decontam_filtered_noblanks_lda # show results
malt_species.decontam_filtered_noblanks_lda
plot(malt_species.decontam_filtered_noblanks_lda, dimen = 2)

lda.data <- cbind(malt_species.decontam_filtered_noblanks_lda_table, predict(malt_species.decontam_filtered_noblanks_lda)$x)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_economy, Sex, Tooth_site, Village
ggplot(lda.data, aes(LD1, LD2, colour = Village, shape = Village)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = Village_colors) +
                        scale_shape_manual(values = Village_shapes) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

```


Now without Yanomami samples
```{r fa_species_decontam_noblanks_noyanomami}
malt_species.decontam_filtered_noblanks_noyanomami.fa <- fa(malt_species.decontam_filtered_noblanks_noyanomami_clr, nfactors = 7, rotate = "varimax", fm = "minres") # rotate = "Varimax" "oblimin" # fm = "ols" "minres"
print(malt_species.decontam_filtered_noblanks_noyanomami.fa)
# print(malt_species.decontam_filtered_noblanks_noyanomami.fa$loadings,cutoff = 0.7)
# print(malt_species.decontam_filtered_noblanks_noyanomami.fa,sort=TRUE)

malt_species.decontam_filtered_noblanks_noyanomami.fa$values %>%
  as.data.frame(.) %>%
  rename(Eigenvalues = ".") %>%
  arrange(desc(Eigenvalues))%>%
  rownames_to_column("xvalue") %>%
  mutate(xvalue = as.numeric(xvalue)) %>%
  filter(xvalue <= 20) %>%
  ggplot(., aes(xvalue, Eigenvalues)) +
         geom_point() +
         theme_minimal() +
         geom_hline(yintercept = 10, linetype = "dashed", color = "red")

# get the minimum residual values into a datatable
graphinfo <- as.data.frame(factor2cluster(malt_species.decontam_filtered_noblanks_noyanomami.fa))
graphinfo <- graphinfo %>%
  rownames_to_column("Species") %>%
  as_tibble(.)

# now make loadings into a table and designate each species in the residual with the highest absolute value
malt_species.decontam_filtered_noblanks_noyanomami.fa.df <- as.data.frame.matrix(malt_species.decontam_filtered_noblanks_noyanomami.fa$loadings)
# b/c of rotation, the MRs may not be in numerial order. Change the column titles to be in numerical order for downstream processing
colnames(malt_species.decontam_filtered_noblanks_noyanomami.fa.df) <- c("MR1","MR2","MR3","MR4","MR5","MR6","MR7")
malt_species.decontam_filtered_noblanks_noyanomami.fa.df <- malt_species.decontam_filtered_noblanks_noyanomami.fa.df %>%
  as_tibble(.) %>%
  bind_cols(., graphinfo) %>%
  mutate(MR11 = as.character(MR11),
         MR21 = as.character(MR21),
         MR31 = as.character(MR31),
         MR41 = as.character(MR41),
         MR51 = as.character(MR51)) %>%
  mutate(MR11 = replace(MR11, MR11 != "0", "MR1"),
         MR21 = replace(MR21, MR21 != "0", "MR2"),
         MR31 = replace(MR31, MR31 != "0", "MR3"),
         MR41 = replace(MR41, MR41 != "0", "MR4"),
         MR51 = replace(MR51, MR51 != "0", "MR5")) %>%
  select(Species,everything(.)) %>%
  gather("Group","Residual", 7:ncol(.)) %>%
  filter(Residual != "0") %>%
  select(Residual, Species, MR1, MR2, MR3, MR4, MR5)

head(malt_species.decontam_filtered_noblanks_noyanomami.fa.df)


# save the species using in downstream discriminant analyses as a table
malt_species.decontam_filtered_noblanks_noyanomami.fa_species <- malt_species.decontam_filtered_noblanks_noyanomami.fa.df %>% 
  filter(Residual %in% c("MR1","MR2")) %>%
  select(Species)

fwrite(malt_species.decontam_filtered_noblanks_noyanomami.fa_species, file = "./05-results.backup/malt_species.decontam_filtered_noblanks_noyanomami.fa_species.tsv", quote = F, sep = "\t")

```

```{r da_species_decontam_noblanks_noyanomami_eg}
# select only the species that make up the 1st 2 residuals, based on the scree plot above
fa_species_noblanks_noyanomami <- malt_species.decontam_filtered_noblanks_noyanomami.fa.df %>% 
  filter(Residual %in% c("MR1","MR2")) %>%
  pull(Species)

# sub-sample the species table for only those species
malt_species.decontam_filtered_noblanks_noyanomami_da_table <- malt_species.decontam_filtered_noblanks_noyanomami_clr %>%
  select(one_of(fa_species_noblanks_noyanomami)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata_noblanks_noyanomami %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_economy, Sex, Tooth_site, Village
               select(SampleID, Ethnic_Group)) %>%
  select(SampleID, Ethnic_Group, everything())

rownames(malt_species.decontam_filtered_noblanks_noyanomami_da_table) <- malt_species.decontam_filtered_noblanks_noyanomami_da_table$SampleID
malt_species.decontam_filtered_noblanks_noyanomami_da_table <- malt_species.decontam_filtered_noblanks_noyanomami_da_table %>% select(-SampleID)

malt_species.decontam_filtered_noblanks_noyanomami_lda <- lda(Ethnic_Group ~ ., malt_species.decontam_filtered_noblanks_noyanomami_da_table)
malt_species.decontam_filtered_noblanks_noyanomami_lda # show results

plot(malt_species.decontam_filtered_noblanks_noyanomami_lda, dimen = 2)

lda.data <- cbind(malt_species.decontam_filtered_noblanks_noyanomami_da_table, predict(malt_species.decontam_filtered_noblanks_noyanomami_lda)$x)

noyanomami_eg <- ggplot(lda.data, aes(LD1, LD2, colour = Ethnic_Group, shape = Ethnic_Group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = Ethnic_Group_colors) +
                        scale_shape_manual(values = Ethnic_Group_shapes) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

noyanomami_eg

```


```{r da_species_decontam_noblanks_noyanomami_age}
# select only the species that make up the 1st 2 residuals, based on the scree plot above
fa_species_noblanks_noyanomami <- malt_species.decontam_filtered_noblanks_noyanomami.fa.df %>% 
  filter(Residual %in% c("MR1","MR2")) %>%
  pull(Species)

# sub-sample the species table for only those species
malt_species.decontam_filtered_noblanks_noyanomami_da_table <- malt_species.decontam_filtered_noblanks_noyanomami_clr %>%
  select(one_of(fa_species_noblanks_noyanomami)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata_noblanks_noyanomami %>%
               select(SampleID, Age_group)) %>%
  select(SampleID, Age_group, everything())

rownames(malt_species.decontam_filtered_noblanks_noyanomami_da_table) <- malt_species.decontam_filtered_noblanks_noyanomami_da_table$SampleID
malt_species.decontam_filtered_noblanks_noyanomami_da_table <- malt_species.decontam_filtered_noblanks_noyanomami_da_table %>% select(-SampleID)

malt_species.decontam_filtered_noblanks_noyanomami_lda <- lda(Age_group ~ ., malt_species.decontam_filtered_noblanks_noyanomami_da_table)
malt_species.decontam_filtered_noblanks_noyanomami_lda # show results

lda.data <- cbind(malt_species.decontam_filtered_noblanks_noyanomami_da_table, predict(malt_species.decontam_filtered_noblanks_noyanomami_lda)$x)

noyanomami_ag <- ggplot(lda.data, aes(LD1, LD2, colour = Age_group, shape = Age_group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = Age_group_colors) +
                        scale_shape_manual(values = Age_group_shapes) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

noyanomami_ag

```


```{r da_species_decontam_noblanks_noyanomami_me}
# select only the species that make up the 1st 2 residuals, based on the scree plot above
fa_species_noblanks_noyanomami <- malt_species.decontam_filtered_noblanks_noyanomami.fa.df %>% 
  filter(Residual %in% c("MR1","MR2")) %>%
  pull(Species)

# sub-sample the species table for only those species
malt_species.decontam_filtered_noblanks_noyanomami_da_table <- malt_species.decontam_filtered_noblanks_noyanomami_clr %>%
  select(one_of(fa_species_noblanks_noyanomami)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata_noblanks_noyanomami %>%
               select(SampleID, Market_economy)) %>%
  select(SampleID, Market_economy, everything())

rownames(malt_species.decontam_filtered_noblanks_noyanomami_da_table) <- malt_species.decontam_filtered_noblanks_noyanomami_da_table$SampleID
malt_species.decontam_filtered_noblanks_noyanomami_da_table <- malt_species.decontam_filtered_noblanks_noyanomami_da_table %>% select(-SampleID)

malt_species.decontam_filtered_noblanks_noyanomami_lda <- lda(Market_economy ~ ., malt_species.decontam_filtered_noblanks_noyanomami_da_table)
malt_species.decontam_filtered_noblanks_noyanomami_lda # show results

lda.data <- cbind(malt_species.decontam_filtered_noblanks_noyanomami_da_table, predict(malt_species.decontam_filtered_noblanks_noyanomami_lda)$x)

noyanomami_me <- ggplot(lda.data, aes(LD1, LD2, colour = Market_economy, shape = Market_economy, size = Market_economy)) +
                        geom_point() +
                        scale_colour_manual(values = Market_economy_colors) +
                        scale_shape_manual(values = Market_economy_shapes) +
                        scale_size_manual(values = Market_economy_size) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

noyanomami_me

```


```{r da_species_decontam_noblanks_noyanomami_sex}
# select only the species that make up the 1st 2 residuals, based on the scree plot above
fa_species_noblanks_noyanomami <- malt_species.decontam_filtered_noblanks_noyanomami.fa.df %>% 
  filter(Residual %in% c("MR1","MR2")) %>%
  pull(Species)

# sub-sample the species table for only those species
malt_species.decontam_filtered_noblanks_noyanomami_da_table <- malt_species.decontam_filtered_noblanks_noyanomami_clr %>%
  select(one_of(fa_species_noblanks_noyanomami)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata_noblanks_noyanomami %>%
               select(SampleID, Sex)) %>%
  select(SampleID, Sex, everything()) %>%
  filter(Sex != "Unknown")

rownames(malt_species.decontam_filtered_noblanks_noyanomami_da_table) <- malt_species.decontam_filtered_noblanks_noyanomami_da_table$SampleID
malt_species.decontam_filtered_noblanks_noyanomami_da_table <- malt_species.decontam_filtered_noblanks_noyanomami_da_table %>% select(-SampleID)

malt_species.decontam_filtered_noblanks_noyanomami_lda <- lda(Sex ~ ., malt_species.decontam_filtered_noblanks_noyanomami_da_table)
malt_species.decontam_filtered_noblanks_noyanomami_lda # show results

lda.data <- cbind(malt_species.decontam_filtered_noblanks_noyanomami_da_table, predict(malt_species.decontam_filtered_noblanks_noyanomami_lda)$x)

# # works only if the Unknown sample is left in
# ggplot(lda.data, aes(LD1, LD2, colour = Sex, shape = Sex)) +
#                         geom_point(size = 4) +
#                         scale_colour_manual(values = Sex_colors) +
#                         scale_shape_manual(values = Sex_shapes) +
#                         xlab("LD1") +
#                         ylab("LD2") +
#                         theme_minimal(base_size = 10) +
#                         theme(legend.position = "top") +
#                         theme(legend.title = element_blank())

noyanomami_sex <- ggplot(lda.data, aes(x = Sex, y = LD1, group = Sex, fill = Sex)) +
                        geom_boxplot() + 
                        geom_jitter(size = 2, aes(shape = Sex)) +
                        scale_fill_manual(values = Sex_colors) +
                        scale_shape_manual(values = c(3,4)) +
                        xlab("LD1") +
                        ylab("LD2") +
                        geom_hline(yintercept = 0, linetype = "dashed") +
                        coord_flip() +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())


noyanomami_sex

```


```{r da_species_decontam_noblanks_noyanomami_ts}
# select only the species that make up the 1st 2 residuals, based on the scree plot above
fa_species_noblanks_noyanomami <- malt_species.decontam_filtered_noblanks_noyanomami.fa.df %>% 
  filter(Residual %in% c("MR1","MR2")) %>%
  pull(Species)

# sub-sample the species table for only those species
malt_species.decontam_filtered_noblanks_noyanomami_da_table <- malt_species.decontam_filtered_noblanks_noyanomami_clr %>%
  select(one_of(fa_species_noblanks_noyanomami)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata_noblanks_noyanomami %>%
               select(SampleID, Tooth_site)) %>%
  select(SampleID, Tooth_site, everything())

rownames(malt_species.decontam_filtered_noblanks_noyanomami_da_table) <- malt_species.decontam_filtered_noblanks_noyanomami_da_table$SampleID
malt_species.decontam_filtered_noblanks_noyanomami_da_table <- malt_species.decontam_filtered_noblanks_noyanomami_da_table %>% select(-SampleID)

malt_species.decontam_filtered_noblanks_noyanomami_lda <- lda(Tooth_site ~ ., malt_species.decontam_filtered_noblanks_noyanomami_da_table)
malt_species.decontam_filtered_noblanks_noyanomami_lda # show results

lda.data <- cbind(malt_species.decontam_filtered_noblanks_noyanomami_da_table, predict(malt_species.decontam_filtered_noblanks_noyanomami_lda)$x)

noyanomami_ts <- ggplot(lda.data, aes(LD1, LD2, colour = Tooth_site, shape = Tooth_site)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = Tooth_site_colors) +
                        scale_shape_manual(values = Tooth_site_shapes) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

noyanomami_ts

```


```{r da_species_decontam_noblanks_noyanomami_vi}
# select only the species that make up the 1st 2 residuals, based on the scree plot above
fa_species_noblanks_noyanomami <- malt_species.decontam_filtered_noblanks_noyanomami.fa.df %>% 
  filter(Residual %in% c("MR1","MR2")) %>%
  pull(Species)

# sub-sample the species table for only those species
malt_species.decontam_filtered_noblanks_noyanomami_da_table <- malt_species.decontam_filtered_noblanks_noyanomami_clr %>%
  select(one_of(fa_species_noblanks_noyanomami)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata_noblanks_noyanomami %>%
               select(SampleID, Village)) %>%
  select(SampleID, Village, everything())

rownames(malt_species.decontam_filtered_noblanks_noyanomami_da_table) <- malt_species.decontam_filtered_noblanks_noyanomami_da_table$SampleID
malt_species.decontam_filtered_noblanks_noyanomami_da_table <- malt_species.decontam_filtered_noblanks_noyanomami_da_table %>% select(-SampleID)

malt_species.decontam_filtered_noblanks_noyanomami_lda <- lda(Village ~ ., malt_species.decontam_filtered_noblanks_noyanomami_da_table)
malt_species.decontam_filtered_noblanks_noyanomami_lda # show results

lda.data <- cbind(malt_species.decontam_filtered_noblanks_noyanomami_da_table, predict(malt_species.decontam_filtered_noblanks_noyanomami_lda)$x)

noyanomami_vi <- ggplot(lda.data, aes(LD1, LD2, colour = Village, shape = Village)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = Village_colors) +
                        scale_shape_manual(values = Village_shapes) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

noyanomami_vi

```

```{r noblanks_images}

allplots_noyanomami <- plot_grid(noyanomami_eg, noyanomami_ag,
          noyanomami_me, noyanomami_sex,
          noyanomami_ts, noyanomami_vi,
          nrow=3, labels = c("A", "B", "C", "D", "E", "F"), rel_widths = c(1, 1))


ggsave("05-results.backup/allplots_noyanomami.pdf", plot = allplots_noyanomami, device = "pdf",
       scale = 1, width = 10, height = 10, units = c("in"),
       dpi = 300)

```


Now only CMC samples
```{r fa_species_decontam_cmc}
# fa.parallel(malt_species.decontam_filtered_cmc, fm = "minres", n.iter = 19)

malt_species.decontam_filtered_cmc.fa <- fa(malt_species.decontam_filtered_cmc_clr, nfactors = 6, rotate = "varimax", fm = "minres") # rotate = "oblimin" "Varimax" # fm = "ols" "minres"
print(malt_species.decontam_filtered_cmc.fa)
# print(malt_species.decontam_filtered_cmc.fa$loadings,cutoff = 0.7)
# print(malt_species.decontam_filtered_cmc.fa,sort=TRUE)

malt_species.decontam_filtered_cmc.fa$values %>%
  as.data.frame(.) %>%
  rename(Eigenvalues = ".") %>%
  arrange(desc(Eigenvalues))%>%
  rownames_to_column("xvalue") %>%
  mutate(xvalue = as.numeric(xvalue)) %>%
  filter(xvalue <= 20) %>%
  ggplot(., aes(xvalue, Eigenvalues)) +
         geom_point() +
         theme_minimal() +
         geom_hline(yintercept = 10, linetype = "dashed", color = "red")

# get the minimum residual values into a datatable
graphinfo <- as.data.frame(factor2cluster(malt_species.decontam_filtered_cmc.fa))
graphinfo <- graphinfo %>%
  rownames_to_column("Species") %>%
  as_tibble(.)

# now make loadings into a table and designate each species in the residual with the highest absolute value
malt_species.decontam_filtered_cmc.fa.df <- as.data.frame.matrix(malt_species.decontam_filtered_cmc.fa$loadings)
# b/c of rotation, the MRs may not be in numerial order. Change the column titles to be in numerical order for downstream processing
colnames(malt_species.decontam_filtered_cmc.fa.df) <- c("MR1","MR2","MR3","MR4","MR5","MR6")
malt_species.decontam_filtered_cmc.fa.df <- malt_species.decontam_filtered_cmc.fa.df %>%
  as_tibble(.) %>%
  bind_cols(., graphinfo) %>%
  mutate(MR11 = as.character(MR11),
         MR21 = as.character(MR21),
         MR31 = as.character(MR31)) %>%
  mutate(MR11 = replace(MR11, MR11 != "0", "MR1"),
         MR21 = replace(MR21, MR21 != "0", "MR2"),
         MR31 = replace(MR31, MR31 != "0", "MR2")) %>%
  select(Species,everything(.)) %>%
  gather("Group","Residual", 5:ncol(.)) %>%
  filter(Residual != "0") %>%
  select(Residual, Species, MR1, MR3, MR2)

head(malt_species.decontam_filtered_cmc.fa.df)

# now run a discriminant analysis

# save the species using in downstream discriminant analyses as a table
malt_species.decontam_filtered_cmc.fa_species <- malt_species.decontam_filtered_cmc.fa.df %>% 
  filter(Residual %in% c("MR1","MR2","MR3")) %>%
  select(Species)

fwrite(malt_species.decontam_filtered_cmc.fa_species, file = "./05-results.backup/malt_species.decontam_filtered_cmc.fa_species.tsv", quote = F, sep = "\t")

```

```{r lda_species_decontam_cmc_eg}

# select only the species that make up the 1st 2 residuals, based on the scree plot above
fa_species_cmc <- malt_species.decontam_filtered_cmc.fa.df %>% 
  filter(Residual %in% c("MR1","MR2","MR3")) %>%
  pull(Species)

# sub-sample the species table for only those species
malt_species.decontam_filtered_cmc_lda_table <- malt_species.decontam_filtered_cmc_clr %>%
  select(one_of(fa_species_cmc)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata_cmc %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_economy, Sex, Tooth_site, Village
               select(SampleID, Ethnic_Group)) %>%
  select(SampleID, Ethnic_Group, everything())

rownames(malt_species.decontam_filtered_cmc_lda_table) <- malt_species.decontam_filtered_cmc_lda_table$SampleID
malt_species.decontam_filtered_cmc_lda_table <- malt_species.decontam_filtered_cmc_lda_table %>% select(-SampleID)

malt_species.decontam_filtered_cmc_lda <- lda(Ethnic_Group ~ ., malt_species.decontam_filtered_cmc_lda_table)
malt_species.decontam_filtered_cmc_lda # show results

plotit <- plot(malt_species.decontam_filtered_cmc_lda)
plotit

lda.data <- cbind(malt_species.decontam_filtered_cmc_lda_table, predict(malt_species.decontam_filtered_cmc_lda)$x)
# lda.data <- lda.data %>%
#   rownames_to_column("SampleID")


cmc_eg <- ggplot(lda.data, aes(x = Ethnic_Group, y = LD1, group = Ethnic_Group, fill = Ethnic_Group)) +
                        geom_boxplot() + 
                        geom_jitter(size = 2, aes(shape = Ethnic_Group)) +
                        scale_fill_manual(values = Ethnic_Group_colors) +
                        scale_shape_manual(values = c(3,4)) +
                        xlab("Ethnic Group") +
                        ylab("LD1") +
                        geom_hline(yintercept = 0, linetype = "dashed") +
                        coord_flip() +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

cmc_eg

```


```{r lda_species_decontam_cmc_age}

# select only the species that make up the 1st 2 residuals, based on the scree plot above
fa_species_cmc <- malt_species.decontam_filtered_cmc.fa.df %>% 
  filter(Residual %in% c("MR1","MR2","MR3")) %>%
  pull(Species)

# sub-sample the species table for only those species
malt_species.decontam_filtered_cmc_lda_table <- malt_species.decontam_filtered_cmc_clr %>%
  select(one_of(fa_species_cmc)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata_cmc %>%
               select(SampleID, Age_group)) %>%
  select(SampleID, Age_group, everything())

rownames(malt_species.decontam_filtered_cmc_lda_table) <- malt_species.decontam_filtered_cmc_lda_table$SampleID
malt_species.decontam_filtered_cmc_lda_table <- malt_species.decontam_filtered_cmc_lda_table %>% select(-SampleID)

malt_species.decontam_filtered_cmc_lda <- lda(Age_group ~ ., malt_species.decontam_filtered_cmc_lda_table)
malt_species.decontam_filtered_cmc_lda # show results

lda.data <- cbind(malt_species.decontam_filtered_cmc_lda_table, predict(malt_species.decontam_filtered_cmc_lda)$x)

cmc_ag <- ggplot(lda.data, aes(LD1, LD2, colour = Age_group, shape = Age_group, size = Age_group)) +
                        geom_point() +
                        scale_colour_manual(values = Age_group_colors) +
                        scale_shape_manual(values = Age_group_shapes) +
                        scale_size_manual(values = Age_group_size) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

cmc_ag
```


```{r lda_species_decontam_cmc_me}

# select only the species that make up the 1st 2 residuals, based on the scree plot above
fa_species_cmc <- malt_species.decontam_filtered_cmc.fa.df %>% 
  filter(Residual %in% c("MR1","MR2","MR3")) %>%
  pull(Species)

# sub-sample the species table for only those species
malt_species.decontam_filtered_cmc_lda_table <- malt_species.decontam_filtered_cmc_clr %>%
  select(one_of(fa_species_cmc)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata_cmc %>%
               select(SampleID, Market_economy)) %>%
  select(SampleID, Market_economy, everything())

rownames(malt_species.decontam_filtered_cmc_lda_table) <- malt_species.decontam_filtered_cmc_lda_table$SampleID
malt_species.decontam_filtered_cmc_lda_table <- malt_species.decontam_filtered_cmc_lda_table %>% select(-SampleID)

malt_species.decontam_filtered_cmc_lda <- lda(Market_economy ~ ., malt_species.decontam_filtered_cmc_lda_table)
malt_species.decontam_filtered_cmc_lda # show results

lda.data <- cbind(malt_species.decontam_filtered_cmc_lda_table, predict(malt_species.decontam_filtered_cmc_lda)$x)

cmc_me <- ggplot(lda.data, aes(LD1, LD2, colour = Market_economy, shape = Market_economy, size = Market_economy)) +
                        geom_point() +
                        scale_colour_manual(values = Market_economy_colors) +
                        scale_size_manual(values = Market_economy_size) +
                        scale_shape_manual(values = Market_economy_shapes) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

cmc_me

```


```{r lda_species_decontam_cmc_sex}

# select only the species that make up the 1st 2 residuals, based on the scree plot above
fa_species_cmc <- malt_species.decontam_filtered_cmc.fa.df %>% 
  filter(Residual %in% c("MR1","MR2","MR3")) %>%
  pull(Species)

# sub-sample the species table for only those species
malt_species.decontam_filtered_cmc_lda_table <- malt_species.decontam_filtered_cmc_clr %>%
  select(one_of(fa_species_cmc)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata_cmc %>%
               select(SampleID, Sex)) %>%
  select(SampleID, Sex, everything())

rownames(malt_species.decontam_filtered_cmc_lda_table) <- malt_species.decontam_filtered_cmc_lda_table$SampleID
malt_species.decontam_filtered_cmc_lda_table <- malt_species.decontam_filtered_cmc_lda_table %>% select(-SampleID)

malt_species.decontam_filtered_cmc_lda <- lda(Sex ~ ., malt_species.decontam_filtered_cmc_lda_table)
malt_species.decontam_filtered_cmc_lda # show results

plot(malt_species.decontam_filtered_cmc_lda)

lda.data <- cbind(malt_species.decontam_filtered_cmc_lda_table, predict(malt_species.decontam_filtered_cmc_lda)$x)

cmc_sex <- ggplot(lda.data, aes(x = Sex, y = LD1, group = Sex, fill = Sex)) +
                        geom_boxplot() + 
                        geom_jitter(size = 2, aes(shape = Sex)) +
                        scale_fill_manual(values = Sex_colors) +
                        scale_shape_manual(values = c(3,4)) +
                        xlab("Ethnic Group") +
                        ylab("LD1") +
                        geom_hline(yintercept = 0, linetype = "dashed") +
                        coord_flip() +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

cmc_sex

```


```{r lda_species_decontam_cmc_ts}

# select only the species that make up the 1st 2 residuals, based on the scree plot above
fa_species_cmc <- malt_species.decontam_filtered_cmc.fa.df %>% 
  filter(Residual %in% c("MR1","MR2","MR3")) %>%
  pull(Species)

# sub-sample the species table for only those species
malt_species.decontam_filtered_cmc_lda_table <- malt_species.decontam_filtered_cmc_clr %>%
  select(one_of(fa_species_cmc)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata_cmc %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_economy, Sex, Tooth_site, Village
               select(SampleID, Tooth_site)) %>%
  select(SampleID, Tooth_site, everything())

rownames(malt_species.decontam_filtered_cmc_lda_table) <- malt_species.decontam_filtered_cmc_lda_table$SampleID
malt_species.decontam_filtered_cmc_lda_table <- malt_species.decontam_filtered_cmc_lda_table %>% select(-SampleID)

malt_species.decontam_filtered_cmc_lda <- lda(Tooth_site ~ ., malt_species.decontam_filtered_cmc_lda_table)
malt_species.decontam_filtered_cmc_lda # show results

lda.data <- cbind(malt_species.decontam_filtered_cmc_lda_table, predict(malt_species.decontam_filtered_cmc_lda)$x)

plot(malt_species.decontam_filtered_cmc_lda)

cmc_ts <- ggplot(lda.data, aes(x = Tooth_site, y = LD1, group = Tooth_site, fill = Tooth_site)) +
                        geom_boxplot() + 
                        geom_jitter(size = 2, aes(shape = Tooth_site)) +
                        scale_fill_manual(values = Tooth_site_colors) +
                        scale_shape_manual(values = c(3,4)) +
                        xlab("Ethnic Group") +
                        ylab("LD1") +
                        geom_hline(yintercept = 0, linetype = "dashed") +
                        coord_flip() +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

cmc_ts

```


```{r lda_species_decontam_cmc_village}

# select only the species that make up the 1st 2 residuals, based on the scree plot above
fa_species_cmc <- malt_species.decontam_filtered_cmc.fa.df %>% 
  filter(Residual %in% c("MR1","MR2","MR3")) %>%
  pull(Species)

# sub-sample the species table for only those species
malt_species.decontam_filtered_cmc_lda_table <- malt_species.decontam_filtered_cmc_clr %>%
  select(one_of(fa_species_cmc)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata_cmc %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_economy, Sex, Tooth_site, Village
               select(SampleID, Village)) %>%
  select(SampleID, Village, everything())

rownames(malt_species.decontam_filtered_cmc_lda_table) <- malt_species.decontam_filtered_cmc_lda_table$SampleID
malt_species.decontam_filtered_cmc_lda_table <- malt_species.decontam_filtered_cmc_lda_table %>% select(-SampleID)

malt_species.decontam_filtered_cmc_lda <- lda(Village ~ ., malt_species.decontam_filtered_cmc_lda_table)
malt_species.decontam_filtered_cmc_lda # show results

lda.data <- cbind(malt_species.decontam_filtered_cmc_lda_table, predict(malt_species.decontam_filtered_cmc_lda)$x)

cmc_vi <- ggplot(lda.data, aes(LD1, LD2, colour = Village, shape = Village, size = Village)) +
                        geom_point() +
                        scale_colour_manual(values = Village_colors) +
                        scale_shape_manual(values = Village_shapes) +
                        scale_size_manual(values = Village_size) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

cmc_vi

```

```{r cmc_images}

allplots_cmc <- plot_grid(cmc_eg, cmc_ag,
          cmc_me, cmc_sex,
          cmc_ts, cmc_vi,
          nrow=3, labels = c("A", "B", "C", "D", "E", "F"), rel_widths = c(1, 1))

ggsave("05-results.backup/allplots_cmc.pdf", plot = allplots_cmc, device = "pdf",
       scale = 1, width = 9, height = 10, units = c("in"),
       dpi = 300)

```


# Not using, SI-PCA and sPLS-DA
```{r sipca_species_decontam_cmc, eval = F}
# first run a pca to get the clr-transformed matrix
malt_species.decontam_filtered_cmc.pca <- mixOmics::pca(malt_species.decontam_filtered_cmc, ncomp = 3, logratio = 'CLR')

# then run sipca with the clr-transformed matrix
malt_species.decontam_filtered_cmc.sipca <- sipca(malt_species.decontam_filtered_cmc.pca$X, scale = F, ncomp = 3, keepX = c(100,100,100)) # 10,10,10 50,50,50 100,100,100

malt_species.decontam_filtered_cmc.sipca.table <- as.data.frame(malt_species.decontam_filtered_cmc.sipca$x)
malt_species.decontam_filtered_cmc.sipca.table <- malt_species.decontam_filtered_cmc.sipca.table %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>% select(SampleID, Ethnic_Group, Description, Age_group, Market_economy, Sex, Tooth_site, Village))
  
malt_species.decontam_filtered_cmc.sipca.table %>%
  ggplot(., aes(`IPC 1`, `IPC 2`, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("sIPCA")

head(selectVar(malt_species.decontam_filtered_cmc.sipca, comp = 1)$value)
malt_species.decontam_filtered_cmc.sipca$loadings$X %>% as.data.frame(.) %>% rownames_to_column("species") %>% filter(V1 > 0) %>% arrange(desc(V1))

```


```{r plsda_species_decontam_cmc, eval = F}
metadata_cmc <- metadata %>%
                       filter(!str_detect(SampleID, outliersF)) %>%
                       filter(str_detect(SampleID, "CMC"))

malt_species.decontam_filtered_cmc.plsda <- plsda(malt_species.decontam_filtered_cmc, metadata_cmc$Tooth_site, ncomp = 3, logratio = "CLR")

malt_species.decontam_filtered_cmc.plsda.table <- as.data.frame(malt_species.decontam_filtered_cmc.plsda$variates$X)
malt_species.decontam_filtered_cmc.plsda.table <- malt_species.decontam_filtered_cmc.plsda.table %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>% select(SampleID, Ethnic_Group, Description, Age_group, Market_economy, Sex, Tooth_site, Village))

malt_species.decontam_filtered_cmc.plsda.table %>%
  ggplot(., aes(comp1, comp2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  # stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("PLS-DA")

malt_species.decontam_filtered_cmc.plsda$loadings$X %>%
  as.data.frame(.) %>%
  bind_cols(., malt_species.decontam_filtered_cmc.plsda$names$colnames$X %>%
              as.data.frame(.)) %>%
  rename(Species = ".") %>%
  select(Species, everything()) %>%
  arrange(desc(comp1))

```


























