---
title: "Cameroon hunter-gatherer calculus/plaque HUMAnN2"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---

```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(psych)
library(vegan)
library(ape)
library(mixOmics)
library(compositions)
library(MASS)
library(janitor)
library(rstatix)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

```{r load_data}
load("./05-results.backup/CMC_humann2_tables.RData")
load("./05-results.backup/CMC_metadata.RData")

# remove unmapped and ungrouped reads
humann2_KOs <- humann2_KOs_full %>% filter(!str_detect(Ortholog, "UNMAPPED|UNGROUPED"))
humann2_path.decontam <- humann2_path.decontam %>% filter(!str_detect(Pathway, "UNMAPPED|UNINTEGRATED"))

```

```{r pca_plotting_functions}
# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))

    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Industry, Ethnic_Group, Age_group, Sex, Market_integration, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group, shape = metadata_group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

```

# Pathways
## Industrial and non-industrial samples
Do a factor analysis to reduce the number of proteins we're working with
First create input tables
```{r path_input_tables_noblanks}
# Pathways

humann2_path.decontam_noblanks_presence_more_30 <- humann2_path.decontam %>%
  select(-matches("EXB|LIB|10M|ERR|SRS|SRR164")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Pathway) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the pathways that are present in at least 1/3 of at least 2 groups
  ungroup() %>%
  # pathways present in > 1/3 of samples in each group
  filter(Percent_presence >= 33.33) %>%
  group_by(Pathway) %>%
  count() %>%
  # only pathways that are present in 1/3 of samples of 2 or more groups
  filter(n > 1) %>%
  # pull out these proteins to keep
  select(Pathway)

humann2_path_noblanks_mat <- humann2_path.decontam %>%
  select(-matches("ERR|SRR164|SRS|EXB|LIB|10M")) %>%
  inner_join(., humann2_path.decontam_noblanks_presence_more_30, by = "Pathway") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  # mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")


```

```{r fa_species_decontam_noblanks}
# from the scree plots I selected nfactors = 5
# from the graphs, I selected fm = minres and rotate = oblimin
humann2_path_noblanks.fa <- fa(humann2_path_noblanks_mat, nfactors = 5, rotate = "oblimin", fm = "minres") # rotate = "Varimax" "oblimin" # fm = "ols" "minres"
# if using fm = "ols", the column titles will be 1, 2, etc and not M1, M2, etc as you get from using fm = "minres"

print(humann2_path_noblanks.fa)
# print(humann2_path_noblanks.fa$loadings,cutoff = 0.7)
# print(humann2_path_noblanks.fa,sort=TRUE)

scree_plot <- humann2_path_noblanks.fa$values %>%
  as.data.frame(.) %>%
  rename(Eigenvalues = ".") %>%
  arrange(desc(Eigenvalues))%>%
  rownames_to_column("xvalue") %>%
  mutate(xvalue = as.numeric(xvalue)) %>%
  filter(xvalue <= 20) %>%
  ggplot(., aes(xvalue, Eigenvalues)) +
         geom_point() +
         theme_minimal() +
         geom_hline(yintercept = 10, linetype = "dashed", color = "red")

scree_plot

# ggsave("./05-results.backup/presentation_pdfs/scree_humann2_path_noblanks_oblimin_minres.pdf", plot = scree_plot, device = "pdf",
#        scale = 1, width = 10, height = 10, units = c("in"), dpi = 300)

# get the minimum residual values into a datatable
graphinfo <- as.data.frame(factor2cluster(humann2_path_noblanks.fa))
colnames(graphinfo) <- c("MR11","MR21","MR31","MR41","MR51")
graphinfo <- graphinfo %>%
  rownames_to_column("Ortholog") %>%
  as_tibble(.) 

# now make loadings into a table and designate each species in the residual with the highest absolute value
humann2_path_noblanks.fa.df <- as.data.frame.matrix(humann2_path_noblanks.fa$loadings)
# b/c of rotation, the MRs may not be in numerial order. Change the column titles to be in numerical order for downstream processing
colnames(humann2_path_noblanks.fa.df) <- c("MR1","MR2","MR3","MR4","MR5")
humann2_path_noblanks.fa.df <- humann2_path_noblanks.fa.df %>%
  as_tibble(.) %>%
  bind_cols(., graphinfo) %>%
  mutate(MR11 = as.character(MR11),
         MR21 = as.character(MR21),
         MR31 = as.character(MR31),
         MR41 = as.character(MR41),
         MR51 = as.character(MR51)) %>%
  mutate(MR11 = replace(MR11, MR11 != "0", "MR1"),
         MR21 = replace(MR21, MR21 != "0", "MR2"),
         MR31 = replace(MR31, MR31 != "0", "MR3"),
         MR41 = replace(MR41, MR41 != "0", "MR4"),
         MR51 = replace(MR51, MR51 != "0", "MR5")) %>%
  select(Ortholog,everything(.)) %>%
  gather("Group","Residual", 7:ncol(.)) %>%
  filter(Residual != "0") %>%
  select(Residual, Ortholog, MR1, MR2, MR3, MR4, MR5)

head(humann2_path_noblanks.fa.df)

# save the species using in downstream discriminant analyses as a table
humann2_path_noblanks.fa_path <- humann2_path_noblanks.fa.df %>% 
  # filter(Residual %in% c("MR1","MR2")) %>%
  select(Ortholog, Residual)

fwrite(humann2_path_noblanks.fa_path, file = "./05-results.backup/minres_oblimin_noblanks.fa_path.tsv", quote = F, sep = "\t")

```

Now a linear discriminant analysis using the selected factors
```{r lda_species_decontam_noblanks_env}

# select only the species that make up the 1st 3 residuals, based on the scree plot above
fa_proteins_noblanks <- humann2_path_noblanks.fa.df %>% 
  filter(Residual %in% c("MR1")) %>%
  pull(Ortholog)

# sub-sample the species table for only those species
humann2_path_noblanks_lda_table <- humann2_path_noblanks_mat %>%
  select(one_of(fa_proteins_noblanks)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
               select(SampleID, Env)) %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
  select(SampleID, Env, everything())

rownames(humann2_path_noblanks_lda_table) <- humann2_path_noblanks_lda_table$SampleID
humann2_path_noblanks_lda_table <- humann2_path_noblanks_lda_table %>% select(-SampleID)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_path_noblanks_lda <- lda(Env ~ ., humann2_path_noblanks_lda_table)
# humann2_path_noblanks_lda # show results
humann2_path_noblanks_lda
plot(humann2_path_noblanks_lda, dimen = 2)

lda.data <- cbind(humann2_path_noblanks_lda_table, predict(humann2_path_noblanks_lda)$x)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_path_noblanks_env <- ggplot(lda.data, aes(LD1, LD2, colour = Env, shape = Env)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = Env_colors) +
                        scale_shape_manual(values = Env_shapes) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())
humann2_path_noblanks_env

```

## Cameroon plaque samples only
```{r path_input_tables_cmc}
# Pathways

humann2_path.decontam_noblanks_presence_more_30 <- humann2_path.decontam %>%
  select(matches("Pathway|CMC")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Pathway) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the pathways that are present in at least 1/3 of at least 2 groups
  ungroup() %>%
  # pathways present in > 1/3 of samples in each group
  filter(Percent_presence >= 33.33) %>%
  group_by(Pathway) %>%
  count() %>%
  # only pathways that are present in 1/3 of samples of 2 or more groups
  filter(n > 1) %>%
  # pull out these proteins to keep
  select(Pathway)

humann2_path_cmc_mat <- humann2_path.decontam %>%
  select(matches("Pathway|CMC")) %>%
  inner_join(., humann2_path.decontam_noblanks_presence_more_30, by = "Pathway") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  # mutate(Counts = Counts + 1) %>%
  spread(Pathway,Counts) %>%
  column_to_rownames("SampleID")


```

```{r fa_species_decontam_cmc}
# from the scree plots I selected nfactors = 5
# from the graphs, I selected fm = minres and rotate = oblimin
humann2_path_cmc.fa <- fa(humann2_path_cmc_mat, nfactors = 5, rotate = "oblimin", fm = "minres") # rotate = "Varimax" "oblimin" # fm = "ols" "minres"
# if using fm = "ols", the column titles will be 1, 2, etc and not M1, M2, etc as you get from using fm = "minres"

print(humann2_path_cmc.fa)
# print(humann2_path_cmc.fa$loadings,cutoff = 0.7)
# print(humann2_path_cmc.fa,sort=TRUE)

scree_plot <- humann2_path_cmc.fa$values %>%
  as.data.frame(.) %>%
  rename(Eigenvalues = ".") %>%
  arrange(desc(Eigenvalues))%>%
  rownames_to_column("xvalue") %>%
  mutate(xvalue = as.numeric(xvalue)) %>%
  filter(xvalue <= 20) %>%
  ggplot(., aes(xvalue, Eigenvalues)) +
         geom_point() +
         theme_minimal() +
         geom_hline(yintercept = 10, linetype = "dashed", color = "red")

scree_plot

# ggsave("./05-results.backup/presentation_pdfs/scree_humann2_path_cmc_oblimin_minres.pdf", plot = scree_plot, device = "pdf",
#        scale = 1, width = 10, height = 10, units = c("in"), dpi = 300)

# get the minimum residual values into a datatable
graphinfo <- as.data.frame(factor2cluster(humann2_path_cmc.fa))
colnames(graphinfo) <- c("MR11","MR21","MR31","MR41","MR51")
graphinfo <- graphinfo %>%
  rownames_to_column("Ortholog") %>%
  as_tibble(.) 

# now make loadings into a table and designate each species in the residual with the highest absolute value
humann2_path_cmc.fa.df <- as.data.frame.matrix(humann2_path_cmc.fa$loadings)
# b/c of rotation, the MRs may not be in numerial order. Change the column titles to be in numerical order for downstream processing
colnames(humann2_path_cmc.fa.df) <- c("MR1","MR2","MR3","MR4","MR5")
humann2_path_cmc.fa.df <- humann2_path_cmc.fa.df %>%
  as_tibble(.) %>%
  bind_cols(., graphinfo) %>%
  mutate(MR11 = as.character(MR11),
         MR21 = as.character(MR21),
         MR31 = as.character(MR31),
         MR41 = as.character(MR41),
         MR51 = as.character(MR51)) %>%
  mutate(MR11 = replace(MR11, MR11 != "0", "MR1"),
         MR21 = replace(MR21, MR21 != "0", "MR2"),
         MR31 = replace(MR31, MR31 != "0", "MR3"),
         MR41 = replace(MR41, MR41 != "0", "MR4"),
         MR51 = replace(MR51, MR51 != "0", "MR5")) %>%
  select(Ortholog,everything(.)) %>%
  gather("Group","Residual", 7:ncol(.)) %>%
  filter(Residual != "0") %>%
  select(Residual, Ortholog, MR1, MR2, MR3, MR4, MR5)

head(humann2_path_cmc.fa.df)

# now run a discriminant analysis

# save the species using in downstream discriminant analyses as a table
humann2_path_cmc.fa_path <- humann2_path_cmc.fa.df %>% 
  # filter(Residual %in% c("MR1","MR2")) %>%
  select(Ortholog, Residual)

fwrite(humann2_path_cmc.fa_path, file = "./05-results.backup/minres_oblimin_cmc.fa_path.tsv", quote = F, sep = "\t")

```

Now a linear discriminant analysis using the selected factors
```{r lda_species_decontam_cmc_env}

# select only the species that make up the 1st 3 residuals, based on the scree plot above
fa_proteins_cmc <- humann2_path_cmc.fa.df %>% 
  filter(Residual %in% c("MR1")) %>%
  pull(Ortholog)

# sub-sample the species table for only those species
humann2_path_cmc_lda_table <- humann2_path_cmc_mat %>%
  select(one_of(fa_proteins_cmc)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
               select(SampleID, Ethnic_Group)) %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
  select(SampleID, Ethnic_Group, everything())

rownames(humann2_path_cmc_lda_table) <- humann2_path_cmc_lda_table$SampleID
humann2_path_cmc_lda_table <- humann2_path_cmc_lda_table %>% select(-SampleID)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_path_cmc_lda <- lda(Ethnic_Group ~ ., humann2_path_cmc_lda_table)
# humann2_path_cmc_lda # show results
humann2_path_cmc_lda
plot(humann2_path_cmc_lda, dimen = 2)

lda.data <- cbind(humann2_path_cmc_lda_table, predict(humann2_path_cmc_lda)$x)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_path_cmc_eg <- ggplot(lda.data, aes(x = Ethnic_Group, y = LD1, group = Ethnic_Group, fill = Ethnic_Group)) +
                        geom_boxplot() + 
                        geom_jitter(size = 2, aes(shape = Ethnic_Group)) +
                        scale_fill_manual(values = Ethnic_Group_colors) +
                        scale_shape_manual(values = c(3,4)) +
                        xlab("Ethnic Group") +
                        ylab("LD1") +
                        geom_hline(yintercept = 0, linetype = "dashed") +
                        coord_flip() +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())
                        theme(legend.title = element_blank())
humann2_path_cmc_eg

```


# Gene Families

## Industrial and non-industrial samples
Do a factor analysis to reduce the number of proteins we're working with
First create input tables
```{r input_tables_noblanks}
# Gene families, no blanks 
# reduce the input table to only orthologs present in at least 1/3 of one ethnic group

humann2_KOs.decontam_noblanks_presence_more_30 <- humann2_KOs.decontam %>%
  select(-matches("EXB|LIB|10M|ERR|SRS|SRR164")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Ortholog) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the SEED proteins that are present in at least 1/3 of at least 2 groups
  ungroup() %>%
  # pathways present in > 1/3 of samples in each group
  filter(Percent_presence >= 33.33) %>%
  group_by(Ortholog) %>%
  count() %>%
  # only pathways that are present in 1/3 of samples of 2 or more groups
  filter(n > 1) %>%
  # pull out these proteins to keep
  select(Ortholog)

# SEED proteins, no blanks 
humann2_KO_noblanks_mat <- humann2_KOs.decontam %>%
  select(-matches("ERR|SRR164|SRS|EXB|LIB|10M")) %>%
  inner_join(., humann2_KOs.decontam_noblanks_presence_more_30, by = "Ortholog") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  # mutate(Counts = Counts + 1) %>%
  spread(Ortholog,Counts) %>%
  column_to_rownames("SampleID")

```

```{r fa_species_decontam_noblanks}
# from the scree plots I selected nfactors = 5
# from the graphs, I selected fm = minres and rotate = oblimin
humann2_KO_noblanks.fa <- fa(humann2_KO_noblanks_mat, nfactors = 5, rotate = "oblimin", fm = "minres") # rotate = "Varimax" "oblimin" # fm = "ols" "minres"
# if using fm = "ols", the column titles will be 1, 2, etc and not M1, M2, etc as you get from using fm = "minres"

print(humann2_KO_noblanks.fa)
# print(humann2_KO_noblanks.fa$loadings,cutoff = 0.7)
# print(humann2_KO_noblanks.fa,sort=TRUE)

scree_plot <- humann2_KO_noblanks.fa$values %>%
  as.data.frame(.) %>%
  rename(Eigenvalues = ".") %>%
  arrange(desc(Eigenvalues))%>%
  rownames_to_column("xvalue") %>%
  mutate(xvalue = as.numeric(xvalue)) %>%
  filter(xvalue <= 20) %>%
  ggplot(., aes(xvalue, Eigenvalues)) +
         geom_point() +
         theme_minimal() +
         geom_hline(yintercept = 10, linetype = "dashed", color = "red")

scree_plot

# ggsave("./05-results.backup/presentation_pdfs/scree_humann2_KOs_noblanks_oblimin_minres.pdf", plot = scree_plot, device = "pdf",
#        scale = 1, width = 10, height = 10, units = c("in"), dpi = 300)

# get the minimum residual values into a datatable
graphinfo <- as.data.frame(factor2cluster(humann2_KO_noblanks.fa))
colnames(graphinfo) <- c("MR11","MR21","MR31","MR41","MR51")
graphinfo <- graphinfo %>%
  rownames_to_column("Ortholog") %>%
  as_tibble(.) 

# now make loadings into a table and designate each species in the residual with the highest absolute value
humann2_KO_noblanks.fa.df <- as.data.frame.matrix(humann2_KO_noblanks.fa$loadings)
# b/c of rotation, the MRs may not be in numerial order. Change the column titles to be in numerical order for downstream processing
colnames(humann2_KO_noblanks.fa.df) <- c("MR1","MR2","MR3","MR4","MR5")
humann2_KO_noblanks.fa.df <- humann2_KO_noblanks.fa.df %>%
  as_tibble(.) %>%
  bind_cols(., graphinfo) %>%
  mutate(MR11 = as.character(MR11),
         MR21 = as.character(MR21),
         MR31 = as.character(MR31),
         MR41 = as.character(MR41),
         MR51 = as.character(MR51)) %>%
  mutate(MR11 = replace(MR11, MR11 != "0", "MR1"),
         MR21 = replace(MR21, MR21 != "0", "MR2"),
         MR31 = replace(MR31, MR31 != "0", "MR3"),
         MR41 = replace(MR41, MR41 != "0", "MR4"),
         MR51 = replace(MR51, MR51 != "0", "MR5")) %>%
  select(Ortholog,everything(.)) %>%
  gather("Group","Residual", 7:ncol(.)) %>%
  filter(Residual != "0") %>%
  select(Residual, Ortholog, MR1, MR2, MR3, MR4, MR5)

head(humann2_KO_noblanks.fa.df)

# now run a discriminant analysis

# save the species using in downstream discriminant analyses as a table
humann2_KO_noblanks.fa_KO <- humann2_KO_noblanks.fa.df %>% 
  # filter(Residual %in% c("MR1","MR2")) %>%
  select(Ortholog, Residual)

fwrite(humann2_KO_noblanks.fa_KO, file = "./05-results.backup/minres_oblimin_noblanks.fa_KO.tsv", quote = F, sep = "\t")

```

Now a linear discriminant analysis using the selected factors

```{r lda_species_decontam_noblanks_env}

# select only the species that make up the 1st 3 residuals, based on the scree plot above
fa_proteins_noblanks <- humann2_KO_noblanks.fa.df %>% 
  filter(Residual %in% c("MR1")) %>%
  pull(Ortholog)

# sub-sample the species table for only those species
humann2_KO_noblanks_lda_table <- humann2_KO_noblanks_mat %>%
  select(one_of(fa_proteins_noblanks)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
               select(SampleID, Env)) %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
  select(SampleID, Env, everything())

rownames(humann2_KO_noblanks_lda_table) <- humann2_KO_noblanks_lda_table$SampleID
humann2_KO_noblanks_lda_table <- humann2_KO_noblanks_lda_table %>% select(-SampleID)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KO_noblanks_lda <- lda(Env ~ ., humann2_KO_noblanks_lda_table)
# humann2_KO_noblanks_lda # show results
humann2_KO_noblanks_lda
plot(humann2_KO_noblanks_lda, dimen = 2)

lda.data <- cbind(humann2_KO_noblanks_lda_table, predict(humann2_KO_noblanks_lda)$x)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KOs_noblanks_env <- ggplot(lda.data, aes(LD1, LD2, colour = Env, shape = Env)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = Env_colors) +
                        scale_shape_manual(values = Env_shapes) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())
humann2_KOs_noblanks_env
```


## Cameroon plaque samples only
Do a factor analysis to reduce the number of proteins we're working with
First create input tables
```{r input_tables}

# Gene families, CMC only 

# reduce the input table to only orthologs present in at least 33% of one ethnic group
humann2_KOs.decontam_cmc_presence_more_30 <- humann2_KOs.decontam %>%
  select(matches("Ortholog|CMC")) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Ortholog) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the SEED proteins that are present in at least 30% of at least one group
  ungroup() %>%
  filter(Percent_presence >= 33.33) %>%
  distinct(Ortholog) %>%
  # pull out these proteins to keep
  select(Ortholog)

# SEED proteins, no blanks 
humann2_KO_cmc_mat <- humann2_KOs.decontam %>%
  select(matches("Ortholog|CMC")) %>%
  inner_join(., humann2_KOs.decontam_cmc_presence_more_30, by = "Ortholog") %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  # mutate(Counts = Counts + 1) %>%
  spread(Ortholog,Counts) %>%
  column_to_rownames("SampleID")

```

```{r fa_species_decontam_cmc}
# from the scree plots I selected nfactors = 5
# from the graphs, I selected fm = minres and rotate = oblimin
humann2_KO_cmc.fa <- fa(humann2_KO_cmc_mat, nfactors = 5, rotate = "oblimin", fm = "minres") # rotate = "Varimax" "oblimin" # fm = "ols" "minres"
# if using fm = "ols", the column titles will be 1, 2, etc and not M1, M2, etc as you get from using fm = "minres"

print(humann2_KO_cmc.fa)
# print(humann2_KO_cmc.fa$loadings,cutoff = 0.7)
# print(humann2_KO_cmc.fa,sort=TRUE)

scree_plot <- humann2_KO_cmc.fa$values %>%
  as.data.frame(.) %>%
  rename(Eigenvalues = ".") %>%
  arrange(desc(Eigenvalues))%>%
  rownames_to_column("xvalue") %>%
  mutate(xvalue = as.numeric(xvalue)) %>%
  filter(xvalue <= 20) %>%
  ggplot(., aes(xvalue, Eigenvalues)) +
         geom_point() +
         theme_minimal() +
         geom_hline(yintercept = 10, linetype = "dashed", color = "red")

scree_plot

# ggsave("./05-results.backup/presentation_pdfs/scree_humann2_KOs_cmc_oblimin_minres.pdf", plot = scree_plot, device = "pdf",
#        scale = 1, width = 10, height = 10, units = c("in"), dpi = 300)

# get the minimum residual values into a datatable
graphinfo <- as.data.frame(factor2cluster(humann2_KO_cmc.fa))
colnames(graphinfo) <- c("MR11","MR21","MR31","MR41","MR51")
graphinfo <- graphinfo %>%
  rownames_to_column("Ortholog") %>%
  as_tibble(.) 

# now make loadings into a table and designate each species in the residual with the highest absolute value
humann2_KO_cmc.fa.df <- as.data.frame.matrix(humann2_KO_cmc.fa$loadings)
# b/c of rotation, the MRs may not be in numerial order. Change the column titles to be in numerical order for downstream processing
colnames(humann2_KO_cmc.fa.df) <- c("MR1","MR2","MR3","MR4","MR5")
humann2_KO_cmc.fa.df <- humann2_KO_cmc.fa.df %>%
  as_tibble(.) %>%
  bind_cols(., graphinfo) %>%
  mutate(MR11 = as.character(MR11),
         MR21 = as.character(MR21),
         MR31 = as.character(MR31),
         MR41 = as.character(MR41),
         MR51 = as.character(MR51)) %>%
  mutate(MR11 = replace(MR11, MR11 != "0", "MR1"),
         MR21 = replace(MR21, MR21 != "0", "MR2"),
         MR31 = replace(MR31, MR31 != "0", "MR3"),
         MR41 = replace(MR41, MR41 != "0", "MR4"),
         MR51 = replace(MR51, MR51 != "0", "MR5")) %>%
  select(Ortholog,everything(.)) %>%
  gather("Group","Residual", 7:ncol(.)) %>%
  filter(Residual != "0") %>%
  select(Residual, Ortholog, MR1, MR2, MR3, MR4, MR5)

head(humann2_KO_cmc.fa.df)


# save the species using in downstream discriminant analyses as a table
humann2_KO_cmc.fa_KO <- humann2_KO_cmc.fa.df %>% 
  # filter(Residual %in% c("MR1","MR2")) %>%
  select(Ortholog, Residual)

fwrite(humann2_KO_cmc.fa_KO, file = "./05-results.backup/minres_oblimin_cmc.fa_KO.tsv", quote = F, sep = "\t")

```

Now a linear discriminant analysis using the selected factors
```{r lda_species_decontam_cmc_eg}

# select only the species that make up the 1st 3 residuals, based on the scree plot above
fa_proteins_cmc <- humann2_KO_cmc.fa.df %>% 
  filter(Residual %in% c("MR1")) %>%
  pull(Ortholog)

# sub-sample the species table for only those species
humann2_KO_cmc_lda_table <- humann2_KO_cmc_mat %>%
  select(one_of(fa_proteins_cmc)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
               select(SampleID, Ethnic_Group)) %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
  select(SampleID, Ethnic_Group, everything())

rownames(humann2_KO_cmc_lda_table) <- humann2_KO_cmc_lda_table$SampleID
humann2_KO_cmc_lda_table <- humann2_KO_cmc_lda_table %>% select(-SampleID)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KO_cmc_lda <- lda(Ethnic_Group ~ ., humann2_KO_cmc_lda_table)
# humann2_KO_cmc_lda # show results
humann2_KO_cmc_lda
plot(humann2_KO_cmc_lda, dimen = 2)

lda.data <- cbind(humann2_KO_cmc_lda_table, predict(humann2_KO_cmc_lda)$x)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KOs_cmc_eg <- ggplot(lda.data, aes(x = Ethnic_Group, y = LD1, group = Ethnic_Group, fill = Ethnic_Group)) +
                        geom_boxplot() + 
                        geom_jitter(size = 2, aes(shape = Ethnic_Group)) +
                        scale_fill_manual(values = Ethnic_Group_colors) +
                        scale_shape_manual(values = c(3,4)) +
                        xlab("Ethnic Group") +
                        ylab("LD1") +
                        geom_hline(yintercept = 0, linetype = "dashed") +
                        coord_flip() +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())
                        theme(legend.title = element_blank())
humann2_KOs_cmc_eg

```

```{r lda_species_decontam_cmc_age}

# select only the species that make up the 1st 3 residuals, based on the scree plot above
fa_proteins_cmc <- humann2_KO_cmc.fa.df %>% 
  filter(Residual %in% c("MR1")) %>%
  pull(Ortholog)

# sub-sample the species table for only those species
humann2_KO_cmc_lda_table <- humann2_KO_cmc_mat %>%
  select(one_of(fa_proteins_cmc)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
               select(SampleID, Age_group)) %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
  select(SampleID, Age_group, everything())

rownames(humann2_KO_cmc_lda_table) <- humann2_KO_cmc_lda_table$SampleID
humann2_KO_cmc_lda_table <- humann2_KO_cmc_lda_table %>% select(-SampleID)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KO_cmc_lda <- lda(Age_group ~ ., humann2_KO_cmc_lda_table)
# humann2_KO_cmc_lda # show results
humann2_KO_cmc_lda
plot(humann2_KO_cmc_lda, dimen = 2)

lda.data <- cbind(humann2_KO_cmc_lda_table, predict(humann2_KO_cmc_lda)$x)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KOs_cmc_age <- ggplot(lda.data, aes(LD1, LD2, colour = Age_group, shape = Age_group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = Age_group_colors) +
                        scale_shape_manual(values = Age_group_shapes) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())
humann2_KOs_cmc_age

```

```{r lda_species_decontam_cmc_me}

# select only the species that make up the 1st 3 residuals, based on the scree plot above
fa_proteins_cmc <- humann2_KO_cmc.fa.df %>% 
  filter(Residual %in% c("MR1")) %>%
  pull(Ortholog)

# sub-sample the species table for only those species
humann2_KO_cmc_lda_table <- humann2_KO_cmc_mat %>%
  select(one_of(fa_proteins_cmc)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
               select(SampleID, Market_integration)) %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
  select(SampleID, Market_integration, everything())

rownames(humann2_KO_cmc_lda_table) <- humann2_KO_cmc_lda_table$SampleID
humann2_KO_cmc_lda_table <- humann2_KO_cmc_lda_table %>% select(-SampleID)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KO_cmc_lda <- lda(Market_integration ~ ., humann2_KO_cmc_lda_table)
# humann2_KO_cmc_lda # show results
humann2_KO_cmc_lda
plot(humann2_KO_cmc_lda, dimen = 2)

lda.data <- cbind(humann2_KO_cmc_lda_table, predict(humann2_KO_cmc_lda)$x)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KOs_cmc_me <- ggplot(lda.data, aes(LD1, LD2, colour = Market_integration, shape = Market_integration)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = Market_integration_colors) +
                        scale_shape_manual(values = Market_integration_shapes) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())
humann2_KOs_cmc_me
```

```{r lda_species_decontam_cmc_sex}

# select only the species that make up the 1st 3 residuals, based on the scree plot above
fa_proteins_cmc <- humann2_KO_cmc.fa.df %>% 
  filter(Residual %in% c("MR1")) %>%
  pull(Ortholog)

# sub-sample the species table for only those species
humann2_KO_cmc_lda_table <- humann2_KO_cmc_mat %>%
  select(one_of(fa_proteins_cmc)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
               select(SampleID, Sex)) %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
  select(SampleID, Sex, everything())

rownames(humann2_KO_cmc_lda_table) <- humann2_KO_cmc_lda_table$SampleID
humann2_KO_cmc_lda_table <- humann2_KO_cmc_lda_table %>% select(-SampleID)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KO_cmc_lda <- lda(Sex ~ ., humann2_KO_cmc_lda_table)
# humann2_KO_cmc_lda # show results
humann2_KO_cmc_lda
plot(humann2_KO_cmc_lda, dimen = 2)

lda.data <- cbind(humann2_KO_cmc_lda_table, predict(humann2_KO_cmc_lda)$x)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KOs_cmc_sex <- ggplot(lda.data, aes(x = Sex, y = LD1, group = Sex, fill = Sex)) +
                        geom_boxplot() + 
                        geom_jitter(size = 2, aes(shape = Sex)) +
                        scale_fill_manual(values = Sex_colors) +
                        scale_shape_manual(values = c(3,4)) +
                        xlab("Sex") +
                        ylab("LD1") +
                        geom_hline(yintercept = 0, linetype = "dashed") +
                        coord_flip() +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())
humann2_KOs_cmc_sex

```

```{r lda_species_decontam_cmc_ts}

# select only the species that make up the 1st 3 residuals, based on the scree plot above
fa_proteins_cmc <- humann2_KO_cmc.fa.df %>% 
  filter(Residual %in% c("MR1")) %>%
  pull(Ortholog)

# sub-sample the species table for only those species
humann2_KO_cmc_lda_table <- humann2_KO_cmc_mat %>%
  select(one_of(fa_proteins_cmc)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
               select(SampleID, Tooth_site)) %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
  select(SampleID, Tooth_site, everything())

rownames(humann2_KO_cmc_lda_table) <- humann2_KO_cmc_lda_table$SampleID
humann2_KO_cmc_lda_table <- humann2_KO_cmc_lda_table %>% select(-SampleID)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KO_cmc_lda <- lda(Tooth_site ~ ., humann2_KO_cmc_lda_table)
# humann2_KO_cmc_lda # show results
humann2_KO_cmc_lda
plot(humann2_KO_cmc_lda, dimen = 2)

lda.data <- cbind(humann2_KO_cmc_lda_table, predict(humann2_KO_cmc_lda)$x)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KOs_cmc_ts <- ggplot(lda.data, aes(x = Tooth_site, y = LD1, group = Tooth_site, fill = Tooth_site)) +
                        geom_boxplot() + 
                        geom_jitter(size = 2, aes(shape = Tooth_site)) +
                        scale_fill_manual(values = Tooth_site_colors) +
                        scale_shape_manual(values = c(3,4)) +
                        xlab("Tooth site") +
                        ylab("LD1") +
                        geom_hline(yintercept = 0, linetype = "dashed") +
                        coord_flip() +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())
humann2_KOs_cmc_ts

```

```{r lda_species_decontam_cmc_vi}

# select only the species that make up the 1st 3 residuals, based on the scree plot above
fa_proteins_cmc <- humann2_KO_cmc.fa.df %>% 
  filter(Residual %in% c("MR1")) %>%
  pull(Ortholog)

# sub-sample the species table for only those species
humann2_KO_cmc_lda_table <- humann2_KO_cmc_mat %>%
  select(one_of(fa_proteins_cmc)) %>%
  rownames_to_column("SampleID") %>%
  inner_join(., metadata %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
               select(SampleID, Village)) %>%
               # change the column after SampleID to get the metadata category you want 
               # Age_group, Ethnic_Group, Env, Industry, Market_integration, Sex, Tooth_site, Village
  select(SampleID, Village, everything())

rownames(humann2_KO_cmc_lda_table) <- humann2_KO_cmc_lda_table$SampleID
humann2_KO_cmc_lda_table <- humann2_KO_cmc_lda_table %>% select(-SampleID)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KO_cmc_lda <- lda(Village ~ ., humann2_KO_cmc_lda_table)
# humann2_KO_cmc_lda # show results
humann2_KO_cmc_lda
plot(humann2_KO_cmc_lda, dimen = 2)

lda.data <- cbind(humann2_KO_cmc_lda_table, predict(humann2_KO_cmc_lda)$x)

# change the first variable here to get the metadata category you want # Age_group, Ethnic_Group, Env, Market_integration, Sex, Tooth_site, Village
humann2_KOs_cmc_vi <- ggplot(lda.data, aes(LD1, LD2, colour = Village, shape = Village)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = Village_colors) +
                        scale_shape_manual(values = Village_shapes) +
                        xlab("LD1") +
                        ylab("LD2") +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())
humann2_KOs_cmc_vi

```

```{r cmc_images, eval = F}

cmcplots_humann2_KOs <- plot_grid(humann2_KOs_cmc_eg, humann2_KOs_cmc_age,
          humann2_KOs_cmc_me, humann2_KOs_cmc_sex,
          humann2_KOs_cmc_ts, humann2_KOs_cmc_vi,
          nrow=3, labels = c("A", "B", "C", "D", "E", "F"), rel_widths = c(1, 1))

cmcplots_humann2_KOs

# ggsave("05-results.backup/presentation_pdfs/cmcplots_humann2_KOs.pdf", plot = cmcplots_humann2_KOs, device = "pdf",
#        scale = 1, width = 10, height = 10, units = c("in"), dpi = 300)

```






























