---
title: "Cameroon hunter-gatherer calculus/plaque MALT beta diversity"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(psych)
library(ape)
library(mixOmics)
library(compositions)
library(vegan)
library(janitor)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

# MALT read stats
The number of reads assigned in each sample appears very low (~50%). We want to check this so we know how many were assinged in each sample. We used the cleaned and merged reads that didn't align to the human genome (hg19), with the following database: /projects1/malt/databases/indexed/index040/full-nt_2017-10. 
```{r load_data}

# load the taxonomy tables that have been decontaminated and the column headers are cleaned up
load("./05-results.backup/CMC_taxonomy_tables.RData")

# load the metadata file
load("./05-results.backup/CMC_metadata.RData")

# list only the anterior samples
anterior <- metadata %>%
  filter(str_detect(SampleID, "CMC")) %>%
  filter(Tooth_site == "Anterior") %>%
  pull(SampleID)

# list only the posterior samples
posterior <- metadata %>%
  filter(str_detect(SampleID, "CMC")) %>%
  filter(Tooth_site == "Posterior") %>%
  pull(SampleID)

```


```{r pca_plotting_functions}
# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type, ExtractionBatch, LibraryBatch), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group, shape = metadata_group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type, ExtractionBatch, LibraryBatch), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$SampleID), size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}


# plot PCA with colored dots and the title including the # of species or genera
plot_pca_titles <- function(df, pc1, pc2, color_group, shape_group, size_group, ncomps) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))
    metadata_group_size <- get(paste(size_group, "_size", sep = ""))

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$explained_variance * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type, ExtractionBatch, LibraryBatch), by = "SampleID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]
    size_group = df_X[[size_group]]
   
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    title_text <- df %>%
                     rownames_to_column("SampleID") %>%
                     gather("Species","Counts",2:ncol(.)) %>%
                     spread(SampleID, Counts) %>%
                     count
    
    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group, size = size_group)) +
                        geom_point() +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        scale_size_manual(values = metadata_group_size) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        ggtitle(paste(title_text))

    return(pca_plot)
}

```

```{r pca_biplot_functions}
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$explained_variance * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }
    
    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(10, !!arrow_pc)

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(-10, !!arrow_pc)


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 10)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

# pca_test_biplot <- plot_pca_bi(malt_species.pca, "PC3", "PC2", "Village", PC2)
# pca_test_biplot


```


List the outliers from the plot in section `raw_plot` in `Cameroon_h-g_taxonomy.Rmd` (above) and remove them from the taxonomy table. 
```{r outliers}
outliers <- c("CMC002.B0201","CMC032.B0101","CMC037.B0201","CMC047.A0201",
              "EXB059.A1901","EXB059.A1301","EXB059.A1101","EXB059.A0901",
              "SRR1646041","SRR1646032","SRR1646042","SRR1646045","SRR1646030","SRR1646047")

outliersF <- str_c(outliers, collapse = "|")

```

#Sample Clustering with PCA

## Species-level analyses with industrial samples

What is the relationship of the groups in a PCA without the blanks?
```{r pca_noblanks_species}
# make a table that has only samples from one or the other tooth site
malt_species.decontam_filtered_noblanks <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|10M")) %>%
                                         select(-one_of(posterior)) %>% # leave only the anterior samples
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>% mutate(Counts = Counts + 1) %>%
                                         mutate(Counts = Counts + 1) %>%
                                         spread(Species,Counts) %>%
                                        column_to_rownames("SampleID")


# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered_noblanks, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Description (do HMP supragingival and HMP subgingival plaque samples separate?)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Description", "Description", "Description", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_species_decontam_filtered_noblanks.pca <- mixOmics::pca(malt_species.decontam_filtered_noblanks, ncomp = 3, logratio = 'CLR')
plot(malt_species_decontam_filtered_noblanks.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_species_decontam_filtered_noblanks.pca.variates <- as.data.frame(malt_species_decontam_filtered_noblanks.pca$variates$X)
malt_species_decontam_filtered_noblanks.pca.variates <- malt_species_decontam_filtered_noblanks.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_species_decontam_filtered_noblanks.pca.varmet <- inner_join(malt_species_decontam_filtered_noblanks.pca.variates, metadata, by = "SampleID")

pca_malt_species_pc1pc2 <- ggplot(malt_species_decontam_filtered_noblanks.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_species_decontam_filtered_noblanks.pca$explained_variance[2], 2)), "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_species_pc1pc2

# ggsave("05-results.backup/pca_nt_species_pc1pc2.pdf", plot = pca_nt_species_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)

```
_Anterior samples only:_ The CM anterior samples are spread across the plot, but
generally plot closer to the JAE samples than the posterior samples do, and plot
between the HMP and JAE samples, rather than clustering away from both like the
posterior samples. Although the Yanomami samples appear to fall within the
variation of the HMP samples in PC1/CP2 figures, they are clearly separated from
all plaque/calculus in PC3. The CMC samples do not plot with the Yanomami
samples.


```{r pca_noyanomami_species}
# make a table that has only samples from one or the other tooth site
malt_species.decontam_filtered_noyanomami <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|10M|SRR164")) %>%
                                         select(-one_of(posterior)) %>% # leave only the anterior samples
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         mutate(Counts = Counts + 1) %>%
                                         spread(Species,Counts) %>%
                                         column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered_noyanomami, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_species.decontam_filtered_noyanomami, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noyanomami, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_species.decontam_filtered_noyanomami, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noyanomami, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_species.decontam_filtered_noyanomami, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noyanomami, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_species.decontam_filtered_noyanomami, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noyanomami, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_species.decontam_filtered_noyanomami, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noyanomami, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Description (do HMP supragingival and HMP subgingival plaque samples separate?)
plot_pca_titles(malt_species.decontam_filtered_noyanomami, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_species.decontam_filtered_noyanomami, "PC3", "PC2", "Description", "Description", "Description", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_species_decontam_filtered_noyanomami.pca <- mixOmics::pca(malt_species.decontam_filtered_noyanomami, ncomp = 3, logratio = 'CLR')
plot(malt_species_decontam_filtered_noyanomami.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_species_decontam_filtered_noyanomami.pca.variates <- as.data.frame(malt_species_decontam_filtered_noyanomami.pca$variates$X)
malt_species_decontam_filtered_noyanomami.pca.variates <- malt_species_decontam_filtered_noyanomami.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_species_decontam_filtered_noyanomami.pca.varmet <- inner_join(malt_species_decontam_filtered_noyanomami.pca.variates, metadata, by = "SampleID")

pca_malt_species_pc1pc2 <- ggplot(malt_species_decontam_filtered_noyanomami.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_species_decontam_filtered_noyanomami.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_species_decontam_filtered_noyanomami.pca$explained_variance[2], 2)), "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_species_pc1pc2

# ggsave("05-results.backup/pca_nt_species_pc1pc2.pdf", plot = pca_nt_species_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)

```

```{r metadata_betadisper_noblanks}

metadata_noblanks <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata_noblanks <- metadata_noblanks %>%
  rename(SampleID = `#SampleID`)  %>%
  filter(!SampleID %in% outliers) %>%
  filter(!str_detect(SampleID, "SRR16460|EXB|LIB|10M")) %>%
  filter(!SampleID %in% posterior) %>% # this is the group to remove
  as_tibble() %>%
  mutate(Market_economy = str_replace(Market_economy, "HMP","Industrial"))%>%
  mutate(Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime","HMP","JAE"),
         Env = fct_relevel(Env, "HunterGatherer","HMP","JAE"),
         Industry = fct_relevel(Industry, "Non-industrial","Industrial"),
         Market_economy = fct_relevel(Market_economy, "Logging road","Forrest camp","Farming","Industrial"),
         Age_group = fct_relevel(Age_group, "18-29","30-39","40-49","50-59","60+","HMP"),
         Sex = fct_relevel(Sex, "Male","Female"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior","HMP"),
         Village = fct_relevel(Village, "202","205","204","203","206","HMP","JAE")) %>%
  select(SampleID, ExtractionBatch, LibraryBatch, Age_group, Ethnic_Group, Env, Industry, Market_economy, Sex, Tooth_site, Village)

```

Run a permanova with function adonis to test if the clusters are significantly different.
```{r adonis_noblanks_noyanomami_species}
# remove the Yanomami samples from the input table
malt_species.decontam_filtered_noblanks_noyanomami <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|10M|SRR164")) %>%
                                         select(-one_of(posterior)) %>% # leave only the anterior samples
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         mutate(Counts = Counts + 1) %>%
                                         spread(Species,Counts) %>%
                                         column_to_rownames("SampleID")

# perform PCA
malt_species.decontam_filtered_noblanks_noyanomami.pca <- mixOmics::pca(malt_species.decontam_filtered_noblanks_noyanomami, ncomp = 3, logratio = 'CLR')

# use the CLR-transformed matrix from the mixOmics PCA above
# Euclidean distance
malt_species.decontam_filtered_noblanks_noyanomami.euc <- vegdist(malt_species.decontam_filtered_noblanks_noyanomami.pca$X, method = "euclidean", na.rm = T)
malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoa <- ape::pcoa(malt_species.decontam_filtered_noblanks_noyanomami.euc)

malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoavectors <- as.data.frame(malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoa$vectors)
malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoavectors <- malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_species.decontam_filtered_noblanks_noyanomami.euc.vecmet <- inner_join(malt_species.decontam_filtered_noblanks_noyanomami.euc.pcoavectors, metadata, by = "SampleID")

malt_species.decontam_filtered_noblanks_noyanomami.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Industry, shape = Industry)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Industry_colors) +
  scale_shape_manual(values = Industry_shapes) +
  stat_ellipse(aes(colour = Industry, group = Industry)) +
  theme_minimal(base_size = 7) +
  ggtitle("Industry")

# test for difference between different metadata categories 
adonis(malt_species.decontam_filtered_noblanks_noyanomami.euc ~ Industry + Ethnic_Group + Age_group + Market_economy + Village, metadata_noblanks, perm=999)

adonis2(malt_species.decontam_filtered_noblanks_noyanomami.euc ~ ExtractionBatch + Village + Sex + Industry + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, metadata_noblanks, permutations = 999, by = "margin")

```


Then run betadisper to see if the groups with significant differences in
clustering have different dispersion ("tight" vs. "loose" clustering of groups)
- is one group more variable than the other(s)?
```{r betadisper_noblanks_noyanomami_species, eval = F}
# Test the metadata categories that are significant (p < 0.05) based on adonis for significant differences in dispersion
# When only including anterior samples, none are significantly different

# Industry
groups_industry <- metadata_noblanks %>%
  select(Industry) %>% 
  pull()

## Calculate multivariate dispersions
industry_betadisper_noblanks_noyanomami <- betadisper(malt_species.decontam_filtered_noblanks_noyanomami.euc, groups_industry, bias.adjust =TRUE)
industry_betadisper_noblanks_noyanomami

## Perform test
anova(industry_betadisper_noblanks_noyanomami)

## Permutation test for F
permutest(industry_betadisper_noblanks_noyanomami, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
industry_betadisper_noblanks_noyanomami.HSD <- TukeyHSD(industry_betadisper_noblanks_noyanomami)
plot(industry_betadisper_noblanks_noyanomami.HSD)

## Plot with data ellipses instead of hulls
plot(industry_betadisper_noblanks_noyanomami, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(industry_betadisper_noblanks_noyanomami)


# Ethnic group
groups_ethnicity <- metadata_noblanks %>%
  select(Ethnic_Group) %>% 
  pull()

## Calculate multivariate dispersions
ethnicity_betadisper_noblanks_noyanomami <- betadisper(malt_species.decontam_filtered_noblanks_noyanomami.euc, groups_ethnicity, bias.adjust =TRUE)
ethnicity_betadisper_noblanks_noyanomami

## Perform test
anova(ethnicity_betadisper_noblanks_noyanomami)

## Permutation test for F
permutest(ethnicity_betadisper_noblanks_noyanomami, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
ethnicity_betadisper_noblanks_noyanomami.HSD <- TukeyHSD(ethnicity_betadisper_noblanks_noyanomami)
plot(ethnicity_betadisper_noblanks_noyanomami.HSD)

## Plot with data ellipses instead of hulls
plot(ethnicity_betadisper_noblanks_noyanomami, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(ethnicity_betadisper_noblanks_noyanomami)


```

Now we will look at the species that have strongest loadings in PC1, PC2, and
PC3, to see what appears to drive differences between the sample groups. We will
visualize the loadings of these species with biplots. In the first set we
include the Yanomami samples, and in the second set we exclude them.
```{r biplot_noblanks_species, eval = F}

malt_species_decontam_filtered_biplot_121 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_species_decontam_filtered_biplot_121

pandoc.table(malt_species_decontam_filtered_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_species_decontam_filtered_biplot_122 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_biplot_122

pandoc.table(malt_species_decontam_filtered_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_species_decontam_filtered_biplot_322 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_biplot_322

malt_species_decontam_filtered_biplot_323 <- plot_pca_bi(malt_species_decontam_filtered_noblanks.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_species_decontam_filtered_biplot_323

pandoc.table(malt_species_decontam_filtered_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species_decontam_filtered_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")


```
_Anterior samples only:_ n PC1, the negative loading taxa are all anaerobic,
while the positive loading taxa are *Neisseria*, *Streptococcus*, and *Rothia*,
wich are aerobic/facultative. The negative loading taxa are anaerobic. This
distinction doesn't clearly match any of the group plotting areas. In PC2
positive loadings, the taxa are mostly *Prevotella* and *Veillonella*, while in
the negative loadings the taxa are mostly anaerobes - this distinction matches
with the split of plaqe in PC2 positive values and calclus in PC2 negative
values. In PC3, the positivce loading taxa are mostly *Rothia* and
*Actinomyces*, while the negative values are mostly *Neisseria*. This
corresponds with the Yanomami buccal samples plotting in PC3 negative values, as
buccal samples have higher proportions of *Neisseria* than plaque.


```{r biplot_noblanks_noyanomami_species}
# remove the Yanomami samples from the input table
malt_species.decontam_filtered_noblanks_noyanomami <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|10M|SRR164")) %>%
                                         select(-one_of(posterior)) %>% # leave only the anterior samples
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         mutate(Counts = Counts + 1) %>%
                                         spread(Species,Counts) %>%
                                         column_to_rownames("SampleID")

# perform PCA
malt_species.decontam_filtered_noblanks_noyanomami.pca <- mixOmics::pca(malt_species.decontam_filtered_noblanks_noyanomami, ncomp = 3, logratio = 'CLR')

# Now plot biplots, and make tables of the top 10 species in each loading
malt_species_decontam_filtered_noyanomami_biplot_121 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_species_decontam_filtered_noyanomami_biplot_121

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_species_decontam_filtered_noyanomami_biplot_122 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_noyanomami_biplot_122

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_species_decontam_filtered_noyanomami_biplot_322 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_noyanomami_biplot_322

malt_species_decontam_filtered_noyanomami_biplot_323 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_species_decontam_filtered_noyanomami_biplot_323

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
_Anterior samples only:_ In PC1 positive values, the taxa with strongest
loadings are 9 *Rothio* and 1 *Streptococcus*, and the arrows point mostly up
into the PC1 positive/PC2 positive quadrant, where the HMP samples plot. The
taxa with strongest PC1 negative loading values are anaerobic. The taxa with
strongest loadings in PC2 positive values are mostly *Prevotella*, which point
into the PC1 negative/PC2 positive quadrant, away from most HMP samples, and
*Veillonella*, which point into the PC1 positive/PC2 positive quadrant, towards
the HMP samples. The PC2 negative loading taxa are mostly anaerobes. In PC3, the
taxa with strongest positive loadings are a combination of aerobes, facultative
anaerobes, and anaerobes, while the taxa with strongest negative loading values
are 7 *Neisseria* and the rest anaerobes.



## Species-level analyses with only Cameroon samples
What is the relationship of Cameroon plaque samples to eachother without any
other sample types? Do they cluster by any of the metadata categories?
```{r pca_cmc_species}
malt_species.decontam_filtered_cmc <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|SRR|JAE")) %>%
                                         select(-one_of(posterior)) %>% # leave only the anterior samples
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         mutate(Counts = Counts + 1) %>%
                                         spread(Species,Counts) %>%
                                         column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered_cmc, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Extraction batch
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Library batch
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_species.decontam_filtered_cmc.pca <- mixOmics::pca(malt_species.decontam_filtered_cmc, ncomp = 3, logratio = 'CLR')
plot(malt_species.decontam_filtered_cmc.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_species.decontam_filtered_cmc.pca.variates <- as.data.frame(malt_species.decontam_filtered_cmc.pca$variates$X)
malt_species.decontam_filtered_cmc.pca.variates <- malt_species.decontam_filtered_cmc.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_species.decontam_filtered_cmc.pca.varmet <- inner_join(malt_species.decontam_filtered_cmc.pca.variates, metadata, by = "SampleID")

pca_malt_species_pc1pc2 <- ggplot(malt_species.decontam_filtered_cmc.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_species.decontam_filtered_cmc.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_species.decontam_filtered_cmc.pca$explained_variance[2], 2))< "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_species_pc1pc2

# ggsave("05-results.backup/pca_nt_species_pc1pc2.pdf", plot = pca_nt_species_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)
```
There is no obvious clustering of the CMC samples by village, sex, ethnicity,
age group, or market economy for either the posterior samples only or the
anterior samples only. Clustering also doesn't seem to be related to sequencing
depth, extraction batch, or library batch.


```{r metadata_betadisper_cmc}
# Need special metadata that has only the categories of CMC samples, b/c of issues w/ extra factors from removed groups
metadata_cmc <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata_cmc <- metadata_cmc %>%
  rename(SampleID = `#SampleID`) %>%
  filter(!SampleID %in% outliers) %>%
  filter(!SampleID %in% posterior) %>%
  filter(Description == "HunterGatherer") %>%
  select(SampleID, ExtractionBatch, LibraryBatch, Age_group, Ethnic_Group, Market_economy, Sex, Tooth_site, Village) %>%
  as_tibble() %>%
  mutate(Age_group = fct_relevel(Age_group, "18-29","30-39","40-49","50-59","60+"),
         Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime"),
         Market_economy = fct_relevel(Market_economy, "Logging road","Forrest camp","Farming"),
         Sex = fct_relevel(Sex, "Male","Female"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior"),
         Village = fct_relevel(Village, "202","205","204","203","206"))
```


Run a permanova with function adonis to test if the clusters are significantly different.
```{r adonis_cmc_species}
# use the CLR-transformed matrix from the mixOmics PCA in the last section

# Euclidean distance
malt_species.decontam_filtered_cmc.euc <- vegdist(malt_species.decontam_filtered_cmc.pca$X, method = "euclidean", na.rm = T)
malt_species.decontam_filtered_cmc.euc.pcoa <- ape::pcoa(malt_species.decontam_filtered_cmc.euc)

malt_species.decontam_filtered_cmc.euc.pcoavectors <- as.data.frame(malt_species.decontam_filtered_cmc.euc.pcoa$vectors)
malt_species.decontam_filtered_cmc.euc.pcoavectors <- malt_species.decontam_filtered_cmc.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_species.decontam_filtered_cmc.euc.vecmet <- inner_join(malt_species.decontam_filtered_cmc.euc.pcoavectors, metadata, by = "SampleID")

malt_species.decontam_filtered_cmc.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Sex, shape = Sex)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Sex_colors) +
  scale_shape_manual(values = Sex_shapes) +
  stat_ellipse(aes(colour = Sex, group = Sex)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis2(malt_species.decontam_filtered_cmc.euc ~ ExtractionBatch + Village + Sex + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, metadata_cmc, permutations = 999, by = "margin")

```

Then run betadisper to see if the groups with significant differences in
clustering have different dispersion ("tight" vs. "loose" clustering of groups)
- is one group more variable than the other(s)?
```{r betadisper_cmc_species}
# Test the 1 metadata category that are significant (p < 0.05) based on adonis2 for significant differences in dispersion
# When including only anterior samples, only Sex is significantly different

# Sex
groups_sex <- metadata_cmc %>%
  select(Sex) %>% 
  pull()

## Calculate multivariate dispersions
sex_betadisper_cmc <- betadisper(malt_species.decontam_filtered_cmc.euc, groups_sex, bias.adjust =TRUE)
sex_betadisper_cmc

## Perform test
anova(sex_betadisper_cmc)

## Permutation test for F
permutest(sex_betadisper_cmc, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
sex_betadisper_cmc.HSD <- TukeyHSD(sex_betadisper_cmc)
plot(sex_betadisper_cmc.HSD)

## Plot with data ellipses instead of hulls
plot(sex_betadisper_cmc, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(sex_betadisper_cmc)



```
Point dispersion from the centroids is not significantly different between Males and Females. 

```{r biplots_cmc_species}

malt_species.decontam_filtered_nocontrols <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|SRR|JAE")) %>%
                                         select(-one_of(posterior)) %>% # leave only the anterior samples
                                         adorn_totals(where = "col", name = "Sum_species") %>%
                                         mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         mutate(Counts = Counts + 1) %>%
                                         spread(Species,Counts) %>%
                                         column_to_rownames("SampleID")

malt_species.decontam_filtered_nocontrols.pca <- mixOmics::pca(malt_species.decontam_filtered_nocontrols, ncomp = 3, logratio = 'CLR')
plot(malt_species.decontam_filtered_nocontrols.pca)

malt_species.decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_species.decontam_filtered_nocontrols_biplot_121

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_species.decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_species.decontam_filtered_nocontrols_biplot_122

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_species.decontam_filtered_nocontrols_biplot_322 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_species.decontam_filtered_nocontrols_biplot_322

malt_species.decontam_filtered_nocontrols_biplot_323 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_species.decontam_filtered_nocontrols_biplot_323

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
_Anterior samples only:_ In PC1, the taxa with strongest positive loadings are
mostly *Prevotella* and other anaerobes, and the taxa with strongest negative
loadings are mostly aerobes including *Neisseria* and *Rothia*. In PC2, the taxa
with strongest positive loadings are all *Rothia*, while the taxa with strongest
negative loadings are all anaerobes. In PC3, the taxa with the strongest
positive loadings are a variety of early colonizer aerobes/facultative anaerobes
and odd species (*Xylanimonas*, *Sanguibacter*), while the taxa with strongest
negative loadings are a variety of facultative anaerobes and anaerobes.



## Genus-level analyses with industrial samples

What is the relationship of the groups in a PCA without the blanks? Are
relationships similar to those observed using species data? Is there better
resolution with genus data?
```{r pca_noblanks_genus, eval = F}
malt_genus.decontam_filtered <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|10M")) %>%
                                         select(-one_of(posterior)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.01) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         mutate(Counts = Counts + 1) %>%
                                         spread(Species,Counts) %>%
                                         column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_genus.decontam_filtered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Description (do HMP supragingial and HMP subgingival plaque separate?)
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Description", "Description", "Description", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_genus.decontam_filtered.pca <- mixOmics::pca(malt_genus.decontam_filtered, ncomp = 3, logratio = 'CLR')
plot(malt_genus.decontam_filtered.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_genus.decontam_filtered.pca.variates <- as.data.frame(malt_genus.decontam_filtered.pca$variates$X)
malt_genus.decontam_filtered.pca.variates <- malt_genus.decontam_filtered.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_genus.decontam_filtered.pca.varmet <- inner_join(malt_genus.decontam_filtered.pca.variates, metadata, by = "SampleID")

pca_malt_genus_pc1pc2 <- ggplot(malt_genus.decontam_filtered.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_genus.decontam_filtered.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_genus.decontam_filtered.pca$explained_variance[2], 2)), "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_genus_pc1pc2


# ggsave("05-results.backup/pca_malt_genus_pc1pc2.pdf", plot = pca_malt_genus_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)


```
_Anterior samples only:_ Similar to the plots using species assignments, the CMC
anterior samples plot more with JAE than with HMP. It's more of a distinction
here, where the CMC samples fall within the variation of the JAE sampels along
PC1, but only fall within the variation of subgingival plaque of the HMP
samples. Here the HMP supragingival plaque cluster farther away in PC1 negative
values. The CMC samples fall within the variation of both HMP and JAE in PC2. In
PC3, the Yanomami samples separate out from HMP along PC2.

```{r pca_noyanomami_genus}
malt_genus.decontam_filtered <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|10M|SRR164")) %>%
                                         select(-one_of(posterior)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.01) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         mutate(Counts = Counts + 1) %>%
                                         spread(Species,Counts) %>%
                                         column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_genus.decontam_filtered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Description (do HMP supragingial and HMP subgingival plaque separate?)
plot_pca_titles(malt_genus.decontam_filtered, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_genus.decontam_filtered, "PC3", "PC2", "Description", "Description", "Description", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_genus.decontam_filtered_noyanomami.pca <- mixOmics::pca(malt_genus.decontam_filtered, ncomp = 3, logratio = 'CLR')
plot(malt_genus.decontam_filtered_noyanomami.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_genus.decontam_filtered_noyanomami.pca.variates <- as.data.frame(malt_genus.decontam_filtered_noyanomami.pca$variates$X)
malt_genus.decontam_filtered_noyanomami.pca.variates <- malt_genus.decontam_filtered_noyanomami.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_genus.decontam_filtered_noyanomami.pca.varmet <- inner_join(malt_genus.decontam_filtered_noyanomami.pca.variates, metadata, by = "SampleID")

pca_malt_genus_pc1pc2 <- ggplot(malt_genus.decontam_filtered_noyanomami.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_genus.decontam_filtered_noyanomami.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_genus.decontam_filtered_noyanomami.pca$explained_variance[2], 2)), "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_genus_pc1pc2


# ggsave("05-results.backup/pca_malt_genus_pc1pc2.pdf", plot = pca_malt_genus_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)


```

Run a permanova with function adonis to test if the clusters are significantly different.
```{r adonis_noblanks_noyanomami_genus}
# remove the Yanomami samples from the input table
malt_genus.decontam_filtered_noblanks_noyanomami <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|10M|SRR164")) %>%
                                         select(-one_of(posterior)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         mutate(Counts = Counts + 1) %>%
                                         spread(Species,Counts) %>%
                                         column_to_rownames("SampleID")

# perform PCA
malt_genus.decontam_filtered_noblanks_noyanomami.pca <- mixOmics::pca(malt_genus.decontam_filtered_noblanks_noyanomami, ncomp = 3, logratio = 'CLR')

# use the CLR-transformed matrix from the mixOmics PCA above
# Euclidean distance
malt_genus.decontam_filtered_noblanks_noyanomami.euc <- vegdist(malt_genus.decontam_filtered_noblanks_noyanomami.pca$X, method = "euclidean", na.rm = T)
malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoa <- ape::pcoa(malt_genus.decontam_filtered_noblanks_noyanomami.euc)

malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoavectors <- as.data.frame(malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoa$vectors)
malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoavectors <- malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_genus.decontam_filtered_noblanks_noyanomami.euc.vecmet <- inner_join(malt_genus.decontam_filtered_noblanks_noyanomami.euc.pcoavectors, metadata, by = "SampleID")

malt_genus.decontam_filtered_noblanks_noyanomami.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Industry, shape = Industry)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Industry_colors) +
  scale_shape_manual(values = Industry_shapes) +
  stat_ellipse(aes(colour = Industry, group = Industry)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis2(malt_genus.decontam_filtered_noblanks_noyanomami.euc ~ ExtractionBatch + Village + Sex + Industry + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, metadata_noblanks, permutations = 999, by = "margin")

```


Then run betadisper to see if the groups have different dispersion ("tight" vs.
"loose" clustering of groups) - is one group more variable than the other(s)?
```{r betadisper_noblanks_noyanomami_genus, eval = F}
# Test the metadata categories that are significant (p < 0.05) based on adonis2 for significant differences in dispersion
# When including only anterior samples, none are significantly different


# Industry
groups_industry <- metadata_noblanks %>%
  select(Industry) %>% 
  pull()

## Calculate multivariate dispersions
industry_betadisper_noblanks_noyanomami_genus <- betadisper(malt_genus.decontam_filtered_noblanks_noyanomami.euc, groups_industry, bias.adjust =TRUE)
industry_betadisper_noblanks_noyanomami_genus

## Perform test
anova(industry_betadisper_noblanks_noyanomami_genus)

## Permutation test for F
permutest(industry_betadisper_noblanks_noyanomami_genus, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
industry_betadisper_noblanks_noyanomami_genus.HSD <- TukeyHSD(industry_betadisper_noblanks_noyanomami_genus)
plot(industry_betadisper_noblanks_noyanomami_genus.HSD)

## Plot with data ellipses instead of hulls
plot(industry_betadisper_noblanks_noyanomami_genus, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(industry_betadisper_noblanks_noyanomami_genus)


```


Now we will look at the genera that have strongest loadings in PC1, PC2, and
PC3, to see what appears to drive differences between the sample groups. We will
visualize the loadings of these species with biplots. In the first set we
include the Yanomami samples, and in the second set we exclude them. Do the
genera with strongest loadings include the species with strongest loadings in
sections `biplots_samples_species` and `biplots_samples_species_no_yanomami`
above?
```{r biplot_noblanks_genus, eval = F}

malt_genus.decontam_filtered_biplot_121 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_genus.decontam_filtered_biplot_121

pandoc.table(malt_genus.decontam_filtered_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")



malt_genus.decontam_filtered_biplot_122 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_genus.decontam_filtered_biplot_122

pandoc.table(malt_genus.decontam_filtered_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_genus.decontam_filtered_biplot_322 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_genus.decontam_filtered_biplot_322

malt_genus.decontam_filtered_biplot_323 <- plot_pca_bi(malt_genus.decontam_filtered.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_genus.decontam_filtered_biplot_323

pandoc.table(malt_genus.decontam_filtered_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_genus.decontam_filtered_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
_Anterior samples only:_ In PC1, the taxa with the strongest positive loadings
are mostly strict anaerobes, while the taxa with strongest negative loadings are
a mix of aerobes and facultative anaerobes, including *Neisseria* and relatives
*Eikenella*, *Haemophilus*, *Kingella*, and *Aggregatibacter*. In PC2, the taxa
with strongest positive loadings are split, mostly pointing either strongly into
the PC1 negative quadrant (anaerobes/unusual oral) or the PC1 positive quadrant
(anaerobes). In PC3, the taxa with the strongest postive loadings are anaerobes,
but also genera that are less common in the mouth, as are the taxa with
strongest negative loadings. The Yanomami samples again separate from JAE and
CMC along PC3, but separate from HMP on PC2.


```{r biplot_samples_no_yanomami_genus}

# remove the Yanomami samples from the input table
malt_genus.decontam_filtered_noblanks_noyanomami <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|10M|SRR164")) %>%
                                         select(-one_of(posterior)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         mutate(Counts = Counts + 1) %>%
                                         spread(Species,Counts) %>%
                                         column_to_rownames("SampleID")

# perform PCA
malt_genus.decontam_filtered_noblanks_noyanomami.pca <- mixOmics::pca(malt_genus.decontam_filtered_noblanks_noyanomami, ncomp = 3, logratio = 'CLR')

# Now plot biplots, and make tables of the top 10 species in each loading
malt_genus_decontam_filtered_noyanomami_biplot_121 <- plot_pca_bi(malt_genus.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_genus_decontam_filtered_noyanomami_biplot_121

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_genus_decontam_filtered_noyanomami_biplot_122 <- plot_pca_bi(malt_genus.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_genus_decontam_filtered_noyanomami_biplot_122

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_genus_decontam_filtered_noyanomami_biplot_322 <- plot_pca_bi(malt_genus.decontam_filtered_noblanks_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_genus_decontam_filtered_noyanomami_biplot_322

malt_genus_decontam_filtered_noyanomami_biplot_323 <- plot_pca_bi(malt_genus.decontam_filtered_noblanks_noyanomami.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_genus_decontam_filtered_noyanomami_biplot_323

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_genus_decontam_filtered_noyanomami_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
_Anterior samples only:_ In PC1, the taxa with the strongest positive loadings
are mostly anaerobes, and the taxa with the strongest negative loadings are more
aerobes/facultative (*Haemophilus, Kingella, Rothia, Neisseria, Eikenella*).
Most of the arrows in PC1 negative point into the PC2 negative quadrant. In PC2,
the taxa with strongest positive loadings are mostly anaerobes with arrows
directed away from HMP, but *Veillonella* and *Actinobaculum* point towards the
HMP samples. The taxa with strongest PC2 negative loadings are mostly anaerobes.
In PC3, the taxa with the strongest positive loadings are mostly anaerobes,
while the taxa with strongest negative loadings are unusual, rare, or poorly
characterized in the oral biofilm.


## Genus-level analyses with only Cameroon samples
What is the relationship of Cameroon plaque samples to eachother without any
other sample types? Do they cluster by any of the metadata categories?
```{r pca_cmc_genus}
malt_genus.decontam_filtered_cmc <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|SRR|JAE")) %>%
                                         select(-one_of(posterior)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         mutate(Counts = Counts + 1) %>%
                                         spread(Species,Counts) %>%
                                         column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_genus.decontam_filtered_cmc, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# by Sex
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market economy
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Market_economy", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Extraction batch
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Library batch
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC1", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_genus.decontam_filtered_cmc, "PC3", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_genus.decontam_filtered_cmc.pca <- mixOmics::pca(malt_genus.decontam_filtered_cmc, ncomp = 3, logratio = 'CLR')
plot(malt_genus.decontam_filtered_cmc.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_genus.decontam_filtered_cmc.pca.variates <- as.data.frame(malt_genus.decontam_filtered_cmc.pca$variates$X)
malt_genus.decontam_filtered_cmc.pca.variates <- malt_genus.decontam_filtered_cmc.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_genus.decontam_filtered_cmc.pca.varmet <- inner_join(malt_genus.decontam_filtered_cmc.pca.variates, metadata, by = "SampleID")

pca_malt_genus_pc1pc2 <- ggplot(malt_genus.decontam_filtered_cmc.pca.varmet, aes(PC1, PC2, colour = Individual_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_genus.decontam_filtered_cmc.pca$explained_variance[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_genus.decontam_filtered_cmc.pca$explained_variance[2], 2)), "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_genus_pc1pc2


# ggsave("05-results.backup/pca_malt_genus_pc1pc2.pdf", plot = pca_malt_genus_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)
```
There is no obvious clustering of the CMC samples by village, sex, ethnicity,
age group, or market economy for either the posterior samples only or the
anterior samples only. Clustering also doesn't seem to be related to sequencing
depth, extraction batch, or library batch.

Run a permanova with function adonis to test if the clusters are significantly different.
```{r adonis_cmc_genus}
# use the CLR-transformed matrix from the mixOmics PCA in the last section

# Euclidean distance
malt_genus.decontam_filtered_cmc.euc <- vegdist(malt_genus.decontam_filtered_cmc.pca$X, method = "euclidean", na.rm = T)
malt_genus.decontam_filtered_cmc.euc.pcoa <- ape::pcoa(malt_genus.decontam_filtered_cmc.euc)

malt_genus.decontam_filtered_cmc.euc.pcoavectors <- as.data.frame(malt_genus.decontam_filtered_cmc.euc.pcoa$vectors)
malt_genus.decontam_filtered_cmc.euc.pcoavectors <- malt_genus.decontam_filtered_cmc.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_genus.decontam_filtered_cmc.euc.vecmet <- inner_join(malt_genus.decontam_filtered_cmc.euc.pcoavectors, metadata, by = "SampleID")

malt_genus.decontam_filtered_cmc.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Sex, shape = Sex)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Sex_colors) +
  scale_shape_manual(values = Sex_shapes) +
  stat_ellipse(aes(colour = Sex, group = Sex)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis(malt_genus.decontam_filtered_cmc.euc ~ Ethnic_Group + Age_group + Sex + Village + Market_economy, metadata_cmc, perm=999)

adonis2(malt_genus.decontam_filtered_cmc.euc ~ ExtractionBatch + Village + Sex + Ethnic_Group + Age_group +  LibraryBatch + Market_economy, metadata_cmc, permutations = 999, by = "margin")

```

Then run betadisper to see if the groups have different dispersion ("tight" vs.
"loose" clustering of groups) - is one group more variable than the other(s)?
```{r betadisper_cmc_genus}
# Test the 1 metadata category that is significant (p < 0.05) based on adonis2 for significant differences in dispersion
# When including only anterior samples, Sex is significantly different

# Sex
groups_sex <- metadata_cmc %>%
  select(Sex) %>% 
  pull()

## Calculate multivariate dispersions
sex_betadisper_cmc_genus <- betadisper(malt_genus.decontam_filtered_cmc.euc, groups_sex, bias.adjust =TRUE)
sex_betadisper_cmc_genus

## Perform test
anova(sex_betadisper_cmc_genus)

## Permutation test for F
permutest(sex_betadisper_cmc_genus, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
sex_betadisper_cmc_genus.HSD <- TukeyHSD(sex_betadisper_cmc_genus)
plot(sex_betadisper_cmc_genus.HSD)

## Plot with data ellipses instead of hulls
plot(sex_betadisper_cmc_genus, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(sex_betadisper_cmc_genus)



```
Point dispersion from the centroids is not significantly different between Males and Females. 

```{r biplots_cmc_genus}

malt_genus.decontam_filtered_nocontrols <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|JAE|SRR")) %>%
                                         select(-one_of(posterior)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus") %>%
                                         mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         mutate(Counts = Counts + 1) %>%
                                         spread(Species,Counts) %>%
                                         column_to_rownames("SampleID")

malt_genus.decontam_filtered_nocontrols.pca <- mixOmics::pca(malt_genus.decontam_filtered_nocontrols, ncomp = 3, logratio = 'CLR')
plot(malt_genus.decontam_filtered_nocontrols.pca)

malt_genus_decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_genus_decontam_filtered_nocontrols_biplot_121

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")



malt_genus_decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_genus_decontam_filtered_nocontrols_biplot_122

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_genus_decontam_filtered_nocontrols_biplot_322 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_genus_decontam_filtered_nocontrols_biplot_322

malt_genus_decontam_filtered_nocontrols_biplot_323 <- plot_pca_bi(malt_genus.decontam_filtered_nocontrols.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_genus_decontam_filtered_nocontrols_biplot_323

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_genus_decontam_filtered_nocontrols_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
_Anterior samples only:_ In PC1, the positive loading taxa are more unusual or
poorly characterized in the oral biofilm, and anaerobes, while the negative
loading taxa are a mix of anaerobes and aerobes. In PC2, the taxa with strongest
positive loadings are more anerobes (*Haemophilus, Kingella, Rothia, Neisseria,
Eikenella*), while the taxa with strongest negative loadings are more anaerobes.
In PC3, the taxa with strongest positive loadings are these unusual taxa again
(*Xylanimonas*, *Sanguibacter*, *Cellulomonas*, *Kytococus*) but also some
expected oral taxa, while the taxa with the strongest negative loadings are a
variety of anaerobes and other unusual oral taxa.


```{r uky_graphs, eval = F}

split_all_village <- plot_pca_titles(malt_species.decontam_filtered_noyanomami, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backup/split_all_village.pdf", plot = split_all_village, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

split_cmc_village <- plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backup/split_cmc_village.pdf", plot = split_cmc_village, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

```




