---
title: "Cameroon hunter-gatherer calculus/plaque SEED decontam"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(decontam)
library(data.table)
library(mixOmics)
library(janitor)
library(tidyverse)
library(gplots)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```


# SEED category assignment stats
```{r load_data}
## load the species and genus tables generated with HUMAnN2
seed_full <- fread("./05-results.backup/aadder_SEED_counts.txt")
seed_full <- as_tibble(seed_full)

# clean the file names
seed_full <- rename(seed_full, Pathway = `#Datasets`)
colnames(seed_full) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.blastn.sam.gz","", colnames(seed_full))
colnames(seed_full) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.blastn.sam.gz","", colnames(seed_full))
colnames(seed_full) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M.blastn.sam.gz","_10M", colnames(seed_full))
colnames(seed_full) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam.blastn.sam.gz","", colnames(seed_full))
colnames(seed_full) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam.blastn.sam.gz","_10M", colnames(seed_full))

# remove unmapped and ungrouped reads
# humann2_KOs <- humann2_KOs_full %>% filter(Ortholog != "UNMAPPED") %>% filter(!str_detect(Ortholog, "UNGROUPED"))

```

```{r load_metadata}

load("./05-results.backup/CMC_metadata.RData")

```

```{r pca_functions}
# add functions for:
# making the input table for PCA?

# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, color_group, shape_group, size_group) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))
    metadata_group_size <- get(paste(size_group, "_size", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]
    size_group = df_X[[size_group]]
                      
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group, size = size_group)) +
                        geom_point() +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        scale_size_manual(values = metadata_group_size) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$explained_variance * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                      # select the correct PC column and explained variance for PC1
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$SampleID), size = 2) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 10) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}

```


```{r pca_paths_with_controls}
# create the input matrix of only the protein-level assignments
# start with the full table and pull out only the level 4 entries
seed_full_L4 <- seed_full[str_count(seed_full$Pathway, ";") == 4, ]

seed_full_L4 <- seed_full_L4 %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  gather("SampleID","Counts", 2:ncol(.)) %>%
  spread(Pathway, Counts)

rownames(seed_full_L4) <- seed_full_L4$SampleID
seed_full_L4 <- as.matrix(seed_full_L4 %>% 
  select(-SampleID)) + 1 # add +1 to all values to be able to do a CLR transformation before running PCA

# check the number of components to retain by tuning the PCA
tune.seed_full_L4_pca <- tune.pca(seed_full_L4, logratio = 'CLR')
# perform a PCA to see how the data cluster
seed_full_L4.pca <- pca(seed_full_L4, ncomp = 3, logratio = 'CLR')

# plot the PCA
pca_test_plot <- plot_pca(seed_full_L4.pca, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group")
pca_test_plot

pca_test_plot_names <- plot_pca_names(seed_full_L4.pca, "PC1", "PC2", "Ethnic_Group")
pca_test_plot_names

```
None of the samples appear to be outliers. The extraction gauze plots with the
samples, but is very likely contaminated with spit. This is what we saw in the
taxonomy


# Decontamination with Decontam
We want to know if there are any species that can be removed b/c they appear to be contaminants. Use the package decontam to determine which species may be contaminants. Prepare the matrix for decontam. 
```{r decontam_data_prep}
# list out the samples that have all 0 or NA values to remove from the input matrix
NA_filter_list <- metadata %>% filter((is.na(SumLibraryConcentration_copies_ul) | SumLibraryConcentration_copies_ul <= 0)) %>% select(SampleID)

# remove NA value samples from the input data frame
seed_full_L4 <- seed_full[str_count(seed_full$Pathway, ";") == 4, ]

seed_full_L4 <- seed_full_L4 %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  gather("SampleID","Counts", 2:ncol(.)) %>%
  spread(Pathway, Counts) %>%
  anti_join(., NA_filter_list)

# make a matrix for decontam
seed_full_L4.matrix <- data.matrix(seed_full_L4[2:ncol(seed_full_L4)])
rownames(seed_full_L4.matrix) <- seed_full_L4$SampleID

# make a metadata dataframe with the same NA-containing samples removed
metadata.decontam <- metadata %>%
  anti_join(., NA_filter_list, by = "SampleID") %>%
  filter(SampleID != "CMC032.B0101")

```

Make a protein abundance curve for the samples based on Env to get an idea of
the number of proteins identified in each prior to filtering, and how long the
tail is.
```{r rank_abundance_curves}

seed_full_L4L <- seed_full_L4 %>%
  adorn_percentages(denominator = "row", na.rm=F) %>%
  filter(!str_detect(SampleID, "SRR|JAE")) %>%
  gather("Pathway","Relab",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type))

Envmt <- unique(seed_full_L4L$Description)

pathway_orders <- lapply(Envmt, function(envmt) {
malt_species.dataframeL_check <- seed_full_L4L %>%
  filter(Description == envmt) %>%
  group_by(Pathway) %>%
  summarise(mean_relab = mean(Relab)) %>%
  arrange(desc(mean_relab)) %>%
  filter(mean_relab > 0) %>%
  rownames_to_column("order") %>%
  mutate(order = as.numeric(order)) %>%
  mutate(Description = envmt)
}) %>%
  bind_rows(.)

# define a function to plot the species abundance curves  
pathway_abundance_curves <- function(df, envt, title_text) {
      options(scipen = 10)
      
    species_abundance_plot <- pathway_orders %>%
        filter(Description == envt) %>%
           ggplot(., aes(x=order, y=mean_relab)) +
           geom_point() +
           theme_minimal() +
           scale_y_log10() +
           geom_hline(yintercept=0.0001, linetype="dashed", color = "red") +
           # xlim(NA, 600) +
           # ylim(0.0001, NA) +
           # coord_cartesian(xlim = c(0, 600), ylim = c(0.0001, 1)) +
           ylab("Mean relative abundance") +
           xlab("protein order") +
           ggtitle(title_text)
  
    return(species_abundance_plot)
}

path_curve_HG <- pathway_abundance_curves(species_orders, "HunterGatherer", "Hunter Gatherer - SEED proteins")
path_curve_EG <- pathway_abundance_curves(species_orders, "ExtractionGauze", "Extraction Gauze - SEED proteins")
path_curve_EB <- pathway_abundance_curves(species_orders, "ExtractionBlank", "Extraction Blank - SEED proteins")
path_curve_LB <- pathway_abundance_curves(species_orders, "LibraryBlank", "Library Blank - SEED proteins")

plot_grid(path_curve_HG, path_curve_EG,
          path_curve_EB, path_curve_LB,
          nrow=2, labels = c("A", "B", "C", "D"), rel_widths = c(1, 1))


```



Now we're ready to run Decontam. Start with the frequency method. 
```{r decontam_frequency, eval = F}
# Check for contamination by fequency
contam.freq.species <- isContaminant(seed_full_L4.matrix, method="frequency", 
                                 conc=metadata.decontam$SumLibraryConcentration_copies_ul, 
                                 threshold = 0.9)


# Print the # of species identified as contaminants by frequency
table(contam.freq.species$contaminant)

cont_freq_species_list <- contam.freq.species[which(contam.freq.species$contaminant=="TRUE"), ]

# Make a histogram to look at the distribution of the p-value (probability for classifying contaminants)

ggplot(contam.freq.species, aes(x=p)) +
  geom_histogram(bins=100) +
  scale_y_log10() +
  geom_hline(yintercept=10, linetype="dashed", color = "red") +
  geom_hline(yintercept=2, linetype="dashed", color = "red") +
  geom_vline(xintercept=0.9, linetype="dashed", color = "red") +
  ggtitle('Frequency - SEED proteins')

```

Try the prevalence method to compare. 
```{r decontam_prevalence}
# Assign sample-type status to identify control samples
metadata.decontam$is.control <- metadata.decontam$Sample_or_Control=="Control"

#Try changing the probability threshold cut-off to see how many more or less are ID'd as contaminants
contam.prev.species <- isContaminant(seed_full_L4.matrix, method="prevalence", 
                                 neg=metadata.decontam$is.control, 
                                 threshold = 0.25)

# Check number of contaminants
table(contam.prev.species$contaminant)

# List of contaminants
cont_prev_species_list <- contam.prev.species[which(contam.prev.species$contaminant=="TRUE"), ]

# Look at the histogram - does the probability threshold need to be changed?
ggplot(contam.prev.species, aes(x=p)) +
  geom_histogram(bins=100) +
  scale_y_log10() +
  geom_hline(yintercept=10, linetype="dashed", color = "red") +
  geom_hline(yintercept=2, linetype="dashed", color = "red") +
  geom_vline(xintercept=0.25, linetype="dashed", color = "red") +
  ggtitle('Prevalence - SEED proteins')

```


Check the combined prevalence and frequency method to see how it differs from the individual methods. 
```{r decontam_combined, eval = F}
# Assign sample-type status to identify control samples (same as was done for prevalence)
metadata.decontam$is.control <- metadata.decontam$Sample_or_Control=="Control"

# Assign contaminant status
contam.comb.species <- isContaminant(seed_full_L4.matrix, method="combined", 
                                 conc=metadata.decontam$SumLibraryConcentration_copies_ul, 
                                 neg=metadata.decontam$is.control,
                                 threshold = 0.9)

## Check how many contaminants
table(contam.comb.species$contaminant)

## List of contaminants
cont_comb_species_list <- contam.comb.species[which(contam.comb.species$contaminant=="TRUE"), ]

# Graph it to look at the histogram and pick a suitable probability-values cut-off
ggplot(contam.comb.species, aes(x=p)) +
  geom_histogram(bins=100) +
  scale_y_log10() +
  geom_hline(yintercept=10, linetype="dashed", color = "red") +
  geom_hline(yintercept=2, linetype="dashed", color = "red") +
  geom_vline(xintercept=0.9, linetype="dashed", color = "red") +
  ggtitle('Combined - Species')

```

Check the rank abundance curves to see how changing the cut-off threshold
changes the tail. This set of plots removes the proteins determined to be
"contaminants" by decontam, using the list created from which ever method is
selected.
```{r decontam_rank_abund_curve}

contaminants_seed <- cont_prev_species_list %>% rownames_to_column("Pathway") %>% select(Pathway) # this uses the prevalence method contaminant list
seed_full_L4.decontam <- seed_full[str_count(seed_full$Pathway, ";") == 4, ]
seed_full_L4.decontam <- seed_full_L4.decontam %>%
  select(-CMC032.B0101) %>%  # this sample has very few reads
  anti_join(., contaminants_seed)

# write.table(malt_species.decontam, file = "05-results.backup/malt_species.decontam.tsv", sep="\t", quote=F, row.names = F)
# write.table(malt_species, "05-results.backup/malt_species.tsv", sep="\t", quote=F, row.names = F)

seed_full_L4.decontamL <- seed_full_L4.decontam %>%
  adorn_percentages(denominator = "col", na.rm=F) %>%
  gather("SampleID","Relab",2:ncol(.)) %>%
  inner_join(., metadata %>%
              select(SampleID, Env, Description), by = "SampleID")

Envmt_decontam <- unique(seed_full_L4.decontamL$Description)

seed_decontam_orders <- lapply(Envmt_decontam, function(envmt) {
seed_full_L4.decontamL_check <- seed_full_L4.decontamL %>%
  filter(Description == envmt) %>%
  group_by(Pathway) %>%
  summarise(mean_relab = mean(Relab)) %>%
  arrange(desc(mean_relab)) %>%
  filter(mean_relab > 0) %>%
  rownames_to_column("order") %>%
  mutate(order = as.numeric(order)) %>%
  mutate(Description = envmt)
}) %>%
  bind_rows(.)
  
seed_decontam_abundance_curves <- function(df, envt, title_text) {
      options(scipen = 10)
      
    path_abundance_plot <- seed_decontam_orders %>%
        filter(Description == envt) %>%
           ggplot(., aes(x=order, y=mean_relab)) +
           geom_point() +
           theme_minimal() +
           scale_y_log10() +
           geom_hline(yintercept=0.0001, linetype="dashed", color = "red") +
           ylab("mean relative abundance") +
           xlab("protein order") +
           ggtitle(title_text) + 
           theme(plot.title = element_text(size = 8))
  
    return(path_abundance_plot)
}

seed_decontam_curve_HG <- seed_decontam_abundance_curves(seed_decontam_orders, "HunterGatherer", "HG, decontam")
seed_decontam_curve_EG <- seed_decontam_abundance_curves(seed_decontam_orders, "ExtractionGauze", "EG, decontam")
seed_decontam_curve_EB <- seed_decontam_abundance_curves(seed_decontam_orders, "ExtractionBlank", "EB, decontam")
seed_decontam_curve_LB <- seed_decontam_abundance_curves(seed_decontam_orders, "LibraryBlank", "LB, decontam")
seed_decontam_curve_HMPsup <- seed_decontam_abundance_curves(seed_decontam_orders, "HMP_supPlaque", "HMPsup, decontam")
seed_decontam_curve_HMPsub <- seed_decontam_abundance_curves(seed_decontam_orders, "HMP_subPlaque", "HMPsub, decontam")
seed_decontam_curve_HMP10sup <- seed_decontam_abundance_curves(seed_decontam_orders, "HMP10M_supPlaque", "HMP10Msup, decontam")
seed_decontam_curve_HMP10sub <- seed_decontam_abundance_curves(seed_decontam_orders, "HMP10M_subPlaque", "HMP10Msub, decontam")
seed_decontam_curve_JAE <- seed_decontam_abundance_curves(seed_decontam_orders, "JAE", "JAE, decontam")
seed_decontam_curve_JAE10 <- seed_decontam_abundance_curves(seed_decontam_orders, "JAE10M", "JAE10M, decontam")

plot_grid(seed_decontam_curve_HG, seed_decontam_curve_EG, seed_decontam_curve_EB, seed_decontam_curve_LB, seed_decontam_curve_HMPsup,
          seed_decontam_curve_HMPsub, seed_decontam_curve_HMP10sup, seed_decontam_curve_HMP10sup, seed_decontam_curve_JAE, seed_decontam_curve_JAE10,
          nrow=2, labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I", "J"), rel_widths = c(1, 1))

```

We'll use the prevalence method, because that's what we used for the taxonomic
decontam step. But which cut-off to use? Decided 0.25 because I don't know how
much of a difference it makes when I also filter out low-abundance proteins.
Most of the ones that are considered contaminants are filtered out by abundance
cut-offs anyway. How does changing the cut-off affect the tail of low-abundance
taxa? List out the pathways that are considered contaminants, to be removed from
the table prior to all analyses. Then remove them from the table to make the
decontaminated input table for analysis, and save as an RData object.
```{r contaminants}
contaminants_seed <- cont_prev_species_list %>% rownames_to_column("Pathway") %>% select(Pathway)
seed_L4.decontam <- seed_full[str_count(seed_full$Pathway, ";") == 4, ]
seed_L4.decontam <- seed_L4.decontam %>%
  anti_join(., contaminants_seed)

save(seed_full, seed_L4.decontam,
     file = "./05-results.backup/CMC_seed_tables.RData")

# write.table(malt_species.decontam, file = "05-results.backup/malt_species.decontam.tsv", sep="\t", quote=F, row.names = F)
# write.table(malt_species, "05-results.backup/malt_species.tsv", sep="\t", quote=F, row.names = F)
# write.table(contaminants_species, "05-results.backup/malt_contaminants_species.tsv", sep="\t", quote=F, row.names = F)

```
