---
title: "Cameroon hunter-gatherer calculus/plaque MALT stats"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(vegan)
library(data.table)
library(tidyverse)
library(gplots)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

# MALT read stats
The number of reads assigned in each sample appears very low (~50%). We want to check this so we know how many were assinged in each sample. We used the cleaned and merged reads that didn't align to the human genome (hg19). 
```{r load_data}
# load the table generated from the MALT log file with the # reads and # aligned reads
maltnt_assignments <- fread("./00-documentation.backup/CMC_nt_aligned_stats.tsv")
maltRSC_assignments <- fread("./00-documentation.backup/CMC_RSC_aligned_stats.tsv")
malt_CMC_HMP_RSC_assignments <- fread("./00-documentation.backup/CMC_HMP_RSC_aligned_stats.tsv")
malt_CMC_HMP_JAE_RSC_assignments <- fread("./00-documentation.backup/CMC_HMP_JAE_RSC_aligned_stats.tsv")
malt_CMC_HMP_JAE_Yano_RSC_assignments <- fread("./00-documentation.backup/CMC_HMP_JAE_Yano_RSC_aligned_stats.tsv")

# clean the file names
maltnt_assignments$SampleID <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.rma6","", maltnt_assignments$SampleID)
maltRSC_assignments$SampleID <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.rma6","", maltRSC_assignments$SampleID)
malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.rma6","", malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID)
malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam.rma6","", malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID)
malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam.rma6","_10M", malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID)
malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.rma6","", malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID)
malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M.rma6","_10M", malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID)
malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID <- gsub(".fastq.truncated.rma6","", malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID)

maltnt_assignments$SampleID <- gsub("../04-analysis/malt/nt/","", maltnt_assignments$SampleID)
maltRSC_assignments$SampleID <- gsub("../04-analysis/malt/RefSeqCustom/","", maltRSC_assignments$SampleID)
malt_CMC_HMP_RSC_assignments$SampleID <- gsub("../04-analysis/malt/RefSeqCustom/","", malt_CMC_HMP_RSC_assignments$SampleID)
malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID <- gsub("../04-analysis/malt/RefSeqCustom/","", malt_CMC_HMP_JAE_Yano_RSC_assignments$SampleID)

# load the metadata file
load("./05-results.backup/CMC_metadata.RData")

```


```{r outliers}
# these sampls are outliers based on PCA of the taxonomic profile in Cameroon_h-g_taxonomy.Rmd
outliers <- c("CMC002.B0201","CMC032.B0101","CMC037.B0201","CMC047.A0201","EXB059.A1901","EXB059.A1301","EXB059.A1101","EXB059.A0901")

```


```{r percents}
maltnt_assignments <- maltnt_assignments %>%
  filter(!str_detect(SampleID, outliers %>% str_c(collapse = "|"))) %>%
  mutate(Percent = Aligned/Queries*100)

maltRSC_assignments <- maltRSC_assignments %>%
  filter(!str_detect(SampleID, outliers %>% str_c(collapse = "|"))) %>%
  mutate(Percent = Aligned/Queries*100)

malt_CMC_HMP_RSC_assignments <- malt_CMC_HMP_RSC_assignments %>%
  filter(!str_detect(SampleID, outliers %>% str_c(collapse = "|"))) %>%
  mutate(Percent = Aligned/Queries*100)

malt_CMC_HMP_JAE_Yano_RSC_assignments <- malt_CMC_HMP_JAE_Yano_RSC_assignments %>%
  filter(!str_detect(SampleID, outliers %>% str_c(collapse = "|"))) %>%
  mutate(Percent = Aligned/Queries*100)
 
```

```{r graphing_functions}

# plotting box and whisker
plot_box_whisker <- function(df, hasmtdta = c("TRUE","FALSE"), stat_info, metadata_group, fill_group, title_text) {
    fill_group_colors <- get(paste(fill_group, "_colors", sep = ""))

    if (hasmtdta == T) { 
      df_X <- df
      fill_group = df_X[[fill_group]]
    } else {
      df_X <- df %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group,
                                  Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")
      fill_group = df_X[[metadata_group]]
    }
   
    stat_info_table = df_X[[stat_info]]
    metadata_group = df_X[[metadata_group]]

    bx_wh_plot <- ggplot(df_X, aes(x = metadata_group, y = stat_info_table, fill = fill_group)) +
                         geom_boxplot() +
                         scale_fill_manual(values = fill_group_colors) +
                         theme_minimal() +
                         #scale_y_continuous(trans='log10') +
                         #scale_y_continuous(trans='pseudo_log') +
                         theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
                         ylab(stat_info) +
                         ggtitle(title_text)

    return(bx_wh_plot)
}

```


```{r sample_counts}

metadata %>%
  group_by(Ethnic_Group, Tooth_site) %>%
  count()

```


```{r graphs_nt}

plot_box_whisker(maltnt_assignments, hasmtdta = F, "Percent", "Env", "Env", "Percent of reads assigned by MALT with nt2017")
plot_box_whisker(maltnt_assignments, hasmtdta = F, "Percent", "Village", "Village", "Percent of reads assigned by MALT with nt2017")
plot_box_whisker(maltnt_assignments, hasmtdta = F, "Queries", "Env", "Env", "Number of reads assigned by MALT with nt2017")
plot_box_whisker(maltnt_assignments, hasmtdta = F, "Queries", "Village", "Village", "Number of reads assigned by MALT with nt2017")

```

```{r graphs_RSC}

plot_box_whisker(malt_CMC_HMP_JAE_Yano_RSC_assignments, hasmtdta = F, "Queries", "Env", "Env", "Number of reads submitted to MALT for classification")
plot_box_whisker(malt_CMC_HMP_JAE_Yano_RSC_assignments, hasmtdta = F, "Queries", "Village", "Village", "Number of reads submitted to MALT for classification")
plot_box_whisker(malt_CMC_HMP_JAE_Yano_RSC_assignments, hasmtdta = F, "Aligned", "Env", "Env", "Number of reads assigned by MALT with RefSeqCustom")
plot_box_whisker(malt_CMC_HMP_JAE_Yano_RSC_assignments, hasmtdta = F, "Aligned", "Village", "Village", "Number of reads assigned by MALT with RefSeqCustom")
plot_box_whisker(malt_CMC_HMP_JAE_Yano_RSC_assignments, hasmtdta = F, "Percent", "Env", "Env", "Percent of reads assigned by MALT with RefSeqCustom")
plot_box_whisker(malt_CMC_HMP_JAE_Yano_RSC_assignments, hasmtdta = F, "Percent", "Description", "Description", "Percent of reads assigned by MALT with RefSeqCustom")
plot_box_whisker(malt_CMC_HMP_JAE_Yano_RSC_assignments, hasmtdta = F, "Percent", "Village", "Village", "Percent of reads assigned by MALT with RefSeqCustom")

```


Now combine the data to see how much of a difference there is the % assignments between databases
```{r compare}

malt_stats_new <- as_tibble(bind_rows(maltnt_assignments %>%
                          mutate(Database = 'nt') %>%
                          inner_join(., metadata %>%
                                       select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID"),
                        malt_CMC_HMP_JAE_Yano_RSC_assignments %>%
                          mutate(Database = 'RefSeqCustom') %>%
                          inner_join(., metadata %>%
                                       select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID"))) %>%
  mutate(Database = fct_relevel(Database, "nt", "RefSeqCustom"))

db_plot_env <- plot_box_whisker(malt_stats_new, hasmtdta = T, "Percent", "Env", "Database", "Percent of reads assigned by MALT")
db_plot_env

db_plot_description <- plot_box_whisker(malt_stats_new, hasmtdta = T, "Percent", "Description", "Database", "Percent of reads assigned by MALT")
db_plot_description

db_plot_village <- plot_box_whisker(malt_stats_new, hasmtdta = T, "Percent", "Village", "Database", "Percent of reads assigned by MALT")
db_plot_village

# ggsave("05-results.backup/malt_assignments_plots_pct_village.pdf", plot = malt_assignments_plots_pct_village, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)

```


```{r uky_graphs, eval = F}

malt_CMC_HMP_JAE_Yano_RSC_assignments_uky <- malt_CMC_HMP_JAE_Yano_RSC_assignments %>%
  filter(!str_detect(SampleID, outliers %>% str_c(collapse = "|"))) %>%
  mutate(Percent = Aligned/Queries*100) %>%
  filter(!str_detect(SampleID, "10M|EXB|LIB"))
  
assigned <- plot_box_whisker(malt_CMC_HMP_JAE_Yano_RSC_assignments_uky, hasmtdta = F, "Percent", "Village", "Village", "Percent of reads assigned by MALT with RefSeqCustom")
assigned

ggsave("05-results.backup/malt_assignments_plots_pct_uky.pdf", plot = assigned, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

```


