---
title: "Cameroon hunter-gatherer calculus/plaque MALT differentially abundant taxa"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---

```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(janitor)
library(pander)
library(exactRankTests)
library(nlme)
library(compositions)
library(tidyverse)
library(gplots)
library(viridis)
library(cowplot)
# source("./02-scripts.backup/ancom.R")
source("./02-scripts.backup/ancom_v2.1.R") # source the ancom2 script to be able to use it here
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

```{r load_data}

# load the taxonomy tables that have been decontaminated and the column headers are cleaned up
load("./05-results.backup/CMC_taxonomy_tables.RData")

# load the metadata file
load("./05-results.backup/CMC_metadata.RData") # this sets the metadata as factors. Check if it works before deleting the commented out section below

metadata_factors <- metadata %>%
  rename(Sample.ID = SampleID)  %>%
  as_data_frame() %>%
  select(`Sample.ID`, Description, Industry, Age_group, Env, Ethnic_Group, Market_economy, Sex, Tooth_site, Village, ExtractionBatch, LibraryBatch)

```


List the outliers from the plot in section `raw_plot` in `Cameroon_h-g_taxonomy.Rmd` and remove them from the taxonomy table. 
```{r outliers}
outliers <- c("CMC002.B0201","CMC032.B0101","CMC037.B0201","CMC047.A0201",
              "EXB059.A1901","EXB059.A1301","EXB059.A1101","EXB059.A0901",
              "SRR1646041","SRR1646032","SRR1646042","SRR1646045","SRR1646030","SRR1646047")

outliersF <- str_c(outliers, collapse = "|")

```


```{r input_tables}
# new metadata table without factors, in case that's messing this up (Baka isn't appearing)
metadata <- fread("./00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata <- metadata %>%
  rename(SampleID = `#SampleID`) %>%
  filter(!SampleID %in% outliers,
         !str_detect(SampleID, "EXB|LIB|10M|SRR164")) %>%
  rename(Sample.ID = SampleID) %>%
  as_data_frame() %>%
  select(`Sample.ID`, Description, Industry, Age_group, Env, Ethnic_Group, Market_economy, Sex, Tooth_site, Village, ExtractionBatch, LibraryBatch)


# species, no blanks, no Yanomami
malt_species.decontam_filtered_noblanks_noyanomami <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|10M|SRR164")) 

# species, CMC samples only
malt_species.decontam_filtered_cmc <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         select(-matches("EXB|LIB|JAE|SRR")) 


# malt_species.decontam_filtered_cmc <- malt_species.decontam %>%
#                                          select(-one_of(outliers)) %>%
#                                          select(-starts_with("EXB")) %>%
#                                          select(-starts_with("LIB")) %>%
#                                          select(-starts_with("SRR")) %>%
#                                          select(-starts_with("JAE")) %>%
#                                          select(-contains("10M")) %>%
#                                          adorn_totals(where = "col", name = "Sum_species") %>%
#                                          mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
#                                          filter(Percent > 0.001) %>%
#                                          select(-one_of(c("Sum_species","Percent"))) %>%
#                                          gather("SampleID","Counts",2:ncol(.)) %>%
#                                          spread(Species,Counts)
# 
# rownames(malt_species.decontam_filtered_cmc) <- malt_species.decontam_filtered_cmc$SampleID
# malt_species.decontam_filtered_cmc <- malt_species.decontam_filtered_cmc %>% select(-SampleID)
# malt_species.decontam_filtered_cmc = malt_species.decontam_filtered_cmc+1


# genus, no blanks, no Yanomami
# malt_genus.decontam_filtered_noyanomami <- malt_genus.decontam %>%
#                                          select(-one_of(outliers)) %>%
#                                          select(-starts_with("EXB")) %>%
#                                          select(-starts_with("LIB")) %>%
#                                          select(-contains("10M")) %>%
#                                          # select(-one_of(fullSRR)) %>%
#                                          adorn_totals(where = "col", name = "Sum_genus") %>%
#                                          mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
#                                          filter(Percent > 0.01) %>%
#                                          select(-one_of(c("Sum_genus","Percent"))) %>%
#                                          gather("SampleID","Counts",2:ncol(.)) %>%
#                                          spread(Species,Counts)
# 
# rownames(malt_genus.decontam_filtered_noyanomami) <- malt_genus.decontam_filtered_noyanomami$SampleID
# malt_genus.decontam_filtered_noyanomami <- malt_genus.decontam_filtered_noyanomami %>% select(-SampleID)
# malt_genus.decontam_filtered_noyanomami = malt_genus.decontam_filtered_noyanomami+1


# genus, CMC samples only
# malt_genus.decontam_filtered_cmc <- malt_genus.decontam %>%
#                                          select(-one_of(outliers)) %>%
#                                          select(-starts_with("EXB")) %>%
#                                          select(-starts_with("LIB")) %>%
#                                          select(-starts_with("SRR")) %>%
#                                          select(-starts_with("JAE")) %>%
#                                          adorn_totals(where = "col", name = "Sum_genus") %>%
#                                          mutate(Percent = Sum_genus/sum(Sum_genus)*100) %>%
#                                          filter(Percent > 0.001) %>%
#                                          select(-one_of(c("Sum_genus","Percent"))) %>%
#                                          gather("SampleID","Counts",2:ncol(.)) %>%
#                                          spread(Species,Counts)
# 
# rownames(malt_genus.decontam_filtered_cmc) <- malt_genus.decontam_filtered_cmc$SampleID
# malt_genus.decontam_filtered_cmc <- malt_genus.decontam_filtered_cmc %>% select(-SampleID)
# malt_genus.decontam_filtered_cmc = malt_genus.decontam_filtered_cmc+1

```

#Ancom2
```{r ancom2_example, eval = F}
otu_data <- malt_species.decontam_filtered_cmc
otu_id = otu_data$Species
otu_data <- otu_data %>%
  column_to_rownames("Species")

meta_data <- metadata

# source("scripts/ancom_v2.1.R")

# Step 1: Data preprocessing

feature_table = otu_data
sample_var = "Sample.ID"
group_var = NULL
out_cut = 0.05
zero_cut = 0.90
lib_cut = 0
neg_lb = FALSE
prepro = feature_table_pre_process(feature_table, meta_data, sample_var, group_var, 
                                   out_cut, zero_cut, lib_cut, neg_lb)
feature_table <-  prepro$feature_table # Preprocessed feature table
meta_data <- prepro$meta_data # Preprocessed metadata
struc_zero <- prepro$structure_zeros # Structural zero info

# Step 2: ANCOM

main_var = "Tooth_site" # for the CMC only table and tooth site, this takes 1hr11min
adj_formula = NULL # this controls for tooth site, otherwise use NULL
p_adj_method = "fdr"
alpha = 0.05
rand_formula = NULL
t_start = Sys.time()
res = ANCOM(feature_table, meta_data, struc_zero, main_var, p_adj_method, 
            alpha, adj_formula, rand_formula)
t_end = Sys.time()

head(res$out)
# write_csv(res$out, "outputs/res_moving_pics.csv")

# Step 3: Volcano Plot

# Number of taxa except structural zeros
n_taxa = ifelse(is.null(struc_zero), nrow(feature_table), sum(apply(struc_zero, 1, sum) == 0))
# Cutoff values for declaring differentially abundant taxa
cut_off = c(0.9 * (n_taxa -1), 0.8 * (n_taxa -1), 0.7 * (n_taxa -1), 0.6 * (n_taxa -1))
names(cut_off) = c("detected_0.9", "detected_0.8", "detected_0.7", "detected_0.6")

# Annotation data
dat_ann = data.frame(x = min(res$fig$data$x), y = cut_off["detected_0.7"], label = "W[0.7]")

fig = res$fig +  
  geom_hline(yintercept = cut_off["detected_0.7"], linetype = "dashed") + 
  geom_text(data = dat_ann, aes(x = x, y = y, label = label), 
            size = 4, vjust = -0.5, hjust = 0, color = "orange", parse = TRUE)
fig

```




## Ancom1

```{r input_tables_ancom, eval = F}
# this works
# this converts the rownames to a column with the necessary heder for ANCOM, removes the rownames, then puts them back in from the column with the appropriate header
malt_species.decontam_filtered_noblanks_test <- malt_species.decontam_filtered_noblanks %>% rownames_to_column("Sample.ID") %>% remove_rownames() %>% column_to_rownames(var = "Sample.ID")

# sets the OTU.name variable as the column headers of the input matrix. Don't need this b/c it puts spaces back and won't run
OTU.name = colnames(malt_species.decontam_filtered_noblanks)

# something about setting the rowname variable to the appropriate format for ANCOM again
malt_species.decontam_filtered_noblanks_test <- data.frame(Sample.ID=rownames(malt_species.decontam_filtered_noblanks_test), as.matrix(malt_species.decontam_filtered_noblanks_test), row.names = NULL)

# # this puts back column names that got distorted by the step above, but they have spaces and won't run
# colnames(malt_species.decontam_filtered_noblanks_clr_test)=c("Sample.ID", OTU.name)

# # unnecessary, makes sure the data is a data.frame (but it already is)
# malt_species.decontam_filtered_noblanks_clr_test=as.data.frame(malt_species.decontam_filtered_noblanks_clr_test)



```

```{r input_tables_tidyverse, eval = F}
# this should do the same as above but using fewer steps - except it keeps the spaces in names, which have to be removed
malt_species.decontam_filtered_noblanks_clr_test <- malt_species.decontam_filtered_noblanks_clr %>%
  rownames_to_column("Sample.ID") 


# trying to run this one in the next gives: Error in data.pair[, 1] : incorrect number of dimensions
malt_species.decontam_filtered_noblanks_test <- malt_species.decontam_filtered_noblanks %>%
  rownames_to_column("Sample.ID") %>%
  gather("Species", "Counts", 2:ncol(.)) %>%
  mutate(Species = str_replace_all(Species, " ", "_")) %>%
  mutate(Species = str_replace_all(Species, "\\.", "")) %>%
  mutate(Species = str_replace(Species, "\\[", "")) %>%
  mutate(Species = str_replace(Species, "\\]", "")) %>%
  spread(Species, Counts)


```


```{r run_ancom, eval = F}
malt_species.decontam_filtered_noblanks.ancom <- ANCOM.main(OTUdat = malt_species.decontam_filtered_noblanks_test, Vardat = metadata_new, 
                                                                adjusted = F, repeated = F, main.var = "Village", adj.formula = NULL, repeat.var = "Village", 
                                                                longitudinal = T, random.formula = "~1|Village", multcorr=2, sig = 0.05, prev.cut = 0.90)


```



From Zandra's R notebook:
Prepare data:
```{r zandra_data_prep, eval = F}
metadata_z <- read_excel("~/Dropbox (MPI SHH)/1-Methods_Dental Arcade/3-Results/DA_metadata.xlsx")
colnames(metadata_z)[1] <- "Sample.ID"
metadata_z$Sample.ID <- gsub("01.SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam", "", metadata_z$Sample.ID)

otu_table <- read.delim("~/Dropbox (MPI SHH)/1-Methods_Dental Arcade/3-Results/MALT/malt_refseq_species_decontam_20191203.txt")
colnames(otu_table) <- gsub("01.SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam", "", colnames(otu_table))
# Transpose dataframe
otu_table_t <- otu_table %>%
   gather("Sample.ID", "count", 2:77) %>%
   spread(X.Datasets, count)

# I don't really know what this part is doing, but it's necessary to make the test work
otu_table_t <- otu_table_t %>% remove_rownames %>% column_to_rownames(var="Sample.ID")
OTU.name=colnames(otu_table_t)
otu_table_t=data.frame(Sample.ID=rownames(otu_table_t), 
                         as.matrix(otu_table_t), row.names = NULL)
# colnames(otu_table_t)=c("Sample.ID", OTU.name)
# otu_table_t=as.data.frame(otu_table_t)
```

Running ANCOM with a mixed effects model:
```{r zandra_run_ancom, eval = F}
# Note: this took about 10 minutes to run for species and 3 minutes for genus
res.W=ANCOM.main(OTUdat=otu_table_t, Vardat=metadata_z, adjusted=F, repeated=T,
                 main.var="Weight_class", adj.formula=NULL, repeat.var="Individual", longitudinal=T,
                 random.formula="~1|Individual", multcorr=2, sig=0.05, prev.cut=0.90)

res.ANCOM=res.W$W.taxa

write.table(res.ANCOM, "U:/Dental_arcade/ANCOM/ancom_species_weightclass_mixedeffects_BH_20200113.txt")
```


