---
title: "CMC calculus/plaque HUMAnN3 species contributions"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(janitor)
library(rstatix)
library(vegan)
library(tidyverse)
library(gplots)
library(viridis)
library(cowplot)
library(patchwork)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

```{r load_data}
## load the species and genus tables generated with HUMAnN2
humann2_path <- fread("./05-results.backup/pathabundance_joined_cpm.tsv")
humann2_path <- as_tibble(humann2_path)
humann2_KOs_full <- fread("./05-results.backup/genefamilies_joined_cpm_ur90rxn.tsv.gz")
humann2_KOs_full <- as_tibble(humann2_KOs_full)

# clean the file names
humann2_KOs_full <- rename(humann2_KOs_full, Ortholog = `# Gene Family`)
colnames(humann2_KOs_full) <- gsub(".unmapped_Abundance-RPKs","", colnames(humann2_KOs_full))
colnames(humann2_KOs_full) <- gsub(".SG1","", colnames(humann2_KOs_full))

humann2_path <- rename(humann2_path, Pathway = `# Pathway`)
colnames(humann2_path) <- gsub(".unmapped_Abundance","", colnames(humann2_path))
colnames(humann2_path) <- gsub(".SG1","", colnames(humann2_path))

# remove unmapped and ungrouped reads
humann2_KOs <- humann2_KOs_full %>% filter(Ortholog != "UNMAPPED") %>% filter(!str_detect(Ortholog, "UNGROUPED"))


```


```{r ancomBC_top}
# read in ancom-bc table for CMC vs. Industrial plaque
ancom_path_ip_cmc <- fread("./05-results.backup/ancomBC_h3_noblanks_path_IP_CMC.tsv") %>%
  rename(Pathways = Proteins) %>%
  filter(diff.abn == TRUE) %>%
  filter(`log fold change (Industrial plaque - CMC)` >= 1.0 | `log fold change (Industrial plaque - CMC)` <= -1.0) %>%
  arrange(`log fold change (Industrial plaque - CMC)`) %>%
  mutate(Direction = ifelse(`log fold change (Industrial plaque - CMC)` > 0, "IP","CMC")) 
# %>%
#   select(Pathways, Direction, `log fold change (Industrial plaque - CMC)`)

# read in ancom-bc table for CMC vs. Industrial calculus
ancom_path_ic_cmc <- fread("./05-results.backup/ancomBC_h3_noblanks_path_IC_CMC.tsv") %>%
  filter(diff.abn == TRUE) %>%
  filter(`log fold change (Industrial calculus - CMC)` >= 1.0 | `log fold change (Industrial calculus - CMC)` <= -1.0) %>%
  arrange(`log fold change (Industrial calculus - CMC)`) %>%
  mutate(Direction = ifelse(`log fold change (Industrial calculus - CMC)` > 0, "IC","CMC")) 
  
# read in ancom-bc table for CMC anterior vs. posterior
ancom_path_cmc_ts <- fread("./05-results.backup/ancomBC_h3_paths_cmc_toothsite.tsv") %>%
  filter(diff.abn == TRUE) %>%
  filter(`log fold change (Anterior - Posterior)` >= 1.0 | `log fold change (Anterior - Posterior)` <= -1.0) %>%
  arrange(`log fold change (Anterior - Posterior)`) %>%
  mutate(Direction = ifelse(`log fold change (Anterior - Posterior)` > 0, "Anterior","Posterior"))
  

```


```{r load_metadata}

load("./05-results.backup/CMC_metadata.RData")

```

```{r}
# set colors

species_colors <- c("Other" = "#bdbdbd",
                    "unclassified" = "#737373",
                    "Aggregatibacter" = "#FF3200",
                    "Burkholderia" = "#FF8466",
                    "Campylobacter" = "#BF6746",
                    "Corynebacterium" = "#F5B39A",
                    "Enterobacter" = "#FEB24C",
                    "Haemophilus" = "#FEE0B7",
                    "Lautropia" = "#A2B085",
                    "Neisseria" = "#DFEACA",
                    "Ottowia" = "#2C968A",
                    "Propionibacterium" = "#87D6CD",
                    "Pseudopropionibacterium" = "#C70E7B",
                    "Selenomonas" = "#DD6EAF",
                    "Streptococcus" = "#065FA3",
                    "Tetragenococcus" = "#699FC7")

```


# MetaCyc Pathways
## CMC and Industrial samples ANCOM-BC differentially abundant pathways

## CMC vs IP
```{r path_cmc_ip}
ancom_path_ip_cmc_paths <- ancom_path_ip_cmc %>%
  separate(., Pathways, into = "Pathway", sep = ":", extra = "drop") %>%
  pull(Pathway) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those differentially abundant pathways from the list, and split the column with names into 3 (Pathway, Genus, Species)
humann3_path_ancombc_cmc_ip <- humann2_path %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  filter(str_detect(Path, ancom_path_ip_cmc_paths)) %>%
  select(-Path) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  arrange(Pathway)

# calculate the % for each ortholog contributed by each genus         
humann3_path_ancombc_cmc_ip_stats <- humann3_path_ancombc_cmc_ip %>%
  group_by(Pathway, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.) %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop")

# create the list of 10 orthologs again, but don't collapse the list as above
ancom_path_ip_cmc_paths <- ancom_path_ip_cmc %>%
  separate(., Pathways, into = "Pathway", sep = ":", extra = "drop") %>%
  pull(Pathway)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_cmc_ip_stats_extra <- lapply(ancom_path_ip_cmc_paths, function(eclass) {
 high_percent <- humann3_path_ancombc_cmc_ip_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 5) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_cmc_ip_bar_df <- humann3_path_ancombc_cmc_ip_stats %>%
  select(-c("Sum","Pathway")) %>%
  bind_rows(., humann2_path_cmc_ip_stats_extra %>%
              rename(Percent = Remaining) %>%
              select(Genus, Percent, Path)) %>%
  arrange(Path) %>%
  # unclassified for PWY-6470 got added to "Other" so we can remove "Other" for that path 
  filter(Genus != "Other" | Path != "PWY-6470")

```


```{r, eval = F}

# plot the values in a bar chart
humann2_path_cmc_ip_bar_df %>%
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Streptococcus"),
         Path = fct_relevel(Path, ancom_path_ip_cmc %>%
                              separate(., Pathways, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  ggplot(., aes(x=Path, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF",)) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, vjust = 0.5)) +
    ylab("Percent") +
    ggtitle("Pathways - CMC vs. Industrial plaque") +
    theme(title = element_text(size=10))

```

```{r}
humann2_path_cmc_ip_bar_df_l <- humann2_path_cmc_ip_bar_df %>%
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Streptococcus"),
         Path = fct_relevel(Path, ancom_path_ip_cmc %>%
                              separate(., Pathways, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  full_join(., ancom_path_ip_cmc %>%
              mutate(newcol = Pathways) %>%
              separate(., newcol, into = "Path", sep = ":", extra = "drop")) %>%
  mutate(star_up = ifelse(q.val<0.001 & Direction == "IP", "***", ""),
         star_down = ifelse(q.val<0.001 & Direction == "CMC", "***", ""))

humann2_path_cmc_ip_bar_df_l

```

```{r}
# set colors for the plot

cmcip_colors <- c("unclassified" = "#737373",
                  "Streptococcus" = "#065FA3")

cmcipsp <- humann2_path_cmc_ip_bar_df_l %>%
  mutate(Prop = (Percent / 100) * `log fold change (Industrial plaque - CMC)`) %>%
  ggplot(., aes(x = Path, y = Prop, ymin = ci.lo.adj, ymax = ci.up.adj, fill = Genus)) + 
    # geom_col(position = "stack", color = "black", size = 0.25) +
    geom_bar(position = "stack", stat = "identity", color = "black", size = 0.25) +
    geom_errorbar(width=0.2, size=0.25) +
    geom_abline(slope = 0, intercept = 0, linetype="dashed", alpha=0.5) +
    theme_minimal(base_size = 14) +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, size = 14),
          axis.text.y = element_text(size = 12)) +
    scale_y_continuous(breaks=scales::pretty_breaks(n=6)) +
    scale_fill_manual(values = cmcip_colors) +
    # ylim(-4,4) +
    ylab("Log fold change") +
    xlab("") +
    coord_flip() +
    theme(legend.position="bottom")

cmcipsp

```

```{r}
# now calculate the mean proportion of these pathways in each group
cmcippm <- humann2_path %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  filter(str_detect(Path, ancom_path_ip_cmc_paths %>%
                      str_c(collapse = "|"))) %>%
  select(-Path) %>%
  # mutate(Pathway = str_replace_all(Pathway, "\\|unclassified","")) %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  pivot_longer(!Pathway, names_to = "Library_ID", values_to = "CPM") %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  left_join(., metadata %>%
              select(SampleID, Env, Tooth_site) %>%
              rename(Library_ID = SampleID), by = "Library_ID") %>%
  group_by(Env, Path) %>%
  summarize(Mean_cpm = mean(CPM),
            sd_cpm = sd(CPM)) %>%
  ungroup() %>%
  filter(str_detect(Env, "CMC|plaque")) %>%
  mutate(Path = fct_relevel(Path, ancom_path_ip_cmc %>%
                              separate(., Pathways, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  ggplot(., aes(x = Mean_cpm, y = Path, shape = Env, bg = Env)) + 
    # geom_bar(position = "dodge", stat = "identity", color = "black", size = 0.25) +
    geom_point(position = position_dodge(0.9), size = 3) +
    geom_errorbar(aes(xmax = (Mean_cpm + sd_cpm), xmin = (Mean_cpm - sd_cpm)), width = 0.2, position=position_dodge(0.9), color = "grey20") +
    theme_minimal(base_size = 14) +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, size = 14),
          axis.text.y = element_text(size = 14)) +
    # scale_y_continuous(breaks=scales::pretty_breaks(n=10)) +
    # ylim(-4,4) +
    scale_shape_manual(values = c(21, 21)) +
    ylab("") +
    xlab("Mean CPM") +
    scale_fill_manual(values = c("#FF3200","#969696")) +
    theme(legend.position="bottom",
          axis.text.y = element_blank())
cmcippm

```

```{r}

h3_sp_cmcic_ancombc <- cmcipsp + cmcippm
h3_sp_cmcic_ancombc

# ggsave("./06-publication/main_figures/Figure_XX5/h3_sp_cmcic_ancombc.pdf", plot = h3_sp_cmcic_ancombc,
#        device = "pdf", scale = 1, width = 11, height = 8.5, units = c("in"), dpi = 300)

# ggsave("~/Desktop/h3_species_ancombc_dots.pdf", plot = h3_sp_cmcic_ancombc,
#        device = "pdf", scale = 1, width = 12, height = 6, units = c("in"), dpi = 300)


```

## CMC vs. IC
```{r path_cmc_ic}
ancom_path_ic_cmc_paths <- ancom_path_ic_cmc %>%
  separate(., Pathways, into = "Pathway", sep = ":", extra = "drop") %>%
  pull(Pathway) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those differentially abundant pathways from the list, and split the column with names into 3 (Pathway, Genus, Species)
humann3_path_ancombc_cmc_ic <- humann2_path %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  filter(str_detect(Path, ancom_path_ic_cmc_paths)) %>%
  select(-Path) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  arrange(Pathway)

# calculate the % for each ortholog contributed by each genus         
humann3_path_ancombc_cmc_ic_stats <- humann3_path_ancombc_cmc_ic %>%
  group_by(Pathway, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.) %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop")

# create the list of 10 orthologs again, but don't collapse the list as above
ancom_path_ic_cmc_paths <- ancom_path_ic_cmc %>%
  separate(., Pathways, into = "Pathway", sep = ":", extra = "drop") %>%
  pull(Pathway)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_cmc_ic_stats_extra <- lapply(ancom_path_ic_cmc_paths, function(eclass) {
 high_percent <- humann3_path_ancombc_cmc_ic_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 10) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_cmc_ic_bar_df <- humann3_path_ancombc_cmc_ic_stats %>%
  select(-c("Sum","Pathway")) %>%
  filter(Percent >= 10) %>%
  bind_rows(., humann2_path_cmc_ic_stats_extra %>%
              rename(Percent = Remaining) %>%
              select(Genus, Percent, Path)) %>%
  arrange(Path) 
# %>%
#   # unclassified for PWY-6470 got added to "Other" so we can remove "Other" for that path 
#   filter(Genus != "Other" | Path != "PWY-6470")

```

```{r, eval = F}

# plot the values in a bar chart
humann2_path_cmc_ic_bar_df %>%
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Burkholderia","Corynebacterium","Enterobacter","Haemophilus","Lautropia","Neisseria","Propionibacterium","Pseudopropionibacterium","Streptococcus"),
         Path = fct_relevel(Path, ancom_path_ic_cmc %>%
                              separate(., Pathways, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  ggplot(., aes(x=Path, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF",)) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
    ylab("Percent") +
    ggtitle("Pathways - CMC vs. Industrial plaque") +
    theme(title = element_text(size=10))

```

```{r}
humann2_path_cmc_ic_bar_df_l <- humann2_path_cmc_ic_bar_df %>%
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Burkholderia","Corynebacterium","Enterobacter","Haemophilus","Lautropia","Neisseria","Propionibacterium","Pseudomonas","Pseudopropionibacterium","Streptococcus"),
         Path = fct_relevel(Path, ancom_path_ic_cmc %>%
                              separate(., Pathways, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  full_join(., ancom_path_ic_cmc %>%
              mutate(newcol = Pathways) %>%
              separate(., newcol, into = "Path", sep = ":", extra = "drop")) %>%
  mutate(star_up = ifelse(q.val<0.001 & Direction == "IC", "***", ""),
         star_down = ifelse(q.val<0.001 & Direction == "CMC", "***", ""))

humann2_path_cmc_ic_bar_df_l

```

```{r}
cmcic_colors <- c("Other" = "#bdbdbd",
                    "unclassified" = "#737373",
                    "Burkholderia" = "#FF8466",
                    "Corynebacterium" = "#F5B39A",
                    "Enterobacter" = "#FEB24C",
                    "Haemophilus" = "#FEE0B7",
                    "Lautropia" = "#A2B085",
                    "Neisseria" = "#DFEACA",
                    "Propionibacterium" = "#87D6CD",
                    "Pseudopropionibacterium" = "#C70E7B",
                    "Streptococcus" = "#065FA3")

cmcicsp <- humann2_path_cmc_ic_bar_df_l %>%
  mutate(Prop = (Percent / 100) * `log fold change (Industrial calculus - CMC)`,
         Path = fct_relevel(Path, ancom_path_ic_cmc %>%
                              separate(., Pathways, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  ggplot(., aes(x = Path, y = Prop, ymin = ci.lo.adj, ymax = ci.up.adj, fill = Genus)) + 
    # geom_col(position = "stack", color = "black", size = 0.25) +
    geom_bar(position = "stack", stat = "identity", color = "black", size = 0.25) +
    geom_errorbar(width=0.2, size=0.25) +
    geom_abline(slope = 0, intercept = 0, linetype="dashed", alpha=0.5) +
    theme_minimal(base_size = 14) +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, size = 14),
          axis.text.y = element_text(size = 12)) +
    scale_y_continuous(breaks=scales::pretty_breaks(n=6)) +
    scale_fill_manual(values = cmcic_colors) +
    ylab("Log fold change") +
    xlab("") +
    coord_flip() +
    theme(legend.position="bottom")

cmcicsp

```

```{r}
# now calculate the mean proportion of these pathways in each group
cmcicpm <- humann2_path %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  filter(str_detect(Path, ancom_path_ic_cmc_paths %>%
                      str_c(collapse = "|"))) %>%
  select(-Path) %>%
  # mutate(Pathway = str_replace_all(Pathway, "\\|unclassified","")) %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  pivot_longer(!Pathway, names_to = "Library_ID", values_to = "CPM") %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  left_join(., metadata %>%
              select(SampleID, Env, Tooth_site) %>%
              rename(Library_ID = SampleID), by = "Library_ID") %>%
  group_by(Env, Path) %>%
  summarize(Mean_cpm = mean(CPM),
            sd_cpm = sd(CPM)) %>%
  ungroup() %>%
  filter(str_detect(Env, "CMC|calculus")) %>%
  mutate(Path = fct_relevel(Path, ancom_path_ic_cmc %>%
                              separate(., Pathways, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  ggplot(., aes(x = Mean_cpm, y = Path, shape = Env, bg = Env)) + 
    # geom_bar(position = "dodge", stat = "identity", color = "black", size = 0.25) +
    geom_point(position = position_dodge(0.9), size = 3) +
    geom_errorbar(aes(xmax = (Mean_cpm + sd_cpm), xmin = (Mean_cpm - sd_cpm)), width = 0.2, position=position_dodge(0.9), color = "grey20") +
    theme_minimal(base_size = 14) +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, size = 14),
          axis.text.y = element_text(size = 14)) +
    # scale_y_continuous(breaks=scales::pretty_breaks(n=10)) +
    # ylim(-4,4) +
    scale_shape_manual(values = c(21, 21)) +
    ylab("") +
    xlab("Mean CPM") +
    scale_fill_manual(values = c("#FF3200","#feb24c")) +
    theme(legend.position="bottom",
          axis.text.y = element_blank())

cmcicpm

```

```{r}

# h3_sp_cmcic_ancombc <- (cmcipsp + cmcippm) / (cmcicsp + cmcicpm)
h3_sp_cmcic_ancombc <- cmcipsp + cmcippm + cmcicsp + cmcicpm +
  plot_layout(nrow = 2) + 
  plot_layout(widths = c(2, 1), heights = c(1, 3)) + 
  plot_annotation(tag_levels = 'A')

h3_sp_cmcic_ancombc

# ggsave("./06-publication/main_figures/Figure_XX5/h3_sp_cmcic_ancombc.pdf", plot = h3_sp_cmcic_ancombc,
#        device = "pdf", scale = 1, width = 11, height = 8.5, units = c("in"), dpi = 300)

ggsave("~/Desktop/h3_species_ancombc_dots.pdf", plot = h3_sp_cmcic_ancombc,
       device = "pdf", scale = 1, width = 9, height = 7, units = c("in"), dpi = 300)


```


## CMC anterior vs. posterior
```{r path_cmc_ts}
ancom_path_ts_cmc_paths <- ancom_path_cmc_ts %>%
  separate(., Pathway, into = "Path", sep = ":", extra = "drop") %>%
  pull(Path) %>%
  str_c(., collapse = "|") # need this format for filtering in the next step

# select only those differentially abundant pathways from the list, and split the column with names into 3 (Pathway, Genus, Species)
humann3_path_ancombc_cmc_ic <- humann2_path %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  filter(str_detect(Path, ancom_path_ts_cmc_paths)) %>%
  select(-Path) %>%
  filter(str_detect(Pathway, "\\|")) %>%
  gather("SampleID", "CPM", 2:ncol(.)) %>%
  mutate(Pathway = str_replace_all(Pathway, "\\.s__", "|s__")) %>%
  separate(., Pathway, into = c("Pathway", "Genus", "Species"), sep = "\\|") %>%
  mutate(Species = replace_na(Species, "unclassified"),
         Genus = str_replace_all(Genus, "g__", ""),
         Species = str_replace_all(Species, "s__", "")) %>%
  arrange(Pathway)

# calculate the % for each ortholog contributed by each genus         
humann3_path_ancombc_cmc_ts_stats <- humann3_path_ancombc_cmc_ic %>%
  group_by(Pathway, Genus) %>%
  summarize(Sum = sum(CPM)) %>%
  mutate(Percent = Sum/sum(Sum)*100) %>%
  ungroup(.) %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop")

# create the list of 10 orthologs again, but don't collapse the list as above
ancom_path_ts_cmc_paths <- ancom_path_cmc_ts %>%
  separate(., Pathway, into = "Path", sep = ":", extra = "drop") %>%
  pull(Path)

# calculate the total % of all genera that contribute < X% to each ortholog
humann2_path_cmc_ts_stats_extra <- lapply(ancom_path_ts_cmc_paths, function(eclass) {
 high_percent <- humann3_path_ancombc_cmc_ts_stats %>%
   filter(Path == eclass) %>%
   filter(Percent < 10) %>%
   summarise(Remaining = sum(Percent)) %>%
   mutate(Path = eclass,
          Genus = "Other")
}) %>%
 bind_rows(.)

# add this additional % to the main table
humann2_path_cmc_ts_bar_df <- humann3_path_ancombc_cmc_ts_stats %>%
  select(-c("Sum","Pathway")) %>%
  filter(Percent >= 10) %>%
  bind_rows(., humann2_path_cmc_ts_stats_extra %>%
              rename(Percent = Remaining) %>%
              select(Genus, Percent, Path)) %>%
  arrange(Path) 
# %>%
#   # unclassified for PWY-6470 got added to "Other" so we can remove "Other" for that path 
#   filter(Genus != "Other" | Path != "PWY-6470")

```

```{r, eval = F}

# plot the values in a bar chart
humann2_path_cmc_ts_bar_df %>%
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Aggregatibacter","Campylobacter","Enterobacter","Haemophilus","Neisseria","Ottowia","Selenomonas","Streptococcus","Tetragenococcus"),
         Path = fct_relevel(Path, ancom_path_cmc_ts %>%
                              separate(., Pathway, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  ggplot(., aes(x=Path, y=Percent, fill = Genus)) +
    geom_bar(stat = "identity") +
    theme_minimal() +
    # scale_fill_manual(values = c("#440154FF","#39558CFF","#1F968BFF",)) +
    theme(text = element_text(size=18),
          axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
    ylab("Percent") +
    ggtitle("Pathways - CMC vs. Industrial plaque") +
    theme(title = element_text(size=10))

```

```{r}
humann2_path_cmc_ts_bar_df_l <- humann2_path_cmc_ts_bar_df %>%
  mutate(Genus = fct_relevel(Genus, "Other","unclassified","Aggregatibacter","Campylobacter","Enterobacter","Haemophilus","Neisseria","Ottowia","Selenomonas","Streptococcus","Tetragenococcus"),
         Path = fct_relevel(Path, ancom_path_cmc_ts %>%
                              separate(., Pathway, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  full_join(., ancom_path_cmc_ts %>%
              mutate(newcol = Pathway) %>%
              separate(., newcol, into = "Path", sep = ":", extra = "drop")) %>%
  mutate(star_up = ifelse(q.val<0.001 & Direction == "IC", "***", ""),
         star_down = ifelse(q.val<0.001 & Direction == "CMC", "***", ""))

humann2_path_cmc_ts_bar_df_l

```

```{r}

# set minimal color palette for just this plot
cmcts_colors <- c("Other" = "#bdbdbd",
                    "unclassified" = "#737373",
                    "Aggregatibacter" = "#FF3200",
                    "Campylobacter" = "#BF6746",
                    "Enterobacter" = "#FEB24C",
                    "Haemophilus" = "#FEE0B7",
                    "Neisseria" = "#DFEACA",
                    "Ottowia" = "#2C968A",
                    "Selenomonas" = "#DD6EAF",
                    "Streptococcus" = "#065FA3",
                    "Tetragenococcus" = "#699FC7")

tssp <- humann2_path_cmc_ts_bar_df_l %>%
  mutate(Prop = (Percent / 100) * `log fold change (Anterior - Posterior)`,
         Path = fct_relevel(Path, ancom_path_cmc_ts %>%
                              separate(., Pathway, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  ggplot(., aes(x = Path, y = Prop, ymin = ci.lo.adj, ymax = ci.up.adj, fill = Genus)) + 
    # geom_col(position = "stack", color = "black", size = 0.25) +
    geom_bar(position = "stack", stat = "identity", color = "black", size = 0.25) +
    geom_errorbar(width=0.2, size=0.25) +
    geom_abline(slope = 0, intercept = 0, linetype="dashed", alpha=0.5) +
    theme_minimal(base_size = 14) +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, size = 14),
          axis.text.y = element_text(size = 12)) +
    scale_y_continuous(breaks=scales::pretty_breaks(n=6)) +
    scale_fill_manual(values = cmcts_colors) +
    ylab("Log fold change") +
    xlab("") +
    coord_flip() +
    theme(legend.position="bottom")

tssp

# +
#     geom_text(aes(y=`log fold change (Industrial calculus - CMC)`+4*sign(`log fold change (Industrial calculus - CMC)`), label=star_up),
#               vjust=0, hjust = 0, nudge_y = -3, nudge_x = 0, angle = 90, color="black") +
#     geom_text(aes(y=`log fold change (Industrial calculus - CMC)`+4*sign(`log fold change (Industrial calculus - CMC)`), label=star_down),
#               vjust=0, hjust = 0, nudge_y = 3, nudge_x = 0, angle = 90, color="black")

```

```{r}
# now calculate the mean proportion of these pathways in each group
tscpm <- humann2_path %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  filter(str_detect(Path, ancom_path_ts_cmc_paths %>%
                      str_c(collapse = "|"))) %>%
  select(-Path) %>%
  # mutate(Pathway = str_replace_all(Pathway, "\\|unclassified","")) %>%
  filter(!str_detect(Pathway, "\\|")) %>%
  pivot_longer(!Pathway, names_to = "Library_ID", values_to = "CPM") %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  left_join(., metadata %>%
              select(SampleID, Env, Tooth_site) %>%
              rename(Library_ID = SampleID), by = "Library_ID") %>%
  filter(str_detect(Tooth_site, "Anterior|Posterior"),
         str_detect(Env, "CMC")) %>%
  group_by(Tooth_site, Path) %>%
  summarize(Mean_cpm = mean(CPM),
            sd_cpm = sd(CPM)) %>%
  ungroup() %>%
  mutate(Path = fct_relevel(Path, ancom_path_cmc_ts %>%
                              separate(., Pathway, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>%
  ggplot(., aes(x = Mean_cpm, y = Path, shape = Tooth_site, bg = Tooth_site)) + 
    # geom_bar(position = "dodge", stat = "identity", color = "black", size = 0.25) +
    geom_point(position = position_dodge(0.9), size = 3) +
    geom_errorbar(aes(xmax = (Mean_cpm + sd_cpm), xmin = (Mean_cpm - sd_cpm)), width = 0.2, position=position_dodge(0.9), color = "grey20") +
    theme_minimal(base_size = 14) +
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5, vjust = 0.5, size = 14),
          axis.text.y = element_text(size = 14)) +
    scale_shape_manual(values = c(21,21)) +
    # scale_y_continuous(breaks=scales::pretty_breaks(n=10)) +
    # ylim(-4,4) +
    ylab("") +
    xlab("Mean CPM") +
    scale_fill_manual(values = c("#FF3200","#065FA3")) +
    theme(legend.position="bottom",
          axis.text.y = element_blank())
tscpm

```

```{r, eval = F}

# now cluster by sample
# transpose the table to cluster by sample
h3_path_cmc_ts_table <- humann2_path %>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  filter(str_detect(Path, ancom_path_ts_cmc_paths %>%
                      str_c(collapse = "|"))) %>%
  select(-Path) %>%
  filter(!str_detect(Pathway, "\\|"))%>%
  mutate(newcol = Pathway) %>%
  separate(., newcol, into = "Path", sep = ":", extra = "drop") %>%
  select(-Pathway) %>%
  select(Path, everything()) %>%
  pivot_longer(!Path, names_to = "Library_ID", values_to = "CPM") %>%
  pivot_wider(names_from = "Path", values_from = "CPM") %>%
  filter(str_detect(Library_ID, "CMC")) %>%
  column_to_rownames("Library_ID")

# perform a distance calculation for hierarchical clustering for the species ordering
humann3_path.cmc_dist <- vegdist(h3_path_cmc_ts_table, method = "euclidian")

# cluster the data with hclust
humann3_path.cmc_dist_hr <- hclust(humann3_path.cmc_dist)

# humann2_KOs.decontam_FA_cmc_clr_dist_hr$order gives the dendrogram order from the bottom to the top, but using the row numbers not the row names
# pull this order out and store as a vector to make the order a factor in the next step for plotting
smp_cmc_order <- humann3_path.cmc_dist_hr$order %>%
  as.character.numeric_version(.) %>%
  str_trim(.)

# use the dendrogram order to pull out the species in the correct order for the heatmap
ts_hr_order <- h3_path_cmc_ts_table %>%
  as.data.frame() %>%
  rownames_to_column("SampleID") %>%
  gather("Species","CLR",2:ncol(.)) %>%
  spread(Species,CLR) %>%
  mutate(SmpOrder = as.character(seq(1,85))) %>%  
  select(SampleID, SmpOrder, everything()) %>%
  mutate(SmpOrder = fct_relevel(SmpOrder, smp_cmc_order)) %>%
  arrange(SmpOrder) %>%
  pull(SampleID)

# order by anterior-posterior, then alphabetical
ant_post_order <- h3_path_cmc_ts_table %>%
  rownames_to_column("Library_ID") %>%
  pivot_longer(!Library_ID, names_to = "Path", values_to = "CPM") %>%
  # set sample order
  inner_join(., metadata %>%
               select(SampleID, Tooth_site) %>%
               rename(Library_ID = SampleID), by = "Library_ID") %>%
  mutate(Tooth_site = fct_drop(Tooth_site)) %>%
  arrange(Tooth_site) %>%
  select(Library_ID, Tooth_site) %>%
  distinct() %>%
  pull(Library_ID)

```


```{r, eval = F}
# set colors for anterior/posterior
vector_color = h3_path_cmc_ts_table %>%
  rownames_to_column("Library_ID") %>%
  pivot_longer(!Library_ID, names_to = "Path", values_to = "CPM") %>%
  # set sample order
  inner_join(., metadata %>%
               select(SampleID, Tooth_site) %>%
               rename(Library_ID = SampleID), by = "Library_ID") %>%
  inner_join(., Tooth_site_colors %>%
               as_data_frame(.) %>%
               rename(Color = value) %>%
               mutate(Tooth_site = c("Anterior","Posterior","Buccal_mucosa","HMP","HMP10M","ExtractionGauze","ExtractionBlank","LibraryBlank")) , by = "Tooth_site") %>% # fct_drop(Tooth_site)
  mutate(Library_ID = fct_relevel(Library_ID, ant_post_order)) %>%
  arrange(Library_ID) %>%
  select(Library_ID, Color) %>%
  unique %>%
  pull(Color) %>%
  as.character()


# make a heatmap of the differentially abundant pathways
hmap <- h3_path_cmc_ts_table %>%
  rownames_to_column("Library_ID") %>%
  pivot_longer(!Library_ID, names_to = "Path", values_to = "CPM") %>%
  # filter(str_detect(Library_ID, "CMC")) %>%
  # set sample order
  inner_join(., metadata %>%
               select(SampleID, Tooth_site) %>%
               rename(Library_ID = SampleID), by = "Library_ID") %>%
  inner_join(., Tooth_site_colors %>%
               as_data_frame(.) %>%
               rename(Color = value) %>%
               mutate(Tooth_site = c("Anterior","Posterior","Buccal_mucosa","HMP","HMP10M","ExtractionGauze","ExtractionBlank","LibraryBlank")) , by = "Tooth_site") %>% # fct_drop(Tooth_site)
  mutate(Library_ID = fct_relevel(Library_ID, ant_post_order),
         Path = fct_relevel(Path, ancom_path_cmc_ts %>%
                              separate(., Pathway, into = c("Path"), sep = ":", extra = "drop") %>%
                              pull(Path))) %>% # ant_post_order ts_hr_order
  arrange(Library_ID) %>%
  ggplot(., aes(Library_ID, Path, fill = CPM)) +
    geom_tile() +
    scale_fill_gradient(low = "white", high = "#065FA3") +
    theme_minimal(base_size = 18) +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.95, color = vector_color),
          axis.text.y = element_blank(),
        plot.margin = margin(0, 0, 0, 0, "pt"),
        axis.title.y = element_blank(),
        legend.position = "right")

hmap

```


```{r}

h3_species_ancombc <- tssp + tscpm
h3_species_ancombc


h3_species_ancombc <- cmcipsp + cmcippm + cmcicsp + cmcicpm + tssp + tscpm +
  plot_layout(nrow = 3) + 
  plot_layout(widths = c(2, 1), heights = c(1, 2, 3)) + 
  plot_annotation(tag_levels = 'A')

h3_species_ancombc

ggsave("./06-publication/main_figures/Figure_XX5/panel_parts/h3_species_ancombc.pdf", plot = h3_species_ancombc,
       device = "pdf", scale = 1, width = 11, height = 10, units = c("in"), dpi = 300)


```



