---
title: "Cameroon hunter-gatherer calculus/plaque MALT alpha diversity"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---

```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(vegan)
library(janitor)
library(tidyverse)
library(gplots)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

```{r load_data}
malt_species <- fread("./05-results.backup/CMC_HMP_JAE_Yano_MALT_RSC_species_bacteria_archaea.txt")
malt_genus <- fread("./05-results.backup/CMC_HMP_JAE_Yano_MALT_RSC_genus_bacteria_archaea.txt")

# clean the file names
malt_species <- rename(malt_species, Species = `#Datasets`)
colnames(malt_species) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M","_10M", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam","_10M", colnames(malt_species))
colnames(malt_species) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_species))
colnames(malt_species) <- gsub(".fastq.truncated","", colnames(malt_species))


# remove human and unassigned reads
malt_species <- malt_species %>% filter(!str_detect(Species, "Homo sapiens"))
malt_species <- malt_species %>% filter(!str_detect(Species, "Not assigned"))
# get rid of the extra column
malt_species <- malt_species %>% select(-Comparison_CMC_HMP_JAE_MALT_RSC)

malt_genus <- rename(malt_genus, Species = `#Datasets`)
colnames(malt_genus) <- gsub(".SG1.1_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_000.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_000.10M.fastq.combined.fq.prefixed.extractunmapped.bam","_10M", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam.10M","_10M", colnames(malt_genus))
colnames(malt_genus) <- gsub("_S0_L001_R1_001.fastq.combined.fq.prefixed.extractunmapped.bam","", colnames(malt_genus))
colnames(malt_genus) <- gsub(".fastq.truncated","", colnames(malt_genus))

# remove human and unassigned reads
malt_genus <- malt_genus %>% filter(!str_detect(Species, "Homo"))
malt_genus <- malt_genus %>% filter(!str_detect(Species, "Not assigned"))
# get rid of the extra column at the begining
malt_genus <- malt_genus %>% select(-Comparison_CMC_HMP_JAE_MALT_RSC)

# load the metadata file
metadata <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata <- metadata %>%
  rename(SampleID = `#SampleID`)  %>%
  as_tibble() %>%
  mutate(Description = fct_relevel(Description, "HunterGatherer", "Yanomami", "HMP_supPlaque", "HMP_subPlaque",
                                   "HMP10M_supPlaque", "HMP10M_subPlaque","JAE","JAE10M",
                                   "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime","Yanomami","HMP","HMP10M","JAE","JAE10M",
                                     "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Env = fct_relevel(Env, "HunterGatherer","HMP","HMP10M","JAE","JAE10M",
                                     "ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Market_economy = fct_relevel(Market_economy, "logging_road","forrest_camp","farming","Yanomami",
                                      "Industrial","ExtractionGauze","ExtractionBlank","LibraryBlank"),
         Age_group = fct_relevel(Age_group, "10-18","18-29","30-39","40-49","50-59","60+","HMP","HMP10M"),
         Sex = fct_relevel(Sex, "Male","Female","HMP","HMP10M"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior","Buccal_mucosa","HMP","HMP10M"),
         Type = fct_relevel(Type, "Calculus","Plaque","Blank"),
         Village = fct_relevel(Village, "202","205","204","203","206","Yanomami","HMP","HMP10M","JAE","JAE10M",
                               "ExtractionGauze","ExtractionBlank","LibraryBlank"))

fullSRR <- metadata %>%
  select(SampleID) %>%
  filter(str_detect(SampleID, "SRR")) %>%
  filter(!str_detect(SampleID, "10M")) %>%
  pull
```


```{r colors_shapes_peachpear}
# lacroix_palette("PeachPear", n = 8, type = "continuous")
# "#FF3200" "#EF8158" "#E9BD8E" "#CBDDA7" "#38BCAD" "#0B91B5" "#065FA3" "#172869"

Village_colors <- c(`202` = "#FF3200", `205` = "#EF8158", `204` = "#E9BD8E", `203` = "#CBDDA7", `206` = "#38BCAD",
                    Yanomami = "#C70E7B", HMP = "#172869", HMP10M = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                     ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Village_shapes <- c(`202` = 15, `203` = 16, `205` = 17,`204` = 18, `206` = 25, Yanomami = 14, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9,
                    ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Ethnic_Group_colors <- c(Baka = "#FF3200", Nzime = "#172869", Yanomami = "#C70E7B", HMP = "#969696", HMP10M = "#d9d9d9", 
                         JAE10M = "#ffeda0", JAE = "#feb24c",ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Ethnic_Group_shapes <- c(Baka = 19, Nzime = 17,  Yanomami = 14, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9,
                         ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Env_colors <- c(HunterGatherer = "#FF3200", HMP = "#172869", HMP10M = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Env_shapes <- c(HunterGatherer = 19, HMP = 7, HMP10M = 9, JAE = 7, JAE10M = 9, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Description_colors <- c(HunterGatherer = "#FF3200", Yanomami = "#C70E7B", HMP_supPlaque = "#172869", HMP_subPlaque = "#172869", 
                        HMP10M_supPlaque = "#065FA3", HMP10M_subPlaque = "#065FA3", JAE10M = "#ffeda0", JAE = "#feb24c",
                        ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Description_shapes <- c(HunterGatherer = 19, Yanomami = 14, HMP_supPlaque = 7, HMP_subPlaque = 7, HMP10M_supPlaque = 9, 
                        HMP10M_subPlaque = 9, JAE = 7, JAE10M = 9, ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Market_economy_colors <- c(logging_road = "#FF3200", forrest_camp = "#172869", farming = "#82CDAA", Yanomami = "#C70E7B", Industrial = "#969696",
                           ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Market_economy_shapes <- c(logging_road = 19, forrest_camp = 17, farming = 18, Yanomami = 14, HMP = 7, HMP10M = 9, Industrial = 15,
                           ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Age_group_colors <- c(`18-29` = "#FF3200", `30-39` = "#E9B186", `40-49` = "#82CDAA", `50-59` = "#0686B8", `60+` = "#172869",
                      `10-18` = "#C70E7B", HMP = "#969696", HMP10M = "#d9d9d9",
                      ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Age_group_shapes <- c(`18-29` = 15, `30-39` = 16, `40-49` = 17, `50-59` = 18, `60+` = 25, `10-18` = 14, HMP = 7, HMP10M = 9,
                      ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Sex_colors <- c(Female = "#FF3200", Male = "#172869", HMP = "#969696", HMP10M = "#d9d9d9",
                ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Sex_shapes <- c(Female = 19, Male = 17,  HMP = 7, HMP10M = 9,ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

Tooth_site_colors <- c(Anterior = "#FF3200", Posterior = "#172869", Buccal_mucosa = "#C70E7B", HMP = "#969696", HMP10M = "#d9d9d9",
                       ExtractionGauze = "#252525", ExtractionBlank = "#737373", LibraryBlank = "#bdbdbd")
Tooth_site_shapes <- c(Anterior = 19, Posterior = 17,  Buccal_mucosa = 14, HMP = 7, HMP10M = 9,
                       ExtractionGauze = 8, ExtractionBlank = 4, LibraryBlank = 3)

```

```{r box_plotting_functions}

plot_box_whisker <- function(df, stat_info, metadata_group, fill_group, title_text) {
    fill_group_colors <- get(paste(fill_group, "_colors", sep = ""))

      df_X <- df %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group,
                                  Sex, Market_economy, Village, Tooth_site, Type), by = "SampleID")

    stat_info_table = df_X[[stat_info]]
    metadata_group = df_X[[metadata_group]]
    fill_group = df_X[[fill_group]]

    bx_wh_plot <- ggplot(df_X, aes(x = metadata_group, y = stat_info_table, fill = fill_group)) +
                         geom_boxplot() +
                         scale_fill_manual(values = fill_group_colors) +
                         theme_minimal() +
                         #scale_y_continuous(trans='log10') +
                         #scale_y_continuous(trans='pseudo_log') +
                         theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
                         ylab(stat_info) +
                         ggtitle(title_text)

    return(bx_wh_plot)
}

```

List the outliers from the plot in section `raw_plot` in `Cameroon_h-g_taxonomy.Rmd` and remove them from the taxonomy table. 
```{r outliers}
outliers <- c("CMC002.B0201","CMC032.B0101","CMC037.B0201","CMC047.A0201",
              "EXB059.A1901","EXB059.A1301","EXB059.A1101","EXB059.A0901",
              "SRR1646041","SRR1646032","SRR1646042","SRR1646045","SRR1646030","SRR1646047")

outliersF <- str_c(outliers, collapse = "|")

```

Read in the contaminant list from `raw_plot` in `Cameroon_h-g_taxonomy.Rmd` and remove them from the taxonomy table. 
```{r contaminants}

contaminants_species <- fread("05-results.backup/malt_contaminants_species.tsv", header = T, sep = "\t")
contaminants_genus <- fread("05-results.backup/malt_contaminants_genus.tsv")

malt_species.decontam <- malt_species %>%
  anti_join(., contaminants_species)

malt_genus.decontam <- malt_genus %>%
  anti_join(., contaminants_genus)


```


# Alpha-diversity
First we want to know if there are differences in the total diveristy of identified species between *XXXMETADATAXXX*. We'll look at alpha-diversity for this, using both the observed species and Shannon diverity metrics. Observed species to tell us which *XXXMETADATAXXX* have more species/genera, and Shannon to tell us which has more even distribution of species/genera. 

## Observed Speceies/Genera
```{r observed_species}
# make Relab binary to make a presence/absence table
presabs_table <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         adorn_totals(where = "col", name = "Sum_species.decontam") %>%
                                         mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
                                         select(-one_of(c("Sum_species.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts) %>%
                                         gather("Species","Counts",2:ncol(.)) %>%
                                         mutate(Counts = ifelse(Counts > 0, 1, 0)) %>%
                                         spread(Species, Counts) %>%
                                         column_to_rownames("SampleID")

# calculate observed species for each sample with janitor 'adorn_totals'
malt_species.decontam.os <- presabs_table %>%
  adorn_totals(., where = "col", name = "Observed_species") %>%
  select(Observed_species) %>%
  rownames_to_column("SampleID") 

plot_box_whisker(malt_species.decontam.os, "Observed_species", "Env", "Env", "Number of species")
plot_box_whisker(malt_species.decontam.os, "Observed_species", "Village", "Village", "Number of species")


# now on filtered tables
presabs_table_f <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         adorn_totals(where = "col", name = "Sum_species.decontam") %>%
                                         mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts) %>%
                                         gather("Species","Counts",2:ncol(.)) %>%
                                         mutate(Counts = ifelse(Counts > 0, 1, 0)) %>%
                                         spread(Species, Counts) %>%
                                         column_to_rownames("SampleID")
  
# calculate observed species for each sample with janitor 'adorn_totals'
malt_species.decontam_filtered.os <- presabs_table_f %>%
  adorn_totals(., where = "col", name = "Observed_species") %>%
  select(Observed_species) %>%
  rownames_to_column("SampleID") 

plot_box_whisker(malt_species.decontam_filtered.os, "Observed_species", "Env", "Env", "Number of species")
plot_box_whisker(malt_species.decontam_filtered.os, "Observed_species", "Village", "Village", "Number of species")

```

Now at the genus level
```{r observed_genera}
presabs_table <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus.decontam") %>%
                                         mutate(Percent = Sum_genus.decontam/sum(Sum_genus.decontam)*100) %>%
                                         select(-one_of(c("Sum_genus.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts) %>%
                                         gather("Species","Counts",2:ncol(.)) %>%
                                         mutate(Counts = ifelse(Counts > 0, 1, 0)) %>%
                                         spread(Species, Counts) %>%
                                         column_to_rownames("SampleID")
  
# calculate observed genus for each sample with janitor 'adorn_totals'
malt_genus.decontam.os <- presabs_table %>%
  adorn_totals(., where = "col", name = "Observed_genera") %>%
  select(Observed_genera) %>%
  rownames_to_column("SampleID") 

plot_box_whisker(malt_genus.decontam.os, "Observed_genera", "Env", "Env", "Number of genera")
plot_box_whisker(malt_genus.decontam.os, "Observed_genera", "Village", "Village", "Number of genera")


# now on the filtered table
presabs_table_filtered <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus.decontam") %>%
                                         mutate(Percent = Sum_genus.decontam/sum(Sum_genus.decontam)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts) %>%
                                         gather("Species","Counts",2:ncol(.)) %>%
                                         mutate(Counts = ifelse(Counts > 0, 1, 0)) %>%
                                         spread(Species, Counts) %>%
                                         column_to_rownames("SampleID")
  
# calculate observed genus for each sample with janitor 'adorn_totals'
malt_genus.decontam_filtered.os <- presabs_table_filtered %>%
  adorn_totals(., where = "col", name = "Observed_genera") %>%
  select(Observed_genera) %>%
  rownames_to_column("SampleID")

plot_box_whisker(malt_genus.decontam_filtered.os, "Observed_genera", "Env", "Env", "Number of genera")
plot_box_whisker(malt_genus.decontam_filtered.os, "Observed_genera", "Village", "Village", "Number of genera")

```


## Shannon Index
```{r shannon_diversity}
# Shannon diversity with the package vegan
malt_species.decontam.matrix <- as.matrix(malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         adorn_totals(where = "col", name = "Sum_species.decontam") %>%
                                         mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
                                         select(-one_of(c("Sum_species.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts) %>% 
                                         column_to_rownames("SampleID"))
 
malt_species.decontam.shannon <- as.data.frame(diversity(malt_species.decontam.matrix, index = "shannon"))
names(malt_species.decontam.shannon)[1] <- "Shannon_index"

# merge with the metadata file and plot
malt_species.decontam.shannon <- malt_species.decontam.shannon %>%
  rownames_to_column("SampleID") 

plot_box_whisker(malt_species.decontam.shannon, "Shannon_index", "Env", "Env", "Shannon Index")
plot_box_whisker(malt_species.decontam.shannon, "Shannon_index", "Village", "Village", "Shannon Index")


# now with the filtered data
malt_species.decontam_filtered.matrix <- as.matrix(malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         adorn_totals(where = "col", name = "Sum_species.decontam") %>%
                                         mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_species.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts) %>% 
                                         column_to_rownames("SampleID"))

malt_species.decontam_filtered.shannon <- as.data.frame(diversity(malt_species.decontam_filtered.matrix, index = "shannon"))
names(malt_species.decontam_filtered.shannon)[1] <- "Shannon_index"

# merge with the metadata file and plot
malt_species.decontam_filtered.shannon <- malt_species.decontam_filtered.shannon %>%
  rownames_to_column("SampleID") 

plot_box_whisker(malt_species.decontam_filtered.shannon, "Shannon_index", "Env", "Env", "Shannon Index")
plot_box_whisker(malt_species.decontam_filtered.shannon, "Shannon_index", "Village", "Village", "Shannon Index")


```

At the genus level
```{r shannon_genera}
# Shannon diversity with the package vegan
malt_genus.decontam.matrix <- as.matrix(malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus.decontam") %>%
                                         mutate(Percent = Sum_genus.decontam/sum(Sum_genus.decontam)*100) %>%
                                         select(-one_of(c("Sum_genus.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts) %>% 
                                         column_to_rownames("SampleID"))

malt_genus.decontam.shannon <- as.data.frame(diversity(malt_genus.decontam.matrix, index = "shannon"))
names(malt_genus.decontam.shannon)[1] <- "Shannon_index"

# merge with the metadata file and plot
malt_genus.decontam.shannon <- malt_genus.decontam.shannon %>%
  rownames_to_column("SampleID") 

plot_box_whisker(malt_genus.decontam.shannon, "Shannon_index", "Env", "Env", "Shannon Index")
plot_box_whisker(malt_genus.decontam.shannon, "Shannon_index", "Village", "Village", "Shannon Index")


# now with the filtered data
malt_genus.decontam_filtered.matrix <- as.matrix(malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         adorn_totals(where = "col", name = "Sum_genus.decontam") %>%
                                         mutate(Percent = Sum_genus.decontam/sum(Sum_genus.decontam)*100) %>%
                                         filter(Percent > 0.001) %>%
                                         select(-one_of(c("Sum_genus.decontam","Percent"))) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts) %>% 
                                         column_to_rownames("SampleID"))

malt_genus.decontam_filtered.shannon <- as.data.frame(diversity(malt_genus.decontam_filtered.matrix, index = "shannon"))
names(malt_genus.decontam_filtered.shannon)[1] <- "Shannon_index"

# merge with the metadata file and plot
malt_genus.decontam_filtered.shannon <- malt_genus.decontam_filtered.shannon %>%
  rownames_to_column("SampleID") 

plot_box_whisker(malt_genus.decontam_filtered.shannon, "Shannon_index", "Env", "Env", "Shannon Index")
plot_box_whisker(malt_genus.decontam_filtered.shannon, "Shannon_index", "Village", "Village", "Shannon Index")

```


## Pielou's Evenness
Let's also look specifically at evenness of the samples using Pielou's evenness. This calculation (derived from Shannon's index) gives us a number from 0 to 1, where 0 is no evenness and 1 is perfect evenness. We want to see if there are any groups that have substantially greater species/genera evenness than the others. We're not interested in what the actual number is, or how even the samples appear to be, except relative to eachother. 
```{r evenness_pielou}
evenness.species <- diversity(malt_species.decontam.matrix)
evenness.species <- as.data.frame(evenness.species/log(specnumber(malt_species.decontam_filtered.matrix)))
names(evenness.species)[1] <- "Pielou_evenness"

# merge with the metadata file and plot
evenness.species <- evenness.species %>%
  rownames_to_column("SampleID") 

plot_box_whisker(evenness.species, "Pielou_evenness", "Env", "Env", "Pielou's evenness")
plot_box_whisker(evenness.species, "Pielou_evenness", "Village", "Village", "Pielou's evenness")


# now with filtered data
evenness.species_filtered <- diversity(malt_species.decontam_filtered.matrix)
evenness.species_filtered <- as.data.frame(evenness.species_filtered/log(specnumber(malt_species.decontam_filtered.matrix)))
names(evenness.species_filtered)[1] <- "Pielou_evenness"

# merge with the metadata file and plot
evenness.species_filtered <- evenness.species_filtered %>%
  rownames_to_column("SampleID") 

plot_box_whisker(evenness.species_filtered, "Pielou_evenness", "Env", "Env", "Pielou's evenness")
plot_box_whisker(evenness.species_filtered, "Pielou_evenness", "Village", "Village", "Pielou's evenness")

```
In this dataset EXB059.A1601 has no detected genera, and this is the samples that was removed in the warning notice (it ends up with NA for the evenness calculation).

At the genus level
```{r evenness_pielou_genus}
evenness.genus <- diversity(malt_genus.decontam.matrix)
evenness.genus <- as.data.frame(evenness.genus/log(specnumber(malt_genus.decontam.matrix)))
names(evenness.genus)[1] <- "Pielou_evenness"

# merge with the metadata file and plot
evenness.genus <- evenness.genus %>%
  rownames_to_column("SampleID") 

plot_box_whisker(evenness.genus, "Pielou_evenness", "Env", "Env", "Pielou's evenness")
plot_box_whisker(evenness.genus, "Pielou_evenness", "Village", "Village", "Pielou's evenness")


# now with the filtered data
evenness_filtered.genus <- diversity(malt_genus.decontam_filtered.matrix)
evenness_filtered.genus <- as.data.frame(evenness_filtered.genus/log(specnumber(malt_genus.decontam_filtered.matrix)))
names(evenness_filtered.genus)[1] <- "Pielou_evenness"

# merge with the metadata file and plot
evenness_filtered.genus <- evenness_filtered.genus %>%
  rownames_to_column("SampleID") 

plot_box_whisker(evenness_filtered.genus, "Pielou_evenness", "Env", "Env", "Pielou's evenness")
plot_box_whisker(evenness_filtered.genus, "Pielou_evenness", "Village", "Village", "Pielou's evenness")

```
In this dataset EXB059.A1601 and LIB050.A0105 have no detected genera, and these are the 2 samples that were removed in the warning notice (they end up with NA for the evenness calculation).


## Heatmaps
Now make a heatmap of the top 50 most abundant species and genera (as an arbitrary cut-off)
```{r heatmaps, eval = F}
malt_species.decontam_top50 <- malt_species.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("SRR")) %>%
                                         select(-contains("10M")) %>%
                                         select(-contains("EXB")) %>%
                                         select(-contains("LIB")) %>%
                                         adorn_totals(where = "col", name = "Sum_species.decontam") %>%
                                         mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
                                         top_n(50, Percent) %>%
                                         select(-one_of("Sum_species.decontam","Percent")) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_species.decontam_top50) <- malt_species.decontam_top50$SampleID
malt_species.decontam_top50 <- malt_species.decontam_top50 %>% select(-SampleID)

# CLR-transform the matrix for plotting the heatmap
malt_species.decontam_top50_clr = malt_species.decontam_top50+1
#malt_species.decontam_top50_clr <- as.matrix(logratio.transfo(malt_species.decontam_top50_clr, logratio = "CLR"))
malt_species.decontam_top50_clr <- clr(malt_species.decontam_top50_clr)
malt_species.decontam_top50_clr <- as.matrix(malt_species.decontam_top50_clr)


malt_species.decontam_top50_clr.heatmap <- heatmap.2(malt_species.decontam_top50_clr, Colv = T, Rowv = T, dendrogram = "both",
         trace = "none", col = viridis_pal(),
         margins = c(8,6), srtCol = 30, cexCol = 0.6, cexRow = 0.5,
          key = T, keysize = 0.8, main = "50 most abundant species")


malt_genus.decontam_top50 <- malt_genus.decontam %>%
                                         select(-one_of(outliers)) %>%
                                         # select(-contains("SRR")) %>%
                                         select(-contains("10M")) %>%
                                         select(-contains("EXB")) %>%
                                         select(-contains("LIB")) %>%
                                         adorn_totals(where = "col", name = "Sum_genus.decontam") %>%
                                         mutate(Percent = Sum_genus.decontam/sum(Sum_genus.decontam)*100) %>%
                                         top_n(50, Percent) %>%
                                         select(-one_of("Sum_genus.decontam","Percent")) %>%
                                         gather("SampleID","Counts",2:ncol(.)) %>%
                                         spread(Species,Counts)

rownames(malt_genus.decontam_top50) <- malt_genus.decontam_top50$SampleID
malt_genus.decontam_top50 <- malt_genus.decontam_top50 %>% select(-SampleID)

# CLR-transform the matrix for plotting the heatmap
malt_genus.decontam_top50_clr = malt_genus.decontam_top50+1
#malt_genus.decontam_top50_clr <- as.matrix(logratio.transfo(malt_genus.decontam_top50_clr, logratio = "CLR"))
malt_genus.decontam_top50_clr <- clr(malt_genus.decontam_top50_clr)
malt_genus.decontam_top50_clr <- as.matrix(malt_genus.decontam_top50_clr)


malt_genus.decontam_top50_clr.heatmap <- heatmap.2(malt_genus.decontam_top50_clr, Colv = T, Rowv = T, dendrogram = "both",
         trace = "none", col = viridis_pal(),
         margins = c(8,6), srtCol = 30, cexCol = 0.6, cexRow = 0.5,
          key = T, keysize = 0.8, main = "50 most abundant genus")

```

































