---
title: "Cameroon hunter-gatherer calculus/plaque MALT beta diversity"
author: "Irina Velsko"
date: "`r format(Sys.time(), '%b %d, %Y')`"
output:
    github_document:
        fig_width: 8
        fig_height: 6
        toc: true
---


```{r load_libraries, echo = F, message = F}
library(knitr)
library(data.table)
library(psych)
library(ape)
library(mixOmics)
library(compositions)
library(vegan)
library(janitor)
library(pander)
library(tidyverse)
library(gplots)
library(ggrepel)
library(viridis)
library(cowplot)
opts_chunk$set(echo=F, warning=F, message=F, dpi=150)
```

Set the working directory of the notebook to the parent directory of the 
notebook itself.  

```{r setup}
knitr::opts_knit$set(root.dir = normalizePath(".."))
```

# MALT read stats

```{r load_data}

# load the taxonomy tables that have been decontaminated and the column headers are cleaned up
load("./05-results.backup/CMC_taxonomy_tables.RData")

# load the metadata file
load("./05-results.backup/CMC_metadata.RData")

# load the lists of species/genera that are in MR1-MR3 from the factor analysis
mr_species <- fread("./05-results.backup/minres_oblimin_noblanks.fa_species.tsv")
mr_cmc_species <- fread("./05-results.backup/minres_oblimin_cmc.fa_species.tsv")
# mr_genus <- fread("./05-results.backup/minres_oblimin_noyanomami.fa_genus.tsv")

```


```{r pca_plotting_functions}
# plotting PCA with colored dots
plot_pca <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))

    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_integration, Village, Tooth_site, Type, ExtractionBatch, LibraryBatch), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group, shape = metadata_group)) +
                        geom_point(size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 16) +
                        theme(text = element_text(size=16)) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot)
}

```

```{r pca_names}
# this function plots the PCA with the sample names instead of points
plot_pca_names <- function(df, pc1, pc2, metadata_group) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))

    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%")
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_integration, Village, Tooth_site, Type, ExtractionBatch, LibraryBatch), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    pca_plot_names <- ggplot(df_X, aes(pc1, pc2, colour = metadata_group)) +
                        geom_text(aes(label = df_X$SampleID), size = 4) +
                        scale_colour_manual(values = metadata_group_colors) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 16) +
                        theme(text = element_text(size=16)) +
                        theme(legend.position = "top") +
                        theme(legend.title = element_blank())

    return(pca_plot_names)
}

```

```{r pca_titles}
# plot PCA with colored dots and the title including the # of species or genera
plot_pca_titles <- function(df, pc1, pc2, color_group, shape_group, size_group, ncomps) {
    metadata_group_colors <- get(paste(color_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(shape_group, "_shapes", sep = ""))
    metadata_group_size <- get(paste(size_group, "_size", sep = ""))

    pca.list <- mixOmics::pca(df, ncomp = ncomps, logratio = 'CLR')

    exp_var <- paste0(round(pca.list$prop_expl_var$X * 100, 2), "%")
    df_X <- pca.list$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Env, Description, Ethnic_Group, Age_group, Sex, Market_integration, Village, Tooth_site, Type, ExtractionBatch, LibraryBatch), by = "SampleID")
    
    color_group = df_X[[color_group]]
    shape_group = df_X[[shape_group]]
    size_group = df_X[[size_group]]
   
                       if (pc1 == 'PC1') {
                           pc1 <- df_X$PC1
                           exp_var_pc1 <- exp_var[1]
                           xaxis <- c("PC1")
                       }  else if (pc1 == 'PC2') {
                           pc1 <- df_X$PC2
                           exp_var_pc1 <- exp_var[2]
                           xaxis <- c("PC2")
                       } else if (pc1 == 'PC3') {
                           pc1 <- df_X$PC3
                           exp_var_pc1 <- exp_var[3]
                           xaxis <- c("PC3")
                       }

                       if (pc2 == 'PC1') {
                           pc2 <- df_X$PC1
                           exp_var_pc2 <- exp_var[1]
                           yaxis <- c("PC1")
                       }  else if (pc2 == 'PC2') {
                           pc2 <- df_X$PC2
                           exp_var_pc2 <- exp_var[2]
                           yaxis <- c("PC2")
                       } else if (pc2 == 'PC3') {
                           pc2 <- df_X$PC3
                           exp_var_pc2 <- exp_var[3]
                           yaxis <- c("PC3")
                       }

    title_text <- df %>%
                     rownames_to_column("SampleID") %>%
                     gather("Species","Counts",2:ncol(.)) %>%
                     spread(SampleID, Counts) %>%
                     count
    
    pca_plot <- ggplot(df_X, aes(pc1, pc2, colour = color_group, shape = shape_group, size = size_group)) +
                        geom_point() +
                        scale_colour_manual(values = metadata_group_colors) +
                        scale_shape_manual(values = metadata_group_shapes) +
                        scale_size_manual(values = metadata_group_size) +
                        # stat_ellipse() +
                        xlab(paste(xaxis, " - ", exp_var_pc1)) +
                        ylab(paste(yaxis, " - ", exp_var_pc2)) +
                        theme_minimal(base_size = 16) +
                        theme(text = element_text(size=16)) +
                        theme(legend.position = "top") +
                        ggtitle(paste(title_text))

    return(pca_plot)
}

```

```{r pca_biplot_functions}
plot_pca_bi <- function(df, pc1, pc2, metadata_group, columntitle) {
    metadata_group_colors <- get(paste(metadata_group, "_colors", sep = ""))
    metadata_group_shapes <- get(paste(metadata_group, "_shapes", sep = ""))
   
    arrow_pc <- enquo(columntitle)
    
    exp_var <- paste0(round(df$prop_expl_var$X * 100, 2), "%") # explained variance for x- and y-labels
    
    # select only the PCs from the PCA and add metadata
    df_X <- df$variates$X %>%
              as.data.frame() %>%
              rownames_to_column("SampleID") %>%
              inner_join(metadata %>%
                           select(SampleID, Description, Ethnic_Group, Age_group, Sex, Market_integration, Village, Tooth_site, Type), by = "SampleID")
    
    metadata_group = df_X[[metadata_group]]
    
    corr_lam <- df$sdev[c("PC1", "PC2", "PC3")] * sqrt(nrow(df_X))

    df_X <- df_X %>%
      mutate(PC1 = PC1 / corr_lam[1],
             PC2 = PC2 / corr_lam[2], 
             PC3 = PC3 / corr_lam[3])

    # select the correct PC column and explained variance for PC1
    if (pc1 == 'PC1') {
        Pc1 <- df_X$PC1
        exp_var_pc1 <- exp_var[1]
        xaxis <- c("PC1")
    } else if (pc1 == 'PC2') {
        Pc1 <- df_X$PC2
        exp_var_pc1 <- exp_var[2]
        xaxis <- c("PC2")
    } else if (pc1 == 'PC3') {
        Pc1 <- df_X$PC3
        exp_var_pc1 <- exp_var[3]
        xaxis <- c("PC3")
   }
    
    # select the correct PC column and explained variance for PC2
    if (pc2 == 'PC1') {
        Pc2 <- df_X$PC1
        exp_var_pc2 <- exp_var[1]
        yaxis <- c("PC1")
 } else if (pc2 == 'PC2') {
       Pc2 <- df_X$PC2
       exp_var_pc2 <- exp_var[2]
       yaxis <- c("PC2")
    } else if (pc2 == 'PC3') {
       Pc2 <- df_X$PC3
       exp_var_pc2 <- exp_var[3]
       yaxis <- c("PC3")
   }
    
    # Identify the 10 pathways that have highest positive and negative loadings in the selected PC
    pws_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(10, !!arrow_pc)

    neg_10 <- df$loadings$X %>%
      as.data.frame(.) %>%
      rownames_to_column(var = "Taxon") %>%
      top_n(-10, !!arrow_pc)


    pca_plot_bi <- ggplot(df_X, aes(x = Pc1, y = Pc2, colour = metadata_group)) +
      geom_point(size = 2, aes(shape = metadata_group) )+
      geom_segment(data = pws_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "black",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = pws_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      geom_segment(data = neg_10,
                   aes(xend = get(paste(pc1)), yend = get(paste(pc2))),
                   x = 0, y = 0, colour = "grey50",
                   size = 0.5,
                   arrow = arrow(length = unit(0.03, "npc"))) +
      geom_label_repel(data = neg_10,
                   aes(x = get(paste(pc1)), y = get(paste(pc2)), label = Taxon),
                   size = 1.8, colour = "grey20", label.padding = 0.2) +
      labs(x = paste(xaxis, " - ", exp_var_pc1),
           y = paste(yaxis, " - ", exp_var_pc2)) +
      scale_colour_manual(values = metadata_group_colors) +
      scale_shape_manual(values = metadata_group_shapes) +
      theme_minimal() + theme(text = element_text(size = 16)) +
      theme(text = element_text(size=16)) +
      theme(legend.position = "top")

    return(pca_plot_bi)
}

# pca_test_biplot <- plot_pca_bi(malt_species.pca, "PC3", "PC2", "Village", PC2)
# pca_test_biplot


```


Before running decontam, let's plot the samples in a PCA to see if there are any that are clearly not good-quality
```{r raw_plot}

malt_species_raw_pcain <- malt_species_raw %>%
  select(-matches("SRS|ERR|SRR164|10M")) %>%
  gather("SampleID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species_raw_pcain, logratio = 'CLR')

# perform a PCA to see how the data cluster
malt_species.pca <- mixOmics::pca(malt_species_raw_pcain, ncomp = 3, logratio = 'CLR')
plot(malt_species.pca)


raw_sp_pca12 <- plot_pca(malt_species.pca, "PC1", "PC2", "Description")
raw_sp_pca12
raw_sp_pca32 <- plot_pca(malt_species.pca, "PC3", "PC2", "Description")
raw_sp_pca32
plot_pca_names(malt_species.pca,  "PC1", "PC2", "Env")

# ggsave("./06-publication/supplemental_figures/SXX29/SXX29_raw_sp_pca12.pdf", plot = raw_sp_pca12, device = "pdf",
#        scale = 1, width = 8, height = 7, units = c("in"), dpi = 300)
# ggsave("./06-publication/supplemental_figures/SXX29/SXX29_raw_sp_pca32.pdf", plot = raw_sp_pca32, device = "pdf",
#        scale = 1, width = 7.5, height = 7, units = c("in"), dpi = 300)


```


List the outliers from the plot in section `raw_plot` in `Cameroon_h-g_taxonomy.Rmd` (above) and remove them from the taxonomy table. 
```{r outliers}
outliers <- c("CMC002.B0201",
              "EXB059.A1901","EXB059.A1301","EXB059.A1101","EXB059.A0901",
              "SRR1646041")

outliersF <- str_c(outliers, collapse = "|")

```

Read in the contaminant list from `Cameroon_taxonomy_decontam.Rmd` and remove them from the taxonomy table. 
```{r contaminants}

# contaminants_species <- fread("05-results.backup/malt_contaminants_species.tsv", header = T, sep = "\t")
# contaminants_genus <- fread("05-results.backup/malt_contaminants_genus.tsv")
# 
# malt_species.decontam <- malt_species %>%
#   anti_join(., contaminants_species)
# 
# malt_genus.decontam <- malt_genus %>%
#   anti_join(., contaminants_genus)
# 

```

#Sample Clustering with PCA

Step-by-step removal of taxa based on decontam and a filter threshold, followed
by PCA to see how contaminants/low abundance species affect the relationships of
samples to eachother between groups.
```{r pca_all_species}
# unfiltered and not decontaminated
malt_species_unfiltered <- malt_species_raw %>%
  select(-matches("SRR164|ERR|SRS")) %>%
  adorn_totals(where = "col", name = "Sum_species") %>%
  select(-one_of(outliers)) %>%
  # select(-contains("10M")) %>%
  mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
  select(-one_of(c("Sum_species","Percent"))) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species_unfiltered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_species_unfiltered, "PC1", "PC2", "Description", "Env", "Env", 3)
plot_pca_titles(malt_species_unfiltered, "PC3", "PC2", "Description", "Env", "Env", 3)


# ggsave("05-results.backup/pca_species_unfiltered_pc1pc2.pdf", plot = pca_species_unfiltered_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)


# filtered only
malt_species_filtered <- malt_species_raw %>%
  select(-matches("SRR164|ERR|SRS")) %>%
  adorn_totals(where = "col", name = "Sum_species") %>%
  select(-one_of(outliers)) %>%
  # select(-contains("10M")) %>%
  mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
  filter(Percent > 0.005) %>%
  select(-one_of(c("Sum_species","Percent"))) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species_filtered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_species_filtered, "PC1", "PC2", "Description", "Env", "Env", 3)
plot_pca_titles(malt_species_filtered, "PC3", "PC2", "Description", "Env", "Env", 3)


# decontaminated only
malt_species.decontam_unfilt <- malt_species.decontam %>%
  select(-matches("SRR164|ERR|SRS")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam") %>%
  select(-one_of(outliers)) %>%
  # select(-contains("10M")) %>%
  mutate(Percent = Sum_species.decontam/sum(Sum_species.decontam)*100) %>%
  select(-one_of(c("Sum_species.decontam","Percent"))) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_unfilt, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
plot_pca_titles(malt_species.decontam_unfilt, "PC1", "PC2", "Description", "Env", "Env", 3)
plot_pca_titles(malt_species.decontam_unfilt, "PC3", "PC2", "Description", "Env", "Env", 3)

# decontaminated and filtered
malt_species.decontam_filtered <- malt_species.decontam %>%
  select(-matches("SRR164|ERR|SRS")) %>%
  adorn_totals(where = "col", name = "Sum_species.decontam_filtered") %>%
  select(-one_of(outliers)) %>%
  # select(-contains("10M")) %>%
  mutate(Percent = Sum_species.decontam_filtered/sum(Sum_species.decontam_filtered)*100) %>%
  filter(Percent > 0.005) %>%
  select(-one_of(c("Sum_species.decontam_filtered","Percent"))) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
all_samples_sp_pca12 <- plot_pca_titles(malt_species.decontam_filtered, "PC1", "PC2", "Description", "Env", "Env", 3)
all_samples_sp_pca12
all_samples_sp_pca32 <- plot_pca_titles(malt_species.decontam_filtered, "PC3", "PC2", "Description", "Env", "Env", 3)
all_samples_sp_pca32

# ggsave("./06-publication/supplemental_figures/SXX28/SXX28_all_samples_sp_pca12.pdf", plot = all_samples_sp_pca12, device = "pdf",
#        scale = 1, width = 7, height = 7, units = c("in"), dpi = 300)
# ggsave("./06-publication/supplemental_figures/SXX28/SXX28_all_samples_sp_pca32.pdf", plot = all_samples_sp_pca32, device = "pdf",
#        scale = 1, width = 6, height = 7, units = c("in"), dpi = 300)


```
The 10M read subsampled HMP and JAE samples behave similarly to their equivalent
full samples, so for downstream species analysis we will use the full samples.
For the JAE samples, there is still a small but clear "movement" of 10M read-subsampled
samples toward the CMC samples, which were sequenced to a depth of 3M-8M reads,
so there may be some diversity lost in calculus samples that are not deeply
sequenced. We would have to sequence the CMC samples deeper to see if this is
the case for these mature plaque samples as well, but I suspect it would be.


# Species-level analyses
What is the relationship of the groups in a PCA without the blanks? 

## Industrial and Cameroon samples
What is the relationship of the groups in a PCA without the blanks or the
Yanomami samples? Here we'll use a cut-off of 10% prevalence in any group to
keep the species (this means that a species must be present in at least one of
the following: 6 Baka, 3 Nzime, 2 HMP, 1 JAE). If a species is not present in at
least 10% of at least 1 group, it will be removed from analysis.
```{r pca_pct10_noblanks}

# Determine the prevalence of species in each group
malt_species.decontam_noblanks_presence_more_10 <- malt_species.decontam %>%
  select(-matches("EXB|LIB|10M|SRR164|ERR|SRS")) %>%
  select(-one_of(outliers)) %>%
  mutate_if(is.numeric, ~1 * (. > 0)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  inner_join(., metadata %>%
               select(SampleID, Ethnic_Group), by = "SampleID") %>%
  group_by(Ethnic_Group, Species) %>%
  summarize(Percent_presence = sum(Counts)/length(SampleID)*100) %>%
  # select the species that are present in at least 10% of at least one group
  ungroup() %>%
  filter(Percent_presence >= 10) %>%
  distinct(Species) %>%
  select(Species)


# Species, no blanks 
malt_species.decontam_noblanks <- malt_species.decontam %>%
                               select(-matches("EXB|LIB|10M|SRR164")) %>%
                               select(-one_of(outliers)) %>%
                               inner_join(., malt_species.decontam_noblanks_presence_more_10, by = "Species") %>%
                               gather("SampleID","Counts",2:ncol(.)) %>%
                               mutate(Counts = Counts +1) %>%
                               spread(Species,Counts) %>%
                               column_to_rownames("SampleID")

# perform PCA
plot_pca_titles(malt_species.decontam_noblanks, "PC1", "PC2", "Description", "Env", "Env", 3)
plot_pca_titles(malt_species.decontam_noblanks, "PC3", "PC2", "Description", "Env", "Env", 3)

plot_pca_titles(malt_species.decontam_noblanks, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_noblanks, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_noblanks, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_noblanks, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_noblanks, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_noblanks, "PC1", "PC2", "Market_integration", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_noblanks, "PC1", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_noblanks, "PC1", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)

```


Filter the species table by just the species in MR1 and MR2 from
the factor analysis (`Cameroon_taxonomy_discrimspecies.Rmd`)
```{r pca_mr_noblanks_species}
# Now keep only the proteins that are in MR1 from the factor analysis
malt_species.decontam_filtered_noblanks_mr <- malt_species.decontam %>%
  select(-one_of(outliers)) %>%
  select(-matches("10M|SRR164|EXB|LIB|SRS|ERR")) %>%
  inner_join(., mr_species %>%
               filter(str_detect(Residual, "MR1")) %>%
               select(-Residual)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = as.numeric(Counts),
         Counts = Counts + 1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered_noblanks_mr, logratio = 'CLR')

# perform a PCA to see how the data cluster
malt_species.decontam_filtered_noblanks_mr.pca <- pca(malt_species.decontam_filtered_noblanks_mr, ncomp = 3, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
pca_noyanomami_village <- plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
pca_noyanomami_village
plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# ggsave("./06-publication/supplemental_figures/SXX35/panel_parts/SXX35_species_village_pc12.pdf", plot = pca_noyanomami_village,
#        device = "pdf", scale = 1, width = 8, height = 6, units = c("in"), dpi = 300)

# by Sex
species_sex_pc12 <- plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
species_sex_pc12
plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# ggsave("./06-publication/supplemental_figures/SXX35/panel_parts/SXX35_species_sex_pc12.pdf", plot = species_sex_pc12,
#        device = "pdf", scale = 1, width = 8, height = 6, units = c("in"), dpi = 300)

# by Age group
species_age_pc12 <- plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
species_age_pc12
plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# ggsave("./06-publication/supplemental_figures/SXX35/panel_parts/SXX35_species_age_pc12.pdf", plot = species_age_pc12,
#        device = "pdf", scale = 1, width = 8, height = 6, units = c("in"), dpi = 300)

# by Market integration
species_me_pc12 <- plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC1", "PC2", "Market_integration", "Ethnic_Group", "Ethnic_Group", 3)
species_me_pc12
plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC3", "PC2", "Market_integration", "Ethnic_Group", "Ethnic_Group", 3)


# by Tooth site
ts <- plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# ggsave("./06-publication/supplemental_figures/SXX35/panel_parts/SXX35_species_ts_pc12.pdf", plot = ts,
#        device = "pdf", scale = 1, width = 8, height = 6, units = c("in"), dpi = 300)

# by Description (do HMP supragingival and HMP subgingival plaque samples separate?)
plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC3", "PC2", "Description", "Description", "Description", 3)

# by Extraction batch 
plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC1", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC3", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Library batch 
plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC1", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC3", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_species_decontam_filtered_noblanks_mr.pca <- mixOmics::pca(malt_species.decontam_filtered_noblanks_mr, ncomp = 3, logratio = 'CLR')
plot(malt_species_decontam_filtered_noblanks_mr.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_species_decontam_filtered_noblanks_mr.pca.variates <- as.data.frame(malt_species_decontam_filtered_noblanks_mr.pca$variates$X)
malt_species_decontam_filtered_noblanks_mr.pca.variates <- malt_species_decontam_filtered_noblanks_mr.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_species_decontam_filtered_noblanks_mr.pca.varmet <- inner_join(malt_species_decontam_filtered_noblanks_mr.pca.variates, metadata, by = "SampleID")

pca_malt_species_pc1pc2 <- ggplot(malt_species_decontam_filtered_noblanks_mr.pca.varmet, aes(PC1, PC2, colour = Combined_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_species_decontam_filtered_noblanks_mr.pca$prop_expl_var$X[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_species_decontam_filtered_noblanks_mr.pca$prop_expl_var$X[2], 2)), "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_species_pc1pc2

# ggsave("05-results.backup/librarybatch_noyanomami.pdf", plot = extractionbatch, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

```


Make biplots of the top 10 loading species in positive and negative directions
of PC1 and PC2, using the MR1 species only.
```{r biplot_noblanks_noyanomami_species_mr}
malt_species.decontam_filtered_noblanks_mr <- malt_species.decontam %>%
  select(-one_of(outliers)) %>%
  select(-matches("10M|SRR164|EXB|LIB|SRS|ERR")) %>%
  inner_join(., mr_species %>%
               filter(str_detect(Residual, "MR1")) %>%
               select(-Residual)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = as.numeric(Counts),
         Counts = Counts + 1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered_noblanks_mr, logratio = 'CLR')

# perform a PCA to see how the data cluster
malt_species.decontam_filtered_noblanks_mr.pca <- pca(malt_species.decontam_filtered_noblanks_mr, ncomp = 3, logratio = 'CLR')

# Then plot by Tooth_site (do anterior teeth and posterior teeth samples separate?) to see what it looks like before adding the biplot over the points
plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_noblanks_mr, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# Now plot biplots, and make tables of the top 10 species in each loading
malt_species_decontam_filtered_noyanomami_biplot_121 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_mr.pca, "PC1", "PC2", "Ethnic_Group", PC1)
malt_species_decontam_filtered_noyanomami_biplot_121

# ggsave("./06-publication/supplemental_figures/SXX19/panel_parts/SXX19_species_all_biplot_121.pdf", plot = malt_species_decontam_filtered_noyanomami_biplot_121,
#        device = "pdf", scale = 1, width = 6, height = 7, units = c("in"), dpi = 300)

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings")

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_species_decontam_filtered_noyanomami_biplot_122 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_mr.pca, "PC1", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_noyanomami_biplot_122

cmc_hmp_jae_sp_pc_list <- malt_species_decontam_filtered_noyanomami_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)) %>% mutate(Direction = "PC1+", Samples = "Industrial_CMC", Filter = "MR1")
cmc_hmp_jae_sp_pc_list <- cmc_hmp_jae_sp_pc_list %>%
  bind_rows(malt_species_decontam_filtered_noyanomami_biplot_121$plot_env$neg_10 %>% arrange(desc(PC1)) %>% mutate(Direction = "PC1-", Samples = "Industrial_CMC", Filter = "MR1"))
cmc_hmp_jae_sp_pc_list <- cmc_hmp_jae_sp_pc_list %>%
  bind_rows(malt_species_decontam_filtered_noyanomami_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)) %>% mutate(Direction = "PC2+", Samples = "Industrial_CMC", Filter = "MR1"))
cmc_hmp_jae_sp_pc_list <- cmc_hmp_jae_sp_pc_list %>%
  bind_rows(malt_species_decontam_filtered_noyanomami_biplot_122$plot_env$neg_10 %>% arrange(desc(PC2)) %>% mutate(Direction = "PC2-", Samples = "Industrial_CMC", Filter = "MR1"))

# fwrite(cmc_hmp_jae_sp_pc_list, "./05-results.backup/cmc_hmp_jae_sp_pc_list.tsv", sep = "\t", quote = F)

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_species_decontam_filtered_noyanomami_biplot_322 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_mr.pca, "PC3", "PC2", "Ethnic_Group", PC2)
malt_species_decontam_filtered_noyanomami_biplot_322

malt_species_decontam_filtered_noyanomami_biplot_323 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_mr.pca, "PC3", "PC2", "Ethnic_Group", PC3)
malt_species_decontam_filtered_noyanomami_biplot_323

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species_decontam_filtered_noyanomami_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```



Run a permanova with function adonis to test if the clusters are significantly
different. First load the metadata again so the blanks and Yanomami samples can
be removed and then make them factors.
```{r metadata_betadisper_noblanks}

metadata_noblanks <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata_noblanks <- metadata_noblanks %>%
  rename(SampleID = `#SampleID`)  %>%
  filter(SampleID != "CMC032.B0101") %>%
  filter(!SampleID %in% outliers) %>%
  filter(!str_detect(SampleID, "SRR16460|EXB|LIB|10M|SRS|ERR")) %>%
  as_tibble() %>%
  # mutate(Market_integration_spli = str_replace(Market_integration, "HMP","Industrial"))%>%
  mutate(Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime","Industrial plaque","Industrial calculus"),
         Env = fct_relevel(Env, "CMC","Industrial plaque","Industrial calculus"),
         Industry = fct_relevel(Industry, "Non-industrial","Industrial"),
         Market_integration_split = fct_relevel(Market_integration_split, "Logging road","Forrest camp","Farming","Industrial calculus","Industrial plaque"),
         Age_group = fct_relevel(Age_group, "18-29","30-39","40-49","50-59","60+","HMP"),
         Sex = fct_relevel(Sex, "Male","Female"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior"),
         Village = fct_relevel(Village, "202","205","204","203","206","Industrial plaque","Industrial calculus")) %>%
  select(SampleID, Age_group, Ethnic_Group, Env, Industry, Market_integration_split, Sex, Tooth_site, Village, ExtractionBatch, LibraryBatch)

```

Run a PERMANOVA with adonis2 on the abundance-filtered clr-transformed species matrix.
```{r adonis_noblanks_noyanomami_species}
# Euclidean distance, uses the clr-transformed matrix from the mixOmics PCA above
malt_species.decontam_filtered_noblanks_mr.euc <- vegdist(malt_species.decontam_filtered_noblanks_mr.pca$X, method = "euclidean", na.rm = T)
malt_species.decontam_filtered_noblanks_mr.euc.pcoa <- ape::pcoa(malt_species.decontam_filtered_noblanks_mr.euc)

malt_species.decontam_filtered_noblanks_mr.euc.pcoavectors <- as.data.frame(malt_species.decontam_filtered_noblanks_mr.euc.pcoa$vectors)
malt_species.decontam_filtered_noblanks_mr.euc.pcoavectors <- malt_species.decontam_filtered_noblanks_mr.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_species.decontam_filtered_noblanks_mr.euc.vecmet <- inner_join(malt_species.decontam_filtered_noblanks_mr.euc.pcoavectors, metadata, by = "SampleID")

metadata_noblanks %>% select(SampleID) %>% anti_join(., malt_species.decontam_filtered_noblanks_mr.euc.vecmet %>% select(SampleID))

malt_species.decontam_filtered_noblanks_mr.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Industry, shape = Industry)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Industry_colors) +
  scale_shape_manual(values = Industry_shapes) +
  stat_ellipse(aes(colour = Industry, group = Industry)) +
  theme_minimal(base_size = 7) +
  ggtitle("Industry")

# test for difference between different metadata categories 
adonis2(malt_species.decontam_filtered_noblanks_mr.euc ~ ExtractionBatch +  LibraryBatch, malt_species.decontam_filtered_noblanks_mr.euc.vecmet, permutations = 999, by = "margin")
adonis2(malt_species.decontam_filtered_noblanks_mr.euc ~ Ethnic_Group + Market_integration, malt_species.decontam_filtered_noblanks_mr.euc.vecmet, permutations = 999, by = "margin")

# test for difference between different metadata categories excluding Village, which influences Ethnic_Group, Industry, and Market_integration
# this still has Ethnic_Group and Indistry as -Inf
adonis2(malt_species.decontam_filtered_noblanks_mr.euc ~ Industry + Market_integration, malt_species.decontam_filtered_noblanks_mr.euc.vecmet, permutations = 999, by = "margin")

# controlled for tooth site
# controlled for tooth site
# perm <- how(nperm = 999)
# setBlocks(perm) <- with(metadata_noblanks, Tooth_site)
# adonis2(malt_species.decontam_filtered_noblanks_mr.euc ~ Village, metadata_noblanks, permutations = perm, by = "margin")

```
Village and tooth site are significantly different, unless controlling for tooth
site, where nothing is significantly different.

Then run betadisper to see if the groups with significant differences in
clustering have different dispersion ("tight" vs. "loose" clustering of groups)
- is one group more variable than the other(s)?
```{r betadisper_noblanks_noyanomami_species}
# Test the 3 metadata categories that are significant (p < 0.05) based on adonis for significant differences in dispersion
# Industry, tooth site, village
# Ok, sometimes Village is not significantly different, it fluxuates around 0.05. Maybe better to leave it out as not significant

# Tooth site
groups_tooth_site <- metadata_noblanks %>%
  select(Tooth_site) %>% 
  pull()

## Calculate multivariate dispersions
toothsite_betadisper_noblanks_mr <- betadisper(malt_species.decontam_filtered_noblanks_mr.euc, groups_tooth_site, bias.adjust =TRUE)
toothsite_betadisper_noblanks_mr

## Perform test
anova(toothsite_betadisper_noblanks_mr)

## Permutation test for F
permutest(toothsite_betadisper_noblanks_mr, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
toothsite_betadisper_noblanks_mr.HSD <- TukeyHSD(toothsite_betadisper_noblanks_mr)
plot(toothsite_betadisper_noblanks_mr.HSD)

## Plot the groups and distances to centroids on the
## first two PCoA axes
## with data ellipses instead of hulls
plot(toothsite_betadisper_noblanks_mr, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## can also specify which axes to plot, ordering respected
# plot(toothsite_betadisper_noblanks_mr, axes = c(3,1), seg.col = "forestgreen", seg.lty = "dashed")

## Draw a boxplot of the distances to centroid for each group
boxplot(toothsite_betadisper_noblanks_mr)


# Village
groups_village <- metadata_noblanks %>%
  select(Village) %>% 
  pull()

## Calculate multivariate dispersions
village_betadisper_noblanks_mr <- betadisper(malt_species.decontam_filtered_noblanks_mr.euc, groups_village, bias.adjust =TRUE)
village_betadisper_noblanks_mr

## Perform test
anova(village_betadisper_noblanks_mr)

## Permutation test for F
permutest(village_betadisper_noblanks_mr, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
village_betadisper_noblanks_mr.HSD <- TukeyHSD(village_betadisper_noblanks_mr)
plot(village_betadisper_noblanks_mr.HSD)

## Plot with data ellipses instead of hulls
plot(village_betadisper_noblanks_mr, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(village_betadisper_noblanks_mr)


# Market integration
groups_mi <- metadata_noblanks %>%
  select(Market_integration_split) %>% 
  pull()

## Calculate multivariate dispersions
mi_betadisper_noblanks_mr <- betadisper(malt_species.decontam_filtered_noblanks_mr.euc, groups_mi, bias.adjust =TRUE)
mi_betadisper_noblanks_mr

## Perform test
anova(mi_betadisper_noblanks_mr)

## Permutation test for F
permutest(mi_betadisper_noblanks_mr, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
mi_betadisper_noblanks_mr.HSD <- TukeyHSD(mi_betadisper_noblanks_mr)
plot(mi_betadisper_noblanks_mr.HSD)

## Plot with data ellipses instead of hulls
plot(mi_betadisper_noblanks_mr, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(mi_betadisper_noblanks_mr)


```
Significant differences in disperson are between HMP and Posterior, and HMP and
villages 202, 205, 205, 206, JAE.

## Cameroon samples only
What is the relationship of Cameroon plaque samples to eachother without any
other sample types? Do they cluster by any of the metadata categories?
Filter the species table by % abundance
```{r pca_cmc_species}
malt_species.decontam_filtered_cmc <- malt_species.decontam %>%
  select(-one_of(outliers)) %>%
  select(matches("Species|CMC")) %>%
  adorn_totals(where = "col", name = "Sum_species") %>%
  mutate(Percent = Sum_species/sum(Sum_species)*100) %>%
  filter(Percent > 0.001) %>%
  select(-one_of(c("Sum_species","Percent"))) %>%
  gather("SampleID","Counts",2:ncol(.)) %>% 
  mutate(Counts = Counts +1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered_cmc, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
pca_cmc_village <- plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
pca_cmc_village
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

# ggsave("05-results.backup/pca_cmc_village.pdf", plot = pca_cmc_village, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"), dpi = 300)

# by Sex
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# by Age group
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# by Market integration
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Market_integration", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Market_integration", "Ethnic_Group", "Ethnic_Group", 3)

# by Tooth site
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Extraction batch
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Library batch
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc, "PC3", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_species.decontam_filtered_cmc.pca <- mixOmics::pca(malt_species.decontam_filtered_cmc, ncomp = 3, logratio = 'CLR')
plot(malt_species.decontam_filtered_cmc.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_species.decontam_filtered_cmc.pca.variates <- as.data.frame(malt_species.decontam_filtered_cmc.pca$variates$X)
malt_species.decontam_filtered_cmc.pca.variates <- malt_species.decontam_filtered_cmc.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_species.decontam_filtered_cmc.pca.varmet <- inner_join(malt_species.decontam_filtered_cmc.pca.variates, metadata, by = "SampleID")

pca_malt_species_pc1pc2 <- ggplot(malt_species.decontam_filtered_cmc.pca.varmet, aes(PC1, PC2, colour = Combined_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_species.decontam_filtered_cmc.pca$prop_expl_var$X[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_species.decontam_filtered_cmc.pca$prop_expl_var$X[2], 2))< "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_species_pc1pc2


# with names
pca.list <- mixOmics::pca(malt_species.decontam_filtered_cmc, ncomp = 3, logratio = 'CLR')
plot_pca_names(pca.list, "PC1", "PC2", "Ethnic_Group")

# ggsave("05-results.backup/pca_nt_species_pc1pc2.pdf", plot = pca_nt_species_pc1pc2, device = "pdf",
#        scale = 1, width = 7, height = 5, units = c("in"),
#        dpi = 300)
```

Filter the species table by just the species in MR1 and MR2 from
the factor analysis (`Cameroon_taxonomy_discrimspecies.Rmd`)
```{r pca_mr_noblanks_noyanomami_species}
# Now keep only the proteins that are in MR1 from the factor analysis
malt_species.decontam_filtered_cmc_mr <- malt_species.decontam %>%
  select(-one_of(outliers)) %>%
  select(matches("Species|CMC")) %>%
  inner_join(., mr_cmc_species %>%
               filter(str_detect(Residual, "MR1")) %>%
               select(-Residual)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = as.numeric(Counts),
         Counts = Counts + 1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered_cmc_mr, logratio = 'CLR')

# perform a PCA to see how the data cluster
malt_species.decontam_filtered_cmc_mr.pca <- pca(malt_species.decontam_filtered_cmc_mr, ncomp = 3, logratio = 'CLR')

# perform and plot a PCA to see how the data cluster
# by Village
pca_noyanomami_village <- plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
pca_noyanomami_village
plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)

ggsave("./06-publication/supplemental_figures/SXX35/panel_parts/SXX35_species_cmc_village_pc12.pdf", plot = pca_noyanomami_village,
       device = "pdf", scale = 1, width = 7, height = 5.75, units = c("in"), dpi = 300)

# by Sex
species_sex_pc12 <- plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC1", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)
species_sex_pc12
plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC3", "PC2", "Sex", "Ethnic_Group", "Ethnic_Group", 3)

# ggsave("./06-publication/supplemental_figures/SXX35/panel_parts/SXX35_species_cmc_sex_pc12.pdf", plot = species_sex_pc12,
#        device = "pdf", scale = 1, width = 7, height = 5.75, units = c("in"), dpi = 300)

# by Age group
species_age_pc12 <- plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC1", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)
species_age_pc12
plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC3", "PC2", "Age_group", "Ethnic_Group", "Ethnic_Group", 3)

# ggsave("./06-publication/supplemental_figures/SXX35/panel_parts/SXX35_species_cmc_age_pc12.pdf", plot = species_age_pc12,
#        device = "pdf", scale = 1, width = 7, height = 5.75, units = c("in"), dpi = 300)

# by Market integration
species_me_pc12 <- plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC1", "PC2", "Market_integration", "Ethnic_Group", "Ethnic_Group", 3)
species_me_pc12
plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC3", "PC2", "Market_integration", "Ethnic_Group", "Ethnic_Group", 3)

ggsave("./06-publication/supplemental_figures/SXX35/panel_parts/SXX35_species_cmc_me_pc12.pdf", plot = species_me_pc12,
       device = "pdf", scale = 1, width = 7, height = 5.75, units = c("in"), dpi = 300)

# by Tooth site
plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC3", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)

# by Description (do HMP supragingival and HMP subgingival plaque samples separate?)
plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC1", "PC2", "Description", "Description", "Description", 3)
plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC3", "PC2", "Description", "Description", "Description", 3)

# by Extraction batch 
extractionbatch <- plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC1", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)
extractionbatch
plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC3", "PC2", "ExtractionBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Library batch 
librarybatch <- plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC1", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)
librarybatch
plot_pca_titles(malt_species.decontam_filtered_cmc_mr, "PC3", "PC2", "LibraryBatch", "Ethnic_Group", "Ethnic_Group", 3)

# by Individual sequencing depth
# perform a PCA to see how the data cluster
malt_species_decontam_filtered_cmc_mr.pca <- mixOmics::pca(malt_species.decontam_filtered_cmc_mr, ncomp = 3, logratio = 'CLR')
plot(malt_species_decontam_filtered_cmc_mr.pca)

# Select out the PC variates into a new table and add the metadata for plotting
malt_species_decontam_filtered_cmc_mr.pca.variates <- as.data.frame(malt_species_decontam_filtered_cmc_mr.pca$variates$X)
malt_species_decontam_filtered_cmc_mr.pca.variates <- malt_species_decontam_filtered_cmc_mr.pca.variates %>%
  rownames_to_column("SampleID")
  
malt_species_decontam_filtered_cmc_mr.pca.varmet <- inner_join(malt_species_decontam_filtered_cmc_mr.pca.variates, metadata, by = "SampleID")

pca_malt_species_pc1pc2 <- ggplot(malt_species_decontam_filtered_cmc_mr.pca.varmet, aes(PC1, PC2, colour = Combined_Seq_Depth_NonHuman, shape = Ethnic_Group)) +
  geom_point(size = 4) +
  scale_shape_manual(values = Ethnic_Group_shapes) +
  xlab(paste("PC1 - ", 100*(round(malt_species_decontam_filtered_cmc_mr.pca$prop_expl_var$X[1], 2)), "%", sep = "")) +
  ylab(paste("PC2 - ", 100*(round(malt_species_decontam_filtered_cmc_mr.pca$prop_expl_var$X[2], 2)), "%", sep = "")) +
  theme_minimal(base_size = 14) +
  labs(colour = "Seq. depth\n(reads)", shape = "Ethnic group")

pca_malt_species_pc1pc2

# ggsave("./06-publication/supplemental_figures/SXX2/SXX2_exb_betadiv_mr1_sp.pdf", plot = extractionbatch, device = "pdf",
#        scale = 1, width = 6, height = 6, units = c("in"), dpi = 300)
# 
# ggsave("./06-publication/supplemental_figures/SXX2/SXX2_lib_betadiv_mr1_sp.pdf", plot = librarybatch, device = "pdf",
#        scale = 1, width = 6, height = 6, units = c("in"), dpi = 300)

```


```{r metadata_betadisper_cmc}
# Need special metadata that has only the categories of CMC samples, b/c of issues w/ extra factors from removed groups
metadata_cmc <- fread("00-documentation.backup/01-cameroon_hunter_gatherer_metadata.tsv")
metadata_cmc <- metadata_cmc %>%
  rename(SampleID = `#SampleID`) %>%
  filter(!SampleID %in% outliers) %>%
  filter(SampleID != "CMC032.B0101") %>%
  filter(Description == "CMC") %>%
  select(SampleID, Age_group, Ethnic_Group, Market_integration, Sex, Tooth_site, Village, ExtractionBatch, LibraryBatch) %>%
  as_tibble() %>%
  mutate(Age_group = fct_relevel(Age_group, "18-29","30-39","40-49","50-59","60+"),
         Ethnic_Group = fct_relevel(Ethnic_Group, "Baka","Nzime"),
         Market_integration = fct_relevel(Market_integration, "Logging road","Forrest camp","Farming"),
         Sex = fct_relevel(Sex, "Male","Female"),
         Tooth_site = fct_relevel(Tooth_site, "Anterior","Posterior"),
         Village = fct_relevel(Village, "202","205","204","203","206"))
```


Run a permanova with function adonis to test if the clusters are significantly different.
```{r adonis_cmc_species}
# use the CLR-transformed matrix from the mixOmics PCA in the last section

# Euclidean distance
malt_species.decontam_filtered_cmc.euc <- vegdist(malt_species.decontam_filtered_cmc_mr.pca$X, method = "euclidean", na.rm = T)
malt_species.decontam_filtered_cmc.euc.pcoa <- ape::pcoa(malt_species.decontam_filtered_cmc.euc)

malt_species.decontam_filtered_cmc.euc.pcoavectors <- as.data.frame(malt_species.decontam_filtered_cmc.euc.pcoa$vectors)
malt_species.decontam_filtered_cmc.euc.pcoavectors <- malt_species.decontam_filtered_cmc.euc.pcoavectors %>%
  rownames_to_column("SampleID")

malt_species.decontam_filtered_cmc.euc.vecmet <- left_join(malt_species.decontam_filtered_cmc.euc.pcoavectors, metadata_cmc, by = "SampleID")

metadata_cmc %>% select(SampleID) %>% anti_join(., malt_species.decontam_filtered_cmc.euc.vecmet %>% select(SampleID))

malt_species.decontam_filtered_cmc.euc.vecmet %>%
  ggplot(., aes(Axis.1, Axis.2, colour = Tooth_site, shape = Tooth_site)) +
  geom_point(size = 3) +
  scale_colour_manual(values = Tooth_site_colors) +
  scale_shape_manual(values = Tooth_site_shapes) +
  stat_ellipse(aes(colour = Tooth_site, group = Tooth_site)) +
  theme_minimal(base_size = 7) +
  ggtitle("Tooth site")

# test for difference between different metadata categories 
adonis2(malt_species.decontam_filtered_cmc.euc ~ ExtractionBatch +  LibraryBatch, metadata_cmc, permutations = 999, by = "margin")
adonis2(malt_species.decontam_filtered_cmc.euc ~ Village + Tooth_site, metadata_cmc, permutations = 999, by = "margin")

# controlled for tooth site
perm <- how(nperm = 999)
setBlocks(perm) <- with(metadata_cmc, Tooth_site)
adonis2(malt_species.decontam_filtered_cmc.euc ~ Village, metadata_cmc, permutations = perm, by = "margin")

```
Sex and tooth site are significantly different, unless controlling for tooth
site and then only sex is significantly different.


Then run betadisper to see if the groups with significant differences in
clustering have different dispersion ("tight" vs. "loose" clustering of groups)
- is one group more variable than the other(s)?
```{r betadisper_cmc_species}
# Test the 4 metadata categories that are significant (p < 0.05) based on adonis for significant differences in dispersion
# Tooth site, market integration, village, sex
# Ok, sometimes Sex is not significantly different, it fluxuates around 0.05. Maybe better to leave it out as not significant

# Tooth site
groups_tooth_site <- metadata_cmc %>%
  select(Tooth_site) %>% 
  pull()

## Calculate multivariate dispersions
toothsite_betadisper_cmc <- betadisper(malt_species.decontam_filtered_cmc.euc, groups_tooth_site, bias.adjust =TRUE)
toothsite_betadisper_cmc

## Perform test
anova(toothsite_betadisper_cmc)

## Permutation test for F
permutest(toothsite_betadisper_cmc, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
toothsite_betadisper_cmc.HSD <- TukeyHSD(toothsite_betadisper_cmc)
plot(toothsite_betadisper_cmc.HSD)

## Plot the groups and distances to centroids on the
## first two PCoA axes
## with data ellipses instead of hulls
plot(toothsite_betadisper_cmc, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## can also specify which axes to plot, ordering respected
# plot(toothsite_betadisper_cmc, axes = c(3,1), seg.col = "forestgreen", seg.lty = "dashed")

## Draw a boxplot of the distances to centroid for each group
boxplot(toothsite_betadisper_cmc)


# Sex
groups_sex <- metadata_cmc %>%
  select(Sex) %>% 
  pull()

## Calculate multivariate dispersions
sex_betadisper_cmc <- betadisper(malt_species.decontam_filtered_cmc.euc, groups_sex, bias.adjust =TRUE)
sex_betadisper_cmc

## Perform test
anova(sex_betadisper_cmc)

## Permutation test for F
permutest(sex_betadisper_cmc, pairwise = TRUE, permutations = 99)

## Tukey's Honest Significant Differences
sex_betadisper_cmc.HSD <- TukeyHSD(sex_betadisper_cmc)
plot(sex_betadisper_cmc.HSD)

## Plot with data ellipses instead of hulls
plot(sex_betadisper_cmc, ellipse = TRUE, hull = FALSE) # 1 sd data ellipse

## Draw a boxplot of the distances to centroid for each group
boxplot(sex_betadisper_cmc)

```
There is a significant difference in disperson between anterior and posterior
sites and between male and female samples.


Now biplots on the MR1 species
```{r biplots_cmc_species_mr}

malt_species.decontam_filtered_cmc_mr <- malt_species.decontam %>%
  select(-one_of(outliers)) %>%
  select(-matches("10M|SRR|JAE|EXB|LIB")) %>%
  inner_join(., mr_cmc_species %>%
               filter(str_detect(Residual, "MR1")) %>%
               select(-Residual)) %>%
  gather("SampleID","Counts",2:ncol(.)) %>%
  mutate(Counts = as.numeric(Counts),
         Counts = Counts + 1) %>%
  spread(Species,Counts) %>%
  column_to_rownames("SampleID")

# prepare to run a PCA
# check the number of components to retain by tuning the PCA
tune.pca(malt_species.decontam_filtered_cmc_mr, logratio = 'CLR')


# perform a PCA to see how the data cluster
malt_species.decontam_filtered_cmc_mr.pca <- pca(malt_species.decontam_filtered_cmc_mr, ncomp = 3, logratio = 'CLR')

malt_species.decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(malt_species.decontam_filtered_cmc_mr.pca, "PC1", "PC2", "Tooth_site", PC1)
malt_species.decontam_filtered_nocontrols_biplot_121

ggsave("./06-publication/supplemental_figures/SXX19/panel_parts/SXX19_species_cmc_biplot_121.pdf", plot = malt_species.decontam_filtered_nocontrols_biplot_121,
       device = "pdf", scale = 1, width = 6, height = 7, units = c("in"), dpi = 300)

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 positive loadings") 

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(PC1),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC1 negative loadings")


malt_species.decontam_filtered_nocontrols_biplot_122 <- plot_pca_bi(malt_species.decontam_filtered_cmc_mr.pca, "PC1", "PC2", "Tooth_site", PC2)
malt_species.decontam_filtered_nocontrols_biplot_122

cmc_sp_list <- malt_species.decontam_filtered_nocontrols_biplot_121$plot_env$pws_10 %>% arrange(desc(PC1)) %>% mutate(Direction = "PC1+", Samples = "CMC", Filter = "MR1")
cmc_sp_list <- cmc_sp_list %>%
  bind_rows(malt_species.decontam_filtered_nocontrols_biplot_121$plot_env$neg_10 %>% arrange(desc(PC1)) %>% mutate(Direction = "PC1-", Samples = "CMC", Filter = "MR1"))
cmc_sp_list <- cmc_sp_list %>%
  bind_rows(malt_species.decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)) %>% mutate(Direction = "PC2+", Samples = "CMC", Filter = "MR1"))
cmc_sp_list <- cmc_sp_list %>%
  bind_rows(malt_species.decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(desc(PC2)) %>% mutate(Direction = "PC2-", Samples = "CMC", Filter = "MR1"))

fwrite(cmc_sp_list, "./05-results.backup/cmc_sp_pc_list.tsv", sep = "\t", quote = F)

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_122$plot_env$pws_10 %>% arrange(desc(PC2)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_122$plot_env$neg_10 %>% arrange(PC2),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC2 negative loadings")


malt_species.decontam_filtered_nocontrols_biplot_322 <- plot_pca_bi(malt_species.decontam_filtered_cmc_mr.pca, "PC3", "PC2", "Tooth_site", PC2)
malt_species.decontam_filtered_nocontrols_biplot_322

malt_species.decontam_filtered_nocontrols_biplot_323 <- plot_pca_bi(malt_species.decontam_filtered_cmc_mr.pca, "PC3", "PC2", "Tooth_site", PC3)
malt_species.decontam_filtered_nocontrols_biplot_323

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_323$plot_env$pws_10 %>% arrange(desc(PC3)),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 positive loadings")

pandoc.table(malt_species.decontam_filtered_nocontrols_biplot_323$plot_env$neg_10 %>% arrange(PC3),
             digits = 5,
             keep.trailing.zeros = T,
             justify = "lccr",
             split.tables = Inf,
             caption = "PC3 negative loadings")

```
Some of the PCs have overlapping species between the abundance filtered table
and the MR12 table, while others don't. Those that don't tend to be nearly
entirely one genus, but that genus differs.


```{r uky_graphs, eval = F}

all_village <- plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backup/all_village_uky.pdf", plot = all_village, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

all_village_32 <- plot_pca_titles(malt_species.decontam_filtered_noblanks, "PC3", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backup/all_village_32_uky.pdf", plot = all_village_32, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

noy_desc <- plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Description", "Description", "Description", 3)
ggsave("05-results.backup/noy_desc_uky.pdf", plot = noy_desc, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

noy_eg <- plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Ethnic_Group", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backup/noy_eg_uky.pdf", plot = noy_eg, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

noy_ts<- plot_pca_titles(malt_species.decontam_filtered_noblanks_noyanomami, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backup/noy_ts_uky.pdf", plot = noy_ts, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

malt_species_decontam_filtered_noyanomami_biplot_121 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Tooth_site", PC1)
malt_species_decontam_filtered_noyanomami_biplot_121
ggsave("05-results.backup/malt_species_decontam_filtered_noyanomami_biplot_121_uky.pdf", plot = malt_species_decontam_filtered_noyanomami_biplot_121, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

malt_species_decontam_filtered_noyanomami_biplot_122 <- plot_pca_bi(malt_species.decontam_filtered_noblanks_noyanomami.pca, "PC1", "PC2", "Tooth_site", PC2)
malt_species_decontam_filtered_noyanomami_biplot_122
ggsave("05-results.backup/malt_species_decontam_filtered_noyanomami_biplot_122_uky.pdf", plot = malt_species_decontam_filtered_noyanomami_biplot_122, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

cmc_village <- plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Village", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backupcmc_village_uky.pdf", plot = cmc_village, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

cmc_ts <- plot_pca_titles(malt_species.decontam_filtered_cmc, "PC1", "PC2", "Tooth_site", "Ethnic_Group", "Ethnic_Group", 3)
ggsave("05-results.backup/cmc_ts_uky.pdf", plot = cmc_ts, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)

malt_species.decontam_filtered_nocontrols_biplot_121 <- plot_pca_bi(malt_species.decontam_filtered_nocontrols.pca, "PC1", "PC2", "Tooth_site", PC1)
malt_species.decontam_filtered_nocontrols_biplot_121
ggsave("05-results.backup/malt_species.decontam_filtered_nocontrols_biplot_121_uky.pdf", plot = malt_species.decontam_filtered_nocontrols_biplot_121, device = "pdf",
       scale = 1, width = 7, height = 5, units = c("in"),
       dpi = 300)


```




